<div class="container">

<table style="width: 100%;"><tr>
<td>maaper</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Model-based analysis of alternative polyadenylation using 3’ end-linked reads</h2>

<h3>Description</h3>

<p>Model-based analysis of alternative polyadenylation using 3’ end-linked reads
</p>


<h3>Usage</h3>

<pre><code class="language-R">maaper(
  gtf,
  pas_annotation,
  output_dir,
  bam_c1,
  bam_c2,
  read_len,
  ncores = 1,
  num_pas_thre = 25,
  frac_pas_thre = 0.05,
  dist_thre = 600,
  num_thre = 50,
  run = "all",
  subset = NULL,
  region = "all",
  gtf_rds = NULL,
  verbose = FALSE,
  paired = FALSE,
  bed = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gtf</code></td>
<td>
<p>A character specifying the full path of the GTF file (reference genome);</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pas_annotation</code></td>
<td>
<p>A list containing the pas annotation. MAAPER provides processed annotation
information from PolyA_DB v3 on its Github page.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output_dir</code></td>
<td>
<p>A character specifying the full path of the output directory,
which is used to store all intermdediate and final outputs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bam_c1</code></td>
<td>
<p>A character vector specifying the full paths to the bam files for
condition 1 (control). The length of the vector equals the number of samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bam_c2</code></td>
<td>
<p>A character vector specifying the full paths to the bam files for
condition 2 (experiment). The length of the vector equals the number of samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>read_len</code></td>
<td>
<p>An integer specifying the read length.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>An integer specifying the number of cores used in parallel computation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num_pas_thre</code></td>
<td>
<p>An integer specifying the threhold on PAS's read number. Defaults to 25.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>frac_pas_thre</code></td>
<td>
<p>A numeric specifying the threshold on PAS's fraction. Defaults to 0.05.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dist_thre</code></td>
<td>
<p>An integer specifying the threshold on fragment length. Defaults to 600.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num_thre</code></td>
<td>
<p>An integer specifying the threhold on gene's read number. Defaults to 50.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>run</code></td>
<td>
<p>"all" (default) or "skip-train". For test and debug only.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subset</code></td>
<td>
<p>A character vector specifying genes' Ensembl IDs if only a subset of genes need to be analyzed.
Check the <code>pas_annotation</code> files for ID formats.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>region</code></td>
<td>
<p>"all" (default). For test and debug only.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gtf_rds</code></td>
<td>
<p>NULL (default). For test and debug only.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>FALSE (default). For test and debug only.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>paired</code></td>
<td>
<p>A boolean indicating whether to perform paired test instead of unpaired test (defaults to FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bed</code></td>
<td>
<p>Aboolean indicating whether bedGraph files should be output for visualization in genome browser.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p><code>maaper</code> saves two text files, gene.txt and pas.txt, to <code>out_dir</code>.
pas.txt contains the gene names, predicted PASs, and their corresponding fractions in the two conditions.
gene.txt contains the genes' PAS number, p values, RED, RLDu, and RLDi scores.
</p>


<h3>Author(s)</h3>

<p>Wei Vivian Li, <a href="mailto:vivian.li@rutgers.edu">vivian.li@rutgers.edu</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# data used in this example can be found on the package's Github page
pas_annotation = readRDS("./mouse.PAS.mm9.rds")
gtf = "./gencode.mm9.chr19.gtf"
bam_c1 = "./NT_chr19_example.bam"
bam_c2 = "./AS_4h_chr19_example.bam"
maaper(gtf, pas_annotation, output_dir = "./",
       bam_c1, bam_c2, read_len = 76, ncores = 1)

## End(Not run)
</code></pre>


</div>