<div class="container">

<table style="width: 100%;"><tr>
<td>plot_oxford_grid</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>plot the Macro-synteny as an oxford grid.</h2>

<h3>Description</h3>

<p>This is a function to plot the oxford grided plot to compare the macro synteny of two species. It requires as input an orthologs_df loaded by load_orthologs()
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_oxford_grid(
  orthologs_df,
  sp1_label = "",
  sp2_label = "",
  dot_size = 0.5,
  dot_alpha = 0.4,
  reorder = FALSE,
  keep_only_significant = FALSE,
  color_by = NULL,
  pvalue_threshold = 0.001,
  color_palette = NULL,
  shade_non_significant = TRUE,
  reverse_species = FALSE,
  keep_sp1_raw_order = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>orthologs_df</code></td>
<td>
<p>dataframe. orthologs with genomic coordinates loaded by the load_orthologs()</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sp1_label</code></td>
<td>
<p>character. name of 1st species to display on the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sp2_label</code></td>
<td>
<p>character. name of 2nd species to display on the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dot_size</code></td>
<td>
<p>numeric. (default = 0.5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dot_alpha</code></td>
<td>
<p>numeric. (default = 0.4)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reorder</code></td>
<td>
<p>logical. (default = FALSE) tells whether to reorder the chromosomes in clusters as implemented in reorder_macrosynteny()</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep_only_significant</code></td>
<td>
<p>logical. (default = FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color_by</code></td>
<td>
<p>string/variable name. (default = NULL) column of the orthologs_df to use to color the dots.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvalue_threshold</code></td>
<td>
<p>numeric. (default = 0.001)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color_palette</code></td>
<td>
<p>vector. (default = NULL) list of colors (as string under double quote) for the clusters. The amount of colors must match the amount of clusters.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shade_non_significant</code></td>
<td>
<p>logical. (default = TRUE) When TRUE the orthologs located on non-significant linkage groups are displayed in "grey"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reverse_species</code></td>
<td>
<p>logical. (default = FALSE) When TRUE the x and y axis of the plot are reversed. sp1 is displayed on the y axis and sp2 is displayed on the x axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep_sp1_raw_order</code></td>
<td>
<p>logical.(default equals FALSE) tells if the reordering should be constrained on the species1 and change just the order of the species2</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A ggplot2 object
</p>


<h3>See Also</h3>

<p><code>load_orthologs()</code>
</p>
<p><code>reorder_macrosynteny()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># basic usage of plot_oxford_grid : 

orthologs_table &lt;- system.file("extdata","my_orthologs.tab",package="macrosyntR")

my_orthologs &lt;- read.table(orthologs_table,header=TRUE)

plot_oxford_grid(my_orthologs,
                 sp1_label = "B. floridae",
                 sp2_label = "P. echinospica")

# plot a reordered Oxford Grid and color by cluster :

plot_oxford_grid(my_orthologs,
                 sp1_label = "B. floridae",
                 sp2_label = "P. echinospica",
                 reorder = TRUE,
                 color_by = "clust")
 
</code></pre>


</div>