<div class="container">

<table style="width: 100%;"><tr>
<td>load_orthologs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>load orthologs with their genomic coordinates.</h2>

<h3>Description</h3>

<p>Puts together the table of orthologous genes with their genomic coordinates
in the two or more species. It outputs a data.frame shaped as following :
sp1.ID,sp1.Chr,sp1.Start,sp1.End,sp1.Index,sp2.ID,sp2.Chr,sp2.Start,sp2.End,sp2.Index,...
</p>


<h3>Usage</h3>

<pre><code class="language-R">load_orthologs(
  orthologs_table,
  sp1_bed = NULL,
  sp2_bed = NULL,
  bedfiles = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>orthologs_table</code></td>
<td>
<p>character. Full path to the orthologs table (format : geneID_on_species1   geneID_on_species2   geneID_on_speciesN)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sp1_bed</code></td>
<td>
<p>(deprecated) character. Full path to the genomic coordinates of the genes on species1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sp2_bed</code></td>
<td>
<p>(deprecated) character. Full path to the genomic coordinates of the genes on species2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bedfiles</code></td>
<td>
<p>array. List of full paths to the genomic coordinates ordered as in the appearing order of the orthologs_table (BED format)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>dataframe composed of genomic coordinates and relative index of orthologs on both species
</p>


<h3>Examples</h3>

<pre><code class="language-R"># basic usage of load_orthologs for two species :

orthologs_file &lt;- system.file("extdata","Bflo_vs_Pyes.tab",package="macrosyntR")
bedfile_sp1 &lt;- system.file("extdata","Bflo.bed",package="macrosyntR")
bedfile_sp2 &lt;- system.file("extdata","Pyes.bed",package="macrosyntR")


my_orthologs &lt;- load_orthologs(orthologs_table = orthologs_file,
                               bedfiles = c(bedfile_sp1,bedfile_sp2))
# example with 3 species :
orthologs_file &lt;- system.file("extdata","Single_copy_orthologs.tsv",package="macrosyntR")
bedfile_sp3 &lt;- system.file("extdata","Pech.bed",package="macrosyntR")

my_orthologs &lt;- load_orthologs(orthologs_table = orthologs_file,
                               bedfiles = c(bedfile_sp1,bedfile_sp2,bedfile_sp3))


</code></pre>


</div>