<div class="container">

<table style="width: 100%;"><tr>
<td>plot_chord_diagram</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>plot the Macro-synteny as a chord diagram</h2>

<h3>Description</h3>

<p>This is a function to plot the chord diagrams to compare the macro synteny of two or more species.
It requires as input an orthologs_df loaded by load_orthologs()
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_chord_diagram(
  orthologs_df,
  species_labels = NULL,
  species_labels_size = 5,
  color_by = "sp1.Chr",
  custom_color_palette = NULL,
  reorder_chromosomes = TRUE,
  remove_non_linkage_orthologs = TRUE,
  species_labels_hpos = -400,
  label_size = 2,
  ideogram_fill = "white",
  ideogram_color = "black",
  ideogram_height = 4,
  ribbons_curvature = 0.1,
  ribbons_alpha = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>orthologs_df</code></td>
<td>
<p>dataframe. orthologs with genomic coordinates loaded by the load_orthologs()</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>species_labels</code></td>
<td>
<p>list of characters. names of the species to display on the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>species_labels_size</code></td>
<td>
<p>integer. size of the labels (default = 2)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color_by</code></td>
<td>
<p>string. name of the column in the orthologs_df to color the links by (default = "sp1.Chr")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>custom_color_palette</code></td>
<td>
<p>list of characters. palette to use for the coloring of the links following the argument color_by</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reorder_chromosomes</code></td>
<td>
<p>logical. (default = TRUE) tells whether to reorder the chromosomes in clusters as implemented in reorder_macrosynteny()</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove_non_linkage_orthologs</code></td>
<td>
<p>logical. (default = TRUE) tells wether to remove the orthologs that are not within significant linkage groups as calculated by comput_linkage_groups().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>species_labels_hpos</code></td>
<td>
<p>(default =-400)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_size</code></td>
<td>
<p>integer. size of the labels to display on the ideograms (default = 2)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ideogram_fill</code></td>
<td>
<p>character. name of the colors to fill the ideograms with (default = "white")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ideogram_color</code></td>
<td>
<p>character. name of the colors to draw the borders of the ideograms with (default = "black")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ideogram_height</code></td>
<td>
<p>integer. height of the ideograms (default = 4)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ribbons_curvature</code></td>
<td>
<p>float. curvature of the ribbons (default = 0.1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ribbons_alpha</code></td>
<td>
<p>float. alpha of the ribbons (default = 0.5)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A ggplot2 object
</p>


<h3>See Also</h3>

<p><code>load_orthologs()</code>
</p>
<p><code>reorder_macrosynteny()</code>
</p>
<p><code>compute_linkage_groups()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># basic usage of plot_oxford_grid : 

orthologs_table &lt;- system.file("extdata","my_orthologs.tab",package="macrosyntR")

my_orthologs &lt;- read.table(orthologs_table,header=TRUE)

plot_chord_diagram(my_orthologs,species_labels = c("B. flo","P. ech"))

</code></pre>


</div>