<div class="container">

<table style="width: 100%;"><tr>
<td>get_syntenic_genes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>get the syntenic genes as a table</h2>

<h3>Description</h3>

<p>This is a function to extract all the syntenic genes from an orthologs_df. It requires as input an orthologs_df loaded by load_orthologs().
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_syntenic_genes(orthologs_df)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>orthologs_df</code></td>
<td>
<p>dataframe. orthologs with genomic coordinates loaded by load_orthologs()</p>
</td>
</tr></table>
<h3>Value</h3>

<p>dataframe composed of details for each detected syntenic block of genes. It contains the following columns : sp1.Chr, sp1.Start, sp1.End, sp2.Chr, sp2.Start, sp2.End, size, sp1.IDs, sp2.IDs
</p>


<h3>See Also</h3>

<p><code>load_orthologs()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># basic usage of get_syntenic_genes :

orthologs_table &lt;- system.file("extdata","my_orthologs.tab",package="macrosyntR")

my_orthologs &lt;- read.table(orthologs_table,header=TRUE)
                               
my_syntenic_block_of_genes &lt;- get_syntenic_genes(my_orthologs)

</code></pre>


</div>