<div class="container">

<table style="width: 100%;"><tr>
<td>compute_linkage_groups</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute Linkage groups</h2>

<h3>Description</h3>

<p>This is a function to compute the conserved linkage groups shared between two or more species.
It computes the significant associations between chromosomes of all species versus all (pairwise)
using the fischer test implemented in compute_macrosynteny().
It outputs a dataframe shaped as following :
sp1.Chr,sp2.Chr,..., spN.chr,n,LGs
where n is the number of shared orthologs in the group and LGs are the IDs for the linkage groups
</p>


<h3>Usage</h3>

<pre><code class="language-R">compute_linkage_groups(orthologs_df)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>orthologs_df</code></td>
<td>
<p>dataframe. orthologs with genomic coordinates loaded with load_orthologs()</p>
</td>
</tr></table>
<h3>Value</h3>

<p>A dataframe object
</p>


<h3>Examples</h3>

<pre><code class="language-R"># basic usage of compute_linkage_groups: 

orthologs_table &lt;- system.file("extdata","my_orthologs.tab",package="macrosyntR")

my_orthologs &lt;- read.table(orthologs_table,header=TRUE)
                               
my_macrosynteny &lt;- compute_linkage_groups(my_orthologs)

</code></pre>


</div>