<div class="container">

<table style="width: 100%;"><tr>
<td>glmm.gei.meta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
GLMM based meta-analysis of single variant tests for gene-environment interactions
</h2>

<h3>Description</h3>

<p>Use a glmmkin class object from the null GLMM to perform meta-analysis of single variant main effect score test, gene-environment interaction test, or joint test for association with genotypes in a GDS file .gds.
</p>


<h3>Usage</h3>

<pre><code class="language-R">glmm.gei.meta(files, outfile, interaction, SNPID = rep("SNPID", length(files)), 
              CHR = rep("CHR", length(files)), POS = rep("POS", length(files)), 
              Non_Effect_Allele = rep("Non_Effect_Allele", length(files)), 
              Effect_Allele = rep("Effect_Allele", length(files))) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>files</code></td>
<td>

<p>tab or space delimited plain text files (or compressed files that can be recognized by the R function read.table) with at least the following columns: SNPID, CHR, POS, Non_Effect_Allele, Effect_Allele, N_Samples, AF, Beta_Marginal, SE_Beta_Marginal, P_Value_Marginal, Beta_G, Beta_G_sex, SE_Beta_G, SE_Beta_G_sex, Cov_Beta_G_G.sex, P_Value_Interaction, P_Value_Joint. Generally, if each study performs score tests using genotypes in PLINK binary PED format or GDS format, the score test output from glmm.score can be directly used as input files. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile</code></td>
<td>

<p>the output file name.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>interaction</code></td>
<td>

<p>a numeric or a character vector indicating the environmental factors. If a numeric vector, it represents which indices in the order of covariates are the environmental factors; if a character vector, it represents the variable names of the environmental factors.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SNPID</code></td>
<td>

<p>a character vector of SNPID column names in each input file. The length and order must match the length and order of <code>files</code> (default = rep("SNPID", length(files))).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CHR</code></td>
<td>

<p>a character vector of CHR column names in each input file. The length and order must match the length and order of <code>files</code> (default = rep("CHR", length(files))).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>POS</code></td>
<td>

<p>a character vector of POS column names in each input file. The length and order must match the length and order of <code>files</code> (default = rep("POS", length(files))).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Non_Effect_Allele</code></td>
<td>

<p>a character vector of Non_Effect_Allele column names in each input file. The length and order must match the length and order of <code>files</code> (default = rep("Non_Effect_Allele", length(files))).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Effect_Allele</code></td>
<td>

<p>a character vector of Effect_Allele column names in each input file. The length and order must match the length and order of <code>files</code> (default = rep("Effect_Allele", length(files))).
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a data frame containing the following:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>SNPID</code></td>
<td>
<p>SNP name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CHR</code></td>
<td>
<p>chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>POS</code></td>
<td>
<p>physical position.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Non_Effect_Allele</code></td>
<td>
<p>non_effect allele frequency.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Effect_Allele</code></td>
<td>
<p>effect allele frequency.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N_Samples</code></td>
<td>
<p>number of samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>AF</code></td>
<td>
<p>allele frequency.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Beta_Marginal</code></td>
<td>
<p>coefficient estimate for the marginal genetic effect.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SE_Beta_Marginal</code></td>
<td>
<p>standard error of the marginal genetic effect.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>P_Value_Marginal</code></td>
<td>
<p>marginal effect score test p-value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Beta_G</code></td>
<td>
<p>coefficient estimate for the genetic main effect.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Beta_G-*</code></td>
<td>
<p>coefficient estimate for the interaction terms.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SE_Beta_G</code></td>
<td>
<p>model-based standard error associated with the the genetic main effect.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SE_Beta_G-*</code></td>
<td>
<p>mdel-based standard error associated with any GxE or interaction covariate terms.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Cov_Beta_G_G-*</code></td>
<td>
<p>model-based covariance between the genetic main effect and any GxE or interaction covariate terms.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>P_Value_Interaction</code></td>
<td>
<p>gene-environment interaction test p-value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>P_Value_Joint</code></td>
<td>
<p>joint test p-value.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Xinyu Wang, Han Chen, Duy Pham
</p>


<h3>References</h3>

<p>Chen, H., Wang, C., Conomos, M.P., Stilp, A.M., Li, Z., Sofer, T., Szpiro, A.A., Chen, W., Brehm, J.M., Celed√≥n, J.C., Redline, S., Papanicolaou, G.J., Thornton, T.A., Laurie, C.C., Rice, K. and Lin, X. (2016) Control forpopulation structure and relatedness for binary traits in genetic association studies via logistic mixed models. The American Journal of Human Genetics 98, 653-666.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
  infile1 &lt;- system.file("extdata", "meta1.txt", package = "MAGEE")
  infile2 &lt;- system.file("extdata", "meta2.txt", package = "MAGEE")
  infile3 &lt;- system.file("extdata", "meta3.txt", package = "MAGEE")
  infile4 &lt;- system.file("extdata", "meta4.txt", package = "MAGEE")
  infile5 &lt;- system.file("extdata", "meta5.txt", package = "MAGEE")
  outfile &lt;- tempfile()
  glmm.gei.meta(files = c(infile1, infile2, infile3, infile4, infile5),
		outfile = outfile, interaction="sex")
  unlink(outfile)

</code></pre>


</div>