<div class="container">

<table style="width: 100%;"><tr>
<td>pedigrees_all_populate_autosomal</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Populate 1-locus autosomal DNA profile in pedigrees with single-step mutation model.</h2>

<h3>Description</h3>

<p>Populate 1-locus autosomal DNA profile from founder and down in all pedigrees.
Note, that only alleles from ladder is assigned and
that all founders draw type randomly.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pedigrees_all_populate_autosomal(
  pedigrees,
  allele_dist,
  theta,
  mutation_rate,
  progress = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pedigrees</code></td>
<td>
<p>Pedigree list in which to populate genotypes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele_dist</code></td>
<td>
<p>Allele distribution (probabilities) â€“ gets normalised</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theta</code></td>
<td>
<p>Theta correction between 0 and 1 (both included)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutation_rate</code></td>
<td>
<p>Mutation rate between 0 and 1 (both included)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>progress</code></td>
<td>
<p>Show progress</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Note, that pedigrees must first have been inferred by <code>build_pedigrees()</code>.
</p>


<h3>See Also</h3>

<p><code>pedigrees_all_populate_haplotypes_custom_founders()</code> and
<code>pedigrees_all_populate_haplotypes_ladder_bounded()</code>.
</p>


</div>