<div class="container">

<table style="width: 100%;"><tr>
<td>getVarExp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Calculating liability-scale variance explained by genetic risk variants
</h2>

<h3>Description</h3>

<p>These functions calculate the variance explained on the liability scale by the genetic risk variants described by a <code>MangroveORs</code> object. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">getVarExp(ORs, K)
getVarExpSim(ORs,K,iter=1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ORs</code></td>
<td>

<p>An object of class <code>MangroveORs</code>	
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>

<p>The prevalence of the disease.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iter</code></td>
<td>

<p>The number of individuals sampled for variance estimation.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>getVarExp</code> calculates the variance explained analytically, by converting the odds ratios to liability-scale units, and combining them together additively. <code>getVarExpSim</code> simulates genotypes from the allele frequencies, performs risk prediction on them, converts the resulting posteriors to liabilities, and directly measures the variance of the sample.
</p>


<h3>Value</h3>

<p>The proportion of variance explained, as a single numeric value.</p>


<h3>Examples</h3>

<pre><code class="language-R">data(exampleORs)
getVarExp(exampleORs,0.02)
getVarExpSim(exampleORs,0.02)
</code></pre>


</div>