<div class="container">

<table style="width: 100%;"><tr>
<td>manhattanly</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Creates a plotly manhattan plot</h2>

<h3>Description</h3>

<p>Creates an interactive manhattan plot with multiple annotation options
</p>


<h3>Usage</h3>

<pre><code class="language-R">manhattanly(
  x,
  ...,
  col = c("#969696", "#252525"),
  point_size = 5,
  labelChr = NULL,
  suggestiveline = -log10(1e-05),
  suggestiveline_color = "blue",
  suggestiveline_width = 1,
  genomewideline = -log10(5e-08),
  genomewideline_color = "red",
  genomewideline_width = 1,
  highlight = NULL,
  highlight_color = "#00FF00",
  showlegend = FALSE,
  showgrid = FALSE,
  xlab = NULL,
  ylab = "-log10(p)",
  title = "Manhattan Plot"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Can be an object of class <code>manhattanr</code> produced by the
<code>manhattanr</code> function or a <code>data.frame</code> which must contain
at least the following three columns: </p>
 <ul>
<li>
<p>the chromosome
number </p>
</li>
<li>
<p>genomic base-pair position </p>
</li>
<li>
<p>a numeric quantity to plot
such as a p-value or zscore </p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>other parameters passed to <code>manhattanr</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>A character vector indicating the colors of each chromosome. If
the number of colors specified is less than the number of unique
chromosomes, then the elements will be recycled. Can be
<a href="https://www.rapidtables.com/web/color/RGB_Color.html">Hex Codes</a> as
well.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>point_size</code></td>
<td>
<p>A <code>numeric</code> indicating the size of the points on the
plot. Default is 5</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labelChr</code></td>
<td>
<p>A character vector equal to the number of chromosomes
specifying the chromosome labels (e.g., <code>c(1:22, "X", "Y", "MT")</code>).
Default is <code>NULL</code>, meaning that the actual chromosome numbers will be
used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>suggestiveline</code></td>
<td>
<p>Where to draw a "suggestive" line. Default is
<code>-log10(1e-5)</code>. Set to <code>FALSE</code> to disable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>suggestiveline_color</code></td>
<td>
<p>color of "suggestive" line. Only used if
<code>suggestiveline</code> is not set to <code>FALSE</code>. Default is <code>"blue"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>suggestiveline_width</code></td>
<td>
<p>Width of <code>suggestiveline</code>. Default is 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomewideline</code></td>
<td>
<p>Where to draw a "genome-wide sigificant" line. Default
<code>-log10(5e-8)</code>. Set to <code>FALSE</code> to disable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomewideline_color</code></td>
<td>
<p>color of "genome-wide sigificant" line. Only used
if <code>genomewideline</code> is not set to <code>FALSE</code>. Default is
<code>"red"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomewideline_width</code></td>
<td>
<p>Width of <code>genomewideline</code>. Default is 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highlight</code></td>
<td>
<p>A character vector of SNPs in your dataset to highlight.
These SNPs should all be in your dataset. Default is <code>NULL</code> which
means that nothing is highlighted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highlight_color</code></td>
<td>
<p>Color used to highlight points. Only used if
<code>highlight</code> argument has been specified</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showlegend</code></td>
<td>
<p>Should a legend be shown. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showgrid</code></td>
<td>
<p>Should gridlines be shown. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>X-axis label. Default is <code>NULL</code> which means that the label
is automatically determined by the <code>manhattanr</code> function.
Specify here to overwrite the default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>Y-axis label. Default is <code>"-log10(p)"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>Title of the plot. Default is <code>"Manhattan Plot"</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An interactive manhattan plot.
</p>


<h3>Note</h3>

<p>This package is inspired by the
<a href="https://github.com/stephenturner/qqman"><code>qqman</code></a> package. This
package provides additional annotation options and builds on the
<code>plotly</code> <code>d3.js</code> engine. These plots can be included in
Dash apps, Shiny apps, Rmarkdown documents or embedded in websites using
simple HTML code.
</p>


<h3>See Also</h3>

<p><code>manhattanr</code>, <code>HapMap</code>,
<code>significantSNP</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
library(manhattanly)
manhattanly(HapMap)

# highlight SNPs of interest
# 'signigicantSNP' is a character vector of SNPs included in this package
manhattanly(HapMap, snp = "SNP", highlight = significantSNP)

## End(Not run)
</code></pre>


</div>