<div class="container">

<table style="width: 100%;"><tr>
<td>manhplusplot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate the manhattan++ plot</h2>

<h3>Description</h3>

<p>Generate the manhattan++ plot
</p>


<h3>Usage</h3>

<pre><code class="language-R">manhplusplot(infile, outfile, configfile, snpfile, drawastiff = F,
  GWS = 5e-08, FDR = 0.001, MAF = 0.05, chrname = "chr",
  posname = "pos", pvalname = "pvalue", frqname = "maf",
  conseqname = "conseq", showgenes = F, showrsids = F,
  pos.split = 3e+06, pval.split = 0.125, max.pval = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>infile</code></td>
<td>
<p>Input GWAS summary statistics</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile</code></td>
<td>
<p>Output file prefix for the manhattan++ plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>configfile</code></td>
<td>
<p>Configuration file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpfile</code></td>
<td>
<p>Table of SNPs to visualize</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>drawastiff</code></td>
<td>
<p>If TRUE draw a Tiff file, if FALSE draw a PDF file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GWS</code></td>
<td>
<p>Genome wise significance pvalue threshold (5E-8 by default)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FDR</code></td>
<td>
<p>False discovery Rate pvalue threshold (1E-3 by default)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF</code></td>
<td>
<p>Minor Allele Frequency threshold</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrname</code></td>
<td>
<p>Column name for chromosome in GWAS infile</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>posname</code></td>
<td>
<p>Column name for position in GWAS infile</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvalname</code></td>
<td>
<p>Column name for pvalue in GWAS infile</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>frqname</code></td>
<td>
<p>column name for allele frequency in GWAS infile</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>conseqname</code></td>
<td>
<p>column name for variant annotation consequence in GWAS infile</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showgenes</code></td>
<td>
<p>If T shows known genes as bubbles on main manhattan plot, if F show positions of interest as bubbles</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showrsids</code></td>
<td>
<p>If showgenes is T, then show the rsids, rather than genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos.split</code></td>
<td>
<p>The bin lengths for positions</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pval.split</code></td>
<td>
<p>The bin lengths for pvalues</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.pval</code></td>
<td>
<p>The maximum pvalue to display</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For file formats see github page <a href="https://github.com/cgrace1978/manhplot">https://github.com/cgrace1978/manhplot</a>
</p>


<h3>Author(s)</h3>

<p>Chris Grace
</p>


<h3>Examples</h3>

<pre><code class="language-R">

library(manhplot)
## Load R.utils for gzip functionality
library(R.utils)

## unzip the data included with this package
gunzip(system.file("extdata","cad.add.160614_manhformat.txt.gz",package = "manhplot"))

infile&lt;-system.file("extdata","cad.add.160614_manhformat.txt",package = "manhplot")
configfile&lt;-system.file("extdata","config.txt", package = "manhplot")
snpfile&lt;-system.file("extdata","56cad.add.160614.variants.txt", package = "manhplot")

manhplusplot(infile = infile,outfile = file.path(tempdir(), "default-plot"), 
               configfile = configfile, snpfile = snpfile)

## zip the data
gzip(system.file("extdata","cad.add.160614_manhformat.txt",package = "manhplot"))


</code></pre>


</div>