<div class="container">

<table style="width: 100%;"><tr>
<td>setMeanVarCurve</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Set the Mean-Variance Curve of a Set of <code>bioCond</code> Objects</h2>

<h3>Description</h3>

<p>Given a set of <code>bioCond</code> objects, <code>setMeanVarCurve</code>
associates a common mean-variance curve with each of them, assesses the
overall goodness of fit by estimating the
number of prior degrees of freedom, and
accordingly estimates their variance ratio factors
(see also <code>fitMeanVarCurve</code>).
</p>


<h3>Usage</h3>

<pre><code class="language-R">setMeanVarCurve(
  conds,
  predict,
  occupy.only = TRUE,
  method = "NA",
  ratio.var = estimateVarRatio(conds),
  .call = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>conds</code></td>
<td>
<p>A list of <code>bioCond</code> objects, of which at least one
should contain replicate samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>predict</code></td>
<td>
<p>A function representing the mean-variance curve to be
associated with the <code>bioCond</code>s. It should accept a vector of means
and return the predicted variances.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>occupy.only</code></td>
<td>
<p>A logical scalar. If it is <code>TRUE</code> (default), only
occupied intervals are used to estimate the number of prior degrees of
freedom and the variance ratio factors. Otherwise, all intervals are
used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>A character string giving the method for fitting the
mean-variance curve. Used only for constructing the <code>fit.info</code>
fields (see "Value" below).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ratio.var</code></td>
<td>
<p>Backup variance ratio factors of the <code>bioCond</code>s. Only
used when the estimated number of prior degrees of freedom is 0, which
in practice rarely happens.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.call</code></td>
<td>
<p>Never care about this argument.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The specific behavior of this function is pretty much the same as
<code>fitMeanVarCurve</code>, except that the mean-variance curve is
directly specified by users rather than fitted based on the observed
means and variances. Refer to <code>fitMeanVarCurve</code> for a detailed
description of related terms.
</p>
<p>Interestingly, if a positive constant function is supplied as the
mean-variance curve, the resulting statistical model will be rather similar
to the one implemented in the <code>limma</code> package (see also "References").
Notably, using a constant function as the mean-variance curve is
particularly suited to <code>bioCond</code> objects
that have gone through a variance-stabilizing transformation (see
<code>vstBioCond</code> for details and "Examples" below) as well as
<code>bioCond</code>s whose structure matrices have been specifically
designed (see "References").
</p>


<h3>Value</h3>

<p><code>setMeanVarCurve</code> returns the argument list of
<code>bioCond</code> objects, each of which has an added (updated)
<code>fit.info</code> field constructed based on the supplied
mean-variance curve. The field is itself a list consisting of the
following components:
</p>

<dl>
<dt><code>calls</code></dt>
<dd>
<p>The two function calls for associating a mean
variance curve with this <code>bioCond</code> and estimating the related
parameters, respectively. The latter is only present if you have
made an explicit call to some function
(e.g., <code>estimatePriorDf</code>) for performing the parameter
estimation.</p>
</dd>
<dt><code>method</code></dt>
<dd>
<p>Method used for fitting the mean-variance
curve.</p>
</dd>
<dt><code>predict</code></dt>
<dd>
<p>The supplied mean-variance function.</p>
</dd>
<dt><code>mvcID</code></dt>
<dd>
<p>ID of the mean-variance curve.</p>
</dd>
<dt><code>df.prior</code></dt>
<dd>
<p>Number of prior degrees of freedom
assessing the goodness of fit of the mean-variance curve.</p>
</dd>
<dt><code>ratio.var</code></dt>
<dd>
<p>Variance ratio factor of this
<code>bioCond</code>.</p>
</dd>
</dl>
<p>Each <code>bioCond</code> object in the returned list has the same values of
all these components but the <code>ratio.var</code>. <code>mvcID</code> is
automatically generated by the function to label the supplied
mean-variance curve. Each call to <code>setMeanVarCurve</code> results in a
unique <code>mvcID</code>.
</p>
<p>Besides, if there exist <code>bioCond</code> objects that contain only one
ChIP-seq sample, an attribute named <code>"no.rep.rv"</code> will be added to
the returned list, recording the variance ratio factor of no-replicate
conditions.
</p>


<h3>References</h3>

<p>Smyth, G.K., <em>Linear models and empirical bayes methods for
assessing differential expression in microarray experiments.</em> Stat Appl
Genet Mol Biol, 2004. <strong>3</strong>: p. Article3.
</p>
<p>Law, C.W., et al., <em>voom: Precision weights unlock linear model
analysis tools for RNA-seq read counts.</em> Genome Biol, 2014.
<strong>15</strong>(2): p. R29.
</p>
<p>Tu, S., et al., <em>MAnorm2 for quantitatively comparing groups of
ChIP-seq samples.</em> Genome Res, 2021. <strong>31</strong>(1): p. 131-145.
</p>


<h3>See Also</h3>

<p><code>bioCond</code> for creating a <code>bioCond</code> object from a
set of ChIP-seq samples; <code>fitMeanVarCurve</code> for fitting a
mean-variance curve for a set of <code>bioCond</code> objects;
<code>estimateVarRatio</code> for estimating the relative variance
ratio factors of a set of <code>bioCond</code>s; <code>varRatio</code> for a
formal description of variance ratio factor;
<code>estimatePriorDf</code> for estimating the number of prior degrees
of freedom and the corresponding variance ratio factors;
<code>estimatePriorDfRobust</code> for a <em>robust</em> version of
<code>estimatePriorDf</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(H3K27Ac, package = "MAnorm2")
attr(H3K27Ac, "metaInfo")

## Perform differential analysis on bioConds that have gone through a
## variance-stabilizing transformation.

# Perform MA normalization and construct bioConds to represent cell lines
# (i.e., individuals).
norm &lt;- normalize(H3K27Ac, 4, 9)
norm &lt;- normalize(norm, 5:6, 10:11)
norm &lt;- normalize(norm, 7:8, 12:13)
conds &lt;- list(GM12890 = bioCond(norm[4], norm[9], name = "GM12890"),
              GM12891 = bioCond(norm[5:6], norm[10:11], name = "GM12891"),
              GM12892 = bioCond(norm[7:8], norm[12:13], name = "GM12892"))
autosome &lt;- !(H3K27Ac$chrom %in% c("chrX", "chrY"))
conds &lt;- normBioCond(conds, common.peak.regions = autosome)

# Fit a mean-variance curve.
conds &lt;- fitMeanVarCurve(conds, method = "parametric", occupy.only = TRUE)
plotMeanVarCurve(conds, subset = "occupied")

# Apply a variance-stabilizing transformation.
vst_conds &lt;- list(GM12890 = vstBioCond(conds$GM12890))
vst.func &lt;- attr(vst_conds$GM12890, "vst.func")
temp &lt;- matrix(vst.func(as.numeric(conds$GM12891$norm.signal)),
               nrow = nrow(norm))
vst_conds$GM12891 &lt;- bioCond(temp, norm[10:11], name = "GM12891")
temp &lt;- matrix(vst.func(as.numeric(conds$GM12892$norm.signal)),
               nrow = nrow(norm))
vst_conds$GM12892 &lt;- bioCond(temp, norm[12:13], name = "GM12892")

# Associate a constant function with the resulting bioConds as their
# mean-variance curve.
vst_conds &lt;- setMeanVarCurve(vst_conds, function(x) rep_len(1, length(x)),
                             occupy.only = TRUE, method = "constant prior")
plotMeanVarCurve(vst_conds, subset = "occupied")

# Make a comparison between GM12891 and GM12892.
res1 &lt;- diffTest(conds$GM12891, conds$GM12892)
res2 &lt;- diffTest(vst_conds$GM12891, vst_conds$GM12892)

# Examine the consistency of analysis results between using ordinary and
# VSTed signal intensities. Here we map p-values together with observed
# directions of signal changes to the standard normal distribution.
z1 &lt;- qnorm(res1$pval / 2)
z1[res1$Mval &gt; 0] &lt;- -z1[res1$Mval &gt; 0]
z2 &lt;- qnorm(res2$pval / 2)
z2[res2$Mval &gt; 0] &lt;- -z2[res2$Mval &gt; 0]
plot(z1, z2, xlab = "Ordinary", ylab = "VSTed")
abline(a = 0, b = 1, lwd = 2, lty = 5, col = "red")
cor(z1, z2)
cor(z1, z2, method = "sp")

# Simultaneously compare GM12890, GM12891 and GM12892 cell lines.
res1 &lt;- aovBioCond(conds)
res2 &lt;- aovBioCond(vst_conds)

# Examine the consistency of analysis results between using ordinary and
# VSTed signal intensities by mapping p-values to the standard normal
# distribution.
z1 &lt;- qnorm(res1$pval, lower.tail = FALSE)
z1[z1 == Inf] &lt;- 39
z2 &lt;- qnorm(res2$pval, lower.tail = FALSE)
z2[z2 == Inf] &lt;- 39
plot(z1, z2, xlab = "Ordinary", ylab = "VSTed")
abline(a = 0, b = 1, lwd = 2, lty = 5, col = "red")
cor(z1, z2)
cor(z1, z2, method = "sp")

</code></pre>


</div>