<div class="container">

<table style="width: 100%;"><tr>
<td>MAplot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generic MA Plotting</h2>

<h3>Description</h3>

<p><code>MAplot</code> is a generic function used to produce an MA plot. Described
here is the default method for plotting on (normalized) signal intensities
of two ChIP-seq samples (see also <code>normalize</code>).
</p>


<h3>Usage</h3>

<pre><code class="language-R">MAplot(x, ...)

## Default S3 method:
MAplot(
  x,
  y,
  occupy.x,
  occupy.y,
  col = NULL,
  pch = NULL,
  ylim = c(-6, 6),
  xlab = "A value",
  ylab = "M value",
  args.legend = list(x = "topright", legend = c("common", "y specific", "x specific",
    "others")),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x, y</code></td>
<td>
<p><code>x</code> is any R object for which a <code>MAplot</code> method has
been defined. For the default method, <code>x</code> and <code>y</code> are two
numeric vectors representing signal intensities of the 1st and 2nd
samples, respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Arguments to be passed to specific methods for the S3 generic.
For the default method, <code>...</code> represents further arguments to be
passed to <code>plot</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>occupy.x, occupy.y</code></td>
<td>
<p>Two logical vectors of occupancy indicators of the
two samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col, pch</code></td>
<td>
<p>Optional length-4 vectors specifying the colors and point
characters of 4 types of genomic intervals: common peak regions, peak
regions specific to the 2nd sample, peak regions specific to the 1st
sample, and the others. Elements are recycled if necessary.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylim</code></td>
<td>
<p>A length-two vector specifying the plotting range of Y-axis
(i.e., the M value). Each M value falling outside the range will be
shrunk to the corresponding limit. Setting the option to <code>NULL</code>
to suppress this behavior.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab, ylab</code></td>
<td>
<p>Labels for the X and Y axes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>args.legend</code></td>
<td>
<p>A list of arguments to be passed to
<code>legend</code>. You may want to modify the default to
incorporate actual sample names.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>For the default method, <code>MAplot</code> returns <code>NULL</code>.
</p>


<h3>Note</h3>

<p>While it's not strictly required, one typically normalizes the signal
intensities (using <code>normalize</code>) prior to calling this
function.
</p>
<p>Given the typically large number of points to draw, you may want to
use <code>alpha</code> to adjust color transparency if you
intend to specify <code>col</code> explicitly.
</p>


<h3>See Also</h3>

<p><code>normalize</code> for performing an MA normalization on
ChIP-seq samples; <code>MAplot.bioCond</code> for creating an MA plot
on <code>bioCond</code> objects.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(H3K27Ac, package = "MAnorm2")
attr(H3K27Ac, "metaInfo")

## Create MA scatter plots on normalized ChIP-seq samples.

# Perform MA normalization directly on all ChIP-seq samples. Exclude the
# genomic intervals in sex chromosomes from common peak regions, since these
# samples are from different genders.
autosome &lt;- !(H3K27Ac$chrom %in% c("chrX", "chrY"))
norm &lt;- normalize(H3K27Ac, 4:8, 9:13, common.peak.regions = autosome)

# MA plot on two samples from the same individual.
legend &lt;- c("common", "GM12891_2 specific", "GM12891_1 specific", "others")
MAplot(norm[[5]], norm[[6]], norm[[10]], norm[[11]],
       args.legend = list(x = "topright", legend = legend),
       main = "GM12891_rep1 vs. GM12891_rep2")
abline(h = 0, lwd = 2, lty = 5)

# MA plot on two samples from different individuals.
legend &lt;- c("common", "GM12891_1 specific", "GM12890_1 specific", "others")
MAplot(norm[[4]], norm[[5]], norm[[9]], norm[[10]],
       args.legend = list(x = "topright", legend = legend),
       main = "GM12890_rep1 vs. GM12891_rep1")
abline(h = 0, lwd = 2, lty = 5)

</code></pre>


</div>