<div class="container">

<table style="width: 100%;"><tr>
<td>cmbBioCond</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Combine a Set of <code>bioCond</code> Objects into a Single <code>bioCond</code>
</h2>

<h3>Description</h3>

<p>Given a list of <code>bioCond</code> objects, <code>cmbBioCond</code> combines
them into a single <code>bioCond</code>, by treating each <code>bioCond</code> as an
individual ChIP-seq sample. This function is primarily used to handle
ChIP-seq samples associated with a hierarchical structure (see "Details"
for an example).
</p>


<h3>Usage</h3>

<pre><code class="language-R">cmbBioCond(
  conds,
  occupy.num = 1,
  name = "NA",
  weight = NULL,
  strMatrix = NULL,
  meta.info = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>conds</code></td>
<td>
<p>A list of <code>bioCond</code> objects to be combined.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>occupy.num</code></td>
<td>
<p>For each interval, the minimum number of <code>bioCond</code>s
occupying it required for the interval to be considered as occupied by
the newly constructed <code>bioCond</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>Name of the constructed biological condition, used only for
demonstrating a <code>bioCond</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weight</code></td>
<td>
<p>A matrix or data frame specifying the relative precisions of
signal intensities of the constructed <code>bioCond</code>. Must have the same
number of columns as the number of <code>bioCond</code>s in <code>conds</code>. A
vector is interpreted as a matrix having a single row. Note that rows
of <code>weight</code> are recycled if necessary. By default, the same weight
is assigned to each measurement in the constructed <code>bioCond</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>strMatrix</code></td>
<td>
<p>An optional list of symmetric matrices specifying directly
the structure matrix of each genomic interval in the constructed
<code>bioCond</code>. Elements of it are recycled if necessary. This argument,
if set, overrides the <code>weight</code> argument. See <code>bioCond</code>
and <code>setWeight</code> for a detailed description of structure
matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>meta.info</code></td>
<td>
<p>Optional extra information about the <code>bioCond</code> to be
created. If set, the supplied argument is stored in the <code>meta.info</code>
field of returned <code>bioCond</code>, and shall never be used by other tools
in <code>MAnorm2</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Technically, <code>cmbBioCond</code> treats each <code>bioCond</code> object in
<code>conds</code> as a ChIP-seq sample, taking the <code>sample.mean</code> and
<code>occupancy</code> fields stored in each <code>bioCond</code> to represent its
signal intensities and occupancy indicators, respectively. Then, by grouping
these "samples", a new <code>bioCond</code> object is constructed following the
exact routine as described in <code>bioCond</code>. See
<code>bioCond</code> also for a description of the structure of a
<code>bioCond</code> object.
</p>
<p>Notably, ChIP-seq samples contained in these <code>bioCond</code> objects to be
combined are supposed to have been normalized to the same level, so that
these <code>bioCond</code>s are comparable to each other. For this purpose, you
may choose to normalize the ChIP-seq samples involved all together via
<code>normalize</code>, or to normalize the <code>bioCond</code> objects to be
combined via <code>normBioCond</code>.
</p>
<p><code>cmbBioCond</code> is primarily used to deal with ChIP-seq samples sorted
into a hierarchical structure. For example, suppose ChIP-seq samples are
available for multiple male and female individuals, where each individual
is associated with several replicates. To call differential ChIP-seq signals
between males and females, two <code>bioCond</code> objects representing these two
conditions need to be created. One way to do that is to select one ChIP-seq
sample as representative for each individual, and group male and female
samples, respectively. Alternatively, to leverage all available ChIP-seq
samples, a <code>bioCond</code> object could be constructed for each individual,
consisting of the samples of him (her). Then, the <code>bioCond</code>s of
male and female can be separately created by grouping the corresponding
individuals. See also "Examples" below.
</p>


<h3>Value</h3>

<p>A <code>bioCond</code> object, created by combining all the
supplied <code>bioCond</code> objects.
</p>


<h3>See Also</h3>

<p><code>bioCond</code> for creating a <code>bioCond</code> object from a
set of ChIP-seq samples; <code>normalize</code> for performing an MA
normalization on ChIP-seq samples; <code>normBioCond</code> for
normalizing a set of <code>bioCond</code>s; <code>setWeight</code> for
modifying the structure matrices of a <code>bioCond</code> object.
</p>
<p><code>MAplot.bioCond</code> for creating an MA plot on two
<code>bioCond</code> objects; <code>summary.bioCond</code> for
summarizing a <code>bioCond</code>.
</p>
<p><code>fitMeanVarCurve</code> for modeling
the mean-variance dependence across intervals in <code>bioCond</code> objects;
<code>diffTest</code> for comparing two
<code>bioCond</code> objects; <code>aovBioCond</code> for comparing multiple
<code>bioCond</code> objects; <code>varTestBioCond</code> for calling
hypervariable and invariant intervals across ChIP-seq samples contained
in a <code>bioCond</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(H3K27Ac, package = "MAnorm2")
attr(H3K27Ac, "metaInfo")

## Construct two bioConds comprised of the male and female individuals,
## respectively.

# First, normalize ChIP-seq samples separately for each individual (i.e.,
# cell line).
norm &lt;- normalize(H3K27Ac, 4, 9)
norm &lt;- normalize(norm, 5:6, 10:11)
norm &lt;- normalize(norm, 7:8, 12:13)

# Then, construct separately a bioCond for each individual, and perform MA
# normalization on the resulting bioConds. Genomic intervals in sex
# chromosomes are not allowed to be common peak regions, since the
# individuals are of different genders.
conds &lt;- list(GM12890 = bioCond(norm[4], norm[9], name = "GM12890"),
              GM12891 = bioCond(norm[5:6], norm[10:11], name = "GM12891"),
              GM12892 = bioCond(norm[7:8], norm[12:13], name = "GM12892"))
autosome &lt;- !(H3K27Ac$chrom %in% c("chrX", "chrY"))
conds &lt;- normBioCond(conds, common.peak.regions = autosome)

# Finally, group individuals into bioConds based on their genders.
female &lt;- cmbBioCond(conds[c(1, 3)], name = "female")
male &lt;- cmbBioCond(conds[2], name = "male")
summary(female)
summary(male)

</code></pre>


</div>