<div class="container">

<table style="width: 100%;"><tr>
<td>estimatePriorDf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Assess the Goodness of Fit of Mean-Variance Curves</h2>

<h3>Description</h3>

<p>Given a set of <code>bioCond</code> objects of which each has been
associated with a mean-variance curve, <code>estimatePriorDf</code> derives a
common number of
prior degrees of freedom assessing the overall goodness of fit of the
mean-variance curves and accordingly adjusts the variance ratio factor of
each of the <code>bioCond</code>s.
</p>


<h3>Usage</h3>

<pre><code class="language-R">estimatePriorDf(
  conds,
  occupy.only = TRUE,
  return.d0 = FALSE,
  no.rep.rv = NULL,
  .call = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>conds</code></td>
<td>
<p>A list of <code>bioCond</code> objects, of which each has a
<code>fit.info</code> field describing its mean-variance curve (see also
<code>fitMeanVarCurve</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>occupy.only</code></td>
<td>
<p>A logical scalar. If it is <code>TRUE</code> (default), only
occupied intervals are used to estimate the number of
prior degrees of freedom and adjust the variance ratio factors.
Otherwise, all intervals are used (see also "Details").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.d0</code></td>
<td>
<p>A logical scalar. If set to <code>TRUE</code>, the function
simply returns the estimated number of prior degrees of freedom.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>no.rep.rv</code></td>
<td>
<p>A positive real specifying the variance ratio factor of
those <code>bioCond</code>s without replicate samples, if any. By default,
it's set to the geometric mean of variance ratio factors of the other
<code>bioCond</code>s.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.call</code></td>
<td>
<p>Never care about this argument.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>estimatePriorDf</code> borrows part of the modeling strategy implemented in
<code>limma</code> (see "References"). For each <code>bioCond</code> object, the
predicted variances from its mean-variance curve serve as the prior
variances associated with individual intervals.
The common number of prior degrees of freedom
of the supplied <code>bioCond</code>s
quantifies the confidence we have on the associated mean-variance curves.
Intuitively, the closer the observed mean-variance points are
to the curves, the more prior degrees of freedom there will be.
See <code>estimateD0</code> for technical details about the estimation of
number of prior degrees of freedom.
</p>
<p>According to the estimated number of prior degrees of freedom,
<code>estimatePriorDf</code>
separately adjusts the variance ratio factor of each <code>bioCond</code>.
Intrinsically, this process is to scale the mean-variance curve of each
<code>bioCond</code> so that it passes the "middle" of the observed mean-variance
points. See <code>scaleMeanVarCurve</code> for technical details of
scaling a mean-variance curve.
</p>
<p>ChIP-seq signals located in non-occupied intervals result primarily from
background noise, and are therefore associated with less data regularity
than signals in occupied intervals. Involving non-occupied intervals in the
estimation process may result in an under-estimated number of prior degrees
of freedom. Thus, the recommended usage is to set <code>occupy.only</code> to
<code>TRUE</code> (i.e., the default).
</p>
<p>In most cases, the estimation of number of prior degrees of freedom
is automatically
handled when fitting or setting a mean-variance curve, and you don't need to
call this function explicitly (see also <code>fitMeanVarCurve</code> and
<code>setMeanVarCurve</code>). See "Examples"
below for a practical application of this function. Note also that there is
a <em>robust</em> version of this function that uses Winsorized statistics to
protect the estimation procedure against potential outliers (see
<code>estimatePriorDfRobust</code> for details).
</p>


<h3>Value</h3>

<p>By default, <code>estimatePriorDf</code> returns the argument list of
<code>bioCond</code> objects,
with the estimated number of prior degrees of
freedom substituted for the <code>"df.prior"</code> component of each of them.
Besides, their <code>"ratio.var"</code> components have been adjusted
accordingly, and an attribute named <code>"no.rep.rv"</code> is added to the
list if it's ever been used as the variance ratio factor of the
<code>bioCond</code>s without replicate samples. A special case is that the
estimated number of prior degrees of freedom is 0. In this case,
<code>estimatePriorDf</code> won't adjust existing variance ratio factors,
and you may want to use <code>setPriorDfVarRatio</code> to
explicitly specify variance ratio factors.
</p>
<p>If <code>return.d0</code> is set to <code>TRUE</code>, <code>estimatePriorDf</code> simply
returns the estimated number of prior degrees of freedom.
</p>


<h3>References</h3>

<p>Smyth, G.K., <em>Linear models and empirical bayes methods for
assessing differential expression in microarray experiments.</em> Stat Appl
Genet Mol Biol, 2004. <strong>3</strong>: p. Article3.
</p>
<p>Tu, S., et al., <em>MAnorm2 for quantitatively comparing groups of
ChIP-seq samples.</em> Genome Res, 2021. <strong>31</strong>(1): p. 131-145.
</p>


<h3>See Also</h3>

<p><code>bioCond</code> for creating a <code>bioCond</code> object;
<code>fitMeanVarCurve</code> for fitting a mean-variance curve and
using a <code>fit.info</code> field to characterize it;
<code>estimatePriorDfRobust</code> for a <em>robust</em> version of
<code>estimatePriorDf</code>;
<code>setPriorDf</code> for setting the number of
prior degrees of freedom and accordingly
adjusting the variance ratio factors of a set of <code>bioCond</code>s;
<code>diffTest</code> for calling differential
intervals between two <code>bioCond</code> objects.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(H3K27Ac, package = "MAnorm2")
attr(H3K27Ac, "metaInfo")

## Fit a mean-variance curve treating each gender as a biological condition,
## and each individual (i.e., cell line) a replicate.

# First perform the MA normalization and construct bioConds to represent
# individuals.
norm &lt;- normalize(H3K27Ac, 4, 9)
norm &lt;- normalize(norm, 5:6, 10:11)
norm &lt;- normalize(norm, 7:8, 12:13)
conds &lt;- list(GM12890 = bioCond(norm[4], norm[9], name = "GM12890"),
              GM12891 = bioCond(norm[5:6], norm[10:11], name = "GM12891"),
              GM12892 = bioCond(norm[7:8], norm[12:13], name = "GM12892"))
autosome &lt;- !(H3K27Ac$chrom %in% c("chrX", "chrY"))
conds &lt;- normBioCond(conds, common.peak.regions = autosome)

# Group individuals into bioConds based on their genders.
female &lt;- cmbBioCond(conds[c(1, 3)], name = "female")
male &lt;- cmbBioCond(conds[2], name = "male")

# The dependence of variance of ChIP-seq signal intensity across individuals
# on the mean signal intensity is typically not as regular as could be well
# modeled by an explicit parametric form. Better use the local regression to
# adaptively capture the mean-variance trend.
genders &lt;- list(female = female, male = male)
genders1 &lt;- fitMeanVarCurve(genders, method = "local", occupy.only = TRUE)
genders2 &lt;- fitMeanVarCurve(genders, method = "local", occupy.only = FALSE)

# Suppose the local regression is performed using only the occupied genomic
# intervals as input. Good chances are that the extrapolation algorithm
# implemented in the regression method will produce over-estimated variances
# for the non-occupied intervals.
plotMeanVarCurve(genders1, subset = "all")
plotMeanVarCurve(genders2, subset = "all")
plotMeanVarCurve(genders1, subset = "non-occupied")
plotMeanVarCurve(genders2, subset = "non-occupied")

# On the other hand, applying the local regression on all genomic intervals
# may considerably reduce the estimated number of prior degrees of freedom
# associated with the fitted mean-variance curve, as ChIP-seq signals in the
# non-occupied intervals are generally of less data regularity compared with
# those in the occupied intervals.
summary(genders1$female)
summary(genders2$female)

# To split the difference, fit the mean-variance curve on all genomic
# intervals and re-estimate the number of prior degrees of freedom using
# only the occupied intervals, which is also the most recommended strategy
# in practice.
genders3 &lt;- estimatePriorDf(genders2, occupy.only = TRUE)
plotMeanVarCurve(genders3, subset = "all")
plotMeanVarCurve(genders3, subset = "non-occupied")
summary(genders3$female)

</code></pre>


</div>