<div class="container">

<table style="width: 100%;"><tr>
<td>H3K27Ac</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>ChIP-seq Samples for H3K27Ac in Human Lymphoblastoid Cell Lines</h2>

<h3>Description</h3>

<p>Benefiting from the associated ChIP-seq samples, this dataset profiles
H3K27Ac levels along the whole genome for multiple human
lymphoblastoid cell lines, each derived from a separate person.
Specifically, a set of genomic intervals of around the same size (2 kb)
has been systematically selected to thoroughly cover the part of the genome
that is enriched with reads in at least one of the ChIP-seq samples. And
for each of these intervals, this dataset records its raw read count and
enrichment status in each of the samples.
</p>


<h3>Usage</h3>

<pre><code class="language-R">H3K27Ac
</code></pre>


<h3>Format</h3>

<p><code>H3K27Ac</code> is a data frame that records the features of 73,828
non-overlapping genomic intervals regarding the H3K27Ac ChIP-seq signals
in multiple human lymphoblastoid cell lines. It contains the following
variables:
</p>

<dl>
<dt><code>chrom, start, end</code></dt>
<dd>
<p>Genomic coordinate of each interval.
Note that these coordinates are 0-based and correspond to the hg19
genome assembly.</p>
</dd>
<dt><code>cellLine_H3K27Ac_num.read_cnt</code></dt>
<dd>
<p>Each variable whose name
is of this form records the number of reads from a ChIP-seq sample
that fall within each genomic interval. For example,
<code>GM12891_H3K27Ac_2.read_cnt</code> corresponds to the 2nd
biological replicate of a ChIP-seq experiment that targets H3K27Ac
in a cell line named GM12891.</p>
</dd>
<dt><code>cellLine_H3K27Ac_num.occupancy</code></dt>
<dd>
<p>Each variable whose
name is of this form records the enrichment status of each genomic
interval in a ChIP-seq sample. An enrichment status of 1 indicates
that the interval is enriched with reads in the sample; an
enrichment status of 0 indicates otherwise. In practice, enrichment
status of a genomic interval in a certain ChIP-seq sample could be
determined by its overlap with the peaks (see "References" below) of
the sample. Note also that variables of this class correspond to the
variables of raw read counts one by one.</p>
</dd>
</dl>
<p>Each cell line derives from a separate individual of the Caucasian
population. Use <code>attr(H3K27Ac, "metaInfo")</code> to get a data frame
that records meta information about the involved individuals.
</p>


<h3>Source</h3>

<p>Raw sequencing data were obtained from Kasowski et al., 2013 (see
"References" below). Adapters and low-sequencing-quality bases were
trimmed from 3' ends of reads using <code>trim_galore</code>. The
resulting reads were then aligned to the hg19 reference genome
by <code>bowtie</code>. <code>MACS</code> was utilized to call peaks
for each ChIP-seq sample.
</p>
<p>Finally, <code>MAnorm2_utils</code> was exploited to integrate the alignment
results as well as peaks of ChIP-seq samples into this regular
table. <code>MAnorm2_utils</code> is specifically designed to create input
tables of <code>MAnorm2</code>.
See the <a href="https://github.com/tushiqi/MAnorm2_utils">home page of
MAnorm2_utils</a> for more information about it. It has also been uploaded
to the <a href="https://pypi.org/">PyPI repository</a> as a Python package.
</p>


<h3>References</h3>

<p>Zhang, Y., et al., <em>Model-based analysis of ChIP-Seq (MACS).</em> Genome
Biol, 2008. <strong>9</strong>(9): p. R137.
</p>
<p>Kasowski, M., et al., <em>Extensive variation in chromatin states across
humans.</em> Science, 2013. <strong>342</strong>(6159): p. 750-2.
</p>


</div>