<div class="container">

<table style="width: 100%;"><tr>
<td>MAnorm2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>MAnorm2: a Package for Normalizing and Comparing ChIP-seq Samples</h2>

<h3>Description</h3>

<p><code>MAnorm2</code> provides a robust method for normalizing ChIP-seq signals
across individual samples or groups of samples. It also designs a
self-contained system of statistical models for calling differential
ChIP-seq signals between two or more biological conditions as well as
for calling hypervariable ChIP-seq signals across samples.
</p>


<h3>Details</h3>

<p>For a typical differential analysis between two biological conditions
starting with raw read counts, the standard workflow is to
sequentially call <code>normalize</code>, <code>bioCond</code>,
<code>normBioCond</code>,
<code>fitMeanVarCurve</code>, and <code>diffTest</code>
(see the following sections for a rough description of each of these
functions).
Examples given for <code>diffTest</code> provide
specific demonstrations.
<code>MAnorm2</code> is also capable of calling differential ChIP-seq signals
across multiple
biological conditions. See the section below titled "Comparing ChIP-seq
Signals across Multiple Conditions".
</p>
<p>For a hypervariable ChIP-seq analysis
starting with raw read counts, the standard workflow is to
sequentially call <code>normalize</code>, <code>bioCond</code>,
<code>fitMeanVarCurve</code>, <code>estParamHyperChIP</code>, and
<code>varTestBioCond</code>.
Examples given for <code>estParamHyperChIP</code> provide a
specific demonstration.
</p>
<p>The following sections classify the majority of <code>MAnorm2</code> functions
into different utilities. Basically, these sections also represent the order
in which the functions are supposed to be called for a
differential/hypervariable
analysis. For a complete list of <code>MAnorm2</code> functions, use
<code>library(help = "MAnorm2")</code>.
</p>


<h3>Normalizing ChIP-seq Signals across Individual Samples</h3>


<dl>
<dt><code>normalize</code></dt>
<dd>
<p>Perform MA Normalization on a Set of
ChIP-seq Samples</p>
</dd>
<dt><code>normalizeBySizeFactors</code></dt>
<dd>
<p>Normalize ChIP-seq
Samples by Their Size Factors</p>
</dd>
<dt><code>estimateSizeFactors</code></dt>
<dd>
<p>Estimate Size Factors of
ChIP-seq Samples</p>
</dd>
<dt><code>MAplot.default</code></dt>
<dd>
<p>Create an MA Plot on
Two Individual ChIP-seq Samples</p>
</dd>
</dl>
<h3>Creating <code>bioCond</code> Objects to Represent Biological Conditions</h3>


<dl>
<dt><code>bioCond</code></dt>
<dd>
<p>Create a <code>bioCond</code> Object to
Group ChIP-seq Samples</p>
</dd>
<dt><code>setWeight</code></dt>
<dd>
<p>Set the Weights of Signal Intensities
Contained in a <code>bioCond</code></p>
</dd>
<dt><code>normBioCond</code></dt>
<dd>
<p>Perform MA Normalization on a Set
of <code>bioCond</code> Objects</p>
</dd>
<dt><code>normBioCondBySizeFactors</code></dt>
<dd>
<p>Normalize
<code>bioCond</code> Objects by Their Size Factors</p>
</dd>
<dt><code>cmbBioCond</code></dt>
<dd>
<p>Combine a Set of <code>bioCond</code>
Objects into a Single <code>bioCond</code></p>
</dd>
<dt><code>MAplot.bioCond</code></dt>
<dd>
<p>Create an MA Plot on Two
<code>bioCond</code> Objects</p>
</dd>
<dt><code>summary.bioCond</code></dt>
<dd>
<p>Summarize a <code>bioCond</code>
Object</p>
</dd>
</dl>
<h3>Modeling Mean-Variance Trend</h3>


<dl>
<dt><code>fitMeanVarCurve</code></dt>
<dd>
<p>Fit a Mean-Variance Curve</p>
</dd>
<dt><code>setMeanVarCurve</code></dt>
<dd>
<p>Set the Mean-Variance Curve
of a Set of <code>bioCond</code> Objects</p>
</dd>
<dt><code>extendMeanVarCurve</code></dt>
<dd>
<p>Extend the Application
Scope of a Mean-Variance Curve</p>
</dd>
<dt><code>plotMeanVarCurve</code></dt>
<dd>
<p>Plot a Mean-Variance Curve</p>
</dd>
<dt><code>plotMVC</code></dt>
<dd>
<p>Plot a Mean-Variance Curve on a
Single <code>bioCond</code> Object</p>
</dd>
<dt><code>estimateVarRatio</code></dt>
<dd>
<p>Estimate Relative Variance
Ratio Factors of <code>bioCond</code> Objects</p>
</dd>
<dt><code>varRatio</code></dt>
<dd>
<p>Compare Variance Ratio Factors of
Two <code>bioCond</code> Objects</p>
</dd>
<dt><code>distBioCond</code></dt>
<dd>
<p>Quantify the Distance between
Each Pair of Samples in a <code>bioCond</code></p>
</dd>
<dt><code>vstBioCond</code></dt>
<dd>
<p>Apply a Variance-Stabilizing
Transformation to a <code>bioCond</code></p>
</dd>
</dl>
<h3>Assessing the Goodness of Fit of Mean-Variance Curves</h3>


<dl>
<dt><code>estimatePriorDf</code></dt>
<dd>
<p>Assess the Goodness of Fit of
Mean-Variance Curves</p>
</dd>
<dt><code>estimatePriorDfRobust</code></dt>
<dd>
<p>Assess the Goodness of
Fit of Mean-Variance Curves in a Robust Manner</p>
</dd>
<dt><code>setPriorDf</code></dt>
<dd>
<p>Set the Number of Prior Degrees of
Freedom of Mean-Variance Curves</p>
</dd>
<dt><code>setPriorDfRobust</code></dt>
<dd>
<p>The Robust Counterpart of
<code>setPriorDf</code></p>
</dd>
<dt><code>setPriorDfVarRatio</code></dt>
<dd>
<p>Set the Number of Prior
Degrees of Freedom and Variance Ratio Factors</p>
</dd>
<dt><code>estParamHyperChIP</code></dt>
<dd>
<p>The Parameter Estimation
Framework of HyperChIP</p>
</dd>
</dl>
<h3>Calling Differential ChIP-seq Signals between Two Conditions</h3>


<dl>
<dt><code>diffTest.bioCond</code></dt>
<dd>
<p>Compare Two
<code>bioCond</code> Objects</p>
</dd>
<dt><code>MAplot.diffBioCond</code></dt>
<dd>
<p>Create an MA Plot on
Results of Comparing Two <code>bioCond</code> Objects</p>
</dd>
</dl>
<h3>Comparing ChIP-seq Signals across Multiple Conditions</h3>


<dl>
<dt><code>aovBioCond</code></dt>
<dd>
<p>Perform a Moderated Analysis of
Variance on <code>bioCond</code> Objects</p>
</dd>
<dt><code>plot.aovBioCond</code></dt>
<dd>
<p>Plot an <code>aovBioCond</code>
Object</p>
</dd>
<dt><code>varTestBioCond</code></dt>
<dd>
<p>Call Hypervariable and
Invariant Intervals for a <code>bioCond</code></p>
</dd>
<dt><code>plot.varTestBioCond</code></dt>
<dd>
<p>Plot a
<code>varTestBioCond</code> Object</p>
</dd>
</dl>
<h3>Author and Maintainer</h3>

<p>Shiqi Tu &lt;<a href="mailto:tushiqi@picb.ac.cn">tushiqi@picb.ac.cn</a>&gt;
</p>


<h3>References</h3>

<p>Tu, S., et al., <em>MAnorm2 for quantitatively comparing
groups of ChIP-seq samples.</em> Genome Res, 2021.
<strong>31</strong>(1): p. 131-145.
</p>
<p>Chen, H., et al., <em>HyperChIP for identifying hypervariable signals
across ChIP/ATAC-seq samples.</em> bioRxiv, 2021: p. 2021.07.27.453915.
</p>


</div>