<div class="container">

<table style="width: 100%;"><tr>
<td>meanVarParaFit</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Parametrically Fit a Mean-Variance Curve</h2>

<h3>Description</h3>

<p><code>meanVarParaFit</code> fits a mean-variance curve by applying a robust,
gamma-family <code>glm</code> regression, taking advantage of the
form: <code class="reqn">var = c1 + c2 / (2 ^ mean)</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">meanVarParaFit(
  x,
  y,
  weight,
  range.residual = c(1e-04, 15),
  max.iter = 50,
  init.coef = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Two numeric vectors of (sample) means and sample variances,
respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>Two numeric vectors of (sample) means and sample variances,
respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weight</code></td>
<td>
<p>An optional vector of weights to be used in the fitting
procedure. It's typically used when sample variances in <code>y</code> are
associated with different numbers of degrees of freedom.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>range.residual</code></td>
<td>
<p>A length-two vector specifying the range of residuals
of non-outliers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.iter</code></td>
<td>
<p>Maximum number of iteration times allowed during the
fitting procedure.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>init.coef</code></td>
<td>
<p>An optional length-two vector specifying the initial
values of the coefficients.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Whether to print processing messages about iteratively
fitting the mean-variance curve?</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>meanVarParaFit</code> iteratively detects outliers and fits a generalized
linear model on non-outliers. The procedure converges as soon as the set of
outlier points fixes.
</p>
<p>See "References" for the theoretical foundation of the parametric form.
</p>


<h3>Value</h3>

<p>A prediction function which accepts a vector of means and returns
the predicted variances, with an attribute named <code>"coefficients"</code>
attached.
</p>


<h3>References</h3>

<p>Robinson, M.D. and G.K. Smyth, <em>Small-sample estimation of negative
binomial dispersion, with applications to SAGE data.</em> Biostatistics, 2008.
<strong>9</strong>(2): p. 321-32.
</p>
<p>Love, M.I., W. Huber, and S. Anders, <em>Moderated estimation of fold
change and dispersion for RNA-seq data with DESeq2.</em> Genome Biol, 2014.
<strong>15</strong>(12): p. 550.
</p>
<p>Tu, S., et al., <em>MAnorm2 for quantitatively comparing groups of
ChIP-seq samples.</em> Genome Res, 2021. <strong>31</strong>(1): p. 131-145.
</p>


<h3>See Also</h3>

<p><code>meanVarLocalFit</code> for using local regression to fit
a mean-variance curve; <code>fitMeanVarCurve</code> for an interface
to modeling the mean-variance dependence on <code>bioCond</code>
objects; <code>plotMeanVarCurve</code> for plotting a mean-variance
curve.
</p>


</div>