<div class="container">

<table style="width: 100%;"><tr>
<td>estimateD0</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Workhorse Function for Estimating Number of Prior Degrees of Freedom</h2>

<h3>Description</h3>

<p><code>estimateD0</code> underlies other interface functions for assessing
the goodness of fit of an unadjusted mean-variance curve (or a set of
unadjusted mean-variance curves).
</p>


<h3>Usage</h3>

<pre><code class="language-R">estimateD0(z, m)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>z</code></td>
<td>
<p>A list of which each element is a vector of FZ statistics
corresponding to a <code>bioCond</code> object (see also "Details").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>
<p>A vector of numbers of replicates in <code>bioCond</code>
objects. Must correspond to <code>z</code> one by one in the same
order.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For each <code>bioCond</code> object with replicate samples, a vector of
FZ statistics can be deduced from the unadjusted mean-variance curve
associated with it. More specifically, for each genomic interval in a
<code>bioCond</code> with replicate samples, its FZ statistic is defined to be
<code class="reqn">log(t_hat / v0)</code>, where <code class="reqn">t_hat</code> is the observed variance of signal
intensities of the interval, and <code class="reqn">v0</code> is the interval's prior variance
read from the corresponding mean-variance curve.
</p>
<p>Theoretically, each FZ statistic follows a scaled Fisher's Z distribution
plus a constant (since the mean-variance curve is not adjusted yet), and we
can use the sample variance (plus a constant) of the FZ statistics
of each single <code>bioCond</code> to get an estimate of
<code class="reqn">trigamma(d0 / 2)</code>,
where <code class="reqn">d0</code> is the number of prior degrees of freedom
(see also <code>trigamma</code>).
</p>
<p>The final estimate of <code class="reqn">trigamma(d0 / 2)</code> is a weighted mean of estimates
across <code>bioCond</code> objects, with the weights being their respective
numbers of genomic intervals minus 1 that
are used to deduce the FZ statistics.
This should be appropriate, as Fisher's Z distribution is roughly normal
(see also "References"). The weighted mean is similar to the pooled sample
variance in an ANOVA analysis.
</p>
<p>Finally, an estimate of <code class="reqn">d0</code> can be obtained by taking the inverse of
<code class="reqn">trigamma</code> function, which is achieved by applying Newton iteration
to it. Note that <code class="reqn">d0</code> is considered to be infinite if the estimated
<code class="reqn">trigamma(d0 / 2)</code> is less than or equal to 0.
</p>


<h3>Value</h3>

<p>The estimated number of prior degrees of freedom. Note that the
function returns <code>NA</code> if there are not sufficient genomic intervals
for estimating it.
</p>


<h3>References</h3>

<p>Smyth, G.K., <em>Linear models and empirical bayes methods for
assessing differential expression in microarray experiments.</em> Stat Appl
Genet Mol Biol, 2004. <strong>3</strong>: p. Article3.
</p>
<p>Tu, S., et al., <em>MAnorm2 for quantitatively comparing groups of
ChIP-seq samples.</em> Genome Res, 2021. <strong>31</strong>(1): p. 131-145.
</p>


<h3>See Also</h3>

<p><code>bioCond</code> for creating a <code>bioCond</code> object;
<code>fitMeanVarCurve</code> for fitting a mean-variance curve;
<code>estimatePriorDf</code> for an interface to estimating the
number of prior degrees of freedom on <code>bioCond</code> objects;
<code>varRatio</code> for a description of variance ratio factor;
<code>scaleMeanVarCurve</code> for estimating the variance ratio factor
for adjusting a mean-variance curve (or a set of curves).
</p>
<p><code>estimateD0Robust</code> and <code>scaleMeanVarCurveRobust</code>
for estimating number of prior degrees of freedom and variance ratio
factor <em>in a robust manner</em>, respectively.
</p>


</div>