<div class="container">

<table style="width: 100%;"><tr>
<td>varTestBioCond</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Call Hypervariable and Invariant Intervals for a <code>bioCond</code>
</h2>

<h3>Description</h3>

<p>Given a <code>bioCond</code> object with which a mean-variance curve is
associated (see <code>fitMeanVarCurve</code>), <code>varTestBioCond</code> tests
for each genomic interval if the observed variation of its signal intensity
across ChIP-seq samples in the <code>bioCond</code> is significantly greater or
less than is implied by the curve. This function is typically used
in combination with <code>estParamHyperChIP</code> to call hypervariable
and invariant intervals in a <code>bioCond</code> (see also "Examples").
</p>


<h3>Usage</h3>

<pre><code class="language-R">varTestBioCond(cond, min.var = 0, df.prior = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cond</code></td>
<td>
<p>A <code>bioCond</code> object with which a mean-variance curve
has been associated (see also <code>fitMeanVarCurve</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.var</code></td>
<td>
<p>Lower bound of variances read from the mean-variance
curve. Any variance read from the curve less than <code>min.var</code> will be
adjusted to this value. It's primarily used for safely getting the prior
variances and taking into account the practical significance of a signal
variation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>df.prior</code></td>
<td>
<p>Number of prior degrees of freedom associated with the
mean-variance curve. Must be positive.
Can be set to <code>Inf</code> (see "Details").
The default value should be used in most cases, which is extracted from
the <code>"df.prior"</code> component of <code>cond</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>varTestBioCond</code> adopts the modeling strategy implemented in
<code>limma</code> (see "References"),
except that each genomic interval has its own
prior variance, which is read from the mean-variance curve associated with
the <code>bioCond</code> object. The argument <code>df.prior</code> could be
used to specify the common number of degrees of freedom of all the prior
variances, which also effectively assesses the overall goodness of fit of
the mean-variance curve. Technically, <code>varTestBioCond</code> uses
the ratio of the observed variance of each interval to its prior variance as
key statistic, which under the null hypothesis follows an <em>F</em>
distribution, with
its two numbers of degrees of freedom being those of the two variances,
respectively.
(Hence the statistic follows a scaled chi-squared distribution when the
prior df is <code>Inf</code>.) To be noted, the prior df can be empirically
estimated for each
mean-variance curve by specifically designed statistical methods
(see also <code>fitMeanVarCurve</code>, <code>setMeanVarCurve</code>,
<code>estimatePriorDf</code>, and <code>estParamHyperChIP</code>)
and, by default, the function uses the
estimation result to perform the tests. It's highly not recommended to
specify <code>df.prior</code> explicitly when calling <code>varTestBioCond</code>,
unless you know what you are really doing. Besides, <code>varTestBioCond</code>
won't adjust the variance ratio factor of the provided <code>bioCond</code> based
on the specified prior df (see <code>estimatePriorDf</code>
for a description of variance ratio factor).
</p>
<p>Any <code>bioCond</code> object passed to <code>varTestBioCond</code> must contain at
least two ChIP-seq samples; the observed variances of individual
genomic intervals cannot be calculated otherwise.
Besides, a mean-variance curve must be associated with the <code>bioCond</code>
for deducing the prior variances before
<code>varTestBioCond</code> could work. Notably, when fitting a mean-variance
curve for a <code>bioCond</code> object to be passed to <code>varTestBioCond</code>,
it's recommended to pass it alone to <code>fitMeanVarCurve</code> (not
involving other <code>bioCond</code> objects). Also, if you have set
<code>occupy.only</code> to <code>TRUE</code> when calling
<code>fitMeanVarCurve</code>, you should accordingly inspect only the test
results of those genomic intervals that are occupied by the <code>bioCond</code>,
and should re-adjust
<em>p</em>-values within this set of intervals (see "Examples" below).
</p>
<p><code>varTestBioCond</code> can also be used to call hypervariable and invariant
intervals across biological conditions, by first combining multiple
<code>bioCond</code> objects into a single one (see "Examples" below). Note that
ChIP-seq samples in those <code>bioCond</code>s to be combined must be first
normalized to the same level (see <code>cmbBioCond</code> for details).
</p>


<h3>Value</h3>

<p>This function returns an object of <code>class</code>
<code>c("varTestBioCond", "data.frame")</code>, recording the test results for
each genomic interval by each row. The data frame consists of the
following variables:
</p>

<dl>
<dt><code>observed.mean</code></dt>
<dd>
<p>Sample mean of the observed signal
intensities.</p>
</dd>
<dt><code>observed.var</code></dt>
<dd>
<p>Sample variance of the observed signal
intensities.</p>
</dd>
<dt><code>prior.var</code></dt>
<dd>
<p>Prior variance corresponding to the
observed mean signal intensity.</p>
</dd>
<dt><code>fold.change</code></dt>
<dd>
<p>Ratio of <code>observed.var</code> to
<code>prior.var</code>.</p>
</dd>
<dt><code>pval</code></dt>
<dd>
<p>Two sided <em>p</em>-value for the statistical
significance of this fold change.</p>
</dd>
<dt><code>padj</code></dt>
<dd>
<p><em>P</em>-value adjusted for multiple testing with
the <code>"BH"</code> method (see <code>p.adjust</code>), which
controls false discovery rate.</p>
</dd>
</dl>
<p>Row names of the returned data frame inherit from those of
<code>cond$norm.signal</code>. Besides, several attributes are added to the
returned object:
</p>

<dl>
<dt><code>bioCond.name</code></dt>
<dd>
<p>Name of the <code>bioCond</code>
object.</p>
</dd>
<dt><code>mean.var.curve</code></dt>
<dd>
<p>A function representing the
mean-variance curve. It accepts a vector of mean signal intensities
and returns the corresponding prior variances. Note that this
function has incorporated variance ratio factor of the
<code>bioCond</code> and the <code>min.var</code> argument.</p>
</dd>
<dt><code>df</code></dt>
<dd>
<p>A length-2 vector giving the numbers of degrees of
freedom of the observed and prior variances.</p>
</dd>
</dl>
<h3>References</h3>

<p>Smyth, G.K., <em>Linear models and empirical bayes methods for
assessing differential expression in microarray experiments.</em> Stat Appl
Genet Mol Biol, 2004. <strong>3</strong>: p. Article3.
</p>


<h3>See Also</h3>

<p><code>bioCond</code> for creating a <code>bioCond</code> object;
<code>fitMeanVarCurve</code> for fitting a mean-variance curve for
a set of <code>bioCond</code> objects; <code>setMeanVarCurve</code> for
setting the mean-variance curve of a set of <code>bioCond</code>s;
<code>estimatePriorDf</code> for estimating number of prior degrees of
freedom as well as adjusting variance ratio factors accordingly;
<code>estParamHyperChIP</code> for applying the parameter estimation
framework of HyperChIP;
<code>cmbBioCond</code> for combining multiple <code>bioCond</code>s
into a single one.
</p>
<p><code>plot.varTestBioCond</code> for creating a plot to demonstrate a
<code>varTestBioCond</code> object; <code>diffTest</code>
for calling differential intervals between two <code>bioCond</code> objects;
<code>aovBioCond</code> for calling differential intervals across
multiple <code>bioCond</code>s.
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(scales)
data(H3K27Ac, package = "MAnorm2")
attr(H3K27Ac, "metaInfo")

## Call hypervariable and invariant genomic intervals across biological
## replicates of the GM12891 cell line.

# Perform MA normalization and construct a bioCond to represent GM12891.
norm &lt;- normalize(H3K27Ac, 5:6, 10:11)
GM12891 &lt;- bioCond(norm[5:6], norm[10:11], name = "GM12891")

# Fit a mean-variance curve for GM12891 using the parametric method.
GM12891 &lt;- fitMeanVarCurve(list(GM12891), method = "parametric",
                           occupy.only = TRUE)[[1]]
summary(GM12891)
plotMeanVarCurve(list(GM12891), subset = "occupied")

# Assess the observed variances of ChIP-seq signal intensities in GM12891.
res &lt;- varTestBioCond(GM12891)
head(res)

# Inspect only the test results of occupied genomic intervals.
res &lt;- res[GM12891$occupancy, ]
res$padj &lt;- p.adjust(res$pval, method = "BH")
plot(res, padj = 0.2, col = alpha(c("black", "red"), c(0.04, 0.5)))

## Call hypervariable and invariant genomic intervals across cell lines.

# Perform MA normalization and construct bioConds to represent cell lines.
norm &lt;- normalize(H3K27Ac, 4, 9)
norm &lt;- normalize(norm, 5:6, 10:11)
norm &lt;- normalize(norm, 7:8, 12:13)
conds &lt;- list(GM12890 = bioCond(norm[4], norm[9], name = "GM12890"),
              GM12891 = bioCond(norm[5:6], norm[10:11], name = "GM12891"),
              GM12892 = bioCond(norm[7:8], norm[12:13], name = "GM12892"))

# Normalize the cell lines.
autosome &lt;- !(H3K27Ac$chrom %in% c("chrX", "chrY"))
conds &lt;- normBioCond(conds, common.peak.regions = autosome)

# Combine the cell lines into a single bioCond and use local regression to
# adaptively capture the mean-variance trend. Only genomic intervals that
# are occupied by each of the cell lines are considered to be occupied by
# the combined bioCond, which is for avoiding over-estimation of the prior
# variances.
LCLs &lt;- cmbBioCond(conds, occupy.num = length(conds),
                   name = "lymphoblastoid_cell_lines")
LCLs &lt;- fitMeanVarCurve(list(LCLs), method = "local",
                        occupy.only = FALSE)[[1]]
LCLs &lt;- estimatePriorDf(list(LCLs), occupy.only = TRUE)[[1]]
summary(LCLs)
plotMeanVarCurve(list(LCLs), subset = "all")

# Assess the observed variances of ChIP-seq signal intensities across these
# cell lines.
res &lt;- varTestBioCond(LCLs)
head(res)
plot(res, pval = 0.01, col = alpha(c("black", "red"), c(0.04, 0.5)))

# Non-occupied intervals are occupied by some of the cell lines but not all
# of them. These intervals tend to be more variable across the cell lines
# and more significant in the tests than occupied intervals.
f &lt;- !(LCLs$occupancy)
wilcox.test(res$fold.change[f], res$fold.change[!f],
            alternative = "greater")
wilcox.test(res$pval[f], res$pval[!f], alternative = "less")

# Intervals in Y chromosome tend to be more variable across the cell lines
# and more significant in the tests than the other intervals, since the cell
# lines are of different genders.
f &lt;- H3K27Ac$chrom == "chrY"
wilcox.test(res$fold.change[f], res$fold.change[!f],
            alternative = "greater")
wilcox.test(res$pval[f], res$pval[!f], alternative = "less")

# Make a comparison with HyperChIP.
LCLs2 &lt;- estParamHyperChIP(LCLs, occupy.only = FALSE, prob = 0.1)
summary(LCLs)
summary(LCLs2)
res2 &lt;- varTestBioCond(LCLs2)
hist(res$pval, breaks = 100, col = "red")
hist(res2$pval, breaks = 100, col = "red")

</code></pre>


</div>