<div class="container">

<table style="width: 100%;"><tr>
<td>dat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>simulated data for demonstrating the features of marble.</h2>

<h3>Description</h3>

<p>Simulated gene expression data for demonstrating the features of marble.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data("dat")
</code></pre>


<h3>Format</h3>

<p>dat consists of four components: X, Y, E, clin.
</p>


<h3>Details</h3>

<p><strong>The data model for generating Y</strong>
</p>
<p>Use subscript <code class="reqn">i</code> to denote the <code class="reqn">i</code>th subject. Let <code class="reqn">(Y_{i}, X_{i}, E_{i}, clin_{i})</code> (<code class="reqn">i=1,\ldots,n</code>) be
independent and identically distributed random vectors. <code class="reqn">Y_{i}</code> is a continuous response variable representing the
phenotype. <code class="reqn">X_{i}</code> is the <code class="reqn">p</code>â€“dimensional vector of genetic factors. The environmental factors and clinical factors
are denoted as the <code class="reqn">q</code>-dimensional vector <code class="reqn">E_{i}</code> and the <code class="reqn">m</code>-dimensional vector <code class="reqn">clin_{i}</code>, respectively.
The <code class="reqn">\epsilon</code> follows some heavy-tailed distribution. For <code class="reqn">X_{ij}</code> (<code class="reqn">j = 1,\ldots,p</code>), the measurement of the <code class="reqn">j</code>th genetic factor on the <code class="reqn">j</code>th subject, 
considering the following model:
</p>
<p style="text-align: center;"><code class="reqn">Y_{i} = \alpha_{0} + \sum_{k=1}^{q}\alpha_{k}E_{ik}+\sum_{t=1}^{m}\gamma_{t}clin_{it}+\beta_{j}X_{ij}+\sum_{k=1}^{q}\eta_{jk}X_{ij}E_{ik}+\epsilon_{i},</code>
</p>

<p>where <code class="reqn">\alpha_{0}</code> is the intercept, <code class="reqn">\alpha_{k}</code>'s and <code class="reqn">\gamma_{t}</code>'s are the regression coefficients corresponding to effects of environmental and clinical factors, respectively.
The <code class="reqn">\beta_{j}</code>'s and <code class="reqn">\eta_{jk}</code>'s are the regression coefficients of the genetic variants and G<code class="reqn">\times</code>E interactions effects, correspondingly. 
The G<code class="reqn">\times</code>E interactions effects are defined with <code class="reqn">W_{j} = (X_{j}E_{1},\ldots,X_{j}E_{q}).</code> With a slight abuse of notation, denote <code class="reqn">\tilde{W} = W_{j}.</code>  
Denote <code class="reqn">\alpha=(\alpha_{1}, \ldots, \alpha_{q})^{T}</code>, <code class="reqn">\gamma=(\gamma_{1}, \ldots, \gamma_{m})^{T}</code>, <code class="reqn">\beta=(\beta_{1}, \ldots, \beta_{p})^{T}</code>, <code class="reqn">\eta=(\eta_{1}^{T}, \ldots, \eta_{p}^{T})^{T}</code>, <code class="reqn">\tilde{W} = (\tilde{W_{1}}, \dots, \tilde{W_{p}})</code>. 
Then model can be written as 
</p>
<p style="text-align: center;"><code class="reqn">Y_{i} = E_{i}\alpha + clin_{i}\gamma + X_{ij}\beta_{j} + \tilde{W}_{i}\eta_{j} + \epsilon_{i}.</code>
</p>



<h3>See Also</h3>

<p><code>marble</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(dat)
dim(X)
</code></pre>


</div>