<div class="container">

<table style="width: 100%;"><tr>
<td>ParseGTF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Parse gene transfer file (GTF)</h2>

<h3>Description</h3>

<p>Parses the gene transfer file (GTF) for downstream nonsense-mediated decay (NMD) prediction.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ParseGTF(MarvelObject)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.PSI</code> function.</p>
</td>
</tr></table>
<h3>Details</h3>

<p>This function parses the GTF in order to generate new columns for gene IDs, transcript IDs, and transcript type. These information are extracted from the attribute (9th) column for a standard GTF. These information will be used for downstream NMD prediction.
</p>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$NMD$GTF</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- ParseGTF(MarvelObject=marvel.demo)
</code></pre>


</div>