<div class="container">

<table style="width: 100%;"><tr>
<td>CreateMarvelObject.10x</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create Marvel object for droplet-based RNA-sequencing data</h2>

<h3>Description</h3>

<p>Creates an S3 object named <code>Marvel</code> for downstream analysis, specifically for droplet-based RNA-sequencing data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">CreateMarvelObject.10x(
  gene.norm.matrix = NULL,
  gene.norm.pheno = NULL,
  gene.norm.feature = NULL,
  gene.count.matrix = NULL,
  gene.count.pheno = NULL,
  gene.count.feature = NULL,
  sj.count.matrix = NULL,
  sj.count.pheno = NULL,
  sj.count.feature = NULL,
  pca = NULL,
  gtf = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gene.norm.matrix</code></td>
<td>
<p>Sparse matrix. UMI-collapsed, normalised, non-log2-transformed gene expression matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.norm.pheno</code></td>
<td>
<p>Data frame. Sample metadata for annotating <code>gene.norm.matrix</code> columns with cell IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.norm.feature</code></td>
<td>
<p>Data frame. Gene metadata for annotating <code>gene.norm.matrix</code> rows with gene names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.count.matrix</code></td>
<td>
<p>Sparse matrix. UMI-collapsed, non-normalised (raw counts), non-log2-transformed gene expression matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.count.pheno</code></td>
<td>
<p>Data frame. Sample metadata for annotating <code>gene.count.matrix</code> columsn with cell IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.count.feature</code></td>
<td>
<p>Data frame. Gene metadata for annotating <code>gene.count.matrix</code> rows with gene names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sj.count.matrix</code></td>
<td>
<p>Sparse matrix. UMI-collapsed, non-normalised (raw counts), non-log2-transformed splice junction expression matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sj.count.pheno</code></td>
<td>
<p>Data frame. Sample metadata for annotating <code>sj.count.matrix</code> columsn with cell IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sj.count.feature</code></td>
<td>
<p>Data frame. Splice junction metadata for annotating <code>sj.count.matrix</code> rows with splice junction coordinates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pca</code></td>
<td>
<p>Data frame. Coordinates of PCA/tSNE/UMAP.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gtf</code></td>
<td>
<p>Data frame. GTF used in cellranger. Will be used for annotating splice junctions downstream.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class S3.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Retrieve, observe format of pre-saved input files
marvel.demo.10x.raw &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.raw.rds",
                               package="MARVEL")
                               )
# Gene expression (Normalised)
    # Matrix
    df.gene.norm &lt;- marvel.demo.10x.raw$gene.norm.matrix
    df.gene.norm[1:5, 1:5]

    # phenoData
    df.gene.norm.pheno &lt;- marvel.demo.10x.raw$sample.metadata
    head(df.gene.norm.pheno)

    # featureData
    df.gene.norm.feature &lt;- data.frame("gene_short_name"=rownames(df.gene.norm),
                                       stringsAsFactors=FALSE
                                       )
    head(df.gene.norm.feature)

# Gene expression (Counts)
    # Matrix
    df.gene.count &lt;- marvel.demo.10x.raw$gene.count.matrix
    df.gene.count[1:5, 1:5]

    # phenoData
    df.gene.count.pheno &lt;- data.frame("cell.id"=colnames(df.gene.count),
                                       stringsAsFactors=FALSE
                                       )
    head(df.gene.count.pheno)

    # featureData
    df.gene.count.feature &lt;- data.frame("gene_short_name"=rownames(df.gene.count),
                                       stringsAsFactors=FALSE
                                       )
    head(df.gene.count.feature)

# SJ (Counts)
    # Matrix
    df.sj.count &lt;- marvel.demo.10x.raw$sj.count.matrix
    df.sj.count[1:5, 1:5]

    # phenoData
    df.sj.count.pheno &lt;- data.frame("cell.id"=colnames(df.sj.count),
                                     stringsAsFactors=FALSE
                                     )
    head(df.sj.count.pheno)

    # featureData
    df.sj.count.feature &lt;- data.frame("coord.intron"=rownames(df.sj.count),
                                       stringsAsFactors=FALSE
                                       )
    head(df.sj.count.feature)

# tSNE coordinates
df.coord &lt;- marvel.demo.10x.raw$pca
head(df.coord)

# GTF
gtf &lt;- marvel.demo.10x.raw$gtf
head(gtf)

# Create MARVEL object
marvel.demo.10x &lt;- CreateMarvelObject.10x(gene.norm.matrix=df.gene.norm,
                     gene.norm.pheno=df.gene.norm.pheno,
                     gene.norm.feature=df.gene.norm.feature,
                     gene.count.matrix=df.gene.count,
                     gene.count.pheno=df.gene.count.pheno,
                     gene.count.feature=df.gene.count.feature,
                     sj.count.matrix=df.sj.count,
                     sj.count.pheno=df.sj.count.pheno,
                     sj.count.feature=df.sj.count.feature,
                     pca=df.coord,
                     gtf=gtf
                     )
</code></pre>


</div>