<div class="container">

<table style="width: 100%;"><tr>
<td>FindPTC.RI.NegStrand</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Find premature terminal codon (PTC) for retained-intron (RI) located on the negative strand of the transcript</h2>

<h3>Description</h3>

<p>Finds PTC(s) introduced by alternative exons into protein-coding transcripts.
</p>


<h3>Usage</h3>

<pre><code class="language-R">FindPTC.RI.NegStrand(MarvelObject, tran_id, gene_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>MarvelObject</code></td>
<td>
<p>S3 object generated from <code>CompareValues.PSI</code> and <code>ParseGTF</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tran_id</code></td>
<td>
<p>Character string. Vector of <code>tran_id</code> to look for PTCs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_id</code></td>
<td>
<p>Character string. Vector of <code>gene_id</code> corresponding to the <code>tran_id</code> argument.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame of transcripts containing splicing events meeting the <code>psi.de.sig</code> and <code>psi.de.diff</code> criteria are categorised based on the presence or absence of PTCs.
</p>


<h3>Examples</h3>

<pre><code class="language-R">marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define relevant event type
results &lt;- marvel.demo$DE$PSI$Table[["ad"]]
index.1 &lt;- which(results$event_type=="RI")
index.2 &lt;- grep(":-@", results$tran_id, fixed=TRUE)
index &lt;- intersect(index.1, index.2)
results &lt;- results[index, ]
tran_id &lt;- results$tran_id[1]
gene_id &lt;- results$gene_id[1]

# Find PTC
results &lt;- FindPTC.RI.NegStrand(MarvelObject=marvel.demo,
                                tran_id=tran_id,
                                gene_id=gene_id
                                )

# Check output
head(results)
</code></pre>


</div>