<div class="container">

<table style="width: 100%;"><tr>
<td>adhocGene.PlotSJPosition.10x</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plots the locations of specified splice junction relative to isoforms</h2>

<h3>Description</h3>

<p>Plots the locations of specified splice junction relative to isoforms. List of isoforms are retrieved from GTF.
</p>


<h3>Usage</h3>

<pre><code class="language-R">adhocGene.PlotSJPosition.10x(
  MarvelObject,
  coord.intron,
  coord.intron.ext = 50,
  rescale_introns = FALSE,
  show.protein.coding.only = TRUE,
  anno.label.size = 3,
  anno.colors = c("black", "gray", "red")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CheckAlignment.10x</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coord.intron</code></td>
<td>
<p>Character string. Coordinates of splice junction whose splice junction will be plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coord.intron.ext</code></td>
<td>
<p>Numeric value. Number of bases to extend the splice junction start and end coordinates into the exons. Helpful to enhance splice junction locations on the plot. Default is <code>50</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rescale_introns</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, the intron length will be shorten. Helpful when introns are very long and focus visualisation of exons and splice junctions. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.protein.coding.only</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code> (default), only protein-coding isoforms will be displayed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anno.label.size</code></td>
<td>
<p>Numeric value. Font size of isoform ID labels. Default is <code>3</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anno.colors</code></td>
<td>
<p>Vector of character strings. Colors for non-coding UTRs, coding exons, and splice junctions, respectively. Default is <code>c("black", "gray", "red")</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class S3 with new slots <code>MarvelObject$adhocGene$SJPosition$Plot</code>, <code>MarvelObject$adhocGene$SJPosition$metadata</code>, <code>MarvelObject$adhocGene$SJPosition$exonfile</code>, and <code>MarvelObject$adhocGene$SJPosition$cdsfile</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

marvel.demo.10x &lt;- adhocGene.PlotSJPosition.10x(
                        MarvelObject=marvel.demo.10x,
                        coord.intron="chr1:100:1001",
                        rescale_introns=FALSE,
                        show.protein.coding.only=TRUE,
                        anno.label.size=1.5
                        )
</code></pre>


</div>