<div class="container">

<table style="width: 100%;"><tr>
<td>PlotDEValues.Exp.Spliced</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot differential gene expression analysis of differentially spliced genes</h2>

<h3>Description</h3>

<p>Volcano plot of differential splicing analysis results based on differentially spliced genes between 2 groups of cells. x-axis represents the log2 fold change in gene expression. y-axis represents the adjusted p-values.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PlotDEValues.Exp.Spliced(
  MarvelObject,
  method,
  psi.pval = 0.1,
  psi.delta = 0,
  gene.pval = 0.1,
  gene.log2fc = 0.5,
  point.size = 1,
  anno = FALSE,
  anno.gene_short_name = NULL,
  label.size = 2.5,
  y.upper.offset = 5,
  xlabel.size = 8
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>MarvelObject</code></td>
<td>
<p>S3 object generated from <code>CompareValues</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>(Vector of) Character string(s). The <code>method</code> specified in <code>CompareValues</code> function when <code>level</code> option set to <code>"splicing"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>psi.pval</code></td>
<td>
<p>Numeric value. The adjusted p-value from differential splicing analysis, below which, the splicing event is considered differentially spliced. Default is <code>0.1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>psi.delta</code></td>
<td>
<p>Numeric value. The absolute differences in average PSI value between two cell groups from differential splicing analysis, above which, the splicing event is considered differentially spliced.  Default is <code>0</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.pval</code></td>
<td>
<p>Numeric value. The adjusted p-value from differential gene expression analysis, below which, the gene is considered differentially expressed. Default is <code>0.1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.log2fc</code></td>
<td>
<p>Numeric value. The absolute log2 fold change in gene expression betwene two cell groups from differential splicing analysis, above which, the gene is considered differentially expressed. Default is <code>0.5</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>point.size</code></td>
<td>
<p>Numeric value. Size of data points. Default is <code>1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anno</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, the specific gene names will be annotated on the plot as defined in <code>anno.gene_short_name</code> option.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anno.gene_short_name</code></td>
<td>
<p>Vector of character strings. When <code>anno</code> set to <code>TRUE</code>, the gene names to be annotated on the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label.size</code></td>
<td>
<p>Numeric value. Only applicable if <code>anno</code> set to TRUE. Size of the gene name labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y.upper.offset</code></td>
<td>
<p>Numeric value. The value in -log10(p-value) to increase the upper limit of the y-axis. To be used when <code>anno</code> set to TRUE so that gene labels will not be truncated at the upper limit of the y-axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlabel.size</code></td>
<td>
<p>Numeric value. Font size of the xtick labels. Default is <code>8</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class S3 with new slots <code>MarvelObject$DE$Exp.Spliced$Table</code>, <code>MarvelObject$DE$Exp.Spliced$Summary</code>, and <code>MarvelObject$DE$Exp.Spliced$Plot</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- PlotDEValues.Exp.Spliced(MarvelObject=marvel.demo,
                                        method="ad",
                                        psi.pval=0.1,
                                        psi.delta=0,
                                        gene.pval=0.1,
                                        gene.log2fc=0.5
                                        )
# Check output
marvel.demo$DE$Exp.Spliced$Summary
marvel.demo$DE$Exp.Spliced$Plot
</code></pre>


</div>