<div class="container">

<table style="width: 100%;"><tr>
<td>PlotDEValues.SJ.10x</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot differential splice junction analysis results</h2>

<h3>Description</h3>

<p>Volcano plot of results from differential splice junction analysis. x-axis represents the average normalised gene expression across the two cell groups. y-axis represents the differences or log2 fold change between the two cell groups.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PlotDEValues.SJ.10x(
  MarvelObject,
  pval = 0.05,
  log2fc = NULL,
  delta = 5,
  min.gene.norm = 0,
  anno = FALSE,
  anno.coord.intron = NULL,
  label.size = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.Genes.10x</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pval</code></td>
<td>
<p>Numeric value. p-value, below which, the splice junction is considered differentially spliced. To be used in conjunction with <code>log2fc</code>, <code>delta</code>, and <code>min.gene.norm</code>. Default is <code>0.05</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log2fc</code></td>
<td>
<p>Numeric value. Absolute log2 fold change, above which, the splice junction is considered differentially spliced. This option should be <code>NULL</code> if <code>delta</code> has been specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>delta</code></td>
<td>
<p>Numeric value. Absolute differences in average PSI values between the two cell groups, above which, the splice junction is considered differentially spliced. This option should be <code>NULL</code> if <code>log2fc</code> has been specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.gene.norm</code></td>
<td>
<p>Numeric value. The average normalised gene expression across the two cell groups above which the splice junction is considered differentially spliced. Default is <code>0</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anno</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, user-specific spliced genes in <code>anno.coord.intron</code> will be annotated on the plot. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anno.coord.intron</code></td>
<td>
<p>Vector of character strings. If <code>anno</code> set to <code>TRUE</code>, splice junction coordinates specified here will be annotated on the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label.size</code></td>
<td>
<p>Numeric value. If <code>anno</code> set to <code>TRUE</code>, the font size of the annotations on the plot will be adjusted to the size specified here. Default is <code>2</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class S3 with a new slots <code>MarvelObject$DE$SJ$VolcanoPlot$SJ$Plot</code> and <code>MarvelObject$DE$SJ$VolcanoPlot$SJ$Data</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

marvel.demo.10x &lt;- PlotDEValues.SJ.10x(
                        MarvelObject=marvel.demo.10x,
                        pval=0.05,
                        delta=5,
                        min.gene.norm=1.0,
                        anno=FALSE
                        )

# Check outputs
marvel.demo.10x$DE$SJ$VolcanoPlot$SJ$Plot
head(marvel.demo.10x$DE$SJ$VolcanoPlot$SJ$Data)
</code></pre>


</div>