<div class="container">

<table style="width: 100%;"><tr>
<td>IsoSwitch.10x</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Classify gene-splicing relationship</h2>

<h3>Description</h3>

<p>Classify gene-splicing relative changes to each other from cell group 1 to group 2. Classifications are coordinated, opposing, isoform-switching, and complex. In coordinated relationship, both gene and splicing changes in the same direction from cell group 1 to group 2. In opposing relationship, gene changes in the opposite direction relative to splicing from cell group 1 to group 2. In isoform-switching, there is differential splice junction usage without differential expression of the corresponding gene between cell group 1 and group 2. Complex relationship involves genes with both coordinated and opposing relationships with splicing. Only differentially spliced junctions are included for analysis here.
</p>


<h3>Usage</h3>

<pre><code class="language-R">IsoSwitch.10x(
  MarvelObject,
  pval.sj = 0.05,
  log2fc.sj = NULL,
  delta.sj = 5,
  min.gene.norm = 0,
  pval.adj.gene = 0.05,
  log2fc.gene = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.Genes.10x</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pval.sj</code></td>
<td>
<p>Numeric value. p-value from differential splicing analysis, below which, the splice junction is considered differentially spliced. Default is <code>0.05</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log2fc.sj</code></td>
<td>
<p>Numeric value. Absolute log2 fold change from differential splicing analysis, above which, the splice junction is considered differentially spliced. This option should be <code>NULL</code> if <code>delta.sj</code> has been specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>delta.sj</code></td>
<td>
<p>Numeric value. Absolute difference in average PSI values between the two cell groups, above which, the splice junction is considered differentially spliced. This option should be <code>NULL</code> if <code>log2fc.sj</code> has been specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.gene.norm</code></td>
<td>
<p>Numeric value. The average normalised gene expression across the two cell groups above which the splice junction is considered differentially spliced. Default is <code>0</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pval.adj.gene</code></td>
<td>
<p>Numeric value. Adjusted p-value from differential gene expression analysis, below which, the gene is considered differentially expressed. Default is <code>0.05</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log2fc.gene</code></td>
<td>
<p>Numeric value. Absolute log2 fold change from differential gene expression analysis, above which, the gene is considered differentially expressed. This option should be <code>NULL</code> if <code>delta.sj</code> has been specified.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class S3 containing new slots <code>MarvelObject$SJ.Gene.Cor$Data</code>, <code>MarvelObject$SJ.Gene.Cor$Proportion$Plot</code>, and <code>MarvelObject$SJ.Gene.Cor$Proportion$Table</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

marvel.demo.10x &lt;- IsoSwitch.10x(
                        MarvelObject=marvel.demo.10x,
                        pval.sj=0.05,
                        delta.sj=5,
                        min.gene.norm=1.0,
                        pval.adj.gene=0.05,
                        log2fc.gene=0.5
                        )

# Check outputs
marvel.demo.10x$SJ.Gene.Cor$Proportion$Plot
marvel.demo.10x$SJ.Gene.Cor$Proportion$Table
cols &lt;- c("coord.intron", "gene_short_name", "cor.complete")
head(marvel.demo.10x$SJ.Gene.Cor$Data[,cols])
</code></pre>


</div>