<div class="container">

<table style="width: 100%;"><tr>
<td>PlotDEValues</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot differential splicing and gene expression analysis results</h2>

<h3>Description</h3>

<p>Volcano plot of differential splicing and gene expression analysis results. This is a wrapper function for <code>PlotDEValues.PSI.Mean</code>, <code>PlotDEValues.Exp.Global</code>, and <code>PlotDEValues.Exp.Spliced</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PlotDEValues(
  MarvelObject,
  method = NULL,
  pval,
  level,
  delta = NULL,
  log2fc = NULL,
  psi.pval = NULL,
  psi.delta = NULL,
  gene.pval = NULL,
  gene.log2fc = NULL,
  point.size = 1,
  xlabel.size = 8,
  point.alpha = 1,
  anno = FALSE,
  anno.gene_short_name = NULL,
  anno.tran_id = NULL,
  label.size = 2.5,
  y.upper.offset = 5,
  event.types = c("SE", "MXE", "RI", "A5SS", "A3SS", "AFE", "ALE"),
  event.types.colors = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Character string. The statistical method used for differential splicing analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pval</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"splicing.mean"</code>, <code>"splicing.distance"</code>, and <code>"gene.global"</code>. Adjusted p-value below which the splcing events or genes are considered as statistically significant and will consequently be color-annotated on the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>level</code></td>
<td>
<p>Character string. Indicate <code>"splicing.distance"</code> if the percent spliced-in (PSI) values' distribution was previously tested between 2 groups of cells using the <code>CompareValues</code> function. Statistical tests for distribution include Kolmogorov-Smirnov, Kuiper, and Anderson-Darling test. Indicate <code>"splicing.mean"</code> or <code>gene</code> if the PSI or gene expression values' mean was previously tested between 2 groups of cells using the <code>CompareValues</code> function. Statistical tests for comparing mean are t-test and Wilcoxon rank-sum test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>delta</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"splicing.mean"</code>. The positive (and negative) value specified above (and below) which the splicing events are considered to be statistically significant and will consequently be color-annotated on the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log2fc</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"gene.global"</code>. The positive (and negative) value specified above (and below) which the genes are considered to be statistically significant and will consequently be color-annotated on the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>psi.pval</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"gene.spliced"</code>. The adjusted p-value from differential splicing analysis, below which, the splicing event is considered differentially spliced. Default is <code>0.1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>psi.delta</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"gene.spliced"</code>. The absolute differences in average PSI value between two cell groups from differential splicing analysis, above which, the splicing event is considered differentially spliced.  Default is <code>0</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.pval</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"gene.spliced"</code>. The adjusted p-value from differential gene expression analysis, below which, the gene is considered differentially expressed. Default is <code>0.1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.log2fc</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"gene.spliced"</code>. The absolute log2 fold change in gene expression betwene two cell groups from differential splicing analysis, above which, the gene is considered differentially expressed. Default is <code>0.5</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>point.size</code></td>
<td>
<p>Numeric value. Size of data points. Default is <code>1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlabel.size</code></td>
<td>
<p>Numeric value. Font size of the xtick labels. Default is <code>8</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>point.alpha</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"splicing.mean.g2vsg1"</code>. Transpancy of data points. Default is <code>1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anno</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, the specific gene names or splicing events will be annotated on the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anno.gene_short_name</code></td>
<td>
<p>Vector of character strings. When <code>anno</code> set to <code>TRUE</code>, the gene names to be annotated on the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anno.tran_id</code></td>
<td>
<p>Vector of character strings. When <code>anno</code> set to <code>TRUE</code>, the coordinates of the splicing events to be annotated on the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label.size</code></td>
<td>
<p>Numeric value. Only applicable if <code>anno</code> set to TRUE. Size of the gene name labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y.upper.offset</code></td>
<td>
<p>Numeric value. The value in -log10(p-value) to increase the upper limit of the y-axis. To be used when <code>anno</code> set to TRUE so that gene labels will not be truncated at the upper limit of the y-axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>event.types</code></td>
<td>
<p>Vector of character string(s). Only applicable when <code>level</code> set to <code>"splicing.mean.g2vsg1"</code>. The specific splicing event to plot. May take any one or more of the following values <code>"SE"</code>, <code>"MXE"</code>, <code>"RI"</code>, <code>"A5SS"</code>, <code>"A3SS"</code>, <code>"AFE"</code>, and  <code>"ALE"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>event.types.colors</code></td>
<td>
<p>Vector of character string(s). Only applicable when <code>level</code> set to <code>"splicing.mean.g2vsg1"</code>. Customise colors as per splicing event type specified in <code>event.types</code> option. Should be of same length as <code>event.types</code> option.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$DE$PSI$Plot[["method"]]</code> when level set to <code>"splicing.mean"</code> or <code>"splicing.distance"</code> or <code>MarvelObject$DE$Exp.Global$Table</code> and <code>MarvelObject$DE$Exp.Global$Plot</code> when level set to <code>"gene.global"</code> or <code>MarvelObject$DE$Exp.Spliced$Table</code> and <code>MarvelObject$DE$Exp.Spliced$Plot</code> when level set to <code>"gene.spliced"</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- PlotDEValues(MarvelObject=marvel.demo,
                            method="ad",
                            pval=0.10,
                            level="splicing.distance"
                            )

# Check output
marvel.demo$DE$PSI$Plot[["ad"]]
</code></pre>


</div>