<div class="container">

<table style="width: 100%;"><tr>
<td>adhocGene.PlotDEValues.10x</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot differential splice junction analysis results for a specified gene</h2>

<h3>Description</h3>

<p>Scatterplot of results from differential gene and splice junction analysis. x-axis represents the gene expression log2 fold change between the different pairs of cell groups. y-axis represents the PSI differences or log2 fold change between the different pairs of cell groups.
</p>


<h3>Usage</h3>

<pre><code class="language-R">adhocGene.PlotDEValues.10x(
  MarvelObject,
  coord.intron,
  log2fc.gene = 0.5,
  delta.sj = 5,
  label.size = 2,
  point.size = 2,
  xmin = NULL,
  xmax = NULL,
  ymin = NULL,
  ymax = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>adhocGene.DE.Gene.10x</code> and <code>adhocGene.DE.PSI.10x</code> functions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coord.intron</code></td>
<td>
<p>Character string. Coordinates of splice junction whose differential splice junction results will be plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log2fc.gene</code></td>
<td>
<p>Numeric value. Absolute log2 fold change, above which, the gene is considered differentially expressed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>delta.sj</code></td>
<td>
<p>Numeric value. Absolute differences in average PSI values between the two cell groups, above which, the splice junction is considered differentially spliced.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label.size</code></td>
<td>
<p>Numeric value. The font size of the group comparison labels on the plot will be adjusted to the size specified here. Default is <code>2</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>point.size</code></td>
<td>
<p>Numeric value. Size of data points. Default is <code>2</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xmin</code></td>
<td>
<p>Numeric value. Minimum x-axis value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xmax</code></td>
<td>
<p>Numeric value. Maximum x-axis value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ymin</code></td>
<td>
<p>Numeric value. Minimum y-axis value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ymax</code></td>
<td>
<p>Numeric value. Maximum y-axis value.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class S3 with a new slots <code>MarvelObject$adhocGene$DE$VolcanoPlot$Plot</code> and <code>MarvelObject$adhocGene$DE$VolcanoPlot$Table</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

# Define SJ to plot
coord.intron &lt;- marvel.demo.10x$adhocGene$DE$PSI$Data$coord.intron[1]

# Plot SJ vs gene
marvel.demo.10x &lt;- adhocGene.PlotDEValues.10x(
                        MarvelObject=marvel.demo.10x,
                        coord.intron=coord.intron,
                        log2fc.gene=0.5,
                        delta.sj=5,
                        label.size=2,
                        point.size=2,
                        xmin=-2.0,
                        xmax=2.0,
                        ymin=-25,
                        ymax=25
                        )

# Check output
marvel.demo.10x$adhocGene$DE$VolcanoPlot$Plot
</code></pre>


</div>