<div class="container">

<table style="width: 100%;"><tr>
<td>SlicedData-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Class <code>SlicedData</code> for storing large matrices</h2>

<h3>Description</h3>

<p>This class is created for fast and memory efficient manipulations with
large datasets presented in matrix form.
It is used to load, store, and manipulate large datasets,
e.g. genotype and gene expression matrices.
When a dataset is loaded,
it is sliced in blocks of 1,000 rows (default size).
This allows imputing, standardizing, and performing other operations
with the data with minimal memory overhead.
</p>


<h3>Usage</h3>

<pre><code class="language-R"># x[[i]] indexing allows easy access to individual slices.
# It is equivalent to x$GetSlice(i) and x$SetSlice(i,value)
## S4 method for signature 'SlicedData'
x[[i]]
## S4 replacement method for signature 'SlicedData'
x[[i]] &lt;- value

# The following commands work as if x was a simple matrix object
## S4 method for signature 'SlicedData'
nrow(x)
## S4 method for signature 'SlicedData'
ncol(x)
## S4 method for signature 'SlicedData'
dim(x)
## S4 method for signature 'SlicedData'
rownames(x)
## S4 method for signature 'SlicedData'
colnames(x)
## S4 replacement method for signature 'SlicedData'
rownames(x) &lt;- value
## S4 replacement method for signature 'SlicedData'
colnames(x) &lt;- value

# SlicedData object can be easily transformed into a matrix
# preserving row and column names
## S4 method for signature 'SlicedData'
as.matrix(x)

# length(x) can be used in place of x$nSlices()
# to get the number of slices in the object
## S4 method for signature 'SlicedData'
length(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>

<p><code>SlicedData</code> object.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>i</code></td>
<td>

<p>Number of a slice.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>value</code></td>
<td>

<p>New content for the slice / new row or column names.
</p>
</td>
</tr>
</table>
<h3>Extends</h3>

<p><code>SlicedData</code> is a reference classes (<code>envRefClass</code>).
Its methods can change the values of the fields of the class.
</p>


<h3>Fields</h3>


<dl>
<dt>
<code>dataEnv</code>:</dt>
<dd>
<p><code>environment</code>. Stores the slices of the data matrix.
The slices should be accessed via
<code>getSlice()</code> and <code>setSlice()</code> methods.
</p>
</dd>
<dt>
<code>nSlices1</code>:</dt>
<dd>
<p><code>numeric</code>. Number of slices. For internal use.
The value should be access via <code>nSlices()</code> method.
</p>
</dd>
<dt>
<code>rowNameSlices</code>:</dt>
<dd>
<p><code>list</code>. Slices of row names.
</p>
</dd>
<dt>
<code>columnNames</code>:</dt>
<dd>
<p><code>character</code>. Column names.
</p>
</dd>
<dt>
<code>fileDelimiter</code>:</dt>
<dd>
<p><code>character</code>. Delimiter separating values in the input file.
</p>
</dd>
<dt>
<code>fileSkipColumns</code>:</dt>
<dd>
<p><code>numeric</code>. Number of columns with row labels in the input file.
</p>
</dd>
<dt>
<code>fileSkipRows</code>:</dt>
<dd>
<p><code>numeric</code>. Number of rows with column labels in the input file.
</p>
</dd>
<dt>
<code>fileSliceSize</code>:</dt>
<dd>
<p><code>numeric</code>. Maximum number of rows in a slice.
</p>
</dd>
<dt>
<code>fileOmitCharacters</code>:</dt>
<dd>
<p><code>character</code>. 
Missing value (NaN) representation in the input file.
</p>
</dd>
</dl>
<h3>Methods</h3>


<dl>
<dt>
<code>initialize(mat)</code>:</dt>
<dd>
<p>Create the object from a matrix.
</p>
</dd>
<dt>
<code>nSlices()</code>:</dt>
<dd>
<p>Returns the number of slices.
</p>
</dd>
<dt>
<code>nCols()</code>:</dt>
<dd>
<p>Returns the number of columns in the matrix.
</p>
</dd>
<dt>
<code>nRows()</code>:</dt>
<dd>
<p>Returns the number of rows in the matrix.
</p>
</dd>
<dt>
<code>Clear()</code>:</dt>
<dd>
<p>Clears the object.
Removes the data slices and row and column names.
</p>
</dd>
<dt>
<code>Clone()</code>:</dt>
<dd>
<p>Makes a copy of the object.
Changes to the copy do not affect the source object.
</p>
</dd>
<dt>
<code>CreateFromMatrix(mat)</code>:</dt>
<dd>
<p>Creates <code>SlicedData</code> object from a <code>matrix</code>.
</p>
</dd>
<dt>
<code>LoadFile(filename, skipRows = NULL, skipColumns = NULL,</code> <br><code>sliceSize = NULL, omitCharacters = NULL,
            delimiter = NULL, rowNamesColumn = 1)</code>:</dt>
<dd>
<p>Loads data matrix from a file.
<code>filename</code> should be a character string.
The remaining parameters specify the file format and have the same
meaning as <code>file*</code> fields.
Additional <code>rowNamesColumn</code> parameter specifies which of the
columns of row labels to use as row names.
</p>
</dd>
<dt>
<code>SaveFile(filename)</code>:</dt>
<dd>
<p>Saves the data to a file.
<code>filename</code> should be a character string.
</p>
</dd>
<dt>
<code>getSlice(sl)</code>:</dt>
<dd>
<p>Retrieves <code>sl</code>-th slice of the matrix.
</p>
</dd>
<dt>
<code>setSlice(sl, value)</code>:</dt>
<dd>
<p>Set <code>sl</code>-th slice of the matrix.
</p>
</dd>
<dt>
<code>ColumnSubsample(subset)</code>:</dt>
<dd>
<p>Reorders/subsets the columns according to <code>subset</code>. <br>
Acts as <code>M = M[ ,subset]</code> for a matrix <code>M</code>.
</p>
</dd>
<dt>
<code>RowReorder(ordr)</code>:</dt>
<dd>
<p>Reorders rows according to <code>ordr</code>. <br>
Acts as <code>M = M[ordr, ]</code> for a matrix <code>M</code>.
</p>
</dd>
<dt>
<code>RowMatrixMultiply(multiplier)</code>:</dt>
<dd>
<p>Multiply each row by the <code>multiplier</code>. <br>
Acts as <code>M = M %*% multiplier</code> for a matrix <code>M</code>.
</p>
</dd>
<dt>
<code>CombineInOneSlice()</code>:</dt>
<dd>
<p>Combines all slices into one.
The whole matrix can then be obtained via <code>$getSlice(1)</code>.
</p>
</dd>
<dt>
<code>IsCombined()</code>:</dt>
<dd>
<p>Returns <code>TRUE</code> if the number of slices is 1 or 0.
</p>
</dd>
<dt>
<code>ResliceCombined(sliceSize = -1)</code>:</dt>
<dd>
<p>Cuts the data into slices of <code>sliceSize</code> rows.
If <code>sliceSize</code> is not defined,
the value of <code>fileSliceSize</code> field is used.
</p>
</dd>
<dt>
<code>GetAllRowNames()</code>:</dt>
<dd>
<p>Returns all row names in one vector.
</p>
</dd>    
<dt>
<code>RowStandardizeCentered()</code>:</dt>
<dd>
<p>Set the mean of each row to zero and the sum of squares to one.
</p>
</dd>
<dt>
<code>SetNanRowMean()</code>:</dt>
<dd>
<p>Impute rows with row mean.
Rows full of NaN values are imputed with zeros.
</p>
</dd>
<dt>
<code>RowRemoveZeroEps()</code>:</dt>
<dd>
<p>Removes rows of zeros and those that are nearly zero.
</p>
</dd>
<dt>
<code>FindRow(rowname)</code>:</dt>
<dd>
<p>Finds row by name.
Returns a pair of slice number an row number within the slice.
If no row is found, the function returns <code>NULL</code>.
</p>
</dd>
<dt>
<code>rowMeans(x, na.rm = FALSE, dims = 1L)</code>:</dt>
<dd>
<p>Returns a vector of row means.
Works as rowMeans but
requires <code>dims</code> to be equal to <code>1L</code>.
</p>
</dd>
<dt>
<code>rowSums(x, na.rm = FALSE, dims = 1L)</code>:</dt>
<dd>
<p>Returns a vector of row sums.
Works as rowSums but
requires <code>dims</code> to be equal to <code>1L</code>.
</p>
</dd>
<dt>
<code>colMeans(x, na.rm = FALSE, dims = 1L)</code>:</dt>
<dd>
<p>Returns a vector of column means.
Works as colMeans but
requires <code>dims</code> to be equal to <code>1L</code>.
</p>
</dd>
<dt>
<code>colSums(x, na.rm = FALSE, dims = 1L)</code>:</dt>
<dd>
<p>Returns a vector of column sums.
Works as colSums but
requires <code>dims</code> to be equal to <code>1L</code>.
</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Andrey A Shabalin <a href="mailto:andrey.shabalin@gmail.com">andrey.shabalin@gmail.com</a>
</p>


<h3>References</h3>

<p>The package website:
<a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a>
</p>


<h3>See Also</h3>

<p>This class is used to load data for eQTL analysis by
<code>Matrix_eQTL_engine</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Create a SlicedData variable
sd = SlicedData$new()    

# Show the details of the empty object
show(sd)    

# Create a matrix of values and assign to sd
mat = matrix(1:12, 3, 4)    
rownames(mat) = c("row1", "row2", "row3")    
colnames(mat) = c("col1", "col2", "col3", "col4")    
sd$CreateFromMatrix( mat )    

# Show the detail of the object (one slice)
show(sd)    

# Slice it in pieces of 2 rows
sd$ResliceCombined(sliceSize = 2L)    

# Show the number of slices (equivalent function calls)
sd$nSlices()
length(sd)

# Is it all in one slice? (No)
sd$IsCombined()

# Show the column names (equivalent function calls)
sd$columnNames
colnames(sd)

# Show row name slices
sd$rowNameSlices

# Show all row names (equivalent function calls)
sd$GetAllRowNames()
rownames(sd)

# Print the second slice
print(sd[[2]])

# Reorder and subset columns
sd$ColumnSubsample( c(1,3,4) )    

# Reorder and subset rows
sd$RowReorder( c(3,1) )    

# Show the detail of the object (one slice again)
show(sd)    

# Is it all in one slice? (Yes)
sd$IsCombined()

# Find the row with name "row1" (it is second in the first slice)
sd$FindRow("row1")    

</code></pre>


</div>