<div class="container">

<table style="width: 100%;"><tr>
<td>Matrix_eQTL_main</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Main function for fast eQTL analysis in MatrixEQTL package
</h2>

<h3>Description</h3>

<p><code>Matrix_eQTL_engine</code> function tests association of
every row of the <code>snps</code> dataset with every row of the
<code>gene</code> dataset using a linear regression model
defined by the <code>useModel</code> parameter (see below).
</p>
<p>The testing procedure accounts for extra 
covariates specified by the <code>cvrt</code> parameter.
</p>
<p>The <code>errorCovariance</code> parameter can be set to the 
error variance-covariance matrix to account 
for heteroskedastic and/or correlated errors.
</p>
<p>Associations significant at <code>pvOutputThreshold</code> 
(<code>pvOutputThreshold.cis</code>) levels are saved to 
<code>output_file_name</code> (<code>output_file_name.cis</code>), 
with corresponding estimates of effect size (slope coefficient),
test statistics, p-values, and q-values (false discovery rate).
</p>
<p>Matrix eQTL can perform separate analysis for
local (cis) and distant (trans) eQTLs.
For such analysis one has to set the cis-analysis specific
parameters <code>pvOutputThreshold.cis &gt; 0</code>,
<code>cisDist</code>, <code>snpspos</code> and 
genepos in the call of <code>Matrix_eQTL_main</code> function.
A gene-SNP pair is considered local if the 
distance between them is less or equal to <code>cisDist</code>.
The genomic location of genes and SNPs is defined by 
the data frames <code>snpspos</code> and genepos.
Depending on p-value thresholds <code>pvOutputThreshold</code> and
<code>pvOutputThreshold.cis</code> Matrix eQTL runs in 
one of three different modes:
</p>

<ul>
<li>
<p> Set <code>pvOutputThreshold &gt; 0</code> and 
<code>pvOutputThreshold.cis = 0</code> (or use <code>Matrix_eQTL_engine</code>)
to perform eQTL analysis without using gene/SNP locations.
Associations significant at the <code>pvOutputThreshold</code> level
are be recorded in <code>output_file_name</code>
and in the returned object.
</p>
</li>
<li>
<p> Set <code>pvOutputThreshold = 0</code> and
<code>pvOutputThreshold.cis &gt; 0</code> to perform eQTL analysis for
local gene-SNP pairs only. Local associations significant at
<code>pvOutputThreshold.cis</code> level will be recorded in
<code>output_file_name.cis</code> and in the returned object.
</p>
</li>
<li>
<p> Set <code>pvOutputThreshold &gt; 0</code> and
<code>pvOutputThreshold.cis &gt; 0</code> to perform eQTL analysis
with separate p-value thresholds for local and distant eQTLs.
Distant and local associations significant at corresponding
thresholds are recorded in <code>output_file_name</code> and
<code>output_file_name.cis</code> respectively and in the returned object.
In this case the false discovery rate is calculated
separately for these two sets of eQTLs.
</p>
</li>
</ul>
<p><code>Matrix_eQTL_engine</code> is a wrapper for <code>Matrix_eQTL_main</code>
for eQTL analysis without regard to gene/SNP location and provided
for compatibility with the previous versions of the package.
</p>
<p>The parameter <code>pvalue.hist</code> allows to record information sufficient
to create a histogram or QQ-plot of all the p-values
(see <code>plot</code>).
</p>


<h3>Usage</h3>

<pre><code class="language-R">Matrix_eQTL_main(
        snps, 
        gene, 
        cvrt = SlicedData$new(), 
        output_file_name = "", 
        pvOutputThreshold = 1e-5,
        useModel = modelLINEAR, 
        errorCovariance = numeric(), 
        verbose = TRUE, 
        output_file_name.cis = "", 
        pvOutputThreshold.cis = 0,
        snpspos = NULL, 
        genepos = NULL,
        cisDist = 1e6,
        pvalue.hist = FALSE,
        min.pv.by.genesnp = FALSE,
        noFDRsaveMemory = FALSE)

Matrix_eQTL_engine(
        snps, 
        gene, 
        cvrt = SlicedData$new(), 
        output_file_name, 
        pvOutputThreshold = 1e-5, 
        useModel = modelLINEAR, 
        errorCovariance = numeric(), 
        verbose = TRUE,
        pvalue.hist = FALSE,
        min.pv.by.genesnp = FALSE,
        noFDRsaveMemory = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>snps</code></td>
<td>

<p><code>SlicedData</code> object with genotype information. 
Can be real-valued for linear models and
must take at most 3 distinct values for ANOVA unless
the number of ANOVA categories is set to
a higher number (see <code>useModel</code> parameter).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene</code></td>
<td>

<p><code>SlicedData</code> object with gene expression information.
Must have columns matching those of <code>snps</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cvrt</code></td>
<td>

<p><code>SlicedData</code> object with additional covariates. 
Can be an empty <code>SlicedData</code> object in case of no covariates.
The constant is always included in the model and
would cause an error if included in <code>cvrt</code>.
The order of columns must match those in <code>snps</code> and <code>gene</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output_file_name</code></td>
<td>

<p><code>character</code>, <code>connection</code>, or <code>NULL</code>.
If not <code>NULL</code>, significant associations are saved to
this file (all significant associations if <code>pvOutputThreshold=0</code>
or only distant if <code>pvOutputThreshold&gt;0</code>).
If the file with this name exists, it is overwritten.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output_file_name.cis</code></td>
<td>

<p><code>character</code>, <code>connection</code>, or <code>NULL</code>.
If not <code>NULL</code>, significant local associations
are saved to this file.
If the file with this name exists, it is overwritten.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvOutputThreshold</code></td>
<td>

<p><code>numeric</code>. Significance threshold for all/distant tests.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvOutputThreshold.cis</code></td>
<td>
 
<p><code>numeric</code>. Same as <code>pvOutputThreshold</code>, but for local eQTLs.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useModel</code></td>
<td>

<p><code>integer</code>. 
Eigher <code>modelLINEAR</code>, 
<code>modelANOVA</code>, or
<code>modelLINEAR_CROSS</code>.
</p>

<ol>
<li>
<p> Set <code>useModel = modelLINEAR</code> to model
the effect of the genotype as additive linear and
test for its significance using t-statistic.
</p>
</li>
<li>
<p> Set <code>useModel = modelANOVA</code> to treat genotype
as a categorical variables and use ANOVA model and
test for its significance using F-test.
The default number of ANOVA categories is 3.
Set otherwise like this:
<code>options(MatrixEQTL.ANOVA.categories=4)</code>.
</p>
</li>
<li>
<p> Set <code>useModel = modelLINEAR_CROSS</code> to add
a new term to the model equal to the product of genotype and
the last covariate; the significance of this term is
then tested using t-statistic.
</p>
</li>
</ol>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>errorCovariance</code></td>
<td>

<p><code>numeric</code>. The error covariance matrix.
Use <code>numeric()</code> for homoskedastic independent errors. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p><code>logical</code>. Set to <code>TRUE</code> to display
more detailed report about the progress.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpspos</code></td>
<td>

<p><code>data.frame</code> object with information about SNP locations,
must have 3 columns - SNP name, chromosome, and position, like this:
</p>

<table>
<tr>
<td style="text-align: center;">
            snpid   </td>
<td style="text-align: center;"> chr    </td>
<td style="text-align: center;"> pos     </td>
</tr>
<tr>
<td style="text-align: center;">
            Snp_01  </td>
<td style="text-align: center;"> 1      </td>
<td style="text-align: center;"> 721289  </td>
</tr>
<tr>
<td style="text-align: center;">
            Snp_02  </td>
<td style="text-align: center;"> 1      </td>
<td style="text-align: center;"> 752565  </td>
</tr>
<tr>
<td style="text-align: center;">
            ...  </td>
<td style="text-align: center;"> ... </td>
<td style="text-align: center;"> ...  </td>
</tr>
<tr>
<td style="text-align: center;">
        </td>
</tr>
</table>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genepos</code></td>
<td>

<p><code>data.frame</code> with information about transcript locations,
must have 4 columns - the name, chromosome, and 
positions of the left and right ends, like this:
</p>

<table>
<tr>
<td style="text-align: center;">
            geneid  </td>
<td style="text-align: center;">  chr    </td>
<td style="text-align: center;"> left   </td>
<td style="text-align: center;"> right  </td>
</tr>
<tr>
<td style="text-align: center;">
            Gene_01 </td>
<td style="text-align: center;">   1     </td>
<td style="text-align: center;"> 721289 </td>
<td style="text-align: center;"> 731289 </td>
</tr>
<tr>
<td style="text-align: center;">
            Gene_02 </td>
<td style="text-align: center;">   1     </td>
<td style="text-align: center;"> 752565 </td>
<td style="text-align: center;"> 762565 </td>
</tr>
<tr>
<td style="text-align: center;">
            ...  </td>
<td style="text-align: center;"> ...  </td>
<td style="text-align: center;"> ... </td>
<td style="text-align: center;"> ... </td>
</tr>
<tr>
<td style="text-align: center;">
        </td>
</tr>
</table>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cisDist</code></td>
<td>

<p><code>numeric</code>. SNP-gene pairs within this distance are 
considered local. The distance is measured from the
nearest end of the gene. SNPs within a gene are always considered local.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvalue.hist</code></td>
<td>

<p><code>logical</code>, <code>numerical</code>, or <code>"qqplot"</code>.
Defines whether and how the distribution of p-values is recorded
in the returned object.
If <code>pvalue.hist = FALSE</code>, the information is not recorded
and the analysis is performed faster. 
Alternatively, set <code>pvalue.hist = "qqplot"</code> to record information
sufficient to create a QQ-plot of the p-values
(use <code>plot</code> on the returned
object to create the plot).
To record information for a histogram set <code>pvalue.hist</code>
to the desired number of bins of equal size.
Finally, <code>pvalue.hist</code> can also be set to a
custom set of bin edges.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.pv.by.genesnp</code></td>
<td>

<p><code>logical</code>. Set <code>min.pv.by.genesnp = TRUE</code> to record
the minimum p-value for each SNP and each gene in the returned object.
The minimum p-values are recorded even if if they are above the
corresponding thresholds of <code>pvOutputThreshold</code> and
<code>pvOutputThreshold.cis</code>.
The analysis runs faster when the parameter is set to <code>FALSE</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>noFDRsaveMemory</code></td>
<td>

<p><code>logical</code>. Set <code>noFDRsaveMemory = TRUE</code> to save
significant gene-SNP pairs directly to the output files,
reduce memory footprint and skip FDR calculation.
The eQTLs are not recorded in the returned object
if <code>noFDRsaveMemory = TRUE</code>.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Note that the columns of 
<code>gene</code>, <code>snps</code>, and <code>cvrt</code> must match.
If they do not match in the input files,
use <code>ColumnSubsample</code> method to subset and/or reorder them.
</p>


<h3>Value</h3>

<p>The detected eQTLs are saved in <code>output_file_name</code>
and/or <code>output_file_name.cis</code> if they are not <code>NULL</code>.
The method also returns a list with a summary of the performed analysis.
</p>
<table>
<tr style="vertical-align: top;">
<td><code>param</code></td>
<td>

<p>Keeps all input parameters and also records
the number of degrees of freedom for the full model.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>time.in.sec</code></td>
<td>

<p>Time difference between the start and 
the end of the analysis (in seconds).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>all</code></td>
<td>

<p>Information about all detected eQTLs.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cis</code></td>
<td>

<p>Information about detected local eQTLs.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trans</code></td>
<td>

<p>Information about detected distant eQTLs.
</p>
</td>
</tr>
</table>
<p>The elements <code>all</code>, <code>cis</code>, and <code>trans</code>
may contain the following components
</p>

<dl>
<dt><code>ntests</code></dt>
<dd>
<p>Total number of tests performed. This is used for FDR calculation.
</p>
</dd>
<dt><code>eqtls</code></dt>
<dd>
<p>Data frame with recorded significant associations.
Not available if <code>noFDRsaveMemory=FALSE</code>
</p>
</dd>
<dt><code>neqtls</code></dt>
<dd>
<p>Number of significant associations recorded.
</p>
</dd>
<dt><code>hist.bins</code></dt>
<dd>
<p>Histogram bins used for recording p-value distribution.
See <code>pvalue.hist</code> parameter.
</p>
</dd>
<dt><code>hist.counts</code></dt>
<dd>
<p>Number of p-value that fell in each histogram bin.
See <code>pvalue.hist</code> parameter.
</p>
</dd>
<dt><code>min.pv.snps</code></dt>
<dd>
<p>Vector with the best p-value for each SNP.
See <code>min.pv.by.genesnp</code> parameter.
</p>
</dd>
<dt><code>min.pv.gene</code></dt>
<dd>
<p>Vector with the best p-value for each gene.
See <code>min.pv.by.genesnp</code> parameter.
</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Andrey A Shabalin <a href="mailto:andrey.shabalin@gmail.com">andrey.shabalin@gmail.com</a>
</p>


<h3>References</h3>

<p>The package website:
<a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a>
</p>


<h3>See Also</h3>

<p>The code below is the sample code for eQTL analysis NOT using gene/SNP locations.    
</p>
<p>See <code>MatrixEQTL_cis_code</code> for sample code for
eQTL analysis that separates local and distant tests.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Matrix eQTL by Andrey A. Shabalin
# http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/
# 
# Be sure to use an up to date version of R and Matrix eQTL.

# source("Matrix_eQTL_R/Matrix_eQTL_engine.r");
library(MatrixEQTL)

## Location of the package with the data files.
base.dir = find.package('MatrixEQTL');

## Settings

# Linear model to use, modelANOVA, modelLINEAR, or modelLINEAR_CROSS
useModel = modelLINEAR; # modelANOVA, modelLINEAR, or modelLINEAR_CROSS

# Genotype file name
SNP_file_name = paste0(base.dir, "/data/SNP.txt");

# Gene expression file name
expression_file_name = paste0(base.dir, "/data/GE.txt");

# Covariates file name
# Set to character() for no covariates
covariates_file_name = paste0(base.dir, "/data/Covariates.txt");

# Output file name
output_file_name = tempfile();

# Only associations significant at this level will be saved
pvOutputThreshold = 1e-2;

# Error covariance matrix
# Set to numeric() for identity.
errorCovariance = numeric();
# errorCovariance = read.table("Sample_Data/errorCovariance.txt");


## Load genotype data

snps = SlicedData$new();
snps$fileDelimiter = "\t";      # the TAB character
snps$fileOmitCharacters = "NA"; # denote missing values;
snps$fileSkipRows = 1;          # one row of column labels
snps$fileSkipColumns = 1;       # one column of row labels
snps$fileSliceSize = 2000;      # read file in slices of 2,000 rows
snps$LoadFile(SNP_file_name);

## Load gene expression data

gene = SlicedData$new();
gene$fileDelimiter = "\t";      # the TAB character
gene$fileOmitCharacters = "NA"; # denote missing values;
gene$fileSkipRows = 1;          # one row of column labels
gene$fileSkipColumns = 1;       # one column of row labels
gene$fileSliceSize = 2000;      # read file in slices of 2,000 rows
gene$LoadFile(expression_file_name);

## Load covariates

cvrt = SlicedData$new();
cvrt$fileDelimiter = "\t";      # the TAB character
cvrt$fileOmitCharacters = "NA"; # denote missing values;
cvrt$fileSkipRows = 1;          # one row of column labels
cvrt$fileSkipColumns = 1;       # one column of row labels
if(length(covariates_file_name)&gt;0){
    cvrt$LoadFile(covariates_file_name);
}

## Run the analysis

me = Matrix_eQTL_engine(
    snps = snps,
    gene = gene,
    cvrt = cvrt,
    output_file_name = output_file_name,
    pvOutputThreshold = pvOutputThreshold,
    useModel = useModel, 
    errorCovariance = errorCovariance, 
    verbose = TRUE,
    pvalue.hist = TRUE,
    min.pv.by.genesnp = FALSE,
    noFDRsaveMemory = FALSE);
        
unlink(output_file_name);

## Results:

cat('Analysis done in: ', me$time.in.sec, ' seconds', '\n');
cat('Detected eQTLs:', '\n');
show(me$all$eqtls)

## Plot the histogram of all p-values

plot(me)
</code></pre>


</div>