<div class="container">

<table style="width: 100%;"><tr>
<td>mle_pe</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Find MLEs for microbiome mixture</h2>

<h3>Description</h3>

<p>Find joint MLEs of p and e for microbiome mixture model
</p>


<h3>Usage</h3>

<pre><code class="language-R">mle_pe(
  tab,
  interval = c(0, 1),
  tol = 0.000001,
  check_boundary = FALSE,
  SE = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>tab</code></td>
<td>
<p>Dataset of read counts as 3d array of size 3x3x2, genotype in first sample x genotype in second sample x allele in read.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>interval</code></td>
<td>
<p>Interval to which each parameter should be constrained</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tol</code></td>
<td>
<p>Tolerance for convergence</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check_boundary</code></td>
<td>
<p>If TRUE, explicitly check the boundaries of <code>interval</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SE</code></td>
<td>
<p>If TRUE, get estimated standard errors.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A vector containing the estimates of <code>p</code> and <code>e</code> along with the evaluated log likelihood and likelihood ratio test statistics for the hypotheses p=0 and p=1.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(mbmixdata)
mle_pe(mbmixdata)

</code></pre>


</div>