<div class="container">

<table style="width: 100%;"><tr>
<td>mbmix_loglik</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>log likelihood function for microbiome mixture</h2>

<h3>Description</h3>

<p>Calculate log likelihood function for microbiome sample mixture model at particular values of <code>p</code> and <code>e</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mbmix_loglik(tab, p, e = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>tab</code></td>
<td>
<p>Dataset of read counts as 3d array of size 3x3x2, genotype in first sample x genotype in second sample x allele in read.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>Contaminant probability (proportion of mixture coming from the second sample).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>e</code></td>
<td>
<p>Sequencing error rate.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The log likelihood evaluated at <code>p</code> and <code>e</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(mbmixdata)
mbmix_loglik(mbmixdata, p=0.74, e=0.002)

</code></pre>


</div>