<div class="container">

<table style="width: 100%;"><tr>
<td>mle_p</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>MLE of p for fixed e</h2>

<h3>Description</h3>

<p>Calculate the MLE of the contaminant probability p for a fixed value of the sequencing error rate e.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mle_p(
  tab,
  e = 0.002,
  interval = c(0, 1),
  tol = 0.000001,
  check_boundary = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>tab</code></td>
<td>
<p>Dataset of read counts as 3d array of size 3x3x2, genotype in first sample x genotype in second sample x allele in read.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>e</code></td>
<td>
<p>Assumed value for the sequencing error rate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>interval</code></td>
<td>
<p>Interval to which each parameter should be constrained</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tol</code></td>
<td>
<p>Tolerance for convergence</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check_boundary</code></td>
<td>
<p>If TRUE, explicitly check the boundaries of <code>interval</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A single numeric value, the MLE of <code>p</code>, with the log likelihood as an attribute.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(mbmixdata)
mle_p(mbmixdata, e=0.002)

</code></pre>


</div>