<div class="container">

<table style="width: 100%;"><tr>
<td>mcbette-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>mcbette: Model Comparison Using Babette</h2>

<h3>Description</h3>

<p>'mcbette' does a model comparing using babette,
where the models are Bayesian phylogenetic models,
as created by create_inference_model.
</p>


<h3>Details</h3>

<p>The main function is est_marg_liks,
which estimate the marginal likelihoods (aka evidence)
for one or more inference models, based on a single alignment.
Also, the marginal likelihoods are compared, resulting in a
relative weight for each model, where a relative weight of a model
close to <code>1.0</code> means that that model is way likelier than
the others.
</p>
<p>In the process, multiple (temporary) files are created (where
<code>[x]</code> denotes the index in a list)
</p>

<ul>
<li> <p><code>beast2_optionses[x]$input_filename</code>
path to the the BEAST2 XML input file
</p>
</li>
<li> <p><code>beast2_optionses[x]$output_state_filename</code>
path to the BEAST2 XML state file
</p>
</li>
<li> <p><code>inference_models[x]$mcmc$tracelog$filename</code>
path to the BEAST2 trace file with parameter estimates
</p>
</li>
<li> <p><code>inference_models[x]$mcmc$treelog$filename</code>
path to the BEAST2 <code>trees</code> file with the posterior trees
</p>
</li>
<li> <p><code>inference_models[x]$mcmc$screenlog$filename</code>
path to the BEAST2 screen output file
</p>
</li>
</ul>
<p>These file can be deleted manually by bbt_delete_temp_files,
else these will be deleted automatically by the operating system.
</p>


<h3>Author(s)</h3>

<p>Rich√®l J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use can_run_mcbette to see if 'mcbette' can run.
</p>


<h3>Examples</h3>

<pre><code class="language-R">if (can_run_mcbette()) {

  # An example FASTA file
  fasta_filename &lt;- system.file("extdata", "simple.fas", package = "mcbette")

  inference_model_1 &lt;- beautier::create_ns_inference_model(
    site_model = beautier::create_jc69_site_model()
  )
  inference_model_2 &lt;- beautier::create_ns_inference_model(
    site_model = beautier::create_gtr_site_model()
  )

  # Shorten the run, by doing a short (dirty, unreliable) MCMC
  inference_model_1$mcmc &lt;- beautier::create_test_ns_mcmc()
  inference_model_2$mcmc &lt;- beautier::create_test_ns_mcmc()

  inference_models &lt;- c(list(inference_model_1), list(inference_model_2))

  # Estimate the marginal log-likelihoods of the two models
  marg_liks &lt;- est_marg_liks(
    fasta_filename = fasta_filename,
    inference_models = inference_models
  )

  # Interpret the results
  interpret_marg_lik_estimates(marg_liks)

  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()
}
</code></pre>


</div>