<div class="container">

<table style="width: 100%;"><tr>
<td>metaData</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Results of different GWAS.</h2>

<h3>Description</h3>

<p>A dataset containing the results of 10 different genetic association studies testing the association between a set of 25,436 markers and the grain yield.
The data are extrated from: Drops Amaizing available on the https://doi.org/10.15454/6TL2N4 website.
This dataset were obtained thanks to the <em>metaGE.collect</em> function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">metaData
</code></pre>


<h3>Format</h3>

<p>A data frame with 25,436 rows and 35 variables:
</p>

<ul>
<li>
<p> CHR: chromosome of the marker
</p>
</li>
<li>
<p> POS: position of the marker
</p>
</li>
<li>
<p> MARKER: name of the marker
</p>
</li>
<li>
<p> FREQ.env: maf of the marker in the environment env
</p>
</li>
<li>
<p> EFFECT.env: regression coefficient of the marker in the environment env
</p>
</li>
<li>
<p> EFFECT_SE.env: standard error of the regression coefficient of the marker in the environment env
</p>
</li>
<li>
<p> PVAL.env: pvalue of the marker in the environment env
</p>
</li>
<li>
<p> WEIGHT.env: weight of the marker in the environment env
</p>
</li>
<li>
<p> ALLELE0: allele0
</p>
</li>
<li>
<p> ALLELE1: allele1
</p>
</li>
</ul>
</div>