<div class="container">

<table style="width: 100%;"><tr>
<td>meta2d</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Detect rhythmic signals from time-series datasets with multiple methods</h2>

<h3>Description</h3>

<p>This is a function that incorporates ARSER, JTK_CYCLE and Lomb-Scargle to
detect rhythmic signals from time-series datasets.
</p>


<h3>Usage</h3>

<pre><code class="language-R">meta2d(infile, outdir = "metaout", filestyle, timepoints, minper = 20,
  maxper = 28, cycMethod = c("ARS", "JTK", "LS"),
  analysisStrategy = "auto", outputFile = TRUE,
  outIntegration = "both", adjustPhase = "predictedPer",
  combinePvalue = "fisher", weightedPerPha = FALSE, ARSmle = "auto",
  ARSdefaultPer = 24, outRawData = FALSE, releaseNote = TRUE,
  outSymbol = "", parallelize = FALSE, nCores = 1, inDF = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>infile</code></td>
<td>
<p>a character string. The name of input file containing
time-series data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outdir</code></td>
<td>
<p>a character string. The name of directory used to store
output files.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filestyle</code></td>
<td>
<p>a character vector(length 1 or 3). The data format of
input file, must be <code>"txt"</code>, or <code>"csv"</code>, or a character vector
containing field separator character(<code>sep</code>), quoting character
(<code>quote</code>), and the character used for decimal points(<code>dec</code>,
for details see <code>read.table</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>timepoints</code></td>
<td>
<p>a numeric vector corresponding to sampling time points
of input time-series data; if sampling time points are in the first line
of input file, it could be set as a character sting-"Line1" or "line1".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minper</code></td>
<td>
<p>a numeric value. The minimum period length of interested
rhythms. The default is <code>20</code> for circadian rhythms.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxper</code></td>
<td>
<p>a numeric value. The maximum period length of interested
rhythms. The default is <code>28</code> for circadian rhythms.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cycMethod</code></td>
<td>
<p>a character vector(length 1 or 2 or 3). User-defined
methods for detecting rhythmic signals, must be selected as any one, any
two or all three methods(default) from <code>"ARS"</code>(ARSER),
<code>"JTK"</code>(JTK_CYCLE) and <code>"LS"</code>(Lomb-Scargle).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>analysisStrategy</code></td>
<td>
<p>a character string. The strategy used to select
proper methods from <code>cycMethod</code> for analyzing input time-series
data, must be <code>"auto"</code>(default), or <code>"selfUSE"</code>. See
<code>Details</code> part for more information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outputFile</code></td>
<td>
<p>logical. If <code>TRUE</code>, analysis results will be wrote
in the output files. If <code>FALSE</code>, analysis results will be returned
as an R list.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outIntegration</code></td>
<td>
<p>a character string. This parameter controls what
kinds of analysis results will be outputted, must be one of <code>"both"</code>
(default), <code>"onlyIntegration"</code>(only output integration file), or
<code>"noIntegration"</code>(not output integration file).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adjustPhase</code></td>
<td>
<p>a character string. The method used to adjust original
phase calculated by each method in integration file, must be one of
<code>"predictedPer"</code>(adjust phase with predicted period length) or
<code>"notAdjusted"</code>(not adjust phase).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>combinePvalue</code></td>
<td>
<p>a character string. The method used to integrate
multiple p-values, must be one of <code>"bonferroni"</code>(Bonferroni
correction), or <code>"fisher"</code>(Fisher's method).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weightedPerPha</code></td>
<td>
<p>logical. If <code>TRUE</code>, weighted scores based on
p-value given by each method will be used to calculate the integrated
period length and phase.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ARSmle</code></td>
<td>
<p>a character string. The strategy of using MLE method in
<code>ar</code> fit of <code>"ARS"</code>, must be one of
<code>"auto"</code>(use MLE depending the number of time points), <code>"mle"</code>
(always use MLE), or <code>"nomle"</code>(never use MLE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ARSdefaultPer</code></td>
<td>
<p>a numeric value. The expected period length of
interested rhythm, which is a necessary parameter for <code>ARS</code>. The
default is <code>24</code>(for circadian rhythms). Set it to another proper
numeric value for other rhythms.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outRawData</code></td>
<td>
<p>logical. If <code>TRUE</code>, raw time-series data will be
added in the output files.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>releaseNote</code></td>
<td>
<p>logical. If <code>TRUE</code>, reminding or warning notes
during the analysis will be released on the screen.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outSymbol</code></td>
<td>
<p>a character string. A common prefix exists in the names of
output files.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallelize</code></td>
<td>
<p>logical. If <code>TRUE</code>, computation will be done in paralleL
Doesn't work in windows machine.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nCores</code></td>
<td>
<p>a integer. Bigger or equal to one, number of cores to use.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inDF</code></td>
<td>
<p>data.frame. If <code>!is.null(inDF)</code> and timepoints is a numeric meta2d will use this data.frame instead of loading from <code>infile</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><a href="https://github.com/cauyrd/ARSER">ARSER</a>(Yang, 2010),
<a href="http://openwetware.org/wiki/HughesLab:JTK_Cycle">JTK_CYCLE</a>(
Hughes, 2010), and
<a href="https://academic.oup.com/bioinformatics/article/22/3/310/220284">
Lomb-Scargle</a>(Glynn, 2006) are three popular methods of detecting
rhythmic signals. <code>ARS</code> can not analyze unevenly sampled datasets,
or evenly sampled datasets but with missing values, or with replicate
samples, or with non-integer sampling interval. <code>JTK</code> is not
suitable to analyze unevenly sampled datasets or evenly sampled datasets
but with non-integer sampling interval. If set <code>analysisStrategy</code>
as <code>"auto"</code>(default), <code>meta2d</code> will automatically select
proper method from <code>cycMethod</code> for each input dataset. If the user
clearly know that the dataset could be analyzed by each method defined
by <code>cycMethod</code> and do not hope to output integrated values,
<code>analysisStrategy</code> can be set as <code>"selfUSE"</code>.
</p>
<p><code>ARS</code> used here is translated from its python version which always
uses <code>"yule-walker"</code>, <code>"burg"</code>, and <code>"mle"</code> methods(see
<code>ar</code>) to fit autoregressive models to time-series
data. Fitting by <code>"mle"</code> will be very slow for datasets
with many time points. If <code>ARSmle = "auto"</code> is used,
<code>meta2d</code> will only include <code>"mle"</code> when number of time points
is smaller than 24. In addition, one evaluation work(Wu, 2014) indicates
that <code>ARS</code> shows relative high false positive rate in analyzing
high-resolution datasets (1h/2days and 2h/2days). <code>JTK</code>(version 3)
used here is the latest version, which improves its p-value calculation
in analyzing datasets with missing values.
</p>
<p>The power of detecting rhythmic signals for an algorithm is associated
with the nature of data and interested periodic pattern(Deckard, 2013),
which indicates that integrating analysis results from multiple methods
may be helpful to rhythmic detection. For integrating p-values,
Bonferroni correction(<code>"bonferroni"</code>) and Fisher's method(
<code>"fisher"</code>) (Fisher, 1925; implementation code from <span class="pkg">MADAM</span>)
could be selected, and <code>"bonferroni"</code> is usually more conservative
than <code>"fisher"</code>. The integrated period is arithmetic mean of
multiple periods. For integrating phase, <code>meta2d</code> takes use of
<a href="https://en.wikipedia.org/wiki/Mean_of_circular_quantities">
mean of circular quantities</a>. Integrated period and phase is further
used to calculate the baseline value and amplitude through fitting a
constructed periodic model.
</p>
<p>Phases given by <code>JTK</code> and <code>LS</code> need to be adjusted with their
predicted period (<code>adjustedPhase = "predictedPer"</code>) before
integration. If <code>adjustedPhas = "notAdjusted"</code> is selected, no
integrated phase will be calculated. If set <code>weightedPerPha</code> as
<code>TRUE</code>, weighted scores will be used in averaging periods and
phases. Weighted scores for one method are based on all its reported
p-values, which means a weighted score assigned to any one profile will
be affected by all other profiles. It is always a problem of averaging
phases with quite different period lengths(eg. averaging two phases
with 16-hours' and 30-hours' period length). Currently, setting
<code>minper</code>, <code>maxper</code> and <code>ARSdefaultPer</code> to a same value
may be the only way of completely eliminating such problem.
</p>
<p>This function is originally aimed to analyze large scale periodic data(
eg. circadian transcriptome data) without individual information.
Please pay attention to data format of input file(see <code>Examples</code>
part). Except the first column and first row, others are time-series
experimental values(setting missing values as <code>NA</code>).
</p>


<h3>Value</h3>

<p><code>meta2d</code> will write analysis results in different files under
<code>outdir</code> if set <code>outputFile = TRUE</code>. Files named with
"ARSresult", "JTKresult" and "LSreult" store analysis results from
<code>ARS</code>, <code>JTK</code> and <code>LS</code> respectively. The file named with
"meta2d" is the integration file, and it stores integrated values in
columns with a common name tag-"meta2d". The integration file also
contains p-value, FDR value, period, phase(adjusted phase if
<code>adjustedPhase = "predictedPer"</code>) and amplitude values calculated
by each method.
If <code>outputFile = FALSE</code> is selected, <code>meta2d</code> will return a
list containing the following components:
</p>

<table>
<tr>
<td style="text-align: right;">
       ARS   </td>
<td style="text-align: left;">  analysis results from <code>ARS</code> method</td>
</tr>
<tr>
<td style="text-align: right;">
       JTK   </td>
<td style="text-align: left;">  analysis results from <code>JTK</code> method</td>
</tr>
<tr>
<td style="text-align: right;">
       LS    </td>
<td style="text-align: left;">  analysis results from <code>LS</code> method</td>
</tr>
<tr>
<td style="text-align: right;">
       meta  </td>
<td style="text-align: left;">  the integrated analysis results as mentioned above
  </td>
</tr>
</table>
<h3>References</h3>

<p>Yang R. and  Su Z. (2010). Analyzing circadian expression data by
harmonic regression based on autoregressive spectral estimation.
<em>Bioinformatics</em>, <b>26(12)</b>, i168–i174.
</p>
<p>Hughes M. E., Hogenesch J. B. and Kornacker K. (2010). JTK_CYCLE: an
efficient nonparametric algorithm for detecting rhythmic components in
genome-scale data sets. <em>Journal of Biological Rhythms</em>,
<b>25(5)</b>, 372–380.
</p>
<p>Glynn E. F., Chen J. and Mushegian A. R. (2006). Detecting periodic
patterns in unevenly spaced gene expression time series using
Lomb-Scargle periodograms. <em>Bioinformatics</em>, <b>22(3)</b>,
310–316.
</p>
<p>Wu G., Zhu J., Yu J., Zhou L., Huang J. Z. and  Zhang Z. (2014). Evaluation
of five methods for genome-wide circadian gene identification.
<em>Journal of Biological Rhythms</em>, <b>29(4)</b>, 231–242.
</p>
<p>Deckard A., Anafi R. C., Hogenesch J. B., Haase S.B. and Harer J. (2013).
Design and analysis of large-scale biological rhythm studies:
a comparison of algorithms for detecting periodic signals in biological
data. <em>Bioinformatics</em>, <b>29(24)</b>, 3174–3180.
</p>
<p>Fisher, R.A. (1925). <em>Statistical methods for research workers</em>.
Oliver and Boyd (Edinburgh).
</p>
<p>Kugler K. G., Mueller L.A. and Graber A. (2010). MADAM - an open source
toolbox for meta-analysis. <em>Source Code for Biology and Medicine</em>,
<b>5</b>, 3.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># write 'cycSimu4h2d', 'cycMouseLiverRNA' and 'cycYeastCycle' into three
# 'csv' files
write.csv(cycSimu4h2d, file="cycSimu4h2d.csv", row.names=FALSE)
write.csv(cycMouseLiverRNA, file="cycMouseLiverRNA.csv", row.names=FALSE)
write.csv(cycYeastCycle, file="cycYeastCycle.csv", row.names=FALSE)

# write 'cycMouseLiverProtein' into a 'txt' file
write.table(cycMouseLiverProtein, file="cycMouseLiverProtein.txt",
  sep="\t", quote=FALSE, row.names=FALSE)

# analyze 'cycMouseLiverRNA.csv' with JTK_CYCLE
# this is masked for keeping the total running time within 10s required by CRAN check
# meta2d(infile="cycMouseLiverRNA.csv", filestyle="csv", outdir="example",
#  timepoints=18:65, cycMethod="JTK", outIntegration="noIntegration")

# analyze 'cycMouseLiverProtein.txt' with JTK_CYCLE and Lomb-Scargle
meta2d(infile="cycMouseLiverProtein.txt", filestyle="txt",
  outdir="example", timepoints=rep(seq(0, 45, by=3), each=3),
  cycMethod=c("JTK","LS"), outIntegration="noIntegration")

# analyze 'cycSimu4h2d.csv' with ARSER, JTK_CYCLE and Lomb-Scargle and
# output integration file with analysis results from each method
meta2d(infile="cycSimu4h2d.csv", filestyle="csv", outdir="example",
  timepoints="Line1")

# analyze 'cycYeastCycle.csv' with ARSER, JTK_CYCLE and Lomb-Scargle to
# detect transcripts associated with cell cycle, and only output
# integration file
meta2d(infile="cycYeastCycle.csv",filestyle="csv", outdir="example",
  minper=80, maxper=96, timepoints=seq(2, 162, by=16),
  outIntegration="onlyIntegration", ARSdefaultPer=85,
  outRawData=TRUE)
# return analysis results instead of output them into files
cyc &lt;- meta2d(infile="cycYeastCycle.csv",filestyle="csv",
  minper=80, maxper=96, timepoints=seq(2, 162, by=16),
  outputFile=FALSE, ARSdefaultPer=85, outRawData=TRUE)
head(cyc$ARS)
head(cyc$JTK)
head(cyc$LS)
head(cyc$meta)
</code></pre>


</div>