<div class="container">

<table style="width: 100%;"><tr>
<td>IDDIRR</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Integration-driven Discovery and Integration-driven Revision Rates </h2>

<h3>Description</h3>

<p>Calculates the gain or the loss of differentially expressed genes due to meta-analysis
compared to individual studies.
</p>


<h3>Usage</h3>

<pre><code class="language-R">IDDIRR(finalde, deindst)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>finalde</code></td>
<td>
<p> Vector of indices of differentially expressed genes after meta-analysis </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>deindst</code></td>
<td>
<p> Vector of indices of differentially expressed genes found at least in one study </p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>DE </code></td>
<td>
<p>Number of Differentially Expressed (DE) genes </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>IDD </code></td>
<td>
<p>Integration Driven Discoveries: number of genes that are declared DE in
the meta-analysis that were not identified in any of
the individual studies alone.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Loss</code></td>
<td>
<p>Number of genes that are declared DE in individual studies but not in meta-analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>IDR</code></td>
<td>
<p>Integration-driven Discovery Rate: proportion of genes that are identified as DE in the meta-analysis that were not
identified in any of the individual studies alone.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>IRR</code></td>
<td>
<p>Integration-driven Revision Rate: percentage of genes
that are declared DE in individual studies but not in meta-analysis.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p> Guillemette Marot </p>


<h3>References</h3>

<p>  Marot, G., Foulley, J.-L., Mayer, C.-D., Jaffrezic, F. (2009) Moderated effect size and p-value combinations for microarray meta-analyses. Bioinformatics. 25 (20): 2692-2699. </p>


<h3>Examples</h3>

<pre><code class="language-R">data(Singhdata)
out=EScombination(esets=Singhdata$esets,classes=Singhdata$classes)
IDDIRR(out$Meta,out$AllIndStudies)

## The function is currently defined as
#function(finalde,deindst)
#{
#DE=length(finalde)
#gains=finalde[which(!(finalde %in% deindst))]
#IDD=length(gains)
#IDR=IDD/DE*100
#perte=which(!(deindst %in% finalde))
#Loss=length(perte)
#IRR=Loss/length(deindst)*100
#res=c(DE,IDD,Loss,IDR,IRR)
#names(res)=c("DE","IDD","Loss","IDR","IRR")
#res
#}
</code></pre>


</div>