<div class="container">

<table style="width: 100%;"><tr>
<td>metaGE.lscore</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute the local score from a set of pvalues.</h2>

<h3>Description</h3>

<p>The function metaGE.lscore computes the local score and the significant regions from
a set of pvalues.
</p>


<h3>Usage</h3>

<pre><code class="language-R">metaGE.lscore(Data, PvalName, xi)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Data</code></td>
<td>
<p>A dataset containing the following columns: CHR, POS, MARKER and <code>PvalName</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>PvalName</code></td>
<td>
<p>The name of the column containing the p-value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xi</code></td>
<td>
<p>The threshold of the score, <code>xi</code> = 1,2,3 or 4.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is directly inherited from the scorelocalfunctions.R R code file of Fariello MI, Boitard S, Mercier S, et al.,
as available on the <a href="https://forge-dga.jouy.inra.fr/projects/local-score">https://forge-dga.jouy.inra.fr/projects/local-score</a> website.
The technical details of the computation can be found in Fariello MI, Boitard S, Mercier S, et al.
Accounting for linkage disequilibrium in genome scans for selection without individual genotypes: The local score approach.
<a href="https://doi.org/10.1111/mec.14141">doi:10.1111/mec.14141</a>.
The function computes a local score for the detection of significant regions based on the hypothesis that the H0 distribution of the pvalues is
uniform. Under this hypothesis the local score follows a Gumbel distribution (under H0) whose parameters depend
on the threshold <code>xi</code> and on the autocorrelation between pvalues within each chromosome. The threshold has
to be selected in 1,2,3,4 and the autocorrelation is computed internally.
</p>


<h3>Value</h3>

<p>A list with the following elements:
</p>

<table>
<tr>
<td style="text-align: left;">
<code>Data</code> </td>
<td style="text-align: left;"> The dataset <code>Data</code> with the local score as supplementary column.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>SigZones</code> </td>
<td style="text-align: left;"> A dataset containing information about the significant regions.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>SigMarker</code> </td>
<td style="text-align: left;"> A dataset containing the significant markers. </td>
</tr>
<tr>
<td style="text-align: left;">
<code>ChrThreshold</code> </td>
<td style="text-align: left;"> A dataset containing the chromosome-wide significance thresholds.
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">require(dplyr)
## Not run: 
# Import the data
data("metaData")

# Compute the inter-environment correlation matrix
matCorr &lt;- metaGE.cor(metaData, Threshold = 0.8)

# Fit the Fixed Effect model
FeDF &lt;- metaGE.fit(metaData, matCorr, Method = "Fe")

# Compute the score local
xi &lt;- 2
FeScore &lt;- metaGE.lscore(FeDF,"PVALUE", xi)
#FeScore$SigZones
## End(Not run)
</code></pre>


</div>