<div class="container">

<table style="width: 100%;"><tr>
<td>IDD.IRR</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Integration-driven Discovery and Integration-driven Revision Rates </h2>

<h3>Description</h3>

<p>Calculates the gain or the loss of differentially expressed genes due to meta-analysis
compared to individual studies.
</p>


<h3>Usage</h3>

<pre><code class="language-R">IDD.IRR(meta_de, ind_de)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>meta_de</code></td>
<td>
<p> Vector of differentially expressed tags (or indices of these tags) with the meta-analysis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind_de</code></td>
<td>
<p> List of vectors storing differentially expressed tags (or indices of these tags) in each individual study </p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>DE </code></td>
<td>
<p>Number of Differentially Expressed (DE) genes </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>IDD </code></td>
<td>
<p>Integration Driven Discoveries: number of genes that are declared DE in
the meta-analysis that were not identified in any of
the individual studies alone.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Loss</code></td>
<td>
<p>Number of genes that are declared DE in individual studies but not in meta-analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>IDR</code></td>
<td>
<p>Integration-driven Discovery Rate: proportion of genes that are identified as DE in the meta-analysis that were not
identified in any of the individual studies alone.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>IRR</code></td>
<td>
<p>Integration-driven Revision Rate: percentage of genes
that are declared DE in individual studies but not in meta-analysis.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p> Guillemette Marot </p>


<h3>References</h3>

<p>  Marot, G., Foulley, J.-L., Mayer, C.-D., Jaffrezic, F. (2009) Moderated effect size and p-value combinations for microarray meta-analyses. Bioinformatics. 25 (20): 2692-2699. </p>


<h3>Examples</h3>

<pre><code class="language-R">data(rawpval)
adjpval&lt;-lapply(rawpval, FUN=function(x) p.adjust(x, method="BH"))
ind_smalladjp&lt;-lapply(adjpval, FUN=function(x) which(x &lt;= 0.05))
#indicators corresponding to the inverse normal p-value combination
invnormcomb &lt;- invnorm(rawpval,nrep=c(8,8), BHth = 0.05) 
IDD.IRR(invnormcomb$DEindices,ind_smalladjp)
#indicators corresponding to the p-value combination with Fisher's method
fishcomb &lt;- fishercomb(rawpval, BHth = 0.05)
IDD.IRR(fishcomb$DEindices,ind_smalladjp)
</code></pre>


</div>