<div class="container">

<table style="width: 100%;"><tr>
<td>metalonda</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Metagenomic Longitudinal Differential Abundance Analysis for one feature</h2>

<h3>Description</h3>

<p>Find significant time intervals of the one feature
</p>


<h3>Usage</h3>

<pre><code class="language-R">metalonda(
  Count,
  Time,
  Group,
  ID,
  n.perm = 500,
  fit.method = "nbinomial",
  points,
  text = 0,
  parall = FALSE,
  pvalue.threshold = 0.05,
  adjust.method = "BH",
  time.unit = "days",
  ylabel = "Normalized Count",
  col = c("blue", "firebrick"),
  prefix = "Test"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Count</code></td>
<td>
<p>matrix has the number of reads that mapped to each feature in each sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Time</code></td>
<td>
<p>vector of the time label of each sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Group</code></td>
<td>
<p>vector of the group label of each sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ID</code></td>
<td>
<p>vector of the subject ID label of each sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.perm</code></td>
<td>
<p>number of permutations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fit.method</code></td>
<td>
<p>fitting method (nbinomial, lowess).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>points</code></td>
<td>
<p>points at which the prediction should happen.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>text</code></td>
<td>
<p>Feature's name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parall</code></td>
<td>
<p>boolean to indicate whether to use multicore.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvalue.threshold</code></td>
<td>
<p>p-value threshold cutoff for identifing significant time intervals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adjust.method</code></td>
<td>
<p>multiple testing correction method.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>time.unit</code></td>
<td>
<p>time unit used in the Time vector (hours, days, weeks, months, etc.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylabel</code></td>
<td>
<p>text to be shown on the y-axis of all generated figures (default: "Normalized Count")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>two color to be used for the two groups (eg., c("red", "blue")).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prefix</code></td>
<td>
<p>prefix to be used to create directory for the analysis results</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>returns a list of the significant time intervals for the tested feature.
</p>


<h3>References</h3>

<p>Ahmed Metwally (ametwall@stanford.edu)
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(metalonda_test_data)
n.sample = 5
n.timepoints = 10
n.group = 2
Group = factor(c(rep(0, n.sample*n.timepoints), rep(1,n.sample*n.timepoints)))
Time = rep(rep(1:n.timepoints, times = n.sample), 2)
ID = factor(rep(1:(2*n.sample), each = n.timepoints))
points = seq(1, 10, length.out = 10)
## Not run: 
output.nbinomial = metalonda(Count = metalonda_test_data[1,], Time = Time, Group = Group,
ID = ID, fit.method =  "nbinomial", n.perm = 10, points = points,
text = rownames(metalonda_test_data)[1], parall = FALSE, pvalue.threshold = 0.05, 
adjust.method = "BH", time.unit = "hours", ylabel = "Normalized Count", col = c("black", "green"))

## End(Not run)
</code></pre>


</div>