<div class="container">

<table style="width: 100%;"><tr>
<td>MetaSKAT_MSSD_OneSet</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Meta analysis SKAT with summary data from each study cohort.</h2>

<h3>Description</h3>

<p>Meta analysis SKAT with Meta SSD (MSSD) and Info (MInfo) files. MetaSKAT_MSSD_OneSet computes a p-value for a given set,
and MetaSKAT_MSSD_ALL computes p-values for all sets.
</p>


<h3>Usage</h3>

<pre><code class="language-R">
MetaSKAT_MSSD_OneSet(Cohort.Info, SetID, combined.weight=TRUE, weights.beta=c(1,25),
method="davies", r.corr=0, is.separate = FALSE, Group_Idx=NULL, MAF.cutoff=1, 
missing_cutoff=0.15)

MetaSKAT_MSSD_ALL(Cohort.Info, ...)

 </code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Cohort.Info</code></td>
<td>
<p>output object from Open_MSSD_File_2Read. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SetID</code></td>
<td>
<p>a character value of set id to test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>combined.weight</code></td>
<td>
<p>a logical value (default=TRUE) for a type of weighting. See MetaSKAT_wZ page for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights.beta</code></td>
<td>
<p>a numeric vector of parameters for the beta weights (default=c(1,25))</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>a method to compute a p-value (default= "davies"). See MetaSKAT_wZ page for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r.corr</code></td>
<td>
<p>the <code class="reqn">\rho</code> parameter for the compound symmetric correlation structure kernels (default= 0). See MetaSKAT_wZ page for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is.separate</code></td>
<td>
<p>a logical value (default=FALSE) for homogeneous(=FALSE) or heterogeneous(=TRUE) genetic effects
of a SNP set across studies. See MetaSKAT_wZ page for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Group_Idx</code></td>
<td>
<p>a vector of group indicator (default=NULL). See MetaSKAT_wZ page for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF.cutoff</code></td>
<td>
<p>a cutoff of the MAFs of SNPs (default=1). Any SNPs with MAFs &gt; MAF.cutoff will be excluded from the analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>missing_cutoff</code></td>
<td>
<p>a cutoff of the missing rates of SNPs (default=0.15). See MetaSKAT_wZ page for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>the same parameters of MetaSKAT_MSSD_OneSet after SetID.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Please see MetaSKAT_wZ for details.
</p>


<h3>Value</h3>

<p>MetaSKAT_MSSD_OneSet and MetaSKAT_wZ return the same object. See MetaSKAT_wZ for details.
MetaSKAT_MSSD_ALL returns a dataframe with SetIDs (first column) and p-values (second column). 
</p>


<h3>Author(s)</h3>

<p>Seunggeun Lee</p>


<h3>References</h3>

<p>Lee, S., Teslovich, T., Boehnke, M., Lin, X. (2013)
General framework for meta-analysis of rare variants in sequencing association studies.  <em>American Journal of Human Genetics</em>, 93, 42-53. 
</p>


</div>