<div class="container">

<table style="width: 100%;"><tr>
<td>metaGE.collect</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Collect the results of GWAS data from different files</h2>

<h3>Description</h3>

<p>This function merges files containing the summary statistics of GWAS in different environments (one file per environment).
</p>


<h3>Usage</h3>

<pre><code class="language-R">metaGE.collect(
  FileNames,
  VariableNames,
  MinFreq = 0,
  DropDuplicates = TRUE,
  Verbose = FALSE,
  NA.rmv = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>FileNames</code></td>
<td>
<p>A list containing the file paths to merge (one trait only) or a list of such lists</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>VariableNames</code></td>
<td>
<p>A named list containing the column names in the original files corresponding
to the variables : MARKER, CHR, POS, EFFECT, PVAL (optional: FREQ, ALLELE0, ALLELE1) ; or a list of such lists.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MinFreq</code></td>
<td>
<p>A numeric value allowing to filter markers based on the maf. (optional)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>DropDuplicates</code></td>
<td>
<p>A boolean indicating whether duplicate markers should be removed or not. (<code>TRUE</code> by default)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Verbose</code></td>
<td>
<p>A boolean indicating whether progression messages should be printed or not. (<code>FALSE</code> by default)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>NA.rmv</code></td>
<td>
<p>A boolean indicating if the <code>NA</code> should be removed or not (<code>TRUE</code> by default)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Each file MUST contain the variables below:
</p>

<ul>
<li>
<p> MARKER: the marker name
</p>
</li>
<li>
<p> CHR: the chromosome
</p>
</li>
<li>
<p> POS: the position of the marker
</p>
</li>
<li>
<p> EFFECT: the mean effect of the marker
</p>
</li>
<li>
<p> PVAL: the pvalue
</p>
</li>
</ul>
<p>Each file might contain the variables:
</p>

<ul>
<li>
<p> FREQ: MAF
</p>
</li>
<li>
<p> ALLELE0: Allele coding for allele 0
</p>
</li>
<li>
<p> ALLELE1: Allele coding for allele 1
</p>
</li>
</ul>
<h3>Value</h3>

<p>A list with the following elements:
</p>

<table>
<tr>
<td style="text-align: left;">
<code>Data</code> </td>
<td style="text-align: left;"> A tibble containing all the columns of interest of all the files from FileNames.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>RemovedMarkers</code> </td>
<td style="text-align: left;"> Same kind of tibble, but containing the markers that have been removed
due to unclear allele coding, maf filtering or duplicates dropping.
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">require(dplyr)
require(tibble)
require(stringr)
RepData &lt;- system.file("extdata", package = "metaGE")
# Get the complete list of association files
File.list &lt;- list.files(RepData ,full.names = TRUE) %&gt;%
            tibble(Names = .) %&gt;%
            mutate(ShortNames = Names %&gt;%
                  str_remove(pattern = paste0(RepData,"/")) %&gt;%
                  str_remove(pattern = "_DF.txt"))  %&gt;%
            select(ShortNames,Names) %&gt;%
            deframe
###Build the dataset
## First provide the list of variable names
Names.list &lt;- list(MARKER="Marker_Name",
                  CHR="Chromosome",
                  POS="Marker_Position",
                  FREQ="Maf",
                  EFFECT="SNP_Weight",
                  PVAL="Pvalue",
                  ALLELE0="Allele1",
                  ALLELE1="Allele2")

MinFreq &lt;- 0.07

## Now collect
metaData &lt;- metaGE.collect(File.list, Names.list,MinFreq = MinFreq)
</code></pre>


</div>