<div class="container">

<table style="width: 100%;"><tr>
<td>pvalcombination.paired</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> P-value combination for paired data </h2>

<h3>Description</h3>

<p>Calculates differential expression p-values from paired data either from classical or moderated
t-tests (Limma, SMVar) for each study and combines these p-values by the inverse normal method.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pvalcombination.paired(logratios, moderated = c("limma", "SMVar", "t")[1], BHth = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>logratios</code></td>
<td>
<p> List of matrices. Each matrix has one row per gene and one column per replicate and gives the logratios of one study. All studies must have the same genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>moderated</code></td>
<td>
<p> Method to calculate the test statistic inside each study from which the effect size is computed. <code>moderated</code> has to be chosen between "limma", "SMVar" and "t". </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BHth</code></td>
<td>
<p> Benjamini Hochberg threshold. By default, the False Discovery Rate is controlled at 5%. </p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>List
</p>
<table>
<tr style="vertical-align: top;">
<td><code>Study1</code></td>
<td>
<p>Vector of indices of differentially expressed genes in study 1. Similar names are given for the other individual studies.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>AllIndStudies</code></td>
<td>
<p>Vector of indices of differentially expressed genes found by at least one of the individual studies.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Meta </code></td>
<td>
<p>Vector of indices of differentially expressed genes in the meta-analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>TestStatistic</code></td>
<td>
<p>Vector with test statistics for differential expression in the meta-analysis.</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>While the invisible object resulting from this function contains 
the values described previously, other quantities of interest are printed: 
DE,IDD,Loss,IDR,IRR.
All these quantities are defined in function <code>IDDIDR</code> and in (Marot et al., 2009) 
</p>


<h3>Author(s)</h3>

<p> Guillemette Marot </p>


<h3>References</h3>

<p>  Marot, G., Foulley, J.-L., Mayer, C.-D., Jaffrezic, F. (2009) Moderated effect size and p-value combinations for microarray meta-analyses. Bioinformatics. 25 (20): 2692-2699. </p>


<h3>Examples</h3>

<pre><code class="language-R">data(Singhdata)
#create artificially paired data:
artificialdata=lapply(Singhdata$esets,FUN=function(x) (x[,1:10]-x[,11:20]))
#Meta-analysis
res=pvalcombination.paired(artificialdata)
#Number of differentially expressed genes in the meta-analysis
length(res$Meta)
#To plot an histogram of raw p-values
rawpval=2*(1-pnorm(abs(res$TestStatistic)))
hist(rawpval,nclass=100)
</code></pre>


</div>