<div class="container">

<table style="width: 100%;"><tr>
<td>directEScombi</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Direct effect size combination </h2>

<h3>Description</h3>

<p>Combines effect sizes already calculated.
</p>


<h3>Usage</h3>

<pre><code class="language-R">directEScombi(ES, varES, BHth = 0.05, useREM = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ES</code></td>
<td>
<p>Matrix of effect sizes. Each column of <code>ES</code> corresponds to one study and each row to one gene. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>varES</code></td>
<td>
<p> Matrix of effect size variances. Each column of <code>varES</code> corresponds to one study and each row to one gene. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BHth</code></td>
<td>
<p> Benjamini Hochberg threshold. By default, the False Discovery Rate is controlled at 5%. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useREM</code></td>
<td>
<p> A logical value indicating whether or not to include the between-study variance into the model. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Combines effect sizes with the method presented in (Choi et al., 2003).
</p>


<h3>Value</h3>

<p>List
</p>
<table>
<tr style="vertical-align: top;">
<td><code>DEindices</code></td>
<td>
<p> Indices of differentially expressed genes at the chosen Benjamini Hochberg threshold.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>TestStatistic</code></td>
<td>
<p>Vector with test statistics for differential expression in the meta-analysis.</p>
</td>
</tr>
</table>
<h3>References</h3>

<p> Choi, J. K., Yu, U., Kim, S., and Yoo, O. J. (2003). Combining multiple microarray
studies and modeling interstudy variation. Bioinformatics, 19 Suppl 1. </p>


</div>