<div class="container">

<table style="width: 100%;"><tr>
<td>fishercomb</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
P-value combination using Fisher's method
</h2>

<h3>Description</h3>

<p>Combines one sided p-values using Fisher's method.
</p>


<h3>Usage</h3>

<pre><code class="language-R">fishercomb(indpval, BHth = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>indpval</code></td>
<td>

<p>List of vectors of one sided p-values to be combined.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BHth</code></td>
<td>

<p>Benjamini Hochberg threshold. By default, the False Discovery Rate is controlled at 5%.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The test statistic for each gene <em>g</em> is defined as
</p>
<p style="text-align: center;"><code class="reqn">F_g = -2 \sum_{s=1}^S ln(p_{gs})</code>
</p>

<p>where <code class="reqn">p_{gs}</code> corresponds to the raw <em>p</em>-value obtained for gene <em>g</em> in a differential
analysis for study <em>s</em> (assumed to be uniformly distributed under the null hypothesis). Under the
null hypothesis, the test statistic <code class="reqn">F_g</code> follows a <code class="reqn">\chi^2</code> distribution with <em>2S</em>
degrees of freedom. Classical procedures for the correction of multiple testing, such as that of Benjamini
and Hochberg (1995) may subsequently be applied to the obtained <em>p</em>-values to control the false
discovery rate at a desired rate <code class="reqn">\alpha</code>. 
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>DEindices </code></td>
<td>
<p>Indices of differentially expressed genes at the chosen Benjamini Hochberg threshold.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>TestStatistic </code></td>
<td>
<p>Vector with test statistics for differential expression in the meta-analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rawpval </code></td>
<td>
<p>Vector with raw p-values for differential expression in the meta-analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adjpval </code></td>
<td>
<p>Vector with adjusted p-values for differential expression in the meta-analysis.</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>Y. Benjamini and Y. Hochberg (1995). Controlling the false discovery rate: a pratical and powerful approach
to multiple testing. <em>JRSS B</em> (57): 289-300.
</p>
<p>M. Brown (1975). A method for combining non-independent, one-sided tests of significance. <em>Biometrics</em> <b>31</b>(4): 987-992.
</p>
<p>A. Rau, G. Marot and F. Jaffrezic (2014). Differential meta-analysis of RNA-seq data. <em>BMC Bioinformatics</em> <b>15</b>:91
</p>


<h3>See Also</h3>

<p><code>metaRNASeq</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(rawpval)
fishcomb &lt;- fishercomb(rawpval, BHth = 0.05)
DE &lt;- ifelse(fishcomb$adjpval&lt;=0.05,1,0)
hist(fishcomb$rawpval,nclass=100)

## A more detailed example is given in the vignette of the package:
## vignette("metaRNASeq")
</code></pre>


</div>