<div class="container">

<table style="width: 100%;"><tr>
<td>singleStructureGenerator</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>singleStructureGenerator</h2>

<h3>Description</h3>

<p>an R6 class representing a single genomic structure
</p>


<h3>Methods</h3>



<h4>Public methods</h4>


<ul>
<li> <p><a href="#method-singleStructureGenerator-init_neighbSt"><code>singleStructureGenerator$init_neighbSt()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-initialize_ratetree"><code>singleStructureGenerator$initialize_ratetree()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-new"><code>singleStructureGenerator$new()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_seq"><code>singleStructureGenerator$get_seq()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_seqFirstPos"><code>singleStructureGenerator$get_seqFirstPos()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_seq2ndPos"><code>singleStructureGenerator$get_seq2ndPos()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_seqLastPos"><code>singleStructureGenerator$get_seqLastPos()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_seq2ndButLastPos"><code>singleStructureGenerator$get_seq2ndButLastPos()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_combiStructure_index"><code>singleStructureGenerator$get_combiStructure_index()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-update_interStr_firstNeighbSt"><code>singleStructureGenerator$update_interStr_firstNeighbSt()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-update_interStr_lastNeighbSt"><code>singleStructureGenerator$update_interStr_lastNeighbSt()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_eqFreqs"><code>singleStructureGenerator$get_eqFreqs()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-SSE_evol"><code>singleStructureGenerator$SSE_evol()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_transMat"><code>singleStructureGenerator$get_transMat()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-IWE_evol"><code>singleStructureGenerator$IWE_evol()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_alpha_pI"><code>singleStructureGenerator$get_alpha_pI()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_beta_pI"><code>singleStructureGenerator$get_beta_pI()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_alpha_mI"><code>singleStructureGenerator$get_alpha_mI()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_beta_mI"><code>singleStructureGenerator$get_beta_mI()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_alpha_pNI"><code>singleStructureGenerator$get_alpha_pNI()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_beta_pNI"><code>singleStructureGenerator$get_beta_pNI()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_alpha_mNI"><code>singleStructureGenerator$get_alpha_mNI()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_beta_mNI"><code>singleStructureGenerator$get_beta_mNI()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_alpha_Ri"><code>singleStructureGenerator$get_alpha_Ri()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_iota"><code>singleStructureGenerator$get_iota()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_Ri_values"><code>singleStructureGenerator$get_Ri_values()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_Q"><code>singleStructureGenerator$get_Q()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_siteR"><code>singleStructureGenerator$get_siteR()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-get_neighbSt"><code>singleStructureGenerator$get_neighbSt()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-update_ratetree_otherStr"><code>singleStructureGenerator$update_ratetree_otherStr()</code></a>
</p>
</li>
<li> <p><a href="#method-singleStructureGenerator-clone"><code>singleStructureGenerator$clone()</code></a>
</p>
</li>
</ul>
<hr>
<a id="method-singleStructureGenerator-init_neighbSt"></a>



<h4>Method <code>init_neighbSt()</code>
</h4>

<p>Public method: Initialization of $neighbSt
</p>
<p>This fuction initiates each CpG position $neighbSt as encoded in $mapNeighbSt_matrix
Positions at the edge of the entire simulated sequence use
their only neighbor as both neighbors.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$init_neighbSt()</pre></div>



<h5>Returns</h5>

<p>NULL
</p>


<hr>
<a id="method-singleStructureGenerator-initialize_ratetree"></a>



<h4>Method <code>initialize_ratetree()</code>
</h4>

<p>Public method: Initialization of $ratetree
</p>
<p>This function initializes $ratetree
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$initialize_ratetree()</pre></div>



<h5>Returns</h5>

<p>NULL
</p>


<hr>
<a id="method-singleStructureGenerator-new"></a>



<h4>Method <code>new()</code>
</h4>

<p>Create a new singleStructureGenerator object.
</p>
<p>Note that this object is typically generated withing a combiStructureGenerator object.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$new(
  globalState,
  n,
  eqFreqs = NULL,
  combiStr = NULL,
  combiStr_index = NULL,
  params = NULL,
  testing = FALSE
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>globalState</code></dt>
<dd>
<p>Character. Structure's favored global state: "M" for methylated (island structures) / "U" for unmethylated (non-island structures).</p>
</dd>
<dt><code>n</code></dt>
<dd>
<p>Numerical Value. Number of CpG positions</p>
</dd>
<dt><code>eqFreqs</code></dt>
<dd>
<p>Default NULL. When given: numerical vector with structure's methylation state equilibrium frequencies (for unmethylated, partially methylated and methylated)</p>
</dd>
<dt><code>combiStr</code></dt>
<dd>
<p>Default NULL. When initiated from combiStructureGenerator: object of class combiStructureGenerator containing it</p>
</dd>
<dt><code>combiStr_index</code></dt>
<dd>
<p>Default NULL. When initiated from combiStructureGenerator: index in Object of class combiStructureGenerator</p>
</dd>
<dt><code>params</code></dt>
<dd>
<p>Default NULL. When given: data frame containing model parameters</p>
</dd>
<dt><code>testing</code></dt>
<dd>
<p>Default FALSE. TRUE for testing output</p>
</dd>
</dl>
</div>



<h5>Returns</h5>

<p>A new <code>singleStructureGenerator</code> object.
</p>


<hr>
<a id="method-singleStructureGenerator-get_seq"></a>



<h4>Method <code>get_seq()</code>
</h4>

<p>Public method: Get object's methylation state sequence
</p>
<p>Encoded with 1 for unmethylated, 2 for partially methylated and 3 for methylated
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_seq()</pre></div>



<h5>Returns</h5>

<p>vector with equilibrium frequencies of unmethylated, partially methylated and methylated
</p>


<hr>
<a id="method-singleStructureGenerator-get_seqFirstPos"></a>



<h4>Method <code>get_seqFirstPos()</code>
</h4>

<p>Public method: Get first sequence position methylation state
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_seqFirstPos()</pre></div>



<h5>Returns</h5>

<p>numerical encoding of first position's methylation state
</p>


<hr>
<a id="method-singleStructureGenerator-get_seq2ndPos"></a>



<h4>Method <code>get_seq2ndPos()</code>
</h4>

<p>Public method: Get second sequence position methylation state
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_seq2ndPos()</pre></div>



<h5>Returns</h5>

<p>numerical encoding of second position's methylation state. NULL if position does not exist
</p>


<hr>
<a id="method-singleStructureGenerator-get_seqLastPos"></a>



<h4>Method <code>get_seqLastPos()</code>
</h4>

<p>Public method: Get first sequence position methylation state
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_seqLastPos()</pre></div>



<h5>Returns</h5>

<p>numerical encoding of first position's methylation state
</p>


<hr>
<a id="method-singleStructureGenerator-get_seq2ndButLastPos"></a>



<h4>Method <code>get_seq2ndButLastPos()</code>
</h4>

<p>Public method: Get second but last sequence position methylation state
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_seq2ndButLastPos()</pre></div>



<h5>Returns</h5>

<p>numerical encoding of second but last position's methylation state. NULL if position does not exist
</p>


<hr>
<a id="method-singleStructureGenerator-get_combiStructure_index"></a>



<h4>Method <code>get_combiStructure_index()</code>
</h4>

<p>Public method: Get index in object of class combiStructureGenerator
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_combiStructure_index()</pre></div>



<h5>Returns</h5>

<p>index in object of class combiStructureGenerator
</p>


<hr>
<a id="method-singleStructureGenerator-update_interStr_firstNeighbSt"></a>



<h4>Method <code>update_interStr_firstNeighbSt()</code>
</h4>

<p>Public method: Update neighbSt of next singleStructureGenerator object within combiStructureGenerator object
</p>
<p>This function is used when the last $seq position of a singleStructureGenerator object
changes methylation state to update the neighbSt position
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$update_interStr_firstNeighbSt(
  leftNeighbSt,
  rightNeighbSt
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>leftNeighbSt</code></dt>
<dd>
<p>$seq state of left neighbor (left neighbor is in previous singleStructureGenerator object)</p>
</dd>
<dt><code>rightNeighbSt</code></dt>
<dd>
<p>$seq state of right neighbor</p>
</dd>
</dl>
</div>



<h5>Returns</h5>

<p>NULL
</p>


<hr>
<a id="method-singleStructureGenerator-update_interStr_lastNeighbSt"></a>



<h4>Method <code>update_interStr_lastNeighbSt()</code>
</h4>

<p>Public method: Update neighbSt of previous singleStructureGenerator object within combiStructureGenerator object
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$update_interStr_lastNeighbSt(
  leftNeighbSt,
  rightNeighbSt
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>leftNeighbSt</code></dt>
<dd>
<p>$seq state of right neighbor (left neighbor is in next singleStructureGenerator object)</p>
</dd>
<dt><code>rightNeighbSt</code></dt>
<dd>
<p>$seq state of right neighbor</p>
</dd>
</dl>
</div>



<h5>Returns</h5>

<p>NULL
</p>


<hr>
<a id="method-singleStructureGenerator-get_eqFreqs"></a>



<h4>Method <code>get_eqFreqs()</code>
</h4>

<p>Public method: Get object's equilibrium Frequencies
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_eqFreqs()</pre></div>



<h5>Returns</h5>

<p>vector with equilibrium frequencies of unmethylated, partially methylated and methylated
</p>


<hr>
<a id="method-singleStructureGenerator-SSE_evol"></a>



<h4>Method <code>SSE_evol()</code>
</h4>

<p>Public method. Simulate how CpG dinucleotide methylation state changes due to the SSE process
along a time step of length dt
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$SSE_evol(dt, testing = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>dt</code></dt>
<dd>
<p>time step length.</p>
</dd>
<dt><code>testing</code></dt>
<dd>
<p>logical value for testing purposes. Default FALSE.</p>
</dd>
</dl>
</div>



<h5>Returns</h5>

<p>default NULL. If testing TRUE it returns a list with the number of events sampled and a
dataframe with the position(s) affected, new state and old methylation state.
</p>


<hr>
<a id="method-singleStructureGenerator-get_transMat"></a>



<h4>Method <code>get_transMat()</code>
</h4>

<p>Public Method. Get a transition matrix
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_transMat(
  old_eqFreqs,
  new_eqFreqs,
  info,
  testing = FALSE
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>old_eqFreqs</code></dt>
<dd>
<p>numeric vector with 3 frequency values (for old u, p and m)</p>
</dd>
<dt><code>new_eqFreqs</code></dt>
<dd>
<p>numeric vector with 3 frequency values (for new u, p and m)</p>
</dd>
<dt><code>info</code></dt>
<dd>
<p>character string to indicate where the method is being called</p>
</dd>
<dt><code>testing</code></dt>
<dd>
<p>logical value for testing purposes. Default FALSE.</p>
</dd>
</dl>
</div>



<h5>Details</h5>

<p>Given a tripple of old equilibrium frequencies and new equilibrium frequencies, generates the corresponding transition matrix.
</p>



<h5>Returns</h5>

<p>transMat. The transition matrix. If testing = TRUE it returns a list.
If there was a change in the equilibrium frequencies the list contains the following 7 elements, if not it contains the first 3 elements:
</p>

<dl>
<dt><code>transMat</code></dt>
<dd>
<p>transition matrix</p>
</dd>
<dt><code>case</code></dt>
<dd>
<p>The applied case.</p>
</dd>
</dl>
<hr>
<a id="method-singleStructureGenerator-IWE_evol"></a>



<h4>Method <code>IWE_evol()</code>
</h4>

<p>Public Method. Simulate IWE Events
</p>
<p>Simulates how CpG Islands' methylation state frequencies change and simultaneous sites change methylation state
along a branch of length t according to the SSE-IWE model.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$IWE_evol(testing = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>testing</code></dt>
<dd>
<p>logical value for testing purposes. Default FALSE.</p>
</dd>
</dl>
</div>



<h5>Details</h5>

<p>The function checks if the methylation equilibrium frequencies (<code>eqFreqs</code>) and sequence observed
frequencies (<code>obsFreqs</code>) change after the IWE event. If there is a change in either
frequencies, the corresponding change flag <code>eqFreqsChange</code>
in the <code>infoIWE</code> list will be set to <code>TRUE</code>.
</p>



<h5>Returns</h5>

<p>If testing = TRUE it returns a list.
If there was a change in the equilibrium frequencies the list contains the following 7 elements, if not it contains the first 3 elements:
</p>

<dl>
<dt><code>eqFreqsChange</code></dt>
<dd>
<p>logical indicating if there was a change in the equilibrium frequencies.</p>
</dd>
<dt><code>old_eqFreqs</code></dt>
<dd>
<p>Original equilibrium frequencies before the IWE event.</p>
</dd>
<dt><code>new_eqFreqs</code></dt>
<dd>
<p>New equilibrium frequencies after the IWE event.</p>
</dd>
<dt><code>old_obsFreqs</code></dt>
<dd>
<p>Original observed frequencies before the IWE event.</p>
</dd>
<dt><code>new_obsFreqs</code></dt>
<dd>
<p>New observed frequencies after the IWE event.</p>
</dd>
<dt><code>IWE_case</code></dt>
<dd>
<p>Description of the IWE event case.</p>
</dd>
<dt><code>Mk</code></dt>
<dd>
<p>Transition matrix used for the IWE event.</p>
</dd>
</dl>
<hr>
<a id="method-singleStructureGenerator-get_alpha_pI"></a>



<h4>Method <code>get_alpha_pI()</code>
</h4>

<p>Public Method.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_alpha_pI()</pre></div>



<h5>Returns</h5>

<p>Model parameter alpha_pI for sampling island equilibrium frequencies
</p>


<hr>
<a id="method-singleStructureGenerator-get_beta_pI"></a>



<h4>Method <code>get_beta_pI()</code>
</h4>

<p>Public Method.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_beta_pI()</pre></div>



<h5>Returns</h5>

<p>Model parameter for sampling island equilibrium frequencies
</p>


<hr>
<a id="method-singleStructureGenerator-get_alpha_mI"></a>



<h4>Method <code>get_alpha_mI()</code>
</h4>

<p>Public Method.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_alpha_mI()</pre></div>



<h5>Returns</h5>

<p>Model parameter for sampling island equilibrium frequencies
</p>


<hr>
<a id="method-singleStructureGenerator-get_beta_mI"></a>



<h4>Method <code>get_beta_mI()</code>
</h4>

<p>Public Method.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_beta_mI()</pre></div>



<h5>Returns</h5>

<p>Model parameter for sampling island equilibrium frequencies
</p>


<hr>
<a id="method-singleStructureGenerator-get_alpha_pNI"></a>



<h4>Method <code>get_alpha_pNI()</code>
</h4>

<p>Public Method.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_alpha_pNI()</pre></div>



<h5>Returns</h5>

<p>Model parameter for sampling non-island equilibrium frequencies
</p>


<hr>
<a id="method-singleStructureGenerator-get_beta_pNI"></a>



<h4>Method <code>get_beta_pNI()</code>
</h4>

<p>Public Method.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_beta_pNI()</pre></div>



<h5>Returns</h5>

<p>Model parameter for sampling non-island equilibrium frequencies
</p>


<hr>
<a id="method-singleStructureGenerator-get_alpha_mNI"></a>



<h4>Method <code>get_alpha_mNI()</code>
</h4>

<p>Public Method.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_alpha_mNI()</pre></div>



<h5>Returns</h5>

<p>Model parameter for sampling non-island equilibrium frequencies
</p>


<hr>
<a id="method-singleStructureGenerator-get_beta_mNI"></a>



<h4>Method <code>get_beta_mNI()</code>
</h4>

<p>Public Method.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_beta_mNI()</pre></div>



<h5>Returns</h5>

<p>Model parameter for sampling non-island equilibrium frequencies
</p>


<hr>
<a id="method-singleStructureGenerator-get_alpha_Ri"></a>



<h4>Method <code>get_alpha_Ri()</code>
</h4>

<p>Public Method.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_alpha_Ri()</pre></div>



<h5>Returns</h5>

<p>Model parameter for gamma distribution shape to initialize the 3 $Ri_values
</p>


<hr>
<a id="method-singleStructureGenerator-get_iota"></a>



<h4>Method <code>get_iota()</code>
</h4>

<p>Public Method.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_iota()</pre></div>



<h5>Returns</h5>

<p>Model parameter for gamma distribution expected value to initialize the 3 $Ri_values
</p>


<hr>
<a id="method-singleStructureGenerator-get_Ri_values"></a>



<h4>Method <code>get_Ri_values()</code>
</h4>

<p>Public Method.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_Ri_values()</pre></div>



<h5>Returns</h5>

<p>The 3 $Ri_values
</p>


<hr>
<a id="method-singleStructureGenerator-get_Q"></a>



<h4>Method <code>get_Q()</code>
</h4>

<p>Public Method.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_Q(
  siteR = NULL,
  neighbSt = NULL,
  oldSt = NULL,
  newSt = NULL
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>siteR</code></dt>
<dd>
<p>default NULL. Numerical value encoding for the sites rate of independent SSE (1, 2 or 3)</p>
</dd>
<dt><code>neighbSt</code></dt>
<dd>
<p>default NULL. Numerical value encoding for the sites neighbouring state (as in mapNeighbSt_matrix)</p>
</dd>
<dt><code>oldSt</code></dt>
<dd>
<p>default NULL. Numerical value encoding for the sites old methylation state (1, 2 or 3)</p>
</dd>
<dt><code>newSt</code></dt>
<dd>
<p>default NULL. Numerical value encoding for the sites new methylation state (1, 2 or 3)</p>
</dd>
</dl>
</div>



<h5>Returns</h5>

<p>With NULL arguments, the list of rate matrices. With non NULL arguments, the corresponding rate of change.
</p>


<hr>
<a id="method-singleStructureGenerator-get_siteR"></a>



<h4>Method <code>get_siteR()</code>
</h4>

<p>Public Method.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_siteR(index = NULL)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>index</code></dt>
<dd>
<p>default NULL. Numerical value for the index of the CpG position within the singleStr instance</p>
</dd>
</dl>
</div>



<h5>Returns</h5>

<p>with NULL arguments, siteR vector. non NULL arguments, the corresponding siteR
</p>


<hr>
<a id="method-singleStructureGenerator-get_neighbSt"></a>



<h4>Method <code>get_neighbSt()</code>
</h4>

<p>Public Method.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$get_neighbSt(index = NULL)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>index</code></dt>
<dd>
<p>default NULL. Numerical value for the index of the CpG position within the singleStr instance</p>
</dd>
</dl>
</div>



<h5>Returns</h5>

<p>with NULL arguments, neighbSt vector. non NULL arguments, the corresponding neighbSt
</p>


<hr>
<a id="method-singleStructureGenerator-update_ratetree_otherStr"></a>



<h4>Method <code>update_ratetree_otherStr()</code>
</h4>

<p>Public Method. Update ratetree from another singleStructure instance
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$update_ratetree_otherStr(position, rate)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>position</code></dt>
<dd>
<p>Numerical value for the index of the CpG position within the singleStr instance</p>
</dd>
<dt><code>rate</code></dt>
<dd>
<p>Rate of change to asign to that position</p>
</dd>
</dl>
</div>



<h5>Returns</h5>

<p>NULL
</p>


<hr>
<a id="method-singleStructureGenerator-clone"></a>



<h4>Method <code>clone()</code>
</h4>

<p>The objects of this class are cloneable with this method.
</p>


<h5>Usage</h5>

<div class="r"><pre>singleStructureGenerator$clone(deep = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>deep</code></dt>
<dd>
<p>Whether to make a deep clone.</p>
</dd>
</dl>
</div>




</div>