<div class="container">

<table style="width: 100%;"><tr>
<td>treeMultiRegionSimulator</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>treeMultiRegionSimulator</h2>

<h3>Description</h3>

<p>an R6 class representing the methylation state of GpGs in different genomic
structures in the nodes of a tree.
</p>
<p>The whole CpG sequence is an object of class combiStructureGenerator.
Each genomic structure in it is contained in an object of class singleStructureGenerator.
</p>


<h3>Public fields</h3>

<div class="r6-fields">

<dl>
<dt><code>Branch</code></dt>
<dd>
<p>Public attribute: List containing objects of class combiStructureGenerator</p>
</dd>
<dt><code>branchLength</code></dt>
<dd>
<p>Public attribute: Vector with the corresponding branch lengths of each $Branch element</p>
</dd>
</dl>
</div>


<h3>Methods</h3>



<h4>Public methods</h4>


<ul>
<li> <p><a href="#method-treeMultiRegionSimulator-treeEvol"><code>treeMultiRegionSimulator$treeEvol()</code></a>
</p>
</li>
<li> <p><a href="#method-treeMultiRegionSimulator-new"><code>treeMultiRegionSimulator$new()</code></a>
</p>
</li>
<li> <p><a href="#method-treeMultiRegionSimulator-clone"><code>treeMultiRegionSimulator$clone()</code></a>
</p>
</li>
</ul>
<hr>
<a id="method-treeMultiRegionSimulator-treeEvol"></a>



<h4>Method <code>treeEvol()</code>
</h4>

<p>Simulate CpG dinucleotide methylation state evolution along a tree.
The function splits a given tree and simulates evolution along its
branches. It recursively simulates evolution in all of the subtrees in the given tree
until the tree leafs
</p>


<h5>Usage</h5>

<div class="r"><pre>treeMultiRegionSimulator$treeEvol(
  Tree,
  dt = 0.01,
  parent_index = 1,
  testing = FALSE
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>Tree</code></dt>
<dd>
<p>String. Tree in Newick format. When called recursivelly it is given the corresponding subtree.</p>
</dd>
<dt><code>dt</code></dt>
<dd>
<p>Length of SSE time steps.</p>
</dd>
<dt><code>parent_index</code></dt>
<dd>
<p>Default 1. When called recursivelly it is given the corresponding parent branch index.</p>
</dd>
<dt><code>testing</code></dt>
<dd>
<p>Default FALSE. TRUE for testing purposes.</p>
</dd>
</dl>
</div>



<h5>Returns</h5>

<p>NULL
</p>


<hr>
<a id="method-treeMultiRegionSimulator-new"></a>



<h4>Method <code>new()</code>
</h4>

<p>Create a new treeMultiRegionSimulator object.
$Branch is a list for the tree branches, its first element represents the tree root.
</p>
<p>Note that one of either infoStr or rootData needs to be given. Not both, not neither.
</p>


<h5>Usage</h5>

<div class="r"><pre>treeMultiRegionSimulator$new(
  infoStr = NULL,
  rootData = NULL,
  tree,
  params = NULL,
  dt = 0.01,
  testing = FALSE
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>infoStr</code></dt>
<dd>
<p>A data frame containing columns 'n' for the number of sites, and 'globalState' for the favoured global methylation state.
If initial equilibrium frequencies are given the dataframe must contain 3 additional columns: 'u_eqFreq', 'p_eqFreq' and 'm_eqFreq'</p>
</dd>
<dt><code>rootData</code></dt>
<dd>
<p>combiStructureGenerator object. When given, the simulation uses its parameter values.</p>
</dd>
<dt><code>tree</code></dt>
<dd>
<p>tree</p>
</dd>
<dt><code>params</code></dt>
<dd>
<p>Default NULL. When given: data frame containing model parameters. Note that rootData is given, its parameter values are used.</p>
</dd>
<dt><code>dt</code></dt>
<dd>
<p>length of the dt time steps for the SSE evolutionary process</p>
</dd>
<dt><code>testing</code></dt>
<dd>
<p>Default FALSE. TRUE for testing output.</p>
</dd>
</dl>
</div>



<h5>Returns</h5>

<p>A new <code>treeMultiRegionSimulator</code> object.
</p>


<hr>
<a id="method-treeMultiRegionSimulator-clone"></a>



<h4>Method <code>clone()</code>
</h4>

<p>The objects of this class are cloneable with this method.
</p>


<h5>Usage</h5>

<div class="r"><pre>treeMultiRegionSimulator$clone(deep = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>deep</code></dt>
<dd>
<p>Whether to make a deep clone.</p>
</dd>
</dl>
</div>




</div>