<div class="container">

<table style="width: 100%;"><tr>
<td>cubeClvR2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Inheritance Cube: 2-Locus ClvR (Cleave and Rescue)</h2>

<h3>Description</h3>

<p>Based on the Cleave-and-Rescue system of <a href="https://doi.org/10.1101/2020.07.09.196253">Oberhofer</a>,
this is a 3-locus Cas9-based toxin-antidote system. The first locus carries the
Cas9, the second locus carries the gRNAs, and a recoded copy of an essential gene.
The third locus is the targeted essential gene. This gene can be completely
haplosufficient (<code>hSuf</code> = 1) or completely haploinsufficient (<code>hSuf</code> = 0).
It is assumed that having 2 copies of the gene (be it wild-type at the second
locus or recoded at the first) confers complete viability. Additionally, loci
1 and 2 can be linked, given <code>crM</code> and <code>crF</code>, imitating the original
2-locus ClvR system.
For this construct, the first locus will have 2 alleles, the second will have 2
alleles, and the third will have 3 alleles:
</p>

<ul>
<li>
<p> Locus 1
</p>

<ul>
<li>
<p> W: Wild-type
</p>
</li>
<li>
<p> C: Cas9
</p>
</li>
</ul>
</li>
<li>
<p> Locus 2
</p>

<ul>
<li>
<p> W: Wild-type
</p>
</li>
<li>
<p> G: gRNAs and recoded essential gene
</p>
</li>
</ul>
</li>
<li>
<p> Locus 3
</p>

<ul>
<li>
<p> W: Wild-type
</p>
</li>
<li>
<p> R: Functional resistant
</p>
</li>
<li>
<p> B: Non-functional resistant
</p>
</li>
</ul>
</li>
</ul>
<h3>Usage</h3>

<pre><code class="language-R">cubeClvR2(
  cF = 1,
  crF = 0,
  ccF = cF,
  ccrF = crF,
  cM = 1,
  crM = 0,
  ccM = cM,
  ccrM = crM,
  dW = 0,
  drW = 0,
  ddW = dW,
  ddrW = drW,
  hSuf = 1,
  crossF = 0.5,
  crossM = 0.5,
  eta = NULL,
  phi = NULL,
  omega = NULL,
  xiF = NULL,
  xiM = NULL,
  s = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cF</code></td>
<td>
<p>Female cutting rate, one ClvR allele</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>crF</code></td>
<td>
<p>Female functional resistance rate, one ClvR allele</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ccF</code></td>
<td>
<p>Female cutting rate, two ClvR alleles</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ccrF</code></td>
<td>
<p>Female functional resistance rate, two ClvR alleles</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cM</code></td>
<td>
<p>Male cutting rate, one ClvR allele</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>crM</code></td>
<td>
<p>Male functional resistance rate, one ClvR allele</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ccM</code></td>
<td>
<p>Male cutting rate, two ClvR alleles</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ccrM</code></td>
<td>
<p>Male functional resistance rate, two ClvR alleles</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dW</code></td>
<td>
<p>Female deposition cutting rate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>drW</code></td>
<td>
<p>Female deposition functional resistance rate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ddW</code></td>
<td>
<p>Female deposition (HH) cutting rate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ddrW</code></td>
<td>
<p>Female deposition (HH) functional resistance rate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hSuf</code></td>
<td>
<p>Haplosufficiency level, default is completely sufficient</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>crossF</code></td>
<td>
<p>Crossover rate in females, 0 is completely linked, 0.5 is unlinked, 1.0 is complete divergence</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>crossM</code></td>
<td>
<p>Crossover rate in males, 0 is completely linked, 0.5 is unlinked, 1.0 is complete divergence</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eta</code></td>
<td>
<p>Genotype-specific mating fitness</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phi</code></td>
<td>
<p>Genotype-specific sex ratio at emergence</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>omega</code></td>
<td>
<p>Genotype-specific multiplicative modifier of adult mortality</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xiF</code></td>
<td>
<p>Genotype-specific female pupatory success</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xiM</code></td>
<td>
<p>Genotype-specific male pupatory success</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>s</code></td>
<td>
<p>Genotype-specific fractional reduction(increase) in fertility</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Female deposition is implemented incorrectly. Right now, it is performed on
male alleles prior to zygote formation - it should happen post-zygote formation.
Since this construct doesn't have HDR, this should be fine. <br>
Additionally, it is assumed that deposition requries loaded Cas9-RNP complexes
from the mother, having Cas9 and no maternal gRNA, even in the presence of
paternal gRNA, will not result in maternal deposition mediated cleavage.
</p>
<p>Copy-number dependent rates are based on Cas9, not gRNA. The assumption is that
RNA is easier to produce, and therefore won't limit cleavage by Cas9.
</p>


<h3>Value</h3>

<p>Named list containing the inheritance cube, transition matrix, genotypes, wild-type allele,
and all genotype-specific parameters.
</p>


</div>