<div class="container">

<table style="width: 100%;"><tr>
<td>read_report</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read an MGI report</h2>

<h3>Description</h3>

<p><code>read_report()</code> imports data from an MGI report into R as a tidy data set.
</p>
<p>You may call this function in two alternative ways:
</p>

<ul>
<li>
<p> Using <code>report_key</code>: this is the easiest approach. A report key maps to a
report currently hosted at MGI, e.g. <code>read_report("marker_list2")</code> reads
<code>MRK_List2.rpt</code> directly from MGI server into R. See Supported Reports below
for options.
</p>
</li>
<li>
<p> Using <code>report_file</code> and <code>report_type</code>: this approach is more flexible as
you can read directly from a file or URL.
</p>
</li>
</ul>
<h4>Supported Reports</h4>

<p>The set of currently supported reports:
</p>
<div class="sourceCode r"><pre>reports
#&gt; # A tibble: 13 x 4
#&gt;    report_key                 report_file              report_type   report_name
#&gt;    &lt;chr&gt;                      &lt;chr&gt;                    &lt;chr&gt;         &lt;chr&gt;      
#&gt;  1 marker_list1               MRK_List1.rpt            MRK_List1     Mouse Gene~
#&gt;  2 marker_list2               MRK_List2.rpt            MRK_List2     Mouse Gene~
#&gt;  3 marker_coordinates         MGI_MRK_Coord.rpt        MGI_MRK_Coord MGI Marker~
#&gt;  4 gene_model_coordinates     MGI_Gene_Model_Coord.rpt MGI_Gene_Mod~ MGI Gene M~
#&gt;  5 sequence_coordinates       MGI_GTGUP.gff            MGI_GTGUP     MGI Sequen~
#&gt;  6 genbank_refseq_ensembl_ids MRK_Sequence.rpt         MRK_Sequence  MGI Marker~
#&gt;  7 swiss_trembl_ids           MRK_SwissProt_TrEMBL.rpt MRK_SwissPro~ MGI Marker~
#&gt;  8 swiss_prot_ids             MRK_SwissProt.rpt        MRK_SwissProt MGI Marker~
#&gt;  9 gene_trap_ids              MRK_GeneTrap.rpt         MRK_GeneTrap  MGI Marker~
#&gt; 10 ensembl_ids                MRK_ENSEMBL.rpt          MRK_ENSEMBL   MGI Marker~
#&gt; 11 biotype_conflicts          MGI_BioTypeConflict.rpt  MGI_BioTypeC~ MGI Marker~
#&gt; 12 primers                    PRB_PrimerSeq.rpt        PRB_PrimerSeq MGI Marker~
#&gt; 13 interpro_domains           MGI_InterProDomains.rpt  MGI_InterPro~ InterPro d~
</pre></div>



<h3>Usage</h3>

<pre><code class="language-R">read_report(
  report_key = NULL,
  report_file = NULL,
  report_type = NULL,
  n_max = Inf
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>report_key</code></td>
<td>
<p>A character vector. A key used to uniquely refer to an MGI
report.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>report_file</code></td>
<td>
<p>A character vector. The file path or URL to an MGI report
file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>report_type</code></td>
<td>
<p>A character vector. The type of an MGI report.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_max</code></td>
<td>
<p>Maximum number of lines to read.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A tibble with report data in tidy
format. The set of variables is dependent on the specific report requested:
</p>

<ul>
<li>
<p> For <code>"marker_list1"</code>, see <code>vignette("marker_list1")</code>.
</p>
</li>
<li>
<p> For <code>"marker_list2"</code>, see <code>vignette("marker_list2")</code>.
</p>
</li>
<li>
<p> For <code>"marker_coordinates"</code>, see <code>vignette("marker_coordinates")</code>.
</p>
</li>
<li>
<p> For <code>"gene_model_coordinates"</code>, see <code>vignette("gene_model_coordinates")</code>.
</p>
</li>
<li>
<p> For <code>"sequence_coordinates"</code>, see <code>vignette("sequence_coordinates")</code>.
</p>
</li>
<li>
<p> For <code>"genbank_refseq_ensembl_ids"</code>, see <code>vignette("genbank_refseq_ensembl_ids")</code>.
</p>
</li>
<li>
<p> For <code>"swiss_trembl_ids"</code>, see <code>vignette("swiss_trembl_ids")</code>.
</p>
</li>
<li>
<p> For <code>"swiss_prot_ids"</code>, see <code>vignette("swiss_prot_ids")</code>.
</p>
</li>
<li>
<p> For <code>"gene_trap_ids"</code>, see <code>vignette("gene_trap_ids")</code>.
</p>
</li>
<li>
<p> For <code>"ensembl_ids"</code>, see <code>vignette("ensembl_ids")</code>.
</p>
</li>
<li>
<p> For <code>"biotype_conflicts"</code>, see <code>vignette("biotype_conflicts")</code>.
</p>
</li>
<li>
<p> For <code>"primers"</code>, see <code>vignette("primers")</code>.
</p>
</li>
<li>
<p> For <code>"interpro_domains"</code>, see <code>vignette("interpro_domains")</code>.
</p>
</li>
</ul>
</div>