<div class="container">

<table style="width: 100%;"><tr>
<td>default_params_doc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>This function does nothing. It is intended to inherit the documentation
of the parameters from.</h2>

<h3>Description</h3>

<p>This function does nothing. It is intended to inherit the documentation
of the parameters from.
</p>


<h3>Usage</h3>

<pre><code class="language-R">default_params_doc(
  ba_models,
  folder_name,
  mhc,
  mhcs,
  mhc_class,
  mhcnuggets_options,
  mhcnuggets_url,
  n_aas,
  peptide,
  peptides,
  peptide_length,
  peptides_path,
  protein_sequence,
  verbose
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ba_models</code></td>
<td>
<p>Set to TRUE to use a pure BA model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained byget_default_mhcnuggets_folder</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mhc</code></td>
<td>
<p>the MHC haplotype name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mhcs</code></td>
<td>
<p>the MHC haplotype names</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mhc_class</code></td>
<td>
<p>MHC class. Must be <code>I</code>, <code>II</code>
or NA. Use NA to let the class be deduced automatically</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mhcnuggets_options</code></td>
<td>
<p>options to run MHCnuggets with,
as can be created by create_mhcnuggets_options.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_aas</code></td>
<td>
<p>number of amino acids</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>peptide</code></td>
<td>
<p>one peptide sequence</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>peptides</code></td>
<td>
<p>one of more peptide sequences</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>peptide_length</code></td>
<td>
<p>length of a peptide, in number of amino acids</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>peptides_path</code></td>
<td>
<p>the path to the peptides</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>protein_sequence</code></td>
<td>
<p>protein sequence, in uppercase,
for example <code>FAMILYVW</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>set to TRUE for more debug information</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>This is an internal function, so it should be marked with
<code>@noRd</code>. This is not done, as this will disallow all
functions to find the documentation parameters
</p>


<h3>Author(s)</h3>

<p>Rich√®l J.C. Bilderbeek
</p>


</div>