<div class="container">

<table style="width: 100%;"><tr>
<td>HpltMatch</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>HpltMatch() function</h2>

<h3>Description</h3>

<p>Putative haplotypes may be identical to each other, or they may differ only by
incongruent or unresolved sequences. It is therefore useful to curate putative
haplotypes by comparing them to identify potentially overlapping types as
candidates for further investigation. <code>HpltMatch</code> calculates the
proportion of matching sequences between pairs of haplotypes and produces a
.csv table with values in a lower left matrix. If a threshold value is
specified, a list of haplotype matches where the proportion of matching
sequences exceeds the threshold will be produced.
</p>


<h3>Usage</h3>

<pre><code class="language-R">HpltMatch(hplt_occ_matrix, path_out, threshold = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>hplt_occ_matrix</code></td>
<td>
<p>A binary (logical) occurrence matrix with the data set
sequences in columns and the putative haplotypes in rows, as produced by the
CreateHpltOccTable() function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path_out</code></td>
<td>
<p>a user defined path to the folder where the output file(s) will
be saved.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>a numerical value between 0 and 1 (default NULL) specifying a
threshold for the proportion of matching sequences between haplotypes.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Note: The NestTablesXL() function provides a useful format for further
investigation of potentially overlapping haplotypes.
</p>
<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. 2022. MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J., Hansson, B., Stervander, M., Hasselquist, D., &amp; Westerdahl, H. 2022.
MHCtools - an R package for MHC high-throughput sequencing data: genotyping,
haplotype and supertype inference, and downstream genetic analyses in non-model
organisms. Molecular Ecology Resources. https://doi.org/10.1111/1755-0998.13645
</p>


<h3>Value</h3>

<p>A table specifying the proportions of matching sequences between
pairs of haplotypes (in a lower left matrix). If a threshold value is
specified, a list of haplotype matches where the proportion of matching
sequences exceeds the threshold will be printed to the console. The list will
also be saved in the output path, and can be reopened in R e.g. using the
readRDS() function in the base package.
Note: HpltMatch() will overwrite any existing files with the same output
file names in path_out.
</p>


<h3>See Also</h3>

<p><code>HpltFind</code>; <code>CreateHpltOccTable</code>;
<code>NestTablesXL</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">hplt_occ_matrix &lt;- hplt_occurrence_matrix
path_out &lt;- tempdir()
HpltMatch(hplt_occ_matrix, path_out, threshold=NULL)
</code></pre>


</div>