<div class="container">

<table style="width: 100%;"><tr>
<td>CreateSamplesFas</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>CreateSamplesFas() function</h2>

<h3>Description</h3>

<p><code>CreateSamplesFas</code> creates a set of FASTA files with the
sequences present in each sample in a 'dada2' sequence table.
</p>


<h3>Usage</h3>

<pre><code class="language-R">CreateSamplesFas(seq_table, path_out)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seq_table</code></td>
<td>
<p>seq_table is a sequence table as output by the 'dada2'
pipeline, which has samples in rows and nucleotide sequence variants in
columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path_out</code></td>
<td>
<p>is a user defined path to the folder where the output files
will be saved.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. 2022. MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J., Hansson, B., Stervander, M., Hasselquist, D., &amp; Westerdahl, H. 2022.
MHCtools - an R package for MHC high-throughput sequencing data: genotyping,
haplotype and supertype inference, and downstream genetic analyses in non-model
organisms. Molecular Ecology Resources. https://doi.org/10.1111/1755-0998.13645
</p>


<h3>Value</h3>

<p>A set of FASTA files with the sequences present in each sample in
the sequence table.The sequences are named in the FASTA files by an index
number corresponding to their column number in the sequence table, thus
identical sequences will have identical sample names in all the FASTA
files.
</p>


<h3>See Also</h3>

<p><code>CreateFas</code>; for more information about 'dada2' visit
&lt;https://benjjneb.github.io/dada2/&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R">seq_table &lt;- sequence_table_fas
path_out &lt;- tempdir()
CreateSamplesFas(seq_table, path_out)
</code></pre>


</div>