<div class="container">

<table style="width: 100%;"><tr>
<td>PapaDiv</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>PapaDiv() function</h2>

<h3>Description</h3>

<p><code>PapaDiv</code> calculates the joint major histocompatibility complex
(MHC) diversity in parent pairs, taking into account alleles that are shared
between the parents. The joint diversity in parent pairs is often of interest
in studies of mate choice, fitness, and heritability.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PapaDiv(parents_table, seq_table, path_out)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>parents_table</code></td>
<td>
<p>is a table containing the sample names of the parents in
each nest. This table should be organized so that each row represents one
nest, with the individual names of the mothers in the first column
(Mother), and the individual names of the fathers in the second column
(Father).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq_table</code></td>
<td>
<p>seq_table is a sequence table as output by the 'dada2'
pipeline, which has samples in rows and nucleotide sequence variants in
columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path_out</code></td>
<td>
<p>is a user defined path to the folder where the output files
will be saved.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The PapaDiv() function outputs a set of R lists containing for the joint
diversity of each parent pair, the proportion of sequences that are shared
between the parents, the diversity of each of the parents, the observed sequence
variants in each parent, the matched sequence variants, and the incongruent
sequence variants in each parent.
</p>
<p>In addition, PapaDiv() produces a summary table with the names of the parents in
a pair, their respective MHC diversities, and the joint parent pair diversity.
</p>
<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. 2022. MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J., Hansson, B., Stervander, M., Hasselquist, D., &amp; Westerdahl, H. 2022.
MHCtools - an R package for MHC high-throughput sequencing data: genotyping,
haplotype and supertype inference, and downstream genetic analyses in non-model
organisms. Molecular Ecology Resources. https://doi.org/10.1111/1755-0998.13645
</p>


<h3>Value</h3>

<p>a set of R lists containing for the joint diversity of each parent
pair, the proportion of sequences that are shared between the parents, the
diversity of each of the parents, the observed sequence variants in each
parent, the matched sequence variants, and the incongruent sequence
variants in each parent. The sequences are named in the output by an index
number corresponding to their column number in the sequence table, thus
identical sequences will have identical sample names in all the output
files. These files are saved in a sub folder in the output path called
Parent_pairs (created by PapaDiv()) and can be reopened in R e.g. using
the readRDS() function in the base package. For downstream data analysis,
the PapaDiv() function also produces a summary table with the names of the
parents in a pair, their respective MHC diversities, and the joint parent
pair diversity. This table is saved as a .csv file in the output path.
</p>


<h3>See Also</h3>

<p>For more information about 'dada2' visit
&lt;https://benjjneb.github.io/dada2/&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R">parents_table &lt;- parents_table
seq_table &lt;- sequence_table
path_out &lt;- tempdir()
PapaDiv(parents_table, seq_table, path_out)
</code></pre>


</div>