<div class="container">

<table style="width: 100%;"><tr>
<td>GetReplStats</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>GetReplStats function</h2>

<h3>Description</h3>

<p><code>GetReplStats</code> uses the output files produced by the ReplMatch()
function to calculate statistics on the agreement between replicated samples
in the sequencing experiment.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GetReplStats(filepath)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>filepath</code></td>
<td>
<p>is a user defined path to the folder where the output files
from the ReplMatch() function have been saved.</p>
</td>
</tr></table>
<h3>Details</h3>

<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. 2022. MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J., Hansson, B., Stervander, M., Hasselquist, D., &amp; Westerdahl, H. 2022.
MHCtools - an R package for MHC high-throughput sequencing data: genotyping,
haplotype and supertype inference, and downstream genetic analyses in non-model
organisms. Molecular Ecology Resources. https://doi.org/10.1111/1755-0998.13645
</p>


<h3>Value</h3>

<p>A list containing the number of replicate sets with zero incongruent
sequences, the proportion of replicate sets with zero incongruent
sequences, the mean of the mean proportion of incongruent sequences across
all replicate sets, and the repeatability of the sequencing experiment.
</p>


<h3>See Also</h3>

<p><code>ReplMatch</code>; <code>GetReplTable</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">filepath &lt;- system.file("extdata/ReplMatchOut/", package="MHCtools")
GetReplStats(filepath)
</code></pre>


</div>