<div class="container">

<table style="width: 100%;"><tr>
<td>panPrep</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Preparing FASTA files for pan-genomics</h2>

<h3>Description</h3>

<p>Preparing a FASTA file before starting comparisons of sequences.
</p>


<h3>Usage</h3>

<pre><code class="language-R">panPrep(in.file, genome_id, out.file, protein = TRUE, min.length = 10, discard = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>in.file</code></td>
<td>
<p>The name of a FASTA formatted file with protein or nucleotide sequences for coding
genes in a genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome_id</code></td>
<td>
<p>The Genome Identifier, see below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out.file</code></td>
<td>
<p>Name of file where the prepared sequences will be written.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>protein</code></td>
<td>
<p>Logical, indicating if the ‘<span class="samp">⁠in.file⁠</span>’ contains protein (<code>TRUE</code>) or
nucleotide (<code>FALSE</code>) sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.length</code></td>
<td>
<p>Minimum sequence length</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>discard</code></td>
<td>
<p>A text, a regular expression, and sequences having a match against this in their
headerline will be discarded.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function will read the <code>in.file</code> and produce another, slightly modified, FASTA file
which is prepared for the comparisons using <code>blastpAllAll</code>, <code>hmmerScan</code>
or any other method.
</p>
<p>The main purpose of <code>panPrep</code> is to make certain every sequence is labeled with a tag
called a ‘<span class="samp">⁠genome_id⁠</span>’ identifying the genome from which it comes. This text contains the text
“GID” followed by an integer. This integer can be any integer as long as it is unique to every
genome in the study. If a genome has the text “GID12345” as identifier, then the
sequences in the file produced by <code>panPrep</code> will have headerlines starting with
“GID12345_seq1”, “GID12345_seq2”, “GID12345_seq3”...etc. This makes it possible
to quickly identify which genome every sequence belongs to.
</p>
<p>The ‘<span class="samp">⁠genome_id⁠</span>’ is also added to the file name specified in ‘<span class="samp">⁠out.file⁠</span>’. For this reason the
‘<span class="samp">⁠out.file⁠</span>’ must have a file extension containing letters only. By convention, we expect FASTA
files to have one of the extensions ‘<span class="samp">⁠.fsa⁠</span>’, ‘<span class="samp">⁠.faa⁠</span>’, ‘<span class="samp">⁠.fa⁠</span>’ or ‘<span class="samp">⁠.fasta⁠</span>’.
</p>
<p><code>panPrep</code> will also remove sequences shorter than <code>min.length</code>, removing stop codon
symbols (‘<span class="samp">⁠*⁠</span>’), replacing alien characters with ‘<span class="samp">⁠X⁠</span>’ and converting all sequences to upper-case.
If the input ‘<span class="samp">⁠discard⁠</span>’ contains a regular expression, any sequences having a match to this in their
headerline are also removed. Example: If we use the <code>prodigal</code> software (see <code>findGenes</code>)
to find proteins in a genome, partially predicted genes will have the text ‘<span class="samp">⁠partial=10⁠</span>’ or
‘<span class="samp">⁠partial=01⁠</span>’ in their headerline. Using ‘<span class="samp">⁠discard= "partial=01|partial=10"⁠</span>’ will remove
these from the data set.
</p>


<h3>Value</h3>

<p>This function produces a FASTA formatted sequence file, and returns the name of this file.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen and Kristian Liland.
</p>


<h3>See Also</h3>

<p><code>hmmerScan</code>, <code>blastpAllAll</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Using a protein file in this package
# We need to uncompress it first...
pf &lt;- file.path(path.package("micropan"),"extdata","xmpl.faa.xz")
prot.file &lt;- tempfile(fileext = ".xz")
ok &lt;- file.copy(from = pf, to = prot.file)
prot.file &lt;- xzuncompress(prot.file)

# Prepping it, using the genome_id "GID123"
prepped.file &lt;- panPrep(prot.file, genome_id = "GID123", out.file = tempfile(fileext = ".faa"))

# Reading the prepped file
prepped &lt;- readFasta(prepped.file)
head(prepped)

# ...and cleaning...
ok &lt;- file.remove(prot.file, prepped.file)

</code></pre>


</div>