<div class="container">

<table style="width: 100%;"><tr>
<td>readGFF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reading and writing GFF-tables</h2>

<h3>Description</h3>

<p>Reading or writing a GFF-table from/to file.
</p>


<h3>Usage</h3>

<pre><code class="language-R">readGFF(in.file)
writeGFF(gff.table, out.file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>in.file</code></td>
<td>
<p>Name of file with a GFF-table.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gff.table</code></td>
<td>
<p>A table (<code>tibble</code>) with genomic features information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out.file</code></td>
<td>
<p>Name of file.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>A GFF-table is simply a <code>tibble</code> with columns
adhering to the format specified by the GFF3 format, see
https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md for details. There is
one row for each feature.
</p>
<p>The following columns should always be in a full <code>gff.table</code> of the GFF3 format:
</p>

<ul>
<li>
<p> Seqid. A unique identifier of the genomic sequence on which the feature resides.
</p>
</li>
<li>
<p> Source. A description of the procedure that generated the feature, e.g. <code>"R-package micropan::findOrfs"</code>.
</p>
</li>
<li>
<p> Type The type of feature, e.g. <code>"ORF"</code>, <code>"16S"</code> etc.
</p>
</li>
<li>
<p> Start. The leftmost coordinate. This is the start if the feature is on the Sense strand, but
the end if it is on the Antisense strand.
</p>
</li>
<li>
<p> End. The rightmost coordinate. This is the end if the feature is on the Sense strand, but
the start if it is on the Antisense strand.
</p>
</li>
<li>
<p> Score. A numeric score (E-value, P-value) from the <code>Source</code>. 
</p>
</li>
<li>
<p> Strand. A <code>"+"</code> indicates Sense strand, a <code>"-"</code> Antisense.
</p>
</li>
<li>
<p> Phase. Only relevant for coding genes. the values 0, 1 or 2 indicates the reading frame, i.e. 
the number of bases to offset the <code>Start</code> in order to be in the reading frame.
</p>
</li>
<li>
<p> Attributes. A single string with semicolon-separated tokens prociding additional information.
</p>
</li>
</ul>
<p>Missing values are described by <code>"."</code> in the GFF3 format. This is also done here, except for the
numerical columns Start, End, Score and Phase. Here <code>NA</code> is used, but this is replaced by
<code>"."</code> when writing to file.
</p>
<p>The <code>readGFF</code> function will also read files where sequences in FASTA format are added after
the GFF-table. This file section must always start with the line <code>##FASTA</code>. This fasta object
is added to the GFF-table as an attribute (use <code>attr(gff.tbl, "FASTA")</code> to retrieve it).
</p>


<h3>Value</h3>

<p><code>readGFF</code> returns a <code>gff.table</code> with the columns described above.
</p>
<p><code>writeGFF</code> writes the supplied <code>gff.table</code> to a text-file.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen and Kristian Hovde Liland.
</p>


<h3>See Also</h3>

<p><code>findOrfs</code>, <code>lorfs</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Using a GFF file in this package
gff.file &lt;- file.path(path.package("microseq"),"extdata","small.gff")

# Reading gff-file
gff.tbl &lt;- readGFF(gff.file)

</code></pre>


</div>