<div class="container">

<table style="width: 100%;"><tr>
<td>processSeq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Perform dada2 analysis</h2>

<h3>Description</h3>

<p>Perform dada2 analysis
</p>


<h3>Usage</h3>

<pre><code class="language-R">processSeq(
  path = ".",
  truncLen = c(0, 0),
  trimLeft = 0,
  trimRight = 0,
  minLen = 20,
  maxLen = Inf,
  sample_info = NULL,
  train_data = "silva_nr99_v138_train_set.fa.gz",
  train_species = "silva_species_assignment_v138.fa.gz",
  outpath = NULL,
  saveobj = FALSE,
  buildtree = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>path</code></td>
<td>
<p>working dir for the input reads</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>truncLen</code></td>
<td>
<p>(Optional). Default 0 (no truncation). Truncate reads after truncLen bases. Reads shorter than this are discarded.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trimLeft</code></td>
<td>
<p>(Optional). The number of nucleotides to remove from the start of each read.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trimRight</code></td>
<td>
<p>(Optional). Default 0. The number of nucleotides to remove from the end of each read.
If both truncLen and trimRight are provided, truncation will be performed after trimRight is enforced.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minLen</code></td>
<td>
<p>(Optional). Default 20. Remove reads with length less than minLen. minLen is enforced after trimming and truncation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxLen</code></td>
<td>
<p>Optional). Default Inf (no maximum). Remove reads with length greater than maxLen. maxLen is enforced before trimming and truncation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample_info</code></td>
<td>
<p>(Optional).sample information for the sequence</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>train_data</code></td>
<td>
<p>(Required).training database</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>train_species</code></td>
<td>
<p>(Required). species database</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outpath</code></td>
<td>
<p>(Optional).the path for the filtered reads and th out table</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>saveobj</code></td>
<td>
<p>(Optional).Default FALSE. save the phyloseq object output.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>buildtree</code></td>
<td>
<p>build phylogenetic tree or not(default: FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>(Optional). Default TRUE. Print verbose text output.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>list include count table, summary table, taxonomy information and phyloseq object
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


</div>