<div class="container">

<table style="width: 100%;"><tr>
<td>backTranslate</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Replace amino acids with codons</h2>

<h3>Description</h3>

<p>Replaces aligned amino acids with their original codon triplets.
</p>


<h3>Usage</h3>

<pre><code class="language-R">backTranslate(aa.msa, nuc.ffn)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>aa.msa</code></td>
<td>
<p>A fasta object with a multiple alignment, see <code>msalign</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nuc.ffn</code></td>
<td>
<p>A fasta object with the coding sequences, see <code>readFasta</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function replaces the aligned amino acids in <code>aa.msa</code> with their
original codon triplets. This is possible only when the nucleotide sequences in <code>nuc.ffn</code>
are the exact nucleotide sequences behind the protein sequences that are aligned in <code>aa.msa</code>.
</p>
<p>It is required that the first token of the ‘<span class="samp">⁠Header⁠</span>’ lines is identical for a protein sequence
in <code>aa.msa</code> and its nucleotide version in ‘<span class="samp">⁠nuc.ffn⁠</span>’, otherwise it is impossible to
match them. Thus, they may not appear in the same order in the two input fasta objects.
</p>
<p>When aligning coding sequences, one should in general always align their protein sequences, to
keep the codon structure, and then use <code>backTranslate</code> to convert this into a 
nucleotide alignment, if required.
</p>
<p>If the nuclotide sequences contain the stop codons, these will be removed.
</p>


<h3>Value</h3>

<p>A fasta object similar to <code>aa.msa</code>, but where each amino acid has been replace by 
its corresponding codon. All gaps ‘<span class="samp">⁠"-"⁠</span>’ are replaced by triplets ‘<span class="samp">⁠"---"⁠</span>’.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen.
</p>


<h3>See Also</h3>

<p><code>msalign</code>, <code>readFasta</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">msa.file &lt;- file.path(file.path(path.package("microseq"),"extdata"), "small.msa")
aa.msa &lt;- readFasta(msa.file)
nuc.file &lt;- file.path(file.path(path.package("microseq"),"extdata"), "small.ffn")
nuc &lt;- readFasta(nuc.file)
nuc.msa &lt;- backTranslate(aa.msa, nuc)

</code></pre>


</div>