<div class="container">

<table style="width: 100%;"><tr>
<td>msec_summarized</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>An example mutation file.</h2>

<h3>Description</h3>

<p>A dataset containing the information of eight mutations processed by the
fun_homology function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">msec_summarized
</code></pre>


<h3>Format</h3>

<p>A list with 52 factors, each contains 29 variables
</p>

<dl>
<dt>Sample</dt>
<dd>
<p>sample name</p>
</dd>
<dt>Mut_type</dt>
<dd>
<p>mutation type</p>
</dd>
<dt>Chr</dt>
<dd>
<p>altered chromosome</p>
</dd>
<dt>Pos</dt>
<dd>
<p>altered position</p>
</dd>
<dt>Ref</dt>
<dd>
<p>reference base</p>
</dd>
<dt>Alt</dt>
<dd>
<p>altered base</p>
</dd>
<dt>SimpleRepeat_TRF</dt>
<dd>
<p>mutation locating repeat sequence</p>
</dd>
<dt>Neighborhood_sequence</dt>
<dd>
<p>neighborhood sequence</p>
</dd>
<dt>read_length</dt>
<dd>
<p>read length</p>
</dd>
<dt>mut_type</dt>
<dd>
<p>mutation type</p>
</dd>
<dt>alt_length</dt>
<dd>
<p>length of the mutated bases</p>
</dd>
<dt>total_read</dt>
<dd>
<p>number of mutation supporting reads</p>
</dd>
<dt>soft_clipped_read</dt>
<dd>
<p>number of soft-clipped reads</p>
</dd>
<dt>flag_hairpin</dt>
<dd>
<p>number of reads produced by hairpin structure</p>
</dd>
<dt>hairpin_length</dt>
<dd>
<p>maximum length of palindromic sequences</p>
</dd>
<dt>pre_support_length</dt>
<dd>
<p>maximum 5'-supporting length</p>
</dd>
<dt>post_support_length</dt>
<dd>
<p>maximum 3'-supporting length</p>
</dd>
<dt>short_support_length</dt>
<dd>
<p>minimum supporting length</p>
</dd>
<dt>pre_minimum_length</dt>
<dd>
<p>minimum 5'-supporting length</p>
</dd>
<dt>post_minimum_length</dt>
<dd>
<p>minimum 3'-supporting length</p>
</dd>
<dt>pre_farthest</dt>
<dd>
<p>5'-farthest supported base from the mutated base</p>
</dd>
<dt>post_farthest</dt>
<dd>
<p>3'-farthest supported base from the mutated base</p>
</dd>
<dt>low_quality_base_rate_under_q18</dt>
<dd>
<p>low quality base rate</p>
</dd>
<dt>low_quality_pre</dt>
<dd>
<p>low quality base rate of 5'- side</p>
</dd>
<dt>low_quality_post</dt>
<dd>
<p>low quality base rate of 3'- side</p>
</dd>
<dt>pre_rep_status</dt>
<dd>
<p>5'-repeat sequence length</p>
</dd>
<dt>post_rep_status</dt>
<dd>
<p>3'-repeat sequence length</p>
</dd>
<dt>homopolymer_status</dt>
<dd>
<p>homopolymer sequence length</p>
</dd>
<dt>indel_status</dt>
<dd>
<p>whether the mutation is indel or not</p>
</dd>
<dt>indel_length</dt>
<dd>
<p>length of indel mutation</p>
</dd>
<dt>distant_homology</dt>
<dd>
<p>number of reads derived from homologous regions</p>
</dd>
<dt>penalty_pre</dt>
<dd>
<p>5'-penalty score by the mapper</p>
</dd>
<dt>penalty_post</dt>
<dd>
<p>3'-penalty score by the mapper</p>
</dd>
<dt>caution</dt>
<dd>
<p>comment</p>
</dd>
<dt>distant_homology_rate</dt>
<dd>
<p>rate of reads derived from homologous regions</p>
</dd>
<dt>pre_minimum_length_adj</dt>
<dd>
<p>adjusted pre_minimum_length</p>
</dd>
<dt>post_minimum_length_adj</dt>
<dd>
<p>adjusted pre_minimum_length</p>
</dd>
<dt>pre_support_length_adj</dt>
<dd>
<p>adjusted pre_minimum_length</p>
</dd>
<dt>post_support_length_adj</dt>
<dd>
<p>adjusted pre_minimum_length</p>
</dd>
<dt>shortest_support_length_adj</dt>
<dd>
<p>the shortest short_support_length</p>
</dd>
<dt>minimum_length_1</dt>
<dd>
<p>theoretically minimum 5'-supporting length</p>
</dd>
<dt>minimum_length_2</dt>
<dd>
<p>theoretically minimum 3'-supporting length</p>
</dd>
<dt>minimum_length</dt>
<dd>
<p>theoretically minimum supporting length</p>
</dd>
<dt>short_support_length_adj</dt>
<dd>
<p>adjusted short_support_length</p>
</dd>
<dt>altered_length</dt>
<dd>
<p>substituted/inserted length</p>
</dd>
<dt>half_length</dt>
<dd>
<p>half of the read length</p>
</dd>
<dt>short_support_length_total</dt>
<dd>
<p>range of short_support_length</p>
</dd>
<dt>pre_support_length_total</dt>
<dd>
<p>range of pre_support_length</p>
</dd>
<dt>post_support_length_total</dt>
<dd>
<p>range of post_support_length</p>
</dd>
<dt>half_length_total</dt>
<dd>
<p>range of possible short_support_length</p>
</dd>
<dt>total_length_total</dt>
<dd>
<p>range of possible supporting length</p>
</dd>
<dt>soft_clipped_rate</dt>
<dd>
<p>rate of soft clipped reads</p>
</dd>
</dl>
<p>...

</p>


</div>