<div class="container">

<table style="width: 100%;"><tr>
<td>fun_read_check</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read check function.</h2>

<h3>Description</h3>

<p>This function attempts to check the mutation profile in each read.
</p>


<h3>Usage</h3>

<pre><code class="language-R">fun_read_check(
  df_mutation,
  df_bam,
  ref_genome,
  sample_name,
  read_length,
  adapter_1,
  adapter_2,
  short_homology_search_length,
  min_homology_search,
  progress_bar
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>df_mutation</code></td>
<td>
<p>Mutation information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>df_bam</code></td>
<td>
<p>Data from the BAM file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref_genome</code></td>
<td>
<p>Reference genome for the data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample_name</code></td>
<td>
<p>Sample name (character)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>read_length</code></td>
<td>
<p>The read length in the sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adapter_1</code></td>
<td>
<p>The Read 1 adapter sequence of the library.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adapter_2</code></td>
<td>
<p>The Read 2 adapter sequence of the library.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>short_homology_search_length</code></td>
<td>
<p>Small sequence for homology search.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_homology_search</code></td>
<td>
<p>Minimum length to define "homologous".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>progress_bar</code></td>
<td>
<p>"Y": You can see the progress visually.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>list(msec, homology_search, mut_depth)
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
data(exampleMutation)
data(exampleBam)
fun_read_check(df_mutation = exampleMutation,
df_bam = exampleBam,
ref_genome = BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38,
sample_name = "sample",
read_length = 150,
adapter_1 = "AGATCGGAAGAGCACACGTCTGAACTCCAGTCA",
adapter_2 = "AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT",
short_homology_search_length = 4,
min_homology_search = 40,
progress_bar = "N"
)

## End(Not run)
</code></pre>


</div>