<div class="container">

<table style="width: 100%;"><tr>
<td>orfSignature</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Signature for each ORF</h2>

<h3>Description</h3>

<p>Creates a signature text for orfs in an <code>orf.table</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">orfSignature(orf.table, full = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>orf.table</code></td>
<td>
<p>A <code>tibble</code> with ORF information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>full</code></td>
<td>
<p>Logical indicating type of signature.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>A signature is a text that uniquely identifies each ORF in an
<code>orf.table</code>, which is a GFF-table with columns <code>Seqid</code>, <code>Start</code>,
<code>End</code> and <code>Strand</code>.
</p>
<p>The full signature (<code>full = TRUE</code>) contains the <code>Seqid</code>, <code>Start</code>,
<code>End</code> and <code>Strand</code> information for each ORF, separated by 
semicolon <code>";"</code>. This text is always unique 
to each ORF. If <code>full = FALSE</code> the <code>Signature</code> will not contain 
the starting position information for each ORF. This means all nested ORFs ending 
at the same stop-codon will then get identical <code>Signature</code>s. This is 
useful for identifying which ORFs are nested within the same LORF.
</p>
<p>Note that the signature you get with <code>full = FALSE</code> contains <code>Seqid</code>, 
then <code>End</code> if on the positive <code>Strand</code>, <code>Start</code> otherwise, and then
the <code>Strand</code>.
</p>


<h3>Value</h3>

<p>A text vector with the <code>Signature</code> for each ORF.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen.
</p>


<h3>See Also</h3>

<p><code>findOrfs</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Using a genome file in this package
genome.file &lt;- file.path(path.package("microseq"),"extdata","small.fna")

# Reading genome and finding orfs
genome &lt;- readFasta(genome.file)
orf.tbl &lt;- findOrfs(genome)

# Compute signatures
signature.full &lt;- orfSignature(orf.tbl)
signature.reduced &lt;- orfSignature(orf.tbl, full = FALSE)

</code></pre>


</div>