<div class="container">

<table style="width: 100%;"><tr>
<td>isOrtholog</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Identifies orthologs in gene clusters</h2>

<h3>Description</h3>

<p>Finds the ortholog sequences in every cluster based on pairwise distances.
</p>


<h3>Usage</h3>

<pre><code class="language-R">isOrtholog(clustering, dist.tbl)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>clustering</code></td>
<td>
<p>A vector of integers indicating the cluster for every sequence. Sequences with
the same number belong to the same cluster. The name of each element is the tag identifying the sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dist.tbl</code></td>
<td>
<p>A <code>tibble</code> with pairwise distances. The columns ‘<span class="samp">⁠Query⁠</span>’ and
‘<span class="samp">⁠Hit⁠</span>’ contain tags identifying pairs of sequences. The column ‘<span class="samp">⁠Distance⁠</span>’ contains the
distances, always a number from 0.0 to 1.0.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The input <code>clustering</code> is typically produced by <code>bClust</code>. The input
<code>dist.tbl</code> is typically produced by <code>bDist</code>.
</p>
<p>The concept of orthologs is difficult for prokaryotes, and this function finds orthologs in a
simplistic way. For a given cluster, with members from many genomes, there is one ortholog from every
genome. In cases where a genome has two or more members in the same cluster, only one of these is an
ortholog, the rest are paralogs.
</p>
<p>Consider all sequences from the same genome belonging to the same cluster. The ortholog is defined as
the one having the smallest sum of distances to all other members of the same cluster, i.e. the one
closest to the ‘center’ of the cluster.
</p>
<p>Note that the status as ortholog or paralog depends greatly on how clusters are defined in the first
place. If you allow large and diverse (and few) clusters, many sequences will be paralogs. If you define
tight and homogenous (and many) clusters, almost all sequences will be orthologs.
</p>


<h3>Value</h3>

<p>A vector of logicals with the same number of elements as the input ‘<span class="samp">⁠clustering⁠</span>’, indicating
if the corresponding sequence is an ortholog (<code>TRUE</code>) or not (<code>FALSE</code>). The name of each
element is copied from ‘<span class="samp">⁠clustering⁠</span>’.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen and Kristian Hovde Liland.
</p>


<h3>See Also</h3>

<p><code>bDist</code>, <code>bClust</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# Loading distance data and their clustering results
data(list = c("xmpl.bdist","xmpl.bclst"))

# Finding orthologs
is.ortholog &lt;- isOrtholog(xmpl.bclst, xmpl.bdist)
# The orthologs are
which(is.ortholog)

## End(Not run)

</code></pre>


</div>