<div class="container">

<table style="width: 100%;"><tr>
<td>readFastq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read and write FASTQ files</h2>

<h3>Description</h3>

<p>Reads and writes files in the FASTQ format.
</p>


<h3>Usage</h3>

<pre><code class="language-R">readFastq(in.file)
writeFastq(fdta, out.file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>in.file</code></td>
<td>
<p>url/directory/name of (gzipped) FASTQ file to read.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fdta</code></td>
<td>
<p>FASTQ object to write.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out.file</code></td>
<td>
<p>url/directory/name of (gzipped) FASTQ file to write.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>These functions handle input/output of sequences in the commonly used FASTQ format,
typically used for storing DNA sequences (reads) after sequencing. If
filenames (<code>in.file</code> or <code>out.file</code>) have the extension <code>.gz</code> they will automatically be
compressed/uncompressed. 
</p>
<p>The sequences are stored in a <code>tibble</code>, opening up all the possibilities in R for
fast and easy manipulations. The content of the file is stored as three columns, ‘<span class="samp">⁠Header⁠</span>’,
‘<span class="samp">⁠Sequence⁠</span>’ and ‘<span class="samp">⁠Quality⁠</span>’. If other columns are added, these will be ignored by
<code>writeFastq</code>.
</p>


<h3>Value</h3>

<p><code>readFastq</code> returns a <code>tibble</code> with the contents of the (gzipped) FASTQ
file stored in three columns of text. The first, named ‘<span class="samp">⁠Header⁠</span>’, contains
the headerlines, the second, named ‘<span class="samp">⁠Sequence⁠</span>’, contains the sequences and the third, named 
‘<span class="samp">⁠Quality⁠</span>’ contains the base quality scores.
</p>
<p><code>writeFastq</code> produces a (gzipped) FASTQ file.
</p>


<h3>Note</h3>

<p>These functions will only handle files where each entry spans one single line, i.e. not the
(uncommon) multiline FASTQ format.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen and Kristian Hovde Liland.
</p>


<h3>See Also</h3>

<p>codereadFasta.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# We need a FASTQ-file to read, here is one example file:
fq.file &lt;- file.path(file.path(path.package("microseq"),"extdata"),"small.fastq.gz")

# Read and write
fdta &lt;- readFastq(fq.file)
ok &lt;- writeFastq(fdta[1:3,], out.file = "delete_me.fq")

# Make use of dplyr to copy parts of the file to another file
readFastq(fq.file) %&gt;% 
  mutate(Length = str_length(Sequence)) %&gt;% 
  filter(Length &gt; 200) %&gt;% 
  writeFasta(out.file = "long_reads.fa") # writing to FASTA file

## End(Not run)

</code></pre>


</div>