<div class="container">

<table style="width: 100%;"><tr>
<td>multinomClassify</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Classifying with a Multinomial model</h2>

<h3>Description</h3>

<p>Classifying sequences by a trained Multinomial model.
</p>


<h3>Usage</h3>

<pre><code class="language-R">multinomClassify(sequence, trained.model, post.prob = FALSE, prior = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sequence</code></td>
<td>
<p>Character vector of 16S sequences to classify.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trained.model</code></td>
<td>
<p>A list with a trained model, see <code>multinomTrain</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>post.prob</code></td>
<td>
<p>Logical indicating if posterior log-probabilities should be returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prior</code></td>
<td>
<p>Logical indicating if classification should be done by flat priors (default)
or with empirical priors (prior=TRUE).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The classification step of the Multinomial method (Vinje et al, 2015) means counting 
K-mers on all sequences, and computing the posterior probabilities for each taxon in the trained model.
The predicted taxon for each input sequence is the one with the maximum posterior probability for
that sequence.
</p>
<p>By setting <code>post.prob=TRUE</code> you will get the log-probability of the best and second best taxon
for each sequence. This can be used for evaluating the certainty in the classifications,
see <code>taxMachine</code>.
</p>
<p>The classification is parallelized through RcppParallel
employing Intel TBB and TinyThread. By default all available
processing cores are used. This can be changed using the
function <code>setParallel</code>.
</p>


<h3>Value</h3>

<p>If <code>post.prob=FALSE</code> a character vector of predicted taxa is returned.
</p>
<p>If <code>post.prob=TRUE</code> a <code>data.frame</code> with three columns is returned. Taxon
is the vector of predicted taxa, one for each sequence in <code>sequence</code>. The
Post.prob.1 and Post.prob.2 are vectors with the maximum and second largest posterior
log-probabilities for each sequence.
</p>


<h3>Author(s)</h3>

<p>Kristian Hovde Liland and Lars Snipen.
</p>


<h3>References</h3>

<p>Vinje, H, Liland, KH, Alm√∏y, T, Snipen, L. (2015). Comparing K-mer based methods for
improved classification of 16S sequences. BMC Bioinformatics, 16:205.
</p>


<h3>See Also</h3>

<p><code>KmerCount</code>, <code>multinomTrain</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("small.16S")
seq &lt;- small.16S$Sequence
tax &lt;- sapply(strsplit(small.16S$Header,split=" "),function(x){x[2]})
## Not run: 
trn &lt;- multinomTrain(seq,tax)
primer.515f &lt;- "GTGYCAGCMGCCGCGGTAA"
primer.806rB &lt;- "GGACTACNVGGGTWTCTAAT"
reads &lt;- amplicon(seq, primer.515f, primer.806rB)
predicted &lt;- multinomClassify(unlist(reads[nchar(reads)&gt;0]),trn)
print(predicted)

## End(Not run)

</code></pre>


</div>