<div class="container">

<table style="width: 100%;"><tr>
<td>taxMachine</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Classifying 16S sequences</h2>

<h3>Description</h3>

<p>Optimized classification of 16S sequence data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">taxMachine(
  sequence,
  model.in.memory = TRUE,
  model.on.disk = FALSE,
  verbose = TRUE,
  chunk.size = 10000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sequence</code></td>
<td>
<p>Character vector with DNA sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model.in.memory</code></td>
<td>
<p>Logical indicating if model should be cached in memory (default=TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model.on.disk</code></td>
<td>
<p>Logical or text, for reading/saving models, see Deatils below (default=FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical, if <code>TRUE</code> progress is reported during computations (default=TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chunk.size</code></td>
<td>
<p>The number of sequence to classify in each iteration of the loop (default=10000).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function provides optimized taxonomy classifications from 16S sequence data.
</p>
<p>All sequences are classified to the genus level based on a Multinomial model (see <code>multinomTrain</code>)
trained on the designed consensus taxonomy data set <code>contax.trim</code> found in
the R-package <code>microcontax</code>. The word length K=8 has been used in the model.
</p>
<p>To avoid saving fitted models in the package, a model is trained the first time you run <code>taxMachine</code> in an R session.
This takes only a few seconds, and the result is cached for latter use if <code>model.in.memory</code> is <code>TRUE</code>.
</p>
<p>If a path to an existing file with a trained model is supplied in <code>model.on.disk</code>, this Multinomial model is read
from the file and used. If a path to a new file is supplied, the trained Multinomial model will be saved to that file.
The default (<code>model.on.disk=FALSE</code>), means no files are read/saved, while <code>model.on.disk=TRUE</code> will attempt to load/save models from the
<code>microclass/extdata</code> directory.
</p>
<p>Both <code>verbose</code> and <code>chunk.size</code> are used to monitor the progress, which is nice when classifying huge data sets,
since this will take some time.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> with one row for each sequence. The columns are Genus, D.score, R.score and P.recognize.
</p>
<p>Genus is the predicted genus for each sequence. Note that all sequences get a prediction, but may still be
more or less reliable.
</p>
<p>The D.score is a measure of how the predicted genus wins over all other genera in the race for being the chosen one.
A large D.score means the winner stands out clearly, and we can be confident it is the correct genus. A D.score close to 0
means we have an uncertain classification. Only D.scores below 1.0, should be of any concern, see Liland et al
(2016) for details.
</p>
<p>The R.score is a measure of the models ability to recognize the sequence. The more negative the R.score gets, the more
unusual the sequence is compared to the training set (the <code>contax.trim</code> data set). The P.recognize
is a rough probability of seing an R.score this small, or smaller, given the training data. Thus, a very small P.recognize means
the sequence is not really recognized, and the classification is worthless. A very negative R.score indicates either not 16S at all,
many sequencing errors that has destroyed the read, or a completely new taxon never seen before. See Liland et al (2016) for
details.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen and Kristian Hovde Liland
</p>


<h3>References</h3>

<p>Liland, KH, Vinje, H, Snipen, L (2016). <code>microclass</code> - An 
R-package for 16S taxonomy classification. BMC Bioinformatics, xx:yy.
</p>


<h3>See Also</h3>

<p><code>KmerCount</code>, <code>multinomClassify</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
data(small.16S)
tax.tab &lt;- taxMachine(small.16S$Sequence)

## End(Not run)

</code></pre>


</div>