<div class="container">

<table style="width: 100%;"><tr>
<td>translate</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Translation according to the standard genetic code</h2>

<h3>Description</h3>

<p>The translation from DNA(RNA) to amino acid sequence according to the standard genetic code.
</p>


<h3>Usage</h3>

<pre><code class="language-R">translate(nuc.sequences, M.start = TRUE, no.stop = TRUE, trans.tab = 11)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nuc.sequences</code></td>
<td>
<p>Character vector containing the nucleotide sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>M.start</code></td>
<td>
<p>A logical indicating if the amino acid sequence should start with M regardless of start codon.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>no.stop</code></td>
<td>
<p>A logical indicating if terminal stops (*) should be eliminated from the translated sequence</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trans.tab</code></td>
<td>
<p>Translation table, either 11 or 4</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Codons are by default translated according to translation table 11, i.e. the possible start codons
are ATG, GTG or TTG and stop codons are TAA, TGA and TAG.  The only alternative implemented here is
translation table 4, which is used by some bacteria (e.g. Mycoplasma, Mesoplasma). If <code>trans.tab</code> is 4,
the stop codon TGA istranslated to W (Tryptophan).
</p>


<h3>Value</h3>

<p>A character vector of translated sequences.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen and Kristian Hovde Liland.
</p>


<h3>Examples</h3>

<pre><code class="language-R">fa.file &lt;- file.path(file.path(path.package("microseq"),"extdata"),"small.ffn")
fa &lt;- readFasta(fa.file)
translate(fa$Sequence)

# Or, make use of dplyr to manipulate tables
readFasta(fa.file) %&gt;%
  mutate(Protein = translate(Sequence)) -&gt; fa.tbl

</code></pre>


</div>