<div class="container">

<table style="width: 100%;"><tr>
<td>lorfs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Longest ORF</h2>

<h3>Description</h3>

<p>Filtering an <code>orf.table</code> with ORF information to keep only the LORFs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">lorfs(orf.tbl)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>orf.tbl</code></td>
<td>
<p>A <code>tibble</code> with the nine columns of the GFF-format (see <code>findOrfs</code>).</p>
</td>
</tr></table>
<h3>Details</h3>

<p>For every stop-codon there are usually multiple possible start-codons in the same reading
frame (nested ORFs). The LORF (Longest ORF) is defined as the longest of these nested ORFs,
i.e. the ORF starting at the most upstream start-codon matching the stop-codon.
</p>


<h3>Value</h3>

<p>A <code>tibble</code> with a subset of the rows of the argument <code>orf.tbl</code>. 
After this filtering the Type variable in <code>orf.tbl</code> is changed to <code>"LORF"</code>. If you want to
retrieve the LORF sequences, use <code>gff2fasta</code>.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen and Kristian Hovde Liland.
</p>


<h3>See Also</h3>

<p><code>readGFF</code>, <code>findOrfs</code>, <code>gff2fasta</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># See the example in the Help-file for findOrfs.

</code></pre>


</div>