<div class="container">

<table style="width: 100%;"><tr>
<td>multinomTrain</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Training multinomial model</h2>

<h3>Description</h3>

<p>Training the multinomial K-mer method on sequence data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">multinomTrain(sequence, taxon, K = 8, col.names = FALSE, n.pseudo = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sequence</code></td>
<td>
<p>Character vector of 16S sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxon</code></td>
<td>
<p>Character vector of taxon labels for each sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>Word length (integer).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col.names</code></td>
<td>
<p>Logical indicating if column names should be added to the trained model matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.pseudo</code></td>
<td>
<p>Number of pseudo-counts to use (positive numerics, need not be integer). Special case -1
will only return word counts, not log-probabilities.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The training step of the multinomial method (Vinje et al, 2015) means counting K-mers
on all sequences and compute the multinomial probabilities for each K-mer for each unique taxon. 
<code>n.pseudo</code> pseudo-counts are added, divided equally over all K-mers, before probabilities
are estimated. The optimal choice of <code>n.pseudo</code> will depend on <code>K</code> and the 
training data set. The default value <code>n.pseudo=100</code> has proven good for <code>K=8</code> and the
<code>contax.trim</code> data set (see the <code>microcontax</code> R-package).
</p>
<p>Adding the actual K-mers as column names (<code>col.names=TRUE</code>) will slow down the
computations.
</p>
<p>The relative taxon sizes are also computed, and may be used as an empirical prior in the
classification step (see "prior" below).
</p>


<h3>Value</h3>

<p>A list with two elements. The first element is <code>Method</code>, which is the text 
<code>"multinom"</code> in this case. The second element is <code>Fitted</code>, which is a matrix
of probabilities with one row for each unique <code>taxon</code> and one column for each possible word of
length<code>K</code>. The sum of each row is 1.0. No probabilities are 0 if <code>n.pseudo</code>&gt;0.0.
</p>
<p>The matrix <code>Fitted</code> has an attribute <code>attr("prior",)</code>, that contains the relative
taxon sizes.
</p>


<h3>Author(s)</h3>

<p>Kristian Hovde Liland and Lars Snipen.
</p>


<h3>References</h3>

<p>Vinje, H, Liland, KH, Alm√∏y, T, Snipen, L. (2015). Comparing K-mer based methods for
improved classification of 16S sequences. BMC Bioinformatics, 16:205.
</p>


<h3>See Also</h3>

<p><code>KmerCount</code>, <code>multinomClassify</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># See examples for multinomClassify

</code></pre>


</div>