<div class="container">

<table style="width: 100%;"><tr>
<td>findGenes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Finding coding genes</h2>

<h3>Description</h3>

<p>Finding coding genes in genomic DNA using the Prodigal software.
</p>


<h3>Usage</h3>

<pre><code class="language-R">findGenes(
  genome,
  prodigal.exe = "prodigal",
  faa.file = "",
  ffn.file = "",
  proc = "single",
  trans.tab = 11,
  mask.N = FALSE,
  bypass.SD = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genome</code></td>
<td>
<p>A table with columns Header and Sequence, containing the genome sequence(s).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prodigal.exe</code></td>
<td>
<p>Command to run the external software prodigal on the system (text).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>faa.file</code></td>
<td>
<p>If provided, prodigal will output all proteins to this fasta-file (text).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ffn.file</code></td>
<td>
<p>If provided, prodigal will output all DNA sequences to this fasta-file (text).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>proc</code></td>
<td>
<p>Either <code>"single"</code> or <code>"meta"</code>, see below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trans.tab</code></td>
<td>
<p>Either 11 or 4 (see below).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mask.N</code></td>
<td>
<p>Turn on masking of N's (logical)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bypass.SD</code></td>
<td>
<p>Bypass Shine-Dalgarno filter (logical)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The external software Prodigal is used to scan through a prokaryotic genome to detect the protein
coding genes. The text in <code>prodigal.exe</code> must contain the exact command to invoke barrnap on the system.
</p>
<p>In addition to the standard output from this function, FASTA files with protein and/or DNA sequences may
be produced directly by providing filenames in <code>faa.file</code> and <code>ffn.file</code>.
</p>
<p>The input <code>proc</code> allows you to specify if the input data should be treated as a single genome
(default) or as a metagenome. In the latter case the <code>genome</code> are (un-binned) contigs.
</p>
<p>The translation table is by default 11 (the standard code), but table 4 should be used for Mycoplasma etc.
</p>
<p>The <code>mask.N</code> will prevent genes having runs of N inside. The <code>bypass.SD</code> turn off the search
for a Shine-Dalgarno motif.
</p>


<h3>Value</h3>

<p>A GFF-table (see <code>readGFF</code> for details) with one row for each detected
coding gene.
</p>


<h3>Note</h3>

<p>The prodigal software must be installed on the system for this function to work, i.e. the command
‘<span class="samp">⁠system("prodigal -h")⁠</span>’ must be recognized as a valid command if you run it in the Console window.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen and Kristian Hovde Liland.
</p>


<h3>See Also</h3>

<p><code>readGFF</code>, <code>gff2fasta</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# This example requires the external prodigal software
# Using a genome file in this package.
genome.file &lt;- file.path(path.package("microseq"),"extdata","small.fna")

# Searching for coding sequences, this is Mycoplasma (trans.tab = 4)
genome &lt;- readFasta(genome.file)
gff.tbl &lt;- findGenes(genome, trans.tab = 4)

# Retrieving the sequences
cds.tbl &lt;- gff2fasta(gff.tbl, genome)

# You may use the pipe operator
library(ggplot2)
readFasta(genome.file) %&gt;% 
  findGenes(trans.tab = 4) %&gt;% 
  filter(Score &gt;= 50) %&gt;% 
  ggplot() +
  geom_histogram(aes(x = Score), bins = 25)

## End(Not run)

</code></pre>


</div>