<div class="container">

<table style="width: 100%;"><tr>
<td>gff2fasta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Retrieving annotated sequences</h2>

<h3>Description</h3>

<p>Retrieving from a genome the sequences specified in a <code>gff.table</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gff2fasta(gff.table, genome)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gff.table</code></td>
<td>
<p>A <code>gff.table</code> (<code>tibble</code>) with genomic features information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome</code></td>
<td>
<p>A fasta object (<code>tibble</code>) with the genome sequence(s).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Each row in <code>gff.table</code> (see <code>readGFF</code>) describes a genomic feature
in the <code>genome</code>, which is a <code>tibble</code> with columns ‘<span class="samp">⁠Header⁠</span>’ and
‘<span class="samp">⁠Sequence⁠</span>’. The information in the columns Seqid, Start, End and Strand are used to
retrieve the sequences from the ‘<span class="samp">⁠Sequence⁠</span>’ column of <code>genome</code>. Every Seqid in
the <code>gff.table</code> must match the first token in one of the ‘<span class="samp">⁠Header⁠</span>’ texts, in
order to retrieve from the correct ‘<span class="samp">⁠Sequence⁠</span>’.
</p>


<h3>Value</h3>

<p>A fasta object with one row for each row in <code>gff.table</code>. 
The <code>Header</code> for each sequence is a summary of the information in the
corresponding row of <code>gff.table</code>.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen and Kristian Hovde Liland.
</p>


<h3>See Also</h3>

<p><code>readGFF</code>, <code>findOrfs</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Using two files in this package
gff.file &lt;- file.path(path.package("microseq"),"extdata","small.gff")
genome.file &lt;- file.path(path.package("microseq"),"extdata","small.fna")

# Reading the genome first
genome &lt;- readFasta(genome.file)

# Retrieving sequences
gff.table &lt;- readGFF(gff.file)
fa.tbl &lt;- gff2fasta(gff.table, genome)

# Alternative, using piping
readGFF(gff.file) %&gt;% gff2fasta(genome) -&gt; fa.tbl

</code></pre>


</div>