<div class="container">

<table style="width: 100%;"><tr>
<td>findrRNA</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Finding rRNA genes</h2>

<h3>Description</h3>

<p>Finding rRNA genes in genomic DNA using the barrnap software.
</p>


<h3>Usage</h3>

<pre><code class="language-R">findrRNA(genome, barrnap.exe = "barrnap", bacteria = TRUE, cpu = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genome</code></td>
<td>
<p>A table with columns Header and Sequence, containing the genome sequence(s).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>barrnap.exe</code></td>
<td>
<p>Command to run the external software barrnap on the system (text).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bacteria</code></td>
<td>
<p>Logical, the genome is either a bacteria (default) or an archea.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cpu</code></td>
<td>
<p>Number of CPUs to use, default is 1.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The external software barrnap is used to scan through a prokaryotic genome to detect the
rRNA genes (5S, 16S, 23S).
The text in <code>barrnap.exe</code> must contain the exact command to invoke barrnap on the system.
</p>


<h3>Value</h3>

<p>A GFF-table (see <code>readGFF</code> for details) with one row for each detected
rRNA sequence.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen and Kristian Hovde Liland.
</p>


<h3>See Also</h3>

<p><code>readGFF</code>, <code>gff2fasta</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# This example requires the external barrnap software
# Using a genome file in this package.
genome.file &lt;- file.path(path.package("microseq"),"extdata","small.fna")

# Searching for rRNA sequences, and inspecting
genome &lt;- readFasta(genome.file)
gff.tbl &lt;- findrRNA(genome)
print(gff.table)

# Retrieving the sequences
rRNA &lt;- gff2fasta(gff.tbl, genome)

## End(Not run)

</code></pre>


</div>