<div class="container">

<table style="width: 100%;"><tr>
<td>rdpTrain</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Training the RDP classifier</h2>

<h3>Description</h3>

<p>Training the RDP presence/absence K-mer method on sequence data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rdpTrain(sequence, taxon, K = 8, cnames = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sequence</code></td>
<td>
<p>Character vector of 16S sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxon</code></td>
<td>
<p>Character vector of taxon labels for each sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>Word length (integer).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cnames</code></td>
<td>
<p>Logical indicating if column names should be added to the trained model matrix.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The training step of the RDP method means looking for K-mers on all sequences,
and computing the probability of each K-mer being present for each unique taxon. This is an
attempt to re-implement the method described by Wang et tal (2007), but without the bootstrapping. 
See that publications for all details.
</p>
<p>The word-length <code>K</code> is by default 8, since this is the value used by Wang et al. Larger values
may lead to memory-problems since the trained model is a matrix with 4^K columns. Adding the K-mers 
as column names will slow down all computations.
</p>
<p>The relative taxon sizes are also computed, and returned as an attribute to the model matrix. They may 
be used as empirical priors in the classification step.
</p>


<h3>Value</h3>

<p>A list with two elements. The first element is <code>Method</code>, which is the text 
<code>"RDPclassifier"</code> in this case. The second element is <code>Fitted</code>, which is a 
matrix with one row for each unique <code>taxon</code> and one column for 
each possible word of length <code>K</code>. The value in row i and column j is the probability that
word j is present in taxon i.
</p>


<h3>Author(s)</h3>

<p>Kristian Hovde Liland and Lars Snipen.
</p>


<h3>References</h3>

<p>Wang, Q, Garrity, GM, Tiedje, JM, Cole, JR (2007). Naive Bayesian Classifier for 
Rapid Assignment of rRNA Sequences into the New Bacterial Taxonomy. Applied and Enviromental 
Microbiology, 73: 5261-5267.
</p>


<h3>See Also</h3>

<p><code>rdpClassify</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># See examples for rdpClassify.

</code></pre>


</div>