<div class="container">

<table style="width: 100%;"><tr>
<td>rdpClassify</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Classifying with the RDP classifier</h2>

<h3>Description</h3>

<p>Classifying sequences by a trained presence/absence K-mer model.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rdpClassify(sequence, trained.model, post.prob = FALSE, prior = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sequence</code></td>
<td>
<p>Character vector of sequences to classify.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trained.model</code></td>
<td>
<p>A list with a trained model, see <code>rdpTrain</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>post.prob</code></td>
<td>
<p>Logical indicating if posterior log-probabilities should be returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prior</code></td>
<td>
<p>Logical indicating if classification should be done by flat priors (default)
or with empirical priors (prior=TRUE).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The classification step of the presence/absence method known as the RDP classifier
(Wang et al 2007) means looking for K-mers on all sequences, and computing the posterior
probabilities for each taxon using a trained model and a naive Bayes assumption. The predicted
taxon is the one producing the maximum posterior probability, for each <code>sequence</code>.
</p>
<p>The classification is parallelized through RcppParallel
employing Intel TBB and TinyThread. By default all available
processing cores are used. This can be changed using the
function <code>setParallel</code>.
</p>


<h3>Value</h3>

<p>A character vector with the predicted taxa, one for each <code>sequence</code>.
</p>


<h3>Author(s)</h3>

<p>Kristian Hovde Liland and Lars Snipen.
</p>


<h3>References</h3>

<p>Wang, Q, Garrity, GM, Tiedje, JM, Cole, JR (2007). Naive Bayesian Classifier for 
Rapid Assignment of rRNA Sequences into the New Bacterial Taxonomy. Applied and Enviromental 
Microbiology, 73: 5261-5267.
</p>


<h3>See Also</h3>

<p><code>rdpTrain</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("small.16S")
seq &lt;- small.16S$Sequence
tax &lt;- sapply(strsplit(small.16S$Header,split=" "),function(x){x[2]})
## Not run: 
trn &lt;- rdpTrain(seq,tax)
primer.515f &lt;- "GTGYCAGCMGCCGCGGTAA"
primer.806rB &lt;- "GGACTACNVGGGTWTCTAAT"
reads &lt;- amplicon(seq, primer.515f, primer.806rB)
predicted &lt;- rdpClassify(unlist(reads[nchar(reads)&gt;0]),trn)
print(predicted)

## End(Not run)

</code></pre>


</div>