<div class="container">

<table style="width: 100%;"><tr>
<td>msec_analyzed</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>An example mutation file.</h2>

<h3>Description</h3>

<p>A dataset containing the information of eight mutations processed by the
fun_homology function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">msec_analyzed
</code></pre>


<h3>Format</h3>

<p>A list with 37 factors, each contains 29 variables
</p>

<dl>
<dt>Sample</dt>
<dd>
<p>sample name</p>
</dd>
<dt>Mut_type</dt>
<dd>
<p>mutation type</p>
</dd>
<dt>Chr</dt>
<dd>
<p>altered chromosome</p>
</dd>
<dt>Pos</dt>
<dd>
<p>altered position</p>
</dd>
<dt>Ref</dt>
<dd>
<p>reference base</p>
</dd>
<dt>Alt</dt>
<dd>
<p>altered base</p>
</dd>
<dt>SimpleRepeat_TRF</dt>
<dd>
<p>mutation locating repeat sequence</p>
</dd>
<dt>Neighborhood_sequence</dt>
<dd>
<p>neighborhood sequence</p>
</dd>
<dt>read_length</dt>
<dd>
<p>read length</p>
</dd>
<dt>total_read</dt>
<dd>
<p>number of mutation supporting reads</p>
</dd>
<dt>soft_clipped_read</dt>
<dd>
<p>number of soft-clipped reads</p>
</dd>
<dt>flag_hairpin</dt>
<dd>
<p>number of reads produced by hairpin structure</p>
</dd>
<dt>pre_support_length</dt>
<dd>
<p>maximum 5'-supporting length</p>
</dd>
<dt>post_support_length</dt>
<dd>
<p>maximum 3'-supporting length</p>
</dd>
<dt>short_support_length</dt>
<dd>
<p>minimum supporting length</p>
</dd>
<dt>pre_farthest</dt>
<dd>
<p>5'-farthest supported base from the mutated base</p>
</dd>
<dt>post_farthest</dt>
<dd>
<p>3'-farthest supported base from the mutated base</p>
</dd>
<dt>low_quality_base_rate_under_q18</dt>
<dd>
<p>low quality base rate</p>
</dd>
<dt>low_quality_pre</dt>
<dd>
<p>low quality base rate of 5'- side</p>
</dd>
<dt>low_quality_post</dt>
<dd>
<p>low quality base rate of 3'- side</p>
</dd>
<dt>distant_homology_rate</dt>
<dd>
<p>rate of reads derived from homologous regions</p>
</dd>
<dt>soft_clipped_rate</dt>
<dd>
<p>rate of soft clipped reads</p>
</dd>
<dt>prob_filter_1</dt>
<dd>
<p>possibility of short-supporting length</p>
</dd>
<dt>prob_filter_3_pre</dt>
<dd>
<p>possibility of 5'-supporting length</p>
</dd>
<dt>prob_filter_3_post</dt>
<dd>
<p>possibility of 3'-supporting length</p>
</dd>
<dt>filter_1_mutation_intra_hairpin_loop</dt>
<dd>
<p>filter 1</p>
</dd>
<dt>filter_2_hairpin_structure</dt>
<dd>
<p>filter 2</p>
</dd>
<dt>filter_3_microhomology_induced_mutation</dt>
<dd>
<p>filter 3</p>
</dd>
<dt>filter_4_highly_homologous_region</dt>
<dd>
<p>filter 4</p>
</dd>
<dt>filter_5_soft_clipping</dt>
<dd>
<p>filter 5</p>
</dd>
<dt>filter_6_simple_repeat</dt>
<dd>
<p>filter 6</p>
</dd>
<dt>filter_7_mutation_at_homopolymer</dt>
<dd>
<p>filter 7</p>
</dd>
<dt>filter_8_low_quality</dt>
<dd>
<p>filter 8</p>
</dd>
<dt>msec_filter_123</dt>
<dd>
<p>any of filter 1-3</p>
</dd>
<dt>msec_filter_1234</dt>
<dd>
<p>any of filter 1-4</p>
</dd>
<dt>msec_filter_all</dt>
<dd>
<p>any of filter 1-8</p>
</dd>
<dt>comment</dt>
<dd>
<p>comment</p>
</dd>
</dl>
<p>...

</p>


</div>