<div class="container">

<table style="width: 100%;"><tr>
<td>msa2mat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert alignment to matrix</h2>

<h3>Description</h3>

<p>Converts a FASTA formatted multiple alignment to a matrix.
</p>


<h3>Usage</h3>

<pre><code class="language-R">msa2mat(msa)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>msa</code></td>
<td>
<p>A fasta object with a multiple alignment, see <code>msalign</code>'.</p>
</td>
</tr></table>
<h3>Details</h3>

<p>This function converts the fasta object <code>msa</code>, containing a multiple alignment,
to a matrix. This means each position in the alignment is a column in the matrix, and the
content of the ‘<span class="samp">⁠Header⁠</span>’ column of <code>msa</code> is used as rownames of theh matrix.
</p>
<p>Such a matrix is useful for conversion to a <code>DNAbin</code> object that is used by the <code>ape</code>
package for re-constructing phylogenetic trees.
</p>


<h3>Value</h3>

<p>A <code>matrix</code> where each row is a vector of aligned bases/amino acids.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen.
</p>


<h3>See Also</h3>

<p><code>msalign</code>, <code>readFasta</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">msa.file &lt;- file.path(path.package("microseq"),"extdata", "small.msa")
msa &lt;- readFasta(msa.file)
msa.mat &lt;- msa2mat(msa)  # to use with ape::as.DNAbin(msa.mat)

</code></pre>


</div>