<div class="container">

<table style="width: 100%;"><tr>
<td>blastClassify16S</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Classifying using BLAST</h2>

<h3>Description</h3>

<p>A 16S based classification based on BLAST.
</p>


<h3>Usage</h3>

<pre><code class="language-R">blastClassify16S(sequence, bdb)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sequence</code></td>
<td>
<p>Character vector of 16S sequences to classify.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bdb</code></td>
<td>
<p>Name of BLAST data base, see <code>blastDbase16S</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>A vector of 16S sequences (DNA) are classified by first using BLAST <code>blastn</code> against
a database of 16S DNA sequences, and then classify according to the nearest-neighbour principle.
The nearest neighbour of a query sequence is the hit with the largest bitscore. The blast+
software  <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download">https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download</a>
must be installed on the system. Type <code>system("blastn -help")</code> in the Console window,
and a sensible Help-text should appear.
</p>
<p>The database must contain 16S sequences where the Header starts with a token specifying the taxon. 
More specifically, the tokens must look like:
</p>
<p>&lt;taxon&gt;_1
</p>
<p>&lt;taxon&gt;_2
</p>
<p>...etc
</p>
<p>where &lt;taxon&gt; is some proper taxon name. Use <code>blastDbase16S</code> to make such databases.
</p>
<p>The identity of each alignment is also computed. This should be close to 1.0 for a classification
to be trusted. Identity values below 0.95 could indicate uncertain classifications, but this will
vary between taxa.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> with two columns: Taxon is the predicted taxon for each <code>sequence</code>
and Identity is the corresponding identity-value. If no BLAST hit is seen, the sequence is 
<code>"unclassified"</code>.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen.
</p>


<h3>See Also</h3>

<p><code>blastDbase16S</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("small.16S")
## Not run: 
dbase &lt;- blastDbase16S("test", small.16S$Sequence, word(small.16S$Header, 2, 2))
reads &lt;- str_sub(small.16S$Sequence, 100, 550)
blastClassify16S(reads, dbase) %&gt;% 
  bind_cols(small.16S) -&gt; tbl

## End(Not run)

</code></pre>


</div>