<div class="container">

<table style="width: 100%;"><tr>
<td>entrezDownload</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Downloading genome data</h2>

<h3>Description</h3>

<p>Retrieving genomes from NCBI using the Entrez programming utilities.
</p>


<h3>Usage</h3>

<pre><code class="language-R">entrezDownload(accession, out.file, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>accession</code></td>
<td>
<p>A character vector containing a set of valid accession numbers at the NCBI
Nucleotide database.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out.file</code></td>
<td>
<p>Name of the file where downloaded sequences should be written in FASTA format.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical indicating if textual output should be given during execution, to monitor
the download progress.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The Entrez programming utilities is a toolset for automatic download of data from the
NCBI databases, see <a href="https://www.ncbi.nlm.nih.gov/books/NBK25500/">E-utilities Quick Start</a>
for details. <code>entrezDownload</code> can be used to download genomes from the NCBI Nucleotide
database through these utilities.
</p>
<p>The argument ‘<span class="samp">⁠accession⁠</span>’ must be a set of valid accession numbers at NCBI Nucleotide, typically
all accession numbers related to a genome (chromosomes, plasmids, contigs, etc). For completed genomes,
where the number of sequences is low, ‘<span class="samp">⁠accession⁠</span>’ is typically a single text listing all accession
numbers separated by commas. In the case of some draft genomes having a large number of contigs, the
accession numbers must be split into several comma-separated texts. The reason for this is that Entrez
will not accept too many queries in one chunk. 
</p>
<p>The downloaded sequences are saved in ‘<span class="samp">⁠out.file⁠</span>’ on your system. This will be a FASTA formatted file.
Note that all downloaded sequences end up in this file. If you want to download multiple genomes,
you call <code>entrezDownload</code> multiple times and store in multiple files.
</p>


<h3>Value</h3>

<p>The name of the resulting FASTA file is returned (same as <code>out.file</code>), but the real result of
this function is the creation of the file itself.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen and Kristian Liland.
</p>


<h3>See Also</h3>

<p><code>getAccessions</code>, <code>readFasta</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# Accession numbers for the chromosome and plasmid of Buchnera aphidicola, strain APS
acc &lt;- "BA000003.2,AP001071.1"
genome.file &lt;- tempfile(pattern = "Buchnera_aphidicola", fileext = ".fna")
txt &lt;- entrezDownload(acc, out.file = genome.file)

# ...cleaning...
ok &lt;- file.remove(genome.file)

## End(Not run)

</code></pre>


</div>