<div class="container">

<table style="width: 100%;"><tr>
<td>readBlastSelf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reads BLAST result files</h2>

<h3>Description</h3>

<p>Reads files from a search with blastpAllAll
</p>


<h3>Usage</h3>

<pre><code class="language-R">readBlastSelf(blast.files, e.value = 1, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>blast.files</code></td>
<td>
<p>A text vector of filenames.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>e.value</code></td>
<td>
<p>A threshold E-value to immediately discard (very) poor BLAST alignments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical, indicating if textual output should be given to monitor the progress.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The filenames given as input must refer to BLAST result files produced by <code>blastpAllAll</code>.
</p>
<p>With <code>readBlastSelf</code> you only read the self-alignment results, i.e. blasting a genome against itself. With
<code>readBlastPair</code> you read all the other files, i.e. different genomes compared. You may use all blast file
names as input to both, they will select the proper files based on their names, e.g. GID1_vs_GID1.txt is read
by <code>readBlastSelf</code> while GID2_vs_GID1.txt is read by <code>readBlastPair</code>.
</p>
<p>Setting a small ‘<span class="samp">⁠e.value⁠</span>’ threshold will filter the alignment, and may speed up this and later processing,
but you may also loose some important alignments for short sequences.
</p>
<p>Both these functions are used by <code>bDist</code>. The reason we provide them separately is to allow the user
to complete this file reading before calling <code>bDist</code>. If you have a huge number of files, a
skilled user may utilize parallell processing to speed up the reading. For normal size data sets (e.g. less than 100 genomes)
you should probably use <code>bDist</code> directly.
</p>


<h3>Value</h3>

<p>The functions returns a table with columns ‘<span class="samp">⁠Dbase⁠</span>’, ‘<span class="samp">⁠Query⁠</span>’, ‘<span class="samp">⁠Bitscore⁠</span>’
and ‘<span class="samp">⁠Distance⁠</span>’. Each row corresponds to a pair of sequences (a Dbase and a Query sequence) having at least
one BLAST hit between
them. All pairs <em>not</em> listed have distance 1.0 between them. You should normally bind the output from 
<code>readBlastSelf</code> to the ouptut from <code>readBlastPair</code> and use the result as input to <code>bDist</code>.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen.
</p>


<h3>See Also</h3>

<p><code>bDist</code>, <code>blastpAllAll</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Using BLAST result files in this package...
prefix &lt;- c("GID1_vs_GID1_",
            "GID2_vs_GID1_",
            "GID3_vs_GID1_",
            "GID2_vs_GID2_",
            "GID3_vs_GID2_",
            "GID3_vs_GID3_")
bf &lt;- file.path(path.package("micropan"), "extdata", str_c(prefix, ".txt.xz"))

# We need to uncompress them first...
blast.files &lt;- tempfile(pattern = prefix, fileext = ".txt.xz")
ok &lt;- file.copy(from = bf, to = blast.files)
blast.files &lt;- unlist(lapply(blast.files, xzuncompress))

# Reading self-alignment files, then the other files
self.tbl &lt;- readBlastSelf(blast.files)
pair.tbl &lt;- readBlastPair(blast.files)

# ...and cleaning...
ok &lt;- file.remove(blast.files)

# See also examples for bDist

</code></pre>


</div>