<div class="container">

<table style="width: 100%;"><tr>
<td>readFasta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read and write FASTA files</h2>

<h3>Description</h3>

<p>Reads and writes biological sequences (DNA, RNA, protein) in the FASTA format.
</p>


<h3>Usage</h3>

<pre><code class="language-R">readFasta(in.file)
writeFasta(fdta, out.file, width = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>in.file</code></td>
<td>
<p>url/directory/name of (gzipped) FASTA file to read.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fdta</code></td>
<td>
<p>A <code>tibble</code> with sequence data, see ‘Details’ below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out.file</code></td>
<td>
<p>Name of (gzipped) FASTA file to create.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>width</code></td>
<td>
<p>Number of characters per line, or 0 for no linebreaks.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>These functions handle input/output of sequences in the commonly used FASTA format.
For every sequence it is presumed there is one Header-line starting with a ‘&gt;’. If
filenames (<code>in.file</code> or <code>out.file</code>) have the extension <code>.gz</code> they will automatically be
compressed/uncompressed.
</p>
<p>The sequences are stored in a <code>tibble</code>, opening up all the possibilities in R for
fast and easy manipulations. The content of the file is stored as two columns, ‘<span class="samp">⁠Header⁠</span>’
and ‘<span class="samp">⁠Sequence⁠</span>’. If other columns are added, these will be ignored by <code>writeFasta</code>.
</p>
<p>The default <code>width = 0</code> in <code>writeFasta</code> results in no linebreaks in the sequences
(one sequence per line).
</p>


<h3>Value</h3>

<p><code>readFasta</code> returns a <code>tibble</code> with the contents of the (gzipped) FASTA
file stored in two columns of text. The first, named ‘<span class="samp">⁠Header⁠</span>’, contains
the headerlines and the second, named ‘<span class="samp">⁠Sequence⁠</span>’, contains the sequences.
</p>
<p><code>writeFasta</code> produces a (gzipped) FASTA file.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen and Kristian Hovde Liland.
</p>


<h3>See Also</h3>

<p><code>readFastq</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# We need a FASTA-file to read, here is one example file:
fa.file &lt;- file.path(file.path(path.package("microseq"),"extdata"),"small.ffn")

# Read and write
fdta &lt;- readFasta(fa.file)
ok &lt;- writeFasta(fdta[4:5,], out.file = "delete_me.fasta")

# Make use of dplyr to copy parts of the file to another file
readFasta(fa.file) %&gt;% 
  filter(str_detect(Sequence, "TGA$")) %&gt;% 
  writeFasta(out.file = "TGAstop.fasta", width = 80) -&gt; ok

## End(Not run)

</code></pre>


</div>