<div class="container">

<table style="width: 100%;"><tr>
<td>MiNorm</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Microarray data normalization</h2>

<h3>Description</h3>

<p>Normalizes microarray expression intensities using different methods with or without background correction.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MiNorm(Matrix, posNC, method = "none", leaveNC = TRUE, BGcor = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Matrix</code></td>
<td>
<p>numeric matrix of intensities data where each row corresponds to a probe (gene, transcript),
and each column correspondes to a specimen (patient).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>posNC</code></td>
<td>
<p>numeric vector specifying numbers of rows containing  negative controls (non-coding areas).
Used for <code>method="SQN"</code> only.
Rows with negative controls will be removed from an intensity matrix after the normalization if <code>leaveNC=FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>character string specifying normalization method. Possible values are:
<code>"none"</code> (no normalization)
<code>"center"</code> (subtracting the row mean),
<code>"scale"</code> (dividing by row standard deviation),
<code>"standardize"</code> (subtracting the row mean and dividing by row standard deviation - z-score transformation),
<code>"range"</code> (ranges from 0 to 1),
<code>"QN"</code> (normalization based upon quantiles),
<code>"SQN"</code> (subset quantile normalization using negative control features),
<code>"Loess"</code> (cyclicly applying loess normalization).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>leaveNC</code></td>
<td>
<p>logical value indicating whether rows with negative control should be deleted
from intensity matrix after normalization.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BGcor</code></td>
<td>
<p>logical value indicating whether background correction should be done before normalization.
Could be used for background correction only (without data normalization) if <code>method="none"</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is intended to normalize microarray intensities data between arrays.
Background correction is optional.
<br>
Background correction method is "normexp",  which is based on a convolution model (Ritchie, 2007).
See <code>backgroundCorrect</code> for details.
<br>
Quantile normalization method implies that we can give each array the same distribution
See <code>normalize.quantiles</code> for details.
<br>
Subset quantile normalization is performed based on a subset of negative (or non-coding) controls
according to (Wu and Aryee, 2010). Number of normal distributions in the mixture approximation is 5,
weight given to the parametric normal mixture model is 0.9. See <code>SQN</code> for details.
<br>
Cyclic loess normalization implements method of Ballman et al (2004),
whereby each array is normalized to the average of all the arrays.
See <code>normalizeCyclicLoess</code> for details.
</p>


<h3>Value</h3>

<p>A matrix of the same dimensions as <code>Matrix</code> containing normalized values with or without
background correction. If <code>leaveNC=FALSE</code> the function returns a matrix with
normalized values without rows containing negative controls.
</p>


<h3>Author(s)</h3>

<p>Elena N. Filatova
</p>


<h3>References</h3>

<p>Ballman K.V., Grill D.E., Oberg A.L. and Therneau T.M. (2004). Faster cyclic loess: normalizing RNA arrays via linear models. Bioinformatics 20, 2778-2786.
<a href="https://doi.org/10.1093/bioinformatics/bth327">https://doi.org/10.1093/bioinformatics/bth327</a>
<br><br>
Bolstad B.M., Irizarry R.A., Astrand M. and Speed T.P. (2003) A Comparison of Normalization Methods for High Density Oligonucleotide Array Data Based on Bias and Variance. Bioinformatics 19(2), 185-193.
<a href="https://doi.org/10.1093/bioinformatics/19.2.185">https://doi.org/10.1093/bioinformatics/19.2.185</a>
<br><br>
Ritchie M.E., Silver J., Oshlack A., Silver J., Holmes M., Diyagama D., Holloway A. and Smyth G.K. (2007). A comparison of background correction methods for two-colour microarrays. Bioinformatics 23, 2700-2707.
<a href="https://doi.org/10.1093/bioinformatics/btm412">https://doi.org/10.1093/bioinformatics/btm412</a>
<br><br>
Wu Z and Aryee M. (2010). Subset Quantile Normalization using Negative Control Features. Journal of Computational Biology 17(10), 1385-1395.
<a href="https://doi.org/10.1089/cmb.2010.0049">https://doi.org/10.1089/cmb.2010.0049</a>
</p>


<h3>See Also</h3>

<p><code>backgroundCorrect</code>, <code>normalizeCyclicLoess</code>,
<code>normalize.quantiles</code>, <code>SQN</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("IMexpression")
# Loess normalization
LoMatrix&lt;-MiNorm(IMexpression, method="Loess")
par(mfrow=c(1,2))
boxplot(log2(IMexpression),main="Before normalization")
boxplot(log2(LoMatrix),main="Loess normalization")
par(mfrow=c(1,1))


</code></pre>


</div>