<div class="container">

<table style="width: 100%;"><tr>
<td>MiDataSample</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Select matrix columns based on values of attendant vector</h2>

<h3>Description</h3>

<p>This function selects columns of a matrix that correspond to 1 or 2 factor levels of attendant vector.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MiDataSample(Matrix, specimens, group1, group2 = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Matrix</code></td>
<td>
<p>matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>specimens</code></td>
<td>
<p>factor vector with length equal to number of <code>Matrix</code> columns</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group1</code></td>
<td>
<p>value of factor level to sample</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group2</code></td>
<td>
<p>additional value of factor level to sample</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is ment for sampling specimens in  gene/transcript expression matrix for binary classification
in case when specimens belong to more than two groups. The aim is to create gene/transcript expression
matrix that contains specimens for only 1 or 2 groups.
Groups are specified in corresponding factor vector that contains specimens description.
It should be used together with <code>MiSpecimenSample</code> that samples specimens'
description in the same way.
</p>


<h3>Value</h3>

<p><code>Matrix</code> with rescricted number of columns that correspond to specimens from 1 or 2 groups.
</p>


<h3>Author(s)</h3>

<p>Elena N. Filatova
</p>


<h3>See Also</h3>

<p><code>MiSpecimenSample</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
#get gene expression and specimen data
data("IMexpression");data("IMspecimen")
dim(IMexpression) # 100 columns (genes/transcripts) - 89 specimens
colnames(IMexpression)[1:10] # look at first 10 columns of matrix - specimens IDs
IMspecimen[1:10,] # specimens IDs and group factor - diagnoses in attendant vector
# note that specimens in matrix columns are in the same order as specimens in description data
# select specimens with only EBV and NORM diagnoses (and sample the description data as well)
SampleMatrix&lt;-MiDataSample(IMexpression, IMspecimen$diagnosis, "ebv", "norm")
SampleSamples&lt;-MiSpecimenSample(IMspecimen$diagnosis, "ebv", "norm")
dim(SampleMatrix)# only 68 specimens with EBV and NORM diagnoses left
colnames(SampleMatrix)[1:10]
SampleSamples[1:10] # corresponding diagnoses

</code></pre>


</div>