<div class="container">

<table style="width: 100%;"><tr>
<td>association.test.logistic</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Mixed logistic regression for GWAS</h2>

<h3>Description</h3>

<p>Mixed logistic regression for GWAS
</p>


<h3>Usage</h3>

<pre><code class="language-R">association.test.logistic(
  x,
  Y = x@ped$pheno,
  X = matrix(1, nrow(x)),
  K,
  beg = 1,
  end = ncol(x),
  algorithm = c("amle", "offset"),
  eigenK,
  p = 0,
  model = c("additive", "dominant", "recessive"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a bedmatrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>phenotype vector. Default is column <code>pheno</code> of <code>x@ped</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>A matrix of covariates (defaults to a column of ones for the intercept)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>A genetic relationship matrix (or a list of such matrices)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beg</code></td>
<td>
<p>Index of the first SNP tested for association</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>end</code></td>
<td>
<p>Index of the last SNP tested for association</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>algorithm</code></td>
<td>
<p>Algorithm to use</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eigenK</code></td>
<td>
<p>eigen decomposition of K (only if <code>p</code> &gt; 0)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>Number of principal components to include in the model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>Model for the effect allele (allele A2)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional parameter for <code>gaston::logistic.mm.aireml</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Tests the association between the phenotype and requested SNPs in <code>x</code>.
The phenotype <code>Y</code> is a binary trait. A Wald test is performed using an approximate
method defined by the parameter <code>algorithm</code>. 
</p>
<p>Parameter <code>model</code> allows to specify an additive model (genotypes A1 A1, A1 A2, and A2 A2
are recoded for analysis as 0, 1 and 2 respectively), a dominant model (genotypes recoded as 0, 1, and 1) or a recessive
model (recoded as 0, 0 and 1).
</p>
<p>All other arguments are as in <code>gaston::association.test</code>.
</p>


<h3>Value</h3>

<p>A data frame giving for each SNP the association statistics.
</p>


<h3>See Also</h3>

<p><code>association.test</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(TTN)
x &lt;- as.bed.matrix(TTN.gen, TTN.fam, TTN.bim)
## Simulation data ##
set.seed(1)
# some covariables
X &lt;- cbind(1, runif(nrow(x)))
# A random GRM
ran &lt;- random.pm( nrow(x))
# random effects (tau = 1)
omega &lt;- lmm.simu(1, 0, eigenK=ran$eigen)$omega
# linear term of the model
lin &lt;- X %*% c(0.1,-0.2) + omega
# vector of probabilitues
pi &lt;- 1/(1+exp( -lin ))
# vector of binary phenotypes
y &lt;- rbinom(nrow(x), 1, pi)
# testing association with 1) the score test, 2) the offset algorithm, 3) the 'amle' algorithm
a1 &lt;- association.test(x, y, X, K = ran$K, method = "lmm", response = "bin")
a2 &lt;- association.test.logistic(x, y, X, K = ran$K, algorithm = "offset")
a3 &lt;- association.test.logistic(x, y, X, K = ran$K, algorithm = "amle")

</code></pre>


</div>