<div class="container">

<table style="width: 100%;"><tr>
<td>minimap2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>minimap2</h2>

<h3>Description</h3>

<p>This function is a wrapper for the command line tool minimap2.
minimap2 is a long read sequencing alignment tool that is
used to align long reads to a reference genome.
</p>


<h3>Usage</h3>

<pre><code class="language-R">minimap2(
  reference,
  query_sequences,
  output_file_prefix,
  a = TRUE,
  preset_string = "map-hifi",
  threads = 1,
  return = FALSE,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>reference</code></td>
<td>
<p>Reference genome to align the query sequences</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>query_sequences</code></td>
<td>
<p>Query sequences to align to the reference genome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output_file_prefix</code></td>
<td>
<p>Output file to save the alignment results</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>a</code></td>
<td>
<p>Logical value to use the preset string with the -a flag</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>preset_string</code></td>
<td>
<p>Preset string to use with the -x flag</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threads</code></td>
<td>
<p>Number of threads to use</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return</code></td>
<td>
<p>Logical value to return the alignment results</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical value to print progress of the installation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments to pass to minimap2</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>This function returns the line needed to add minimap2 to PATH
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
reference &lt;- system.file("extdata/S288C_ref_genome.fasta", package = "minimapR")
query_sequences &lt;- system.file("extdata/yeast_sample_hifi.fastq.gz", package = "minimapR")
out_dir &lt;- system.file("extdata/test_out", package = "minimapR")
output_file_prefix &lt;- paste0(out_dir, "/yeast_sample_hifi")
bam_out &lt;- minimap2(reference, 
 query_sequences, 
 output_file_prefix,
 threads = 4,
 preset_string = "map-hifi", 
 return = TRUE, 
 verbose = TRUE)

## End(Not run)

## Not run: 
reference &lt;- system.file("extdata/GRCh38_chr1_50m.fa", package = "minimapR")
query_sequences &lt;- system.file("extdata/ont_hs_sample.fastq.gz", package = "minimapR")
out_dir &lt;- system.file("extdata/test_out", package = "minimapR")
output_file_prefix &lt;- paste0(out_dir, "/ont_hs_sample")
bam_out &lt;- minimap2(reference, 
 query_sequences, 
 output_file_prefix,
 threads = 4,
 preset_string = "map-hifi",
 return = TRUE, 
 verbose = TRUE)

## End(Not run)

</code></pre>


</div>