<div class="container">

<table style="width: 100%;"><tr>
<td>process_allele</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2><code>process_allele</code></h2>

<h3>Description</h3>

<p><code>process_allele</code> is used to returned the processed allelic profiles, by
removing the allele profile with duplicate name and length different
from most. 1st allele profile with the duplicated name is returned,
the longer length is taken as normal should there be 2 modes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">process_allele(
  seqc,
  bp = BiocParallel::SerialParam(),
  check_length = TRUE,
  check_bases = TRUE,
  dash_ignore = TRUE,
  accepted_char = c("A", "C", "T", "G"),
  ignore_case = TRUE,
  remove_invariant = FALSE,
  biallelic_only = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqc</code></td>
<td>
<p>a list containing list of nucleotides. To keep it simple,
use provided read_fasta to import the fasta file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bp</code></td>
<td>
<p>is the biocparallel backend, default to serialParam,
most likely sufficient in most scenario</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check_length</code></td>
<td>
<p>Check the length of each sample in the matrix, default to TRUE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check_bases</code></td>
<td>
<p>Check the bases of each sample in the matrix, default to TRUE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dash_ignore</code></td>
<td>
<p>whether to treat '-' as another type</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>accepted_char</code></td>
<td>
<p>character to accept, default to c("A", "C", "T", "G")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ignore_case</code></td>
<td>
<p>whether to be case insensitive, default to TRUE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove_invariant</code></td>
<td>
<p>whether to remove invariant positions, default to FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>biallelic_only</code></td>
<td>
<p>whether to remove positions with more than 2 alleles, default to FALSE</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Will return the processed allelic profiles.
</p>


</div>