<div class="container">

<table style="width: 100%;"><tr>
<td>flag_allele</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2><code>flag_allele</code></h2>

<h3>Description</h3>

<p><code>flag_allele</code> is used to find out a list of samples that has been
flagged with length different from most and will not be included in
computation of minimum SNPs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">flag_allele(seqc, bp = BiocParallel::SerialParam())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqc</code></td>
<td>
<p>a list containing list of nucleotides. To keep it simple,
use provided read_fasta to import the fasta file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bp</code></td>
<td>
<p>is the biocparallel backend, default to serialParam,
most likely sufficient in most scenario</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Will return a list of ignored samples.
</p>


</div>