<div class="container">

<table style="width: 100%;"><tr>
<td>infer_from_combined</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2><code>infer_from_combined</code></h2>

<h3>Description</h3>

<p><code>infer_from_combined</code> infers the results (presence/absense of genes &amp; CC) from the combined result
</p>


<h3>Usage</h3>

<pre><code class="language-R">infer_from_combined(combined_result, search_table, genome_size, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>combined_result</code></td>
<td>
<p>the combined result from <code>combine_fastq_search_result</code> or equivalent,
with a list containing:
- result: a dataframe containing the following columns:
'sequence', 'search_id', 'reads', 'raw_match', 'mean_qualities', 'indexes', 'id', 'type',
'strand', 'result', 'extra', 'match_ref_seq', 'n_reads'
- read_length:
'reads_id', 'reads_length'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>search_table</code></td>
<td>
<p>a dataframe with the following columns:
- "id","type","sequence","strand","result","extra","match_ref_seq"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome_size</code></td>
<td>
<p>estimated genome size for coverage calculation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional arguments to pass to the process methods</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a dataframe containing the following columns:
- type, rank, result, reads_count, proportion_matched, pass_filter
</p>


</div>