<div class="container">

<table style="width: 100%;"><tr>
<td>inner.KRV</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Kernel RV Coefficient Test; Inner Function</h2>

<h3>Description</h3>

<p>Function called when user calls function KRV. For each kernel matrix inputted into KRV, KRV runs inner.KRV on that 
kernel with the inputted kernel.y outcome matrix.
</p>


<h3>Usage</h3>

<pre><code class="language-R">inner.KRV(
  y = NULL,
  X = NULL,
  adjust.type,
  kernel.otu,
  kernel.y,
  returnKRV = FALSE,
  returnR2 = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>A numeric n by p matrix of p continuous phenotype variables and 
sample size n (default = NULL). If it is NULL, a 
phenotype kernel matrix must be entered for "kernel.y". Defaults to NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>A numeric n by q matrix, containing q additional covariates 
(default = NULL). If NULL, an intercept only model is used. If the first
column of X is not uniformly 1, then an intercept column will be added.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adjust.type</code></td>
<td>
<p>Possible values are "none" (default if X is null), 
"phenotype" to adjust only the y variable (only possible if y is a numeric 
phenotype matrix rather than a pre-computed kernel), or "both" to adjust 
both the X and Y kernels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kernel.otu</code></td>
<td>
<p>A numeric OTU n by n kernel matrix or a list of matrices, 
where n is the sample size. It can be constructed from microbiome data, such 
as by transforming from a distance metric.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kernel.y</code></td>
<td>
<p>Either a numerical n by n kernel matrix for phenotypes or a 
method to compute the kernel of phenotype. Methods are "Gaussian" or "linear". 
A Gaussian kernel (kernel.y="Gaussian") can capture the general relationship 
between microbiome and phenotypes; a linear kernel (kernel.y="linear") 
may be preferred if the underlying relationship is close to linear.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>returnKRV</code></td>
<td>
<p>A logical indicating whether to return the KRV statistic. Defaults to FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>returnR2</code></td>
<td>
<p>A logical indicating whether to return the R-squared coefficient. Defaults to FALSE.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>y and X (if not NULL) should all be numerical matrices or vectors with the same number of rows.
</p>
<p>Ks should be a list of n by n matrices or a single matrix. If you have distance metric from metagenomic data, each kernel can be
constructed through function D2K. Each kernel can also be constructed through other mathematical approaches.
</p>
<p>Missing data is not permitted. Please remove all individuals with missing y, X, Ks prior to analysis
</p>
<p>Parameter "method" only concerns how kernel specific p-values are generated. When Ks is a list of multiple kernels, omnibus
p-value is computed through permutation from each individual p-value, which are calculated through method of choice.
</p>


<h3>Value</h3>

<p>Returns a p-value for the candidate kernel matrix
</p>
<table>
<tr style="vertical-align: top;">
<td><code>pv</code></td>
<td>
<p>p-value for the candidate kernel matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>KRV</code></td>
<td>
<p>KRV statistic for the candidate kernel matrix. Only returned if returnKRV = TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>R2</code></td>
<td>
<p>R-squared for the candidate kernel matrix. Only returned if returnR2 = TRUE.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Haotian Zheng, Xiang Zhan, Ni Zhao
</p>


<h3>References</h3>

<p>Zhan, X., Plantinga, A., Zhao, N., and Wu, M.C. A Fast Small-Sample Kernel Independence Test for Microbiome Community-Level 
Association Analysis. Biometrics. 2017 Mar 10. doi: 10.1111/biom.12684.
</p>


</div>