<div class="container">

<table style="width: 100%;"><tr>
<td>MiRKAT.R</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Robust MiRKAT (robust regression)</h2>

<h3>Description</h3>

<p>A more robust version of MiRKAT utilizing a linear model by robust regression using an M estimator.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MiRKAT.R(y, X, Ks, omnibus = "kernel_om", returnKRV = FALSE, returnR2 = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>A numeric vector of the a continuous or dichotomous outcome variable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>A numerical matrix or data frame, containing additional covariates that you want to adjust for Mustn't be NULL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Ks</code></td>
<td>
<p>list of n by n kernel matrices (or a single n by n kernel matrix), where n is the sample size. It can be constructed from 
microbiome data through distance metric or other approaches, such as linear kernels or Gaussian kernels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>omnibus</code></td>
<td>
<p>A string equal to either "Cauchy" or "kernel_om" (or nonambiguous abbreviations thereof), specifying whether 
to use the Cauchy combination test or an omnibus kernel to generate the omnibus p-value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>returnKRV</code></td>
<td>
<p>A logical indicating whether to return the KRV statistic. Defaults to FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>returnR2</code></td>
<td>
<p>A logical indicating whether to return the R-squared coefficient. Defaults to FALSE.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>MiRKAT.R creates a kernel matrix using the linear model created with the function rlm, a robust regression function, then does 
the KRV analysis on Ks and the newly formed kernel matrix representing the outome traits. 
</p>
<p>y and X should all be numerical matrices or vectors with the same number of rows, and mustn't be NULL.
</p>
<p>Ks should be a list of n by n matrices or a single matrix. If you have distance metric from metagenomic data, each kernel can be 
constructed through function D2K. Each kernel may also be constructed through other mathematical approaches.
</p>
<p>Missing data is not permitted. Please remove all individuals with missing y, X, Ks prior to analysis
</p>


<h3>Value</h3>

<p>Returns p-values for each individual kernel matrix, an omnibus p-value if multiple kernels were provided, and measures of effect size KRV and R2. 
</p>
<table>
<tr style="vertical-align: top;">
<td><code>p_values</code></td>
<td>
<p>labeled individual p-values for each kernel</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>omnibus_p</code></td>
<td>
<p>omnibus p_value, calculated as for the KRV test</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>KRV</code></td>
<td>
<p>A vector of kernel RV statistics (a measure of effect size), one for each candidate kernel matrix. Only returned if returnKRV = TRUE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>R2</code></td>
<td>
<p>A vector of R-squared statistics, one for each candidate kernel matrix. Only returned if returnR2 = TRUE</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Weijia Fu
</p>


<h3>Examples</h3>

<pre><code class="language-R">    


# Generate data
library(GUniFrac)

data(throat.tree)
data(throat.otu.tab)
data(throat.meta)

unifracs = GUniFrac(throat.otu.tab, throat.tree, alpha = c(1))$unifracs
if (requireNamespace("vegan")) {
 library(vegan)
 BC= as.matrix(vegdist(throat.otu.tab, method="bray"))
 Ds = list(w = unifracs[,,"d_1"], uw = unifracs[,,"d_UW"], BC = BC)
} else {
 Ds = list(w = unifracs[,,"d_1"], uw = unifracs[,,"d_UW"])
}
Ks = lapply(Ds, FUN = function(d) D2K(d))

covar = cbind(throat.meta$Age, as.numeric(throat.meta$Sex == "Male"))

# Continuous phenotype
n = nrow(throat.meta)
y = rchisq(n, 2)
MiRKAT.R(y, X = covar, Ks = Ks)

</code></pre>


</div>