<div class="container">

<table style="width: 100%;"><tr>
<td>track_wkflow</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Track the number of reads (= sequences), samples and cluster (e.g. ASV)
from various objects including dada-class and derep-class.</h2>

<h3>Description</h3>


<a href="https://adrientaudiere.github.io/MiscMetabar/articles/Rules.html#lifecycle">
<img src="https://img.shields.io/badge/lifecycle-maturing-blue" alt="lifecycle-maturing"></a>


<ul>
<li>
<p> List of fastq and fastg.gz files -&gt; nb of reads and samples
</p>
</li>
<li>
<p> List of dada-class -&gt; nb of reads, clusters (ASV) and samples
</p>
</li>
<li>
<p> List of derep-class -&gt; nb of reads, clusters (unique sequences)
and samples
</p>
</li>
<li>
<p> Matrix of samples x clusters (e.g. <code>otu_table</code>) -&gt; nb of reads,
clusters and samples
</p>
</li>
<li>
<p> Phyloseq-class -&gt; nb of reads, clusters and samples
</p>
</li>
</ul>
<h3>Usage</h3>

<pre><code class="language-R">track_wkflow(
  list_of_objects,
  obj_names = NULL,
  clean_pq = FALSE,
  taxonomy_rank = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>list_of_objects</code></td>
<td>
<p>(required) a list of objects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>obj_names</code></td>
<td>
<p>A list of names corresponding to the list of objects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clean_pq</code></td>
<td>
<p>(logical)
If set to TRUE, empty samples and empty ASV are discarded
before clustering.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxonomy_rank</code></td>
<td>
<p>A vector of int. Define the column number of
taxonomic rank <code style="white-space: pre;">⁠in physeq@tax_table⁠</code> to compute the number of unique value.
Default is NULL and do not compute values for any taxonomic rank</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Other arguments passed on to <code>clean_pq()</code> function.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The number of sequences, clusters (e.g. OTUs, ASVs) and samples for
each object.
</p>


<h3>See Also</h3>

<p><code>track_wkflow_samples()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(enterotype)
if (requireNamespace("pbapply")) {
  track_wkflow(list(data_fungi, enterotype), taxonomy_rank = c(3, 5))
}

</code></pre>


</div>