<div class="container">

<table style="width: 100%;"><tr>
<td>build_phytree_pq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Build phylogenetic trees from refseq slot of a phyloseq object</h2>

<h3>Description</h3>


<a href="https://adrientaudiere.github.io/MiscMetabar/articles/Rules.html#lifecycle">
<img src="https://img.shields.io/badge/lifecycle-experimental-orange" alt="lifecycle-experimental"></a>

<p>This function build tree phylogenetic tree and if nb_bootstrap is
set, it build also the 3 corresponding bootstrapped tree.
</p>
<p>Default parameters are based on <a href="https://doi.org/10.12688/f1000research.8986.2">doi:10.12688/f1000research.8986.2</a>
and phangorn vignette <a href="https://klausvigo.github.io/phangorn/articles/Trees.html">Estimating phylogenetic trees with phangorn</a>. You should understand your data, especially the markers,
before using this function.
</p>
<p>Note that phylogenetic reconstruction with markers used for metabarcoding are
not robust. You must verify the robustness of your phylogenetic tree using
taxonomic classification (see vignette <a href="https://adrientaudiere.github.io/MiscMetabar/articles/tree_visualization.html">Tree visualization</a>) and bootstrap or multi-tree visualization
</p>


<h3>Usage</h3>

<pre><code class="language-R">build_phytree_pq(
  physeq,
  nb_bootstrap = 0,
  model = "GTR",
  optInv = TRUE,
  optGamma = TRUE,
  rearrangement = "NNI",
  control = phangorn::pml.control(trace = 0),
  optNni = TRUE,
  multicore = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nb_bootstrap</code></td>
<td>
<p>(default 0): If a positive number is set,
the function also build 3 bootstrapped trees using <code>nb_bootstrap</code>
bootstrap samples</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>allows to choose an amino acid models or nucleotide model,
see <code>phangorn::optim.pml()</code> for more details</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>optInv</code></td>
<td>
<p>Logical value indicating whether topology gets optimized
(NNI). See <code>phangorn::optim.pml()</code> for more details</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>optGamma</code></td>
<td>
<p>Logical value indicating whether gamma rate parameter gets
optimized. See <code>phangorn::optim.pml()</code> for more details</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rearrangement</code></td>
<td>
<p>type of tree tree rearrangements to perform, one of
"NNI", "stochastic" or "ratchet"
see <code>phangorn::optim.pml()</code> for more details</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>control</code></td>
<td>
<p>A list of parameters for controlling the fitting process.
see <code>phangorn::optim.pml()</code> for more details</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>optNni</code></td>
<td>
<p>Logical value indicating whether topology gets optimized (NNI).
see <code>phangorn::optim.pml()</code> for more details</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>multicore</code></td>
<td>
<p>(logical) whether models should estimated in parallel.
see <code>phangorn::bootstrap.pml()</code> for more details</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Other params for be passed on to
<code>phangorn::optim.pml()</code> function</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>phangorn</code> package if you
use this function.
</p>


<h3>Value</h3>

<p>A list of phylogenetic tree
</p>


<h3>Author(s)</h3>

<p>Adrien Taudi√®re
</p>


<h3>Examples</h3>

<pre><code class="language-R">

if (requireNamespace("phangorn")) {
  set.seed(22)
  df &lt;- subset_taxa_pq(data_fungi_mini, taxa_sums(data_fungi_mini) &gt; 9000)
  df_tree &lt;- build_phytree_pq(df, nb_bootstrap = 2)
  plot(df_tree$UPGMA)
  phangorn::plotBS(df_tree$UPGMA, df_tree$UPGMA_bs, main = "UPGMA")
  plot(df_tree$NJ, "unrooted")
  plot(df_tree$ML)

  phangorn::plotBS(df_tree$ML$tree, df_tree$ML_bs, p = 20, frame = "circle")
  phangorn::plotBS(
    df_tree$ML$tree,
    df_tree$ML_bs,
    p = 20,
    frame = "circle",
    method = "TBE"
  )
  plot(phangorn::consensusNet(df_tree$ML_bs))
  plot(phangorn::consensusNet(df_tree$NJ_bs))
  ps_tree &lt;- merge_phyloseq(df, df_tree$ML$tree)
}


</code></pre>


</div>