<div class="container">

<table style="width: 100%;"><tr>
<td>swarm_clustering</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Re-cluster sequences of an object of class <code>physeq</code>
or cluster a list of DNA sequences using SWARM</h2>

<h3>Description</h3>


<a href="https://adrientaudiere.github.io/MiscMetabar/articles/Rules.html#lifecycle">
<img src="https://img.shields.io/badge/lifecycle-maturing-blue" alt="lifecycle-maturing"></a>

<p>A wrapper of SWARM software.
</p>


<h3>Usage</h3>

<pre><code class="language-R">swarm_clustering(
  physeq = NULL,
  dna_seq = NULL,
  d = 1,
  swarmpath = "swarm",
  vsearch_path = "vsearch",
  nproc = 1,
  swarm_args = "--fastidious",
  tax_adjust = 0,
  keep_temporary_files = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dna_seq</code></td>
<td>
<p>NOT WORKING FOR THE MOMENT
You may directly use a character vector of DNA sequences
in place of physeq args. When physeq is set, dna sequences take the value of
<code>physeq@refseq</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d</code></td>
<td>
<p>(default: 1) maximum number of differences allowed between two
amplicons, meaning that two amplicons will be grouped if they have <code>d</code>
(or less) differences</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>swarmpath</code></td>
<td>
<p>(default: swarm) path to swarm</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vsearch_path</code></td>
<td>
<p>(default: vsearch) path to vsearch, used only if physeq
is NULL and dna_seq is provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nproc</code></td>
<td>
<p>(default: 1)
Set to number of cpus/processors to use for the clustering</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>swarm_args</code></td>
<td>
<p>(default : "â€“fastidious") a one length character
element defining other parameters to  passed on to swarm See other possible
methods in the <a href="https://github.com/torognes/swarm/blob/master/man/swarm_manual.pdf">SWARM pdf manual</a></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tax_adjust</code></td>
<td>
<p>(Default 0) See the man page
of <code>merge_taxa_vec()</code> for more details.
To conserved the taxonomic rank of the most abundant ASV,</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep_temporary_files</code></td>
<td>
<p>(logical, default: FALSE) Do we keep temporary
files ?
</p>

<ul>
<li>
<p> temp.fasta (refseq in fasta or dna_seq sequences)
</p>
</li>
<li>
<p> temp_output (classical output of SWARM)
</p>
</li>
<li>
<p> temp_uclust (clusters output of SWARM)
</p>
</li>
</ul>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function use the <code>merge_taxa_vec</code> function to
merge taxa into clusters. By default tax_adjust = 0. See the man page
of <code>merge_taxa_vec()</code>.
</p>
<p>This function is mainly a wrapper of the work of others.
Please cite <a href="https://github.com/torognes/swarm">SWARM</a>.
</p>


<h3>Value</h3>

<p>A new object of class <code>physeq</code> or a list of cluster if dna_seq
args was used.
</p>


<h3>References</h3>

<p>SWARM can be downloaded from
<a href="https://github.com/torognes/swarm/">https://github.com/torognes/swarm/</a>.
</p>
<p>SWARM can be downloaded from
<a href="https://github.com/torognes/swarm">https://github.com/torognes/swarm</a>.
More information in the associated publications
<a href="https://doi.org/10.1093/bioinformatics/btab493">doi:10.1093/bioinformatics/btab493</a> and <a href="https://doi.org/10.7717/peerj.593">doi:10.7717/peerj.593</a>
</p>


<h3>See Also</h3>

<p><code>postcluster_pq()</code>, <code>vsearch_clustering()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
summary_plot_pq(data_fungi)
system2("swarm", "-h")

data_fungi_swarm &lt;- swarm_clustering(data_fungi)
summary_plot_pq(data_fungi_swarm)

sequences_ex &lt;- c(
  "TACCTATGTTGCCTTGGCGGCTAAACCTACCCGGGATTTGATGGGGCGAATTAATAACGAATTCATTGAATCA",
  "TACCTATGTTGCCTTGGCGGCTAAACCTACCCGGGATTTGATGGGGCGAATTACCTGGTAAGGCCCACTT",
  "TACCTATGTTGCCTTGGCGGCTAAACCTACCCGGGATTTGATGGGGCGAATTACCTGGTAGAGGTG",
  "TACCTATGTTGCCTTGGCGGCTAAACCTACC",
  "CGGGATTTGATGGCGAATTACCTGGTATTTTAGCCCACTTACCCGGTACCATGAGGTG",
  "GCGGCTAAACCTACCCGGGATTTGATGGCGAATTACCTGG",
  "GCGGCTAAACCTACCCGGGATTTGATGGCGAATTACAAAG",
  "GCGGCTAAACCTACCCGGGATTTGATGGCGAATTACAAAG",
  "GCGGCTAAACCTACCCGGGATTTGATGGCGAATTACAAAG"
)

sequences_ex_swarm &lt;- swarm_clustering(
  dna_seq = sequences_ex
)

</code></pre>


</div>