<div class="container">

<table style="width: 100%;"><tr>
<td>MiSPU</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>microbiome based sum of powered score (MiSPU)</h2>

<h3>Description</h3>

<p>We propose a class of microbiome based sum of powered score (MiSPU) tests based on a newly defined generalized taxon proportion that combines observed microbial composition information with phylogenetic tree information. Different from the existing methods, a MiSPU test is based on a weighted score of the generalized taxon proportion in a general framework of regression, upweighting more likely to be associated microbial lineages. Our simulations demonstrated that one or more MiSPU tests were more powerful than MiRKAT while correctly controlling type I error rates. An adaptive MiSPU (aMiSPU) test is proposed to combine multiple MiSPU tests with various weights, approximating the most powerful MiSPU for a given scenario, consequently being highly adaptive and high powered across various scenarios.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MiSPU(y, X, tree, cov = NULL,model = c("gaussian", "binomial"),
 pow = c(2:8, Inf), n.perm = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>Outcome of interest. It can be a disease indicator; =0 for controls, =1 for cases.
Or it can be a quantitative trait. A vector with length n (number of observations).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>OTU count table, row - n sample, column - q OTU</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree</code></td>
<td>
<p>Rooted phylogenetic tree of R class “phylo”</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cov</code></td>
<td>
<p>Covariates. A matrix with dimension n by p (n :number of observation, p : number of covariates).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>Use "gaussian" for a quantitative trait, and use "binomial" for a binary trait.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pow</code></td>
<td>
<p>The gamma which controls the weight. Larger pow puts more weight on the variables that have larger absolute score.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.perm</code></td>
<td>
<p>number of permutations or bootstraps.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list object, including the results for MiSPU_u, MiSPU_w and aMiSPU.
</p>


<h3>Author(s)</h3>

<p>Chong Wu
</p>


<h3>References</h3>

<p>Pan, W., et al.(2014) A powerful and adaptive association test for rare variants, Genetics, 197(4), 1081-95
</p>
<p>Chong, W., Pan, W. (2015) An Adaptive Association Test for Microbiome Data, submitted. 
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(throat.otu.tab)
data(throat.tree)
data(throat.meta)

Y.tmp =throat.meta[,3]
Y = rep(0,dim(throat.meta)[1])
Y[Y.tmp=="Smoker"] = 1

cov.tmp = throat.meta[,c(10,12)]
cov = matrix(1,dim(throat.meta)[1],2)
cov[cov.tmp[,1]== "None",1] = 0
cov[cov.tmp[,2]== "Male",2] = 0

start.time = proc.time()
X = as.matrix(throat.otu.tab)

out = MiSPU(Y,X, throat.tree,cov,model =  "binomial", pow = c(2:8, Inf), n.perm = 1000)
out
</code></pre>


</div>