<div class="container">

<table style="width: 100%;"><tr>
<td>GUniFrac</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Generalized UniFrac distances for comparing microbial communities.
</h2>

<h3>Description</h3>

<p>A generalized version of commonly used UniFrac distances. It is defined as:
</p>
<p style="text-align: center;"><code class="reqn">d^{(\alpha)} = \frac{\sum_{i=1}^m b_i (p^A_{i} + p^B_{i})^\alpha 
\left\vert \frac{ p^A_{i} - p^B_{i} }{p^A_{i} + p^B_{i}} \right\vert } { 
 \sum_{i=1}^m b_i (p^A_{i} + p^B_{i})^\alpha},</code>
</p>

<p>where <code class="reqn">m</code> is the number of branches, <code class="reqn">b_i</code> is the length of 
<code class="reqn">i</code>th branch, <code class="reqn">p^A_{i}, p^B_{i}</code> are the branch
proportion for community A and B. 
</p>
<p>Generalized UniFrac distance contains an extra parameter <code class="reqn">\alpha</code> 
controlling the weight on abundant lineages so the distance is not dominated 
by highly abundant lineages. <code class="reqn">\alpha=0.5</code> has overall the best 
power. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">GUniFrac(otu.tab, tree,alpha = c(0,0.5,1))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>otu.tab</code></td>
<td>


<p>OTU count table, row - n sample, column - q OTU
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree</code></td>
<td>


<p>Rooted phylogenetic tree of R class “phylo”
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>Parameter controlling weight on abundant lineages</p>
</td>
</tr>
</table>
<h3>Value</h3>






<p>Return a list containing
</p>
<table>
<tr style="vertical-align: top;">
<td><code>d0</code></td>
<td>
<p>UniFrac(0)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d5</code></td>
<td>
<p>UniFrac(0.5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d1</code></td>
<td>
<p>UniFrac(1), weighted UniFrac</p>
</td>
</tr>
</table>
<p>or a list containing
</p>
<table>
<tr style="vertical-align: top;">
<td><code>GUniFrac</code></td>
<td>
<p>The distance matrix for different alpha</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>The weight</p>
</td>
</tr>
</table>
<h3>Note</h3>


<p>The time consuming part is written in C and faster than the original one. The function only accepts rooted tree. 
</p>


<h3>Author(s)</h3>


<p>Chong Wu &lt;chongwu@umn.edu&gt;
</p>


<h3>References</h3>


<p>Chen, Jun, et al (2012). "Associating microbiome composition with environmental covariates using generalized UniFrac distances." Bioinformatics 28(16):2106-2113.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(throat.otu.tab)
data(throat.tree)
data(throat.meta)

groups &lt;- throat.meta$SmokingStatus

# Calculate the UniFracs
unifracs &lt;- GUniFrac(throat.otu.tab, throat.tree)
unifracs
</code></pre>


</div>