<div class="container">

<table style="width: 100%;"><tr>
<td>plot_genome_coverage</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>plot_genome_coverage</h2>

<h3>Description</h3>

<p>plot_genome_coverage
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_genome_coverage(biomart_file, path_fasta, chr_name, heteroplasmy_matrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>biomart_file</code></td>
<td>
<p>Character string with full path to the txt file
downloaded from BioMart <a href="https://m.ensembl.org/info/data/biomart/index.html">https://m.ensembl.org/info/data/biomart/index.html</a> . It must have
the following five columns:Gene.stable.ID, Gene.name, Gene.start..bp.,
Gene.end..bp., Chromosome.scaffold.name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path_fasta</code></td>
<td>
<p>Character string with full path to the fasta file of the
genomic region of interest. It should be the same file used in
<em>get_raw_counts_allele</em>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr_name</code></td>
<td>
<p>Character specifying the name of the chromosome of interest.
It must be one of the names in the <em>Chromosome.scaffold.name</em> column
from the <em>biomart_file</em>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>heteroplasmy_matrix</code></td>
<td>
<p>Third element returned by <em>get_heteroplasmy</em>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>KaryoPlot object as returned by <em>plotKaryotype</em> function from package <em>karyoploteR</em>.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>See Also</h3>

<p><a href="http://bioconductor.org/packages/release/bioc/html/karyoploteR.html">http://bioconductor.org/packages/release/bioc/html/karyoploteR.html</a>
</p>


</div>