<div class="container">

<table style="width: 100%;"><tr>
<td>data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>simulated data for demonstrating the features of mixedBayes</h2>

<h3>Description</h3>

<p>Simulated gene expression data for demonstrating the features of mixedBayes.
</p>


<h3>Format</h3>

<p>The data object consists of seven components: y, e, X, g, w ,k and coeff. coeff contains the true values of parameters used for generating Y.
</p>


<h3>Details</h3>

<p><strong>The data and model setting</strong>
</p>
<p>Consider a longitudinal study on <code class="reqn">n</code> subjects with <code class="reqn">k</code> repeated measurement for each subject. Let <code class="reqn">Y_{ij}</code> be the measurement for the <code class="reqn">i</code>th subject at each time point <code class="reqn">j</code>(<code class="reqn">1\leq i \leq n, 1\leq j \leq k</code>) .We use a <code class="reqn">m</code>-dimensional vector <code class="reqn">G_{ij}</code> to denote the genetics factors, where <code class="reqn">G_{ij} = (G_{ij1},...,G_{ijm})^\top</code>. Also, we use <code class="reqn">p</code>-dimensional vector <code class="reqn">E_{ij}</code> to denote the environment factors, where <code class="reqn">E_{ij} = (E_{ij1},...,E_{ijp})^\top</code>. <code class="reqn">X_{ij} = (1, T_{ij})^\top</code>, where <code class="reqn">T_{ij}^\top</code> is a vector of time effects . <code class="reqn">Z_{ij}</code> is a <code class="reqn">h \times 1</code> covariate associated with random effects and <code class="reqn">\alpha_{i}</code> is a <code class="reqn">h\times 1</code> vector of random effects.  At the beginning,  the interaction effects is modeled as the product of genomics features and environment factors with 4 different levels. After representing the environment factors as three dummy variables, the identification of the gene by environment interaction needs to be performed as group level.  Combing the genetics factors, environment factors and their interactions that associated with the longitudinal phenotype, we have the following mixed-effects model:
</p>
<p style="text-align: center;"><code class="reqn">Y_{ij} = X_{ij}^\top\gamma_{0}+E_{ij}^\top\gamma_{1}+G_{ij}^\top\gamma_{2}+(G_{ij}\bigotimes E_{ij})^\top\gamma_{3}+Z_{ij}^\top\alpha_{i}+\epsilon_{ij}.</code>
</p>

<p>where <code class="reqn">\gamma_{1}</code>,<code class="reqn">\gamma_{2}</code>,<code class="reqn">\gamma_{3}</code> are <code class="reqn">p</code>,<code class="reqn">m</code> and <code class="reqn">mp</code> dimensional vectors that represent the coefficients of the environment effects, the genetics effects and interactions effects, respectively. Accommodating the Kronecker product of the <code class="reqn">m</code> - dimensional vector <code class="reqn">G_{ij}</code> and the <code class="reqn">p</code>-dimensional vector <code class="reqn">E_{ij}</code>, the interactions between genetics and environment factors can be expressed as a <code class="reqn">mp</code>-dimensional vector, denoted as the following form:
</p>
<p style="text-align: center;"><code class="reqn">G_{ij}\bigotimes E_{ij} = [E_{ij1}E_{ij1},E_{ij2}E_{ij2},...,E_{ij1}E_{ijp},E_{ij2}E_{ij1},...,E_{ijm}E_{ijp}]^\top.</code>
</p>

<p>For random intercept and slope model, <code class="reqn">Z_{ij}^\top = (1,j)</code> and  <code class="reqn">\alpha_{i} = (\alpha_{i1},\alpha_{i2})^\top</code>. For random intercept model, <code class="reqn">Z_{ij}^\top = 1</code> and <code class="reqn">\alpha_{i} = \alpha_{i1}</code>.
</p>


<h3>See Also</h3>

<p><code>mixedBayes</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(data)
length(y)
dim(g)
dim(e)
dim(w)
print(k)
print(X)
print(coeff)

</code></pre>


</div>