<div class="container">

<table style="width: 100%;"><tr>
<td>TARAoceans</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>TARA ocean microbiome data</h2>

<h3>Description</h3>

<p>The TARA Oceans expedition facilitated the study of plankton communities by 
providing oceans metagenomic data combined with environmental measures to the 
scientific community. This dataset focuses on 139 prokaryotic-enriched 
samples collected from 68 stations and spread across three depth layers: the 
surface (SRF), the deep chlorophyll maximum (DCM) layer and the mesopelagic 
(MES) zones. Samples were located in height different oceans or seas: Indian 
Ocean (IO), Mediterranean Sea (MS), North Atlantic Ocean (NAO), North Pacific
Ocean (NPO), Red Sea (RS), South Atlantic Ocean (SAO), South Pacific Ocean 
(SPO) and South Ocean (SO). Here, only a subset of the original data is 
provided (1% of the 35,650 prokaryotic operational taxonomic units (OTUs) 
and of the 39,246 bacterial genes (NOGs) (selected at random).
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(TARAoceans)
</code></pre>


<h3>Format</h3>

<p>A list containing the following components:
</p>

<dl>
<dt><code>phychem</code></dt>
<dd>
<p>data matrix with 139 rows and 22 columns. Each row 
represents a sample and each column an environmental variable.</p>
</dd>
<dt><code>pro.phylo</code></dt>
<dd>
<p>data matrix with 139 rows (samples) and 356 columns
(prokaryotic OTUs).</p>
</dd>
<dt><code>taxonomy</code></dt>
<dd>
<p>data matrix with 356 rows (prokaryotic OTUs) and 6 
columns indicating the taxonomy of each OTU.</p>
</dd>
<dt><code>phylogenetic.tree</code></dt>
<dd>
<p>a phylo object (see package 'ape') 
representing the prokaryotic OTUs phylogenetic tree.</p>
</dd>
<dt><code>pro.NOGs</code></dt>
<dd>
<p>data matrix with 139 rows (samples) and 638 columns 
(NOGs).</p>
</dd>
<dt><code>sample</code></dt>
<dd>
<p>a list containing three following entries (all three 
are character vectors): <code>name</code> (sample name), <code>ocean</code> (oceanic 
region of the sample) and <code>depth</code> (sample depth).</p>
</dd>
</dl>
<h3>Source</h3>

<p>The raw data were downloaded from 
<a href="http://ocean-microbiome.embl.de/companion.html">http://ocean-microbiome.embl.de/companion.html</a>.
</p>


<h3>References</h3>

<p>Sunagawa S., Coelho L.P., Chaffron S., Kultima J.R., Labadie K., 
Salazar F., Djahanschiri B., Zeller G., Mende D.R., Alberti A., 
Cornejo-Castillo F., Costea P.I., Cruaud C., d'Oviedo F., Engelen S., 
Ferrera I., Gasol J., Guidi L., Hildebrand F., Kokoszka F., Lepoivre C., 
Lima-Mendez G., Poulain J., Poulos B., Royo-Llonch M., Sarmento H., 
Vieira-Silva S., Dimier C., Picheral M., Searson S., Kandels-Lewis S., 
<em>Tara</em> Oceans coordinators, Bowler C., de Vargas C., Gorsky G., 
Grimsley N., Hingamp P., Iudicone D., Jaillon O., Not F., Ogata H., Pesant 
S., Speich S., Stemmann L., Sullivan M., Weissenbach J., Wincker P., Karsenti 
E., Raes J., Acinas S. and Bork P. (2015). Structure and function of the 
global ocean microbiome. <em>Science</em>, <b>348</b>, 6237.
</p>


</div>