<div class="container">

<table style="width: 100%;"><tr>
<td>select.features</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Select important features</h2>

<h3>Description</h3>

<p>Select features using supervised or unsupervised kernel method. A 
supervised feature selection method is performed if <code>Y</code> is provided.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'features'
select(
  X,
  Y = NULL,
  kx.func = c("linear", "gaussian.radial.basis", "bray"),
  ky.func = c("linear", "gaussian.radial.basis"),
  keepX = NULL,
  method = c("kernel", "kpca", "graph"),
  lambda = NULL,
  n_components = 2,
  Lg = NULL,
  mu = 1,
  max_iter = 100,
  nstep = 50,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>a numeric matrix (or data frame) used to select variables. 
<code>NA</code>s not allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>a numeric matrix (or data frame) used to select variables. 
<code>NA</code>s not allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kx.func</code></td>
<td>
<p>the kernel function name to use on <code>X</code>. Widely used 
kernel functions are pre-implemented, and can be directly used by setting 
<code>kx.func</code> to one of the following values: <code>"linear"</code>, 
<code>"gaussian.radial.basis"</code> or <code>"bray"</code>. Default: <code>"linear"</code>. If 
<code>Y</code> is provided, the kernel <code>"bray"</code> is not allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ky.func</code></td>
<td>
<p>the kernel function name to use on <code>Y</code>. Available 
kernels are: <code>"linear"</code>, and <code>"gaussian.radial.basis"</code>. Default: 
<code>"linear"</code>. This value is ignored when <code>Y</code> is not provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keepX</code></td>
<td>
<p>the number of variables to select.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>the method to use. Either an unsupervised variable selection
method (<code>"kernel"</code>), a kernel PCA oriented variable selection method 
(<code>"kpca"</code>) or a structure driven variable selection selection 
(<code>"graph"</code>). Default: <code>"kernel"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambda</code></td>
<td>
<p>the penalization parameter that controls the trade-off between 
the minimization of the distorsion and the sparsity of the solution 
parameter.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_components</code></td>
<td>
<p>how many principal components should be used with method
<code>"kpca"</code>. Required with method <code>"kpca"</code>. Default: <code>2</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Lg</code></td>
<td>
<p>the Laplacian matrix of the graph representing relations between 
the input dataset variables. Required with method <code>"graph"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu</code></td>
<td>
<p>the penalization parameter that controls the trade-off between the
the distorsion and the influence of the graph. Default: <code>1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_iter</code></td>
<td>
<p>the maximum number of iterations. Default: <code>100</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nstep</code></td>
<td>
<p>the number of values used for the regularization path. Default: 
<code>50</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>the kernel function arguments. In particular, 
<code>sigma</code>(<code>"gaussian.radial.basis"</code>): double. The inverse kernel 
width used by <code>"gaussian.radial.basis"</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p><code>ukfs</code> returns a vector of sorted selected features indexes.
</p>


<h3>Author(s)</h3>

<p>Celine Brouard &lt;celine.brouard@inrae.fr&gt;
Jerome Mariette &lt;jerome.mariette@inrae.fr&gt;
Nathalie Vialaneix &lt;nathalie.vialaneix@inrae.fr&gt;
</p>


<h3>References</h3>

<p>Brouard C., Mariette J., Flamary R. and Vialaneix N. (2022). 
Feature selection for kernel methods in systems biology. <em>NAR Genomics
and Bioinformatics</em>, <b>4</b>(1), lqac014. DOI: <a href="https://doi.org/10.1093/nargab/lqac014">doi:10.1093/nargab/lqac014</a>.
</p>


<h3>See Also</h3>

<p><code>compute.kernel</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## These examples require the installation of python modules
## See installation instruction at: http://mixkernel.clementine.wf

data("Koren.16S")
## Not run: 
 sf.res &lt;- select.features(Koren.16S$data.raw, kx.func = "bray", lambda = 1,
                           keepX = 40, nstep = 1)
 colnames(Koren.16S$data.raw)[sf.res]

## End(Not run)

data("nutrimouse")
## Not run: 
 grb.func &lt;- "gaussian.radial.basis"
 genes &lt;- center.scale(nutrimouse$gene)
 lipids &lt;- center.scale(nutrimouse$lipid)
 sf.res &lt;- select.features(genes, lipids, kx.func = grb.func, 
                           ky.func = grb.func, keepX = 40)
 colnames(nutrimouse$gene)[sf.res]

## End(Not run)

</code></pre>


</div>