<div class="container">

<table style="width: 100%;"><tr>
<td>get_codons</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get Codons From Gene Sequence</h2>

<h3>Description</h3>

<p>Group a gene sequence into codons (triplets) that can be used for subsequent translation. If any elements of the character vector have length &gt;1 (insertions in ALT column of VCF), they are trimmed to the first base. Used by id.snps, id.indels, and create_ref functions.
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_codons(gene.seq, rev = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gene.seq</code></td>
<td>
<p>Character vector containing gene sequence. The length of this vector should be equal to the length of sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rev</code></td>
<td>
<p>Logical indicating whether the reverse complement of the gene.seq should be used.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A character vector containing translated amino acids associated with each nucleotide position in the input vector.
</p>


<h3>Examples</h3>

<pre><code class="language-R">get_codons(gene.seq = c("A","U","G","C","A","T","T","T","A","C","A","G","T","A","A"))
</code></pre>


</div>