<div class="container">

<table style="width: 100%;"><tr>
<td>id_snps</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>ID SNV-based Amino Acid Changes</h2>

<h3>Description</h3>

<p>Identify amino acid changes associated with single nucleotide variation. Changes associated with indels are identified in separate function. Used by call_mutations() function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">id_snps(variant.calls, ref, code.num = "1")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>variant.calls</code></td>
<td>
<p>Data frame with cols POS, REF, ALT, AF (alt freq), DP (total read depth).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref</code></td>
<td>
<p>reference genome in "MixVir" format (from create_ref() function)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>code.num</code></td>
<td>
<p>Number defining the genetic code to use for translation</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Data frame that includes amino acid calls based on SNP/SNV variants observed in sample. Contains cols "POS", "REF_BASE", "GENE", "REF_CODON", "REF_AA", "GENE_AA_POS", "REF_IDENT", "REF", "ALT", "ALT_freq", "ALT_COUNT", "samp_codon", "samp_AA", "samp_identity", "DP"
</p>


</div>