<div class="container">

<table style="width: 100%;"><tr>
<td>add_depths_to_ref</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Add Read Depths To Ref</h2>

<h3>Description</h3>

<p>Add column of total read depths for a given sample to reference df. Run as part of id_mutations().
</p>


<h3>Usage</h3>

<pre><code class="language-R">add_depths_to_ref(ref, samp.variants)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ref</code></td>
<td>
<p>reference genome in "MixVir" format (from create_ref() function)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>samp.variants</code></td>
<td>
<p>Data frame produced by function vcf_to_mixvir(). Contains columns "CHR", "POS", "REF", "ALT", "DP", "ALT_COUNT".</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Original 'ref' data frame with depth (DP) column added: cols "genomic_pos"	"ref_base"	"gene"	"ref_codon"	"ref_AA"	"GENE_AA_POS"	"ref_identity" "DP"
</p>


</div>