<div class="container">

<table style="width: 100%;"><tr>
<td>pairwise.mod.t.test</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Pairwise Moderated t-Tests</h2>

<h3>Description</h3>

<p>Performs pairwise moderated t-tests (unpaired) based on Bioconductor package 
limma.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pairwise.mod.t.test(x, group, adjust.method = "BH", sort.by = "none")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a (non-empty) numeric matrix of data values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group</code></td>
<td>
<p>an optional factor representing the groups.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adjust.method</code></td>
<td>
<p>see <code>p.adjust</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sort.by</code></td>
<td>
<p>see <code>toptable</code></p>
</td>
</tr>
</table>
<p>, where <code>"logFC"</code>
corresponds to difference in means.
</p>


<h3>Details</h3>

<p>The function uses Bioconductor package limma to compute pairwise moderated
t-tests. For more details we refer to <code>ebayes</code>.
</p>


<h3>Value</h3>

<p>A data.frame with the results.
</p>


<h3>References</h3>

<p>Ritchie, M.E., Phipson, B., Wu, D., Hu, Y., Law, C.W., Shi, W., and Smyth, G.K.
(2015). limma powers differential expression analyses for RNA-sequencing and
microarray studies. <em>Nucleic Acids Research</em> 43(7), e47.
</p>


<h3>See Also</h3>

<p><code>oneway.test</code>, <code>mod.t.test</code></p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(123)
X &lt;- rbind(matrix(rnorm(5*20), nrow = 5, ncol = 20),
           matrix(rnorm(5*20, mean = 1), nrow = 5, ncol = 20))
gr &lt;- factor(c(rep("A1", 5), rep("B2", 5), rep("C3", 5), rep("D4", 5)))
mod.oneway.test(X, gr)
pairwise.mod.t.test(X, gr)
</code></pre>


</div>