<div class="container">

<table style="width: 100%;"><tr>
<td>mod.t.test</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Moderated t-Test</h2>

<h3>Description</h3>

<p>Performs moderated t-tests based on Bioconductor package limma.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mod.t.test(x, group = NULL, paired = FALSE, subject, adjust.method = "BH",
           sort.by = "none", na.rm = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a (non-empty) numeric matrix of data values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group</code></td>
<td>
<p>an optional factor representing the groups.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>paired</code></td>
<td>
<p>a logical indicating whether you want a paired test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subject</code></td>
<td>
<p>factor with subject IDs; required if <code>paired = TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adjust.method</code></td>
<td>
<p>see <code>p.adjust</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sort.by</code></td>
<td>
<p>see <code>toptable</code></p>
</td>
</tr>
</table>
<p>, where <code>"logFC"</code>
corresponds to difference in means.
</p>
<table><tr style="vertical-align: top;">
<td><code>na.rm</code></td>
<td>
<p>logical. Should missing values (including NaN) be omitted from 
the calculations of group means?</p>
</td>
</tr></table>
<h3>Details</h3>

<p>The function uses Bioconductor package limma to compute moderated t-tests.
For more details we refer to <code>ebayes</code>.
</p>


<h3>Value</h3>

<p>A data.frame with the results.
</p>


<h3>References</h3>

<p>Ritchie, M.E., Phipson, B., Wu, D., Hu, Y., Law, C.W., Shi, W., and Smyth, G.K.
(2015). limma powers differential expression analyses for RNA-sequencing and
microarray studies. <em>Nucleic Acids Research</em> 43(7), e47.
</p>


<h3>See Also</h3>

<p><code>t.test</code></p>


<h3>Examples</h3>

<pre><code class="language-R">## One-sample test
X &lt;- matrix(rnorm(10*20, mean = 1), nrow = 10, ncol = 20)

mod.t.test(X)
## corresponds to
library(limma)
design &lt;- matrix(1, nrow = ncol(X), ncol = 1)
colnames(design) &lt;- "A"
fit1 &lt;- lmFit(X, design)
fit2 &lt;- eBayes(fit1)
topTable(fit2, coef = 1, number = Inf, confint = TRUE, sort.by = "none")[,-4]

## Two-sample test
set.seed(123)
X &lt;- rbind(matrix(rnorm(5*20), nrow = 5, ncol = 20),
           matrix(rnorm(5*20, mean = 1), nrow = 5, ncol = 20))
g2 &lt;- factor(c(rep("group 1", 10), rep("group 2", 10)))

mod.t.test(X, group = g2)
## corresponds to
design &lt;- model.matrix(~ 0 + g2)
colnames(design) &lt;- c("group1", "group2")
fit1 &lt;- lmFit(X, design)
cont.matrix &lt;- makeContrasts(group1vsgroup2="group1-group2", levels=design)
fit2 &lt;- contrasts.fit(fit1, cont.matrix)
fit3 &lt;- eBayes(fit2)
topTable(fit3, coef = 1, number = Inf, confint = TRUE, sort.by = "none")[,-4]

## Paired two-sample test
subjID &lt;- factor(rep(1:10, 2))
mod.t.test(X, group = g2, paired = TRUE, subject = subjID)
</code></pre>


</div>