<div class="container">

<table style="width: 100%;"><tr>
<td>DNA</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Primate splice-junction gene sequences (DNA)</h2>

<h3>Description</h3>

<p>It consists of 3,186 data points (splice junctions). The
data points are described by 180 indicator binary
variables and the problem is to recognize the 3 classes (ei, ie,
neither), i.e., the boundaries between exons (the parts of the DNA
sequence retained after splicing) and introns (the parts of the DNA
sequence that are spliced out).
</p>
<p>The StaLog dna dataset is a processed version of the Irvine 
database described below. The main difference is that the 
symbolic variables representing the nucleotides (only A,G,T,C) 
were replaced by 3 binary indicator variables. Thus the original 
60 symbolic attributes were changed into 180 binary attributes.  
The names of the examples were removed. The examples with 
ambiguities were removed (there was very few of them, 4).   
The StatLog version of this dataset was produced by Ross King
at Strathclyde University. For original details see the Irvine 
database documentation.
</p>
<p>The nucleotides A,C,G,T were given indicator values as follows:
</p>

<table>
<tr>
<td style="text-align: center;">
    	</td>
<td style="text-align: left;"> A -&gt; 1 0 0</td>
</tr>
<tr>
<td style="text-align: center;">
    	</td>
<td style="text-align: left;"> C -&gt; 0 1 0</td>
</tr>
<tr>
<td style="text-align: center;">
    	</td>
<td style="text-align: left;"> G -&gt; 0 0 1</td>
</tr>
<tr>
<td style="text-align: center;">
    	</td>
<td style="text-align: left;"> T -&gt; 0 0 0</td>
</tr>
<tr>
<td style="text-align: center;">
    </td>
</tr>
</table>
<p>Hint. Much better performance is generally observed if attributes
closest to the junction are used. In the StatLog version, this
means using attributes A61 to A120 only.   
</p>


<h3>Usage</h3>

<pre><code class="language-R">data("DNA", package = "mlbench")
</code></pre>


<h3>Format</h3>

<p>A data frame with 3,186 observations on 180 variables, all
nominal and a target class.</p>


<h3>Source</h3>


<ul>
<li>
<p> Source:<br>
- all examples taken from Genbank 64.1 (ftp site:
genbank.bio.net)<br>
- categories "ei" and "ie" include every "split-gene" 
for primates in Genbank 64.1<br>
- non-splice examples taken from sequences known not to include
a splicing site<br></p>
</li>
<li>
<p> Donor: G. Towell, M. Noordewier, and J. Shavlik, 
towell,shavlik@cs.wisc.edu, noordewi@cs.rutgers.edu
</p>
</li>
</ul>
<p>These data have been taken from: 
</p>

<ul><li>
<p> ftp.stams.strath.ac.uk/pub/Statlog
</p>
</li></ul>
<p>and were converted to R format by Evgenia Dimitriadou.
</p>


<h3>References</h3>

<p>machine learning:<br>
– M. O. Noordewier and G. G. Towell and J. W. Shavlik, 1991; 
"Training Knowledge-Based Neural Networks to Recognize Genes in 
DNA Sequences".  Advances in Neural Information Processing Systems,
volume 3, Morgan Kaufmann.
</p>
<p>– G. G. Towell and J. W. Shavlik and M. W. Craven, 1991;  
"Constructive Induction in Knowledge-Based Neural Networks",  
In Proceedings of the Eighth International Machine Learning
Workshop, Morgan Kaufmann.
</p>
<p>– G. G. Towell, 1991;
"Symbolic Knowledge and Neural Networks: Insertion, Refinement, and
Extraction", PhD Thesis, University of Wisconsin - Madison.
</p>
<p>– G. G. Towell and J. W. Shavlik, 1992;
"Interpretation of Artificial Neural Networks: Mapping 
Knowledge-based Neural Networks into Rules", In Advances in Neural
Information Processing Systems, volume 4, Morgan Kaufmann.  
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("DNA", package = "mlbench")
summary(DNA)
</code></pre>


</div>