<div class="container">

<table style="width: 100%;"><tr>
<td>threshold_allmodels</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Select significant marker thanks to a p-value threshold</h2>

<h3>Description</h3>

<p>Select significant marker at each mlmm step according to a threshold.
</p>


<h3>Usage</h3>

<pre><code class="language-R">threshold_allmodels(threshold=NULL, res_mlmm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>a value to declare the significant p value. The default value is Bonferroni 0.05</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>res_mlmm</code></td>
<td>
<p>a list of p-value for each mlmm step. Use helper function <code>mlmm_allmodels</code> to get this argument.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A matrix with a line for significant SNP at each mlmm step (according to the defined threshold) and 3 columns : SNP, p-value, step
</p>


<h3>Examples</h3>

<pre><code class="language-R">### Additive model ###
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa)
KK = list(K.add)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model and Marker selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))
res.threshold &lt;- threshold_allmodels(threshold, res_mlmm)

# Effects estimations with the selected model
sel_XXclass &lt;- fromeBICtoEstimation(sel_XX, res.eBIC, res.threshold)
eff.estimations &lt;- Estimation_allmodels(floweringDateAD, sel_XXclass, KK)
genotypes.boxplot(Xa, floweringDateAD, effects = eff.estimations)

## End(Not run)

### Additive + dominance model
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa, Xd)
KK = list(K.add, K.dom)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Marker selection
res.threshold &lt;- threshold_allmodels(threshold, res_mlmm)

## End(Not run)


### Female+Male model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm)
KK = list(K.female, K.male)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Marker selection
res.threshold &lt;- threshold_allmodels(threshold, res_mlmm)

## End(Not run)



### Female+Male+Interaction model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm, Xfm)
KK = list(K.female, K.male, K.hybrid)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Marker selection
res.threshold &lt;- threshold_allmodels(threshold, res_mlmm)

## End(Not run)
</code></pre>


</div>