<div class="container">

<table style="width: 100%;"><tr>
<td>fromeBICtoEstimation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Helper function that create the selec_XXclass argument of the Estimation_allmodels function</h2>

<h3>Description</h3>

<p>Function that create the selec_XXclass argument of the <code>Estimation_allmodels</code> function from the output of the <code>eBIC_allmodels</code> function or from the output of the <code>threshold_allmodels</code> function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">fromeBICtoEstimation(XX, res.eBIC, res.threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>XX</code></td>
<td>
<p>A list of length one, two or three matrices depending on the model. Matrices are n by m matrix, where n=number of individuals, m=number of SNPs, with rownames(X)=individual names, and colnames(X)=SNP names.
</p>
<p>- additive: a single matrix
</p>
<p>- additive+dominance: two matrices
</p>
<p>- female+male: two matrices with the female one first
</p>
<p>- female+male+interaction: three matrices with the female first, the male then the interaction</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>res.eBIC</code></td>
<td>
<p>output of the <code>eBIC_allmodels</code> function</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>res.threshold</code></td>
<td>
<p>output of the <code>threshold_allmodels</code> function</p>
</td>
</tr>
</table>
<h3>See Also</h3>

<p><code>eBIC_allmodels</code> <code>Estimation_allmodels</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">### Additive model ###
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa)
KK = list(K.add)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))

# Effects estimations with the selected model
sel_XXclass &lt;- fromeBICtoEstimation(sel_XX, res.eBIC)
eff.estimations &lt;- Estimation_allmodels(floweringDateAD, sel_XXclass, KK)
genotypes.boxplot(Xa, floweringDateAD, effects = eff.estimations)

## End(Not run)

### Additive + dominance model
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa, Xd)
KK = list(K.add, K.dom)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))
#the selected model is the null model

## End(Not run)


### Female+Male model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm)
KK = list(K.female, K.male)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

## End(Not run)



### Female+Male+Interaction model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm, Xfm)
KK = list(K.female, K.male, K.hybrid)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

## End(Not run)
</code></pre>


</div>