<div class="container">

<table style="width: 100%;"><tr>
<td>mlmm.gwas</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>mlmm.gwas</h2>

<h3>Description</h3>

<p>Pipeline for GWAS using Multi Locus Mixed Model (MLMM).
</p>


<h3>Details</h3>

<p>This is a fork of the MLMM / <a href="https://github.com/Gregor-Mendel-Institute/MultLocMixMod">MultLocMixMod</a> package by Vincent Segura and Bjarni J. Vilhjalmsson.
</p>
<p>The main differencies from the original package are:
</p>

<ul>
<li>
<p> abandon of the multi-Bonferroni model selection
</p>
</li>
<li>
<p> abandon of the backward model search
</p>
</li>
<li>
<p> eBIC modified to be adapted to the rate between number of individuals and number of markers.
</p>
</li>
<li>
<p> function added to select significant SNP according to a p-value threshold at each mlmm step
</p>
</li>
<li>
<p> new models supported: additive+dominance, male+female and male+female+interaction. These models are described in <a href="https://doi.org/10.1007/s00122-017-3003-4">Bonnafous et al. (2017)</a>.
</p>
</li>
<li>
<p> graphical functions: a new Manhattan plot and a boxplot representation of markers effects.
</p>
</li>
</ul>
<p>A <a href="../doc/gwas-manual.html">vignette</a> presents the usage of this package with the additive model.
</p>


<h3>References</h3>

<p><a href="https://doi.org/10.1007/s00122-017-3003-4">Bonnafous, F., Fievet, G., Blanchet, N. et al. Theor Appl Genet (2018) 131: 319.</a>
</p>
<p><a href="https://doi.org/10.1038/ng.2314">Segura V, Vilhjálmsson BJ, Platt A, Korte A, Seren Uuml, Long Q, et al. An efficient multi-locus mixed-model approach for genome-wide association studies in structured populations. Nat Genet. 2012;44:825–830.</a>
</p>


</div>