<div class="container">

<table style="width: 100%;"><tr>
<td>manhattan.plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Manhattan plot</h2>

<h3>Description</h3>

<p>Draw a Manhattan plot of the association p-values of the markers.
</p>


<h3>Usage</h3>

<pre><code class="language-R">manhattan.plot(res.mlmm, map = NULL, steps = 1, hideCofactors = FALSE,
  chrToPlot = "all", unit = "cM", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>res.mlmm</code></td>
<td>
<p>Output object from <code>mlmm_allmodels</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>Dataframe with 3 columns : markers names, chromosome or scaffold names and position (any unit is allowed: cM, Mpb etc.).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>steps</code></td>
<td>
<p>An integer. The iteration number of the forward approach. If a vector of length &gt;= 2 is passed, several plots will be drawn. By default, only step 1 is drawn.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hideCofactors</code></td>
<td>
<p>If TRUE, the coffactors (fixed effects) won't be drawn</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrToPlot</code></td>
<td>
<p>Names of the chromosomes or scaffolds to plot. Use this if you want to zoom on a particular chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>unit</code></td>
<td>
<p>Unit of the positions in the map.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional arguments can be passed to the plot function.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Draws a manhattan plot ie. plot -log(p-value) vs marker position
</p>
<p>If a map is passed, markers position will be used as x axis.
If not, the indices of markers inside the res.mlmm object will be used instead
</p>
<p>If there are cofactors (as in all but the first step of the forward approch),
the cofactors markers will be plotted too (symbol: star).
</p>
<p>If a map is passed, markers not in the map or in the map but not assigned to a chromosome
will be assigned to a virtual chromosome 0.
</p>
<p>Markers in the map, assigned to a chromosome, but with missing position, will be
ploted at the end of the chromosome.
</p>


<h3>See Also</h3>

<p><code>mlmm_allmodels</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">### Additive model ###
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa)
KK = list(K.add)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))

# Effects estimations with the selected model
sel_XXclass &lt;- fromeBICtoEstimation(sel_XX, res.eBIC)
eff.estimations &lt;- Estimation_allmodels(floweringDateAD, sel_XXclass, KK)
genotypes.boxplot(Xa, floweringDateAD, effects = eff.estimations)

## End(Not run)

### Additive + dominance model
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa, Xd)
KK = list(K.add, K.dom)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))
#the selected model is the null model

## End(Not run)


### Female+Male model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm)
KK = list(K.female, K.male)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

## End(Not run)



### Female+Male+Interaction model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm, Xfm)
KK = list(K.female, K.male, K.hybrid)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

## End(Not run)
</code></pre>


</div>