<div class="container">

<table style="width: 100%;"><tr>
<td>mlmm_allmodels</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Multi-Locus Mixed-Model</h2>

<h3>Description</h3>

<p>Carry GWAS correcting for population structure while including cofactors through a forward regression approach.
</p>
<p>possible models: additive, additive+dominance, female+male, female+male+interaction
</p>
<p>For additive model, look at the example below or at this <a href="../doc/gwas-manual.html">vignette</a>. For other models, read <a href="https://link.springer.com/article/10.1007/s00122-017-3003-4">Bonnafous et al. (2017)</a>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mlmm_allmodels(Y, XX, KK, nbchunks = 2, maxsteps = 20, cofs = NULL,
  female = NULL, male = NULL, threshold=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>A numeric named vector where the names are individuals' names and the values their phenotype. The names of Y will be matched to the row names of X.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>XX</code></td>
<td>
<p>A list of length one, two or three matrices depending on the model. Matrices are n by m matrix, where n=number of individuals, m=number of SNPs, with rownames(X)=individual names, and colnames(X)=SNP names.
</p>
<p>- additive: a single matrix
</p>
<p>- additive+dominance: two matrices
</p>
<p>- female+male: two matrices with the female one first
</p>
<p>- female+male+interaction: three matrices with the female first, the male then the interaction</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>KK</code></td>
<td>
<p>a list of one, two or three matrices depending on the models
</p>
<p>- additive: a n by n matrix, where n=number of individuals, with rownames()=colnames()=individual names
</p>
<p>- additive+dominance: two n by n matrices, where n=number of individuals, with rownames()=colnames()=individual names
</p>
<p>- female+male: a n.female by n.female matrix, with rownames()=colnames()=female names and a n.male by n.male matrix, with rownames()=colnames()=male names
</p>
<p>- female+male+interaction: the same two matrices as the model female+male and a n by n matrix, where n=number of individuals, with rownames()=colnames()=individual names</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nbchunks</code></td>
<td>
<p>An integer defining the number of chunks of matrices to run the analysis, allows to decrease the memory usage. minimum=2, increase it if you do not have enough memory</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxsteps</code></td>
<td>
<p>An integer &gt;= 3. Maximum number of steps desired in the forward approach. The forward approach breaks automatically once the pseudo-heritability is close to 0, however to avoid doing too many steps in case the pseudo-heritability does not reach a value close to 0, this parameter is also used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cofs</code></td>
<td>
<p>A n by q matrix, where n=number of individuals, q=number of fixed effect, with rownames()=individual names and with column names, forbidden head of column names for this matrix "eff1_" and usage of special characters as "*","/","&amp;"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>female</code></td>
<td>
<p>A factor of levels female names and length n, only for the last two models</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>male</code></td>
<td>
<p>A factor of levels male names and length n, only for the last two models</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>a value to declare the significant p value. The default value is Bonferroni 0.05</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Each of the data arguments must be sorted in the same way, according to the individual name.
</p>


<h3>Value</h3>

<p>a list with one element per step of the forward approach. Each element of this list is a named vector of p-values, the names are the names of the markers, with "selec_" as prefix for the markers used as fixed effects.
</p>


<h3>See Also</h3>

<p><code>manhattan.plot</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">### Data for additive and additive+dominance models

data("mlmm.gwas.AD")

### Additive model ###

XX = list(Xa)
KK = list(K.add)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))

# Marker selection
res.threshold &lt;- threshold_allmodels(threshold=NULL, res_mlmm)

# Effects estimations with the selected model
sel_XXclass &lt;- fromeBICtoEstimation(sel_XX, res.eBIC, res.threshold)
eff.estimations &lt;- Estimation_allmodels(floweringDateAD, sel_XXclass, KK)
genotypes.boxplot(Xa, floweringDateAD, effects = eff.estimations)

## Not run: 
### Additive + dominance model

XX = list(Xa, Xd)
KK = list(K.add, K.dom)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))
#the selected model is the null model

# Marker selection 
res.threshold &lt;- threshold_allmodels(threshold=NULL, res_mlmm)

### Data for female+male and female+male+interaction

data("mlmm.gwas.FMI")

### Female+Male model

XX = list(Xf, Xm)
KK = list(K.female, K.male)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

# Marker selection 
res.threshold &lt;- threshold_allmodels(threshold=NULL, res_mlmm)


### Female+Male+Interaction model

XX = list(Xf, Xm, Xfm)
KK = list(K.female, K.male, K.hybrid)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

# Marker selection 
res.threshold &lt;- threshold_allmodels(threshold=NULL, res_mlmm)


## End(Not run)
</code></pre>


</div>