<div class="container">

<table style="width: 100%;"><tr>
<td>Estimation_allmodels</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute estimated effects</h2>

<h3>Description</h3>

<p>Estimate the effect of selected SNPs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Estimation_allmodels(Y, selec_XXclass, KK, cofs = NULL, female = NULL,
  male = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>A numeric named vector where the names are individuals names and the values their phenotype. The names of Y will be matched to the row names of X.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>selec_XXclass</code></td>
<td>
<p>A n by mk data.frame of factors
with rownames()=individual names, and colnames()=mk selected SNP names
additive+dominance: three levels factor
female+male+interaction: four levels factor
</p>
<p>Use function <code>fromeBICtoEstimation</code> to get this armument.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>KK</code></td>
<td>
<p>a list of one, two or three matrices depending on the models
</p>
<p>- additive: a n by n matrix, where n=number of individuals, with rownames()=colnames()=individual names
</p>
<p>- additive+dominance: two n by n matrices, where n=number of individuals, with rownames()=colnames()=individual names
</p>
<p>- female+male: a n.female by n.female matrix, with rownames()=colnames()=female names and a n.male by n.male matrix, with rownames()=colnames()=male names
</p>
<p>- female+male+interaction: the same two matrices as the model female+male and a n by n matrix, where n=number of individuals, with rownames()=colnames()=individual names</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cofs</code></td>
<td>
<p>A n by q matrix, where n=number of individuals, q=number of fixed effect, with rownames()=individual names and with column names, forbidden head of column names for this matrix "eff1_" and usage of special characters as "*","/","&amp;"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>female</code></td>
<td>
<p>A factor of levels female names and length n, only for the last two models</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>male</code></td>
<td>
<p>A factor of levels male names and length n, only for the last two models</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A dataframe with 3 colum: BLUE, Tukey.Class and Frequency. The firt line name is "mu", the names of the other lines are in the form markername_allele.
</p>


<h3>Examples</h3>

<pre><code class="language-R">### Additive model ###
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa)
KK = list(K.add)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))

# Effects estimations with the selected model
sel_XXclass &lt;- fromeBICtoEstimation(sel_XX, res.eBIC)
eff.estimations &lt;- Estimation_allmodels(floweringDateAD, sel_XXclass, KK)
genotypes.boxplot(Xa, floweringDateAD, effects = eff.estimations)

## End(Not run)

### Additive + dominance model
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa, Xd)
KK = list(K.add, K.dom)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))
#the selected model is the null model

## End(Not run)


### Female+Male model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm)
KK = list(K.female, K.male)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

## End(Not run)



### Female+Male+Interaction model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm, Xfm)
KK = list(K.female, K.male, K.hybrid)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

## End(Not run)
</code></pre>


</div>