<div class="container">

<table style="width: 100%;"><tr>
<td>creating.diploid</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generation of the starting population</h2>

<h3>Description</h3>

<p>Generation of the starting population
</p>


<h3>Usage</h3>

<pre><code class="language-R">creating.diploid(
  dataset = NULL,
  vcf = NULL,
  chr.nr = NULL,
  bp = NULL,
  snp.name = NULL,
  hom0 = NULL,
  hom1 = NULL,
  bpcm.conversion = 0,
  nsnp = 0,
  nindi = 0,
  freq = "beta",
  population = NULL,
  sex.s = "fixed",
  add.chromosome = FALSE,
  generation = 1,
  class = 0L,
  sex.quota = 0.5,
  chromosome.length = NULL,
  length.before = 5,
  length.behind = 5,
  real.bv.add = NULL,
  real.bv.mult = NULL,
  real.bv.dice = NULL,
  snps.equidistant = NULL,
  change.order = FALSE,
  bv.total = 0,
  polygenic.variance = 100,
  bve.mult.factor = NULL,
  bve.poly.factor = NULL,
  base.bv = NULL,
  add.chromosome.ends = TRUE,
  new.phenotype.correlation = NULL,
  new.residual.correlation = NULL,
  new.breeding.correlation = NULL,
  add.architecture = NULL,
  snp.position = NULL,
  position.scaling = FALSE,
  bit.storing = FALSE,
  nbits = 30,
  randomSeed = NULL,
  miraculix = TRUE,
  miraculix.dataset = TRUE,
  n.additive = 0,
  n.equal.additive = 0,
  n.dominant = 0,
  n.equal.dominant = 0,
  n.qualitative = 0,
  n.quantitative = 0,
  dominant.only.positive = FALSE,
  var.additive.l = NULL,
  var.dominant.l = NULL,
  var.qualitative.l = NULL,
  var.quantitative.l = NULL,
  effect.size.equal.add = 1,
  effect.size.equal.dom = 1,
  exclude.snps = NULL,
  replace.real.bv = FALSE,
  shuffle.traits = NULL,
  shuffle.cor = NULL,
  skip.rest = FALSE,
  enter.bv = TRUE,
  name.cohort = NULL,
  template.chip = NULL,
  beta.shape1 = 1,
  beta.shape2 = 1,
  time.point = 0,
  creating.type = 0,
  trait.name = NULL,
  share.genotyped = 1,
  genotyped.s = NULL,
  map = NULL,
  remove.invalid.qtl = TRUE,
  verbose = TRUE,
  bv.standard = FALSE,
  mean.target = NULL,
  var.target = NULL,
  is.maternal = NULL,
  is.paternal = NULL,
  vcf.maxsnp = Inf,
  internal = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dataset</code></td>
<td>
<p>SNP dataset, use "random", "allhetero" "all0" when generating a dataset via nsnp,nindi</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vcf</code></td>
<td>
<p>Path to a vcf-file used as input genotypes (correct haplotype phase is assumed!)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr.nr</code></td>
<td>
<p>Vector containing the assosiated chromosome for each marker (default: all on the same)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bp</code></td>
<td>
<p>Vector containing the physical position (bp) for each marker (default: 1,2,3...)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.name</code></td>
<td>
<p>Vector containing the name of each marker (default ChrXSNPY - XY chosen accordingly)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hom0</code></td>
<td>
<p>Vector containing the first allelic variant in each marker (default: 0)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hom1</code></td>
<td>
<p>Vector containing the second allelic variant in each marker (default: 1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bpcm.conversion</code></td>
<td>
<p>Convert physical position (bp) into a cM position (default: 0 - not done)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsnp</code></td>
<td>
<p>number of markers to generate in a random dataset</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nindi</code></td>
<td>
<p>number of inidividuals to generate in a random dataset</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freq</code></td>
<td>
<p>frequency of allele 1 when randomly generating a dataset</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>population</code></td>
<td>
<p>Population list</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sex.s</code></td>
<td>
<p>Specify which newly added individuals are male (1) or female (2)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add.chromosome</code></td>
<td>
<p>If TRUE add an additional chromosome to the dataset</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>generation</code></td>
<td>
<p>Generation of the newly added individuals (default: 1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>class</code></td>
<td>
<p>Migration level of the newly added individuals</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sex.quota</code></td>
<td>
<p>Share of newly added female individuals (deterministic if sex.s="fixed", alt: sex.s="random")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosome.length</code></td>
<td>
<p>Length of the newly added chromosome (default: 5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>length.before</code></td>
<td>
<p>Length before the first SNP of the dataset (default: 5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>length.behind</code></td>
<td>
<p>Length after the last SNP of the dataset (default: 5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>real.bv.add</code></td>
<td>
<p>Single Marker effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>real.bv.mult</code></td>
<td>
<p>Two Marker effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>real.bv.dice</code></td>
<td>
<p>Multi-marker effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snps.equidistant</code></td>
<td>
<p>Use equidistant markers (computationally faster! ; default: TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>change.order</code></td>
<td>
<p>If TRUE sort markers according to given marker positions</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bv.total</code></td>
<td>
<p>Number of traits (If more than traits via real.bv.X use traits with no directly underlying QTL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>polygenic.variance</code></td>
<td>
<p>Genetic variance of traits with no underlying QTL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bve.mult.factor</code></td>
<td>
<p>Multiplicate trait value times this</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bve.poly.factor</code></td>
<td>
<p>Potency trait value over this</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>base.bv</code></td>
<td>
<p>Average genetic value of a trait</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add.chromosome.ends</code></td>
<td>
<p>Add chromosome ends as recombination points</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>new.phenotype.correlation</code></td>
<td>
<p>(OLD! - use new.residual.correlation) Correlation of the simulated enviromental variance</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>new.residual.correlation</code></td>
<td>
<p>Correlation of the simulated enviromental variance</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>new.breeding.correlation</code></td>
<td>
<p>Correlation of the simulated genetic variance (child share! heritage is not influenced!</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add.architecture</code></td>
<td>
<p>Add genetic architecture (marker positions)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.position</code></td>
<td>
<p>Location of each marker on the genetic map</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>position.scaling</code></td>
<td>
<p>Manual scaling of snp.position</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bit.storing</code></td>
<td>
<p>Set to TRUE if the MoBPS (not-miraculix! bit-storing is used)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nbits</code></td>
<td>
<p>Bits available in MoBPS-bit-storing</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>randomSeed</code></td>
<td>
<p>Set random seed of the process</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>miraculix</code></td>
<td>
<p>If TRUE use miraculix package for data storage, computations and dataset generation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>miraculix.dataset</code></td>
<td>
<p>Set FALSE to deactive miraculix package for dataset generation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.additive</code></td>
<td>
<p>Number of additive QTL with effect size drawn from a gaussian distribution</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.equal.additive</code></td>
<td>
<p>Number of additive QTL with equal effect size (effect.size)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.dominant</code></td>
<td>
<p>Number of dominant QTL with effect size drawn from a gaussian distribution</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.equal.dominant</code></td>
<td>
<p>Number of n.equal.dominant QTL with equal effect size</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.qualitative</code></td>
<td>
<p>Number of qualitative epistatic QTL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.quantitative</code></td>
<td>
<p>Number of quantitative epistatic QTL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dominant.only.positive</code></td>
<td>
<p>Set to TRUE to always asign the heterozygous variant with the higher of the two homozygous effects (e.g. hybrid breeding); default: FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>var.additive.l</code></td>
<td>
<p>Variance of additive QTL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>var.dominant.l</code></td>
<td>
<p>Variance of dominante QTL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>var.qualitative.l</code></td>
<td>
<p>Variance of qualitative epistatic QTL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>var.quantitative.l</code></td>
<td>
<p>Variance of quantitative epistatic QTL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>effect.size.equal.add</code></td>
<td>
<p>Effect size of the QTLs in n.equal.additive</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>effect.size.equal.dom</code></td>
<td>
<p>Effect size of the QTLs in n.equal.dominant</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exclude.snps</code></td>
<td>
<p>Marker were no QTL are simulated on</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replace.real.bv</code></td>
<td>
<p>If TRUE delete the simulated traits added before</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shuffle.traits</code></td>
<td>
<p>Combine different traits into a joined trait</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shuffle.cor</code></td>
<td>
<p>Target Correlation between shuffeled traits</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>skip.rest</code></td>
<td>
<p>Internal variable needed when adding multipe chromosomes jointly</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>enter.bv</code></td>
<td>
<p>Internal parameter</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name.cohort</code></td>
<td>
<p>Name of the newly added cohort</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>template.chip</code></td>
<td>
<p>Import genetic map and chip from a species ("cattle", "chicken", "pig")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta.shape1</code></td>
<td>
<p>First parameter of the beta distribution for simulating allele frequencies</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta.shape2</code></td>
<td>
<p>Second parameter of the beta distribution for simulating allele frequencies</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>time.point</code></td>
<td>
<p>Time point at which the new individuals are generated</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>creating.type</code></td>
<td>
<p>Technique to generate new individuals (usage in web-based application)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trait.name</code></td>
<td>
<p>Name of the trait generated</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>share.genotyped</code></td>
<td>
<p>Share of individuals genotyped in the founders</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotyped.s</code></td>
<td>
<p>Specify with newly added individuals are genotyped (1) or not (0)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>map-file that contains up to 5 colums (Chromsome, SNP-id, M-position, Bp-position, allele freq - Everything not provides it set to NA). A map can be imported via MoBPSmaps::ensembl.map()</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove.invalid.qtl</code></td>
<td>
<p>Set to FALSE to deactive the automatic removal of QTLs on markers that do not exist</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Set to FALSE to not display any prints</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bv.standard</code></td>
<td>
<p>Set TRUE to standardize trait mean and variance via bv.standardization() - automatically set to TRUE when mean/var.target are used</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mean.target</code></td>
<td>
<p>Target mean</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>var.target</code></td>
<td>
<p>Target variance</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is.maternal</code></td>
<td>
<p>Vector coding if a trait is caused by a maternal effect (Default: all FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is.paternal</code></td>
<td>
<p>Vector coding if a trait is caused by a paternal effect (Default: all FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vcf.maxsnp</code></td>
<td>
<p>Maximum number of SNPs to include in the genotype file (default: Inf)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>internal</code></td>
<td>
<p>Dont touch!</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Population-list
</p>


<h3>Examples</h3>

<pre><code class="language-R">population &lt;- creating.diploid(nsnp=1000, nindi=100)
</code></pre>


</div>