<div class="container">

<table style="width: 100%;"><tr>
<td>get.geno</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Derive genotypes of selected individuals</h2>

<h3>Description</h3>

<p>Function to devide genotypes of selected individuals
</p>


<h3>Usage</h3>

<pre><code class="language-R">get.geno(
  population,
  database = NULL,
  gen = NULL,
  cohorts = NULL,
  chromosomen = "all",
  export.alleles = FALSE,
  non.genotyped.as.missing = FALSE,
  use.id = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>population</code></td>
<td>
<p>Population list</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>database</code></td>
<td>
<p>Groups of individuals to consider for the export</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gen</code></td>
<td>
<p>Quick-insert for database (vector of all generations to export)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cohorts</code></td>
<td>
<p>Quick-insert for database (vector of names of cohorts to export)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosomen</code></td>
<td>
<p>Beschraenkung des Genotypen auf bestimmte Chromosomen (default: 1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>export.alleles</code></td>
<td>
<p>If TRUE export underlying alleles instead of just 012</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>non.genotyped.as.missing</code></td>
<td>
<p>Set to TRUE to replace non-genotyped markers with NA</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use.id</code></td>
<td>
<p>Set to TRUE to use MoBPS ids instead of Sex_Nr_Gen based names (default: FALSE)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Genotype data for in gen/database/cohorts selected individuals
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(ex_pop)
geno &lt;- get.geno(ex_pop, gen=2)
</code></pre>


</div>