<div class="container">

<table style="width: 100%;"><tr>
<td>effect.estimate.add</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimation of marker effects</h2>

<h3>Description</h3>

<p>Function to estimate marker effects
</p>


<h3>Usage</h3>

<pre><code class="language-R">effect.estimate.add(geno, pheno, map = NULL, scaling = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>genotype dataset (marker x individuals)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p>phenotype dataset (each phenotype in a row)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>genomic map</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scaling</code></td>
<td>
<p>Set FALSE to not perform variance scaling</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Empirical kinship matrix (IBD-based since Founders)
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(ex_pop)
pheno &lt;- get.pheno(ex_pop, gen=1:5)
geno &lt;- get.geno(ex_pop, gen=1:5)
map &lt;- get.map(ex_pop, use.snp.nr=TRUE)
real.bv.add &lt;- effect.estimate.add(geno, pheno, map)
</code></pre>


</div>