<div class="container">

<table style="width: 100%;"><tr>
<td>ld.decay</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate LD plot</h2>

<h3>Description</h3>

<p>Generate LD pot
</p>


<h3>Usage</h3>

<pre><code class="language-R">ld.decay(
  population,
  genotype.dataset = NULL,
  chromosomen = 1,
  dist = NULL,
  step = 5,
  max = 500,
  max.cases = 100,
  database = NULL,
  gen = NULL,
  cohorts = NULL,
  type = "snp",
  plot = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>population</code></td>
<td>
<p>Population list</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotype.dataset</code></td>
<td>
<p>Genotype dataset (default: NULL - just to save computation time when get.geno was already run)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosomen</code></td>
<td>
<p>Only consider a specific chromosome in calculations (default: 1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dist</code></td>
<td>
<p>Manuel input of marker distances to analyse</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>step</code></td>
<td>
<p>Stepsize to calculate LD</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max</code></td>
<td>
<p>Maximum distance between markers to consider for LD-plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.cases</code></td>
<td>
<p>Maximum number of marker pairs to consider of each distance (default: 100; randomly sampled!)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>database</code></td>
<td>
<p>Groups of individuals to consider for the export</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gen</code></td>
<td>
<p>Quick-insert for database (vector of all generations to export)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cohorts</code></td>
<td>
<p>Quick-insert for database (vector of names of cohorts to export)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Compute LD decay according to following distance measure between markers (default: "snp", alt: "bp", "cM")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>Set to FALSE to not generate an LD plot</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>LD-decay plot for in gen/database/cohorts selected individuals
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(ex_pop)
ld.decay(population=ex_pop, gen=5)
</code></pre>


</div>