<div class="container">

<table style="width: 100%;"><tr>
<td>creating.trait</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generation of genomic traits</h2>

<h3>Description</h3>

<p>Generation of the trait in a starting population
</p>


<h3>Usage</h3>

<pre><code class="language-R">creating.trait(
  population,
  real.bv.add = NULL,
  real.bv.mult = NULL,
  real.bv.dice = NULL,
  bv.total = 0,
  polygenic.variance = 100,
  bve.mult.factor = NULL,
  bve.poly.factor = NULL,
  base.bv = NULL,
  new.phenotype.correlation = NULL,
  new.residual.correlation = NULL,
  new.breeding.correlation = NULL,
  n.additive = 0,
  n.equal.additive = 0,
  n.dominant = 0,
  n.equal.dominant = 0,
  n.qualitative = 0,
  n.quantitative = 0,
  dominant.only.positive = FALSE,
  var.additive.l = NULL,
  var.dominant.l = NULL,
  var.qualitative.l = NULL,
  var.quantitative.l = NULL,
  effect.size.equal.add = 1,
  effect.size.equal.dom = 1,
  exclude.snps = NULL,
  randomSeed = NULL,
  shuffle.traits = NULL,
  shuffle.cor = NULL,
  replace.traits = FALSE,
  trait.name = NULL,
  remove.invalid.qtl = TRUE,
  bv.standard = FALSE,
  mean.target = NULL,
  var.target = NULL,
  verbose = TRUE,
  is.maternal = NULL,
  is.paternal = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>population</code></td>
<td>
<p>Population list</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>real.bv.add</code></td>
<td>
<p>Single Marker effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>real.bv.mult</code></td>
<td>
<p>Two Marker effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>real.bv.dice</code></td>
<td>
<p>Multi-marker effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bv.total</code></td>
<td>
<p>Number of traits (If more than traits via real.bv.X use traits with no directly underlying QTL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>polygenic.variance</code></td>
<td>
<p>Genetic variance of traits with no underlying QTL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bve.mult.factor</code></td>
<td>
<p>Multiplicate trait value times this</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bve.poly.factor</code></td>
<td>
<p>Potency trait value over this</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>base.bv</code></td>
<td>
<p>Average genetic value of a trait</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>new.phenotype.correlation</code></td>
<td>
<p>(OLD! - use new.residual.correlation) Correlation of the simulated enviromental variance</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>new.residual.correlation</code></td>
<td>
<p>Correlation of the simulated enviromental variance</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>new.breeding.correlation</code></td>
<td>
<p>Correlation of the simulated genetic variance (child share! heritage is not influenced!</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.additive</code></td>
<td>
<p>Number of additive QTL with effect size drawn from a gaussian distribution</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.equal.additive</code></td>
<td>
<p>Number of additive QTL with equal effect size (effect.size)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.dominant</code></td>
<td>
<p>Number of dominant QTL with effect size drawn from a gaussian distribution</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.equal.dominant</code></td>
<td>
<p>Number of n.equal.dominant QTL with equal effect size</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.qualitative</code></td>
<td>
<p>Number of qualitative epistatic QTL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.quantitative</code></td>
<td>
<p>Number of quantitative epistatic QTL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dominant.only.positive</code></td>
<td>
<p>Set to TRUE to always asign the heterozygous variant with the higher of the two homozygous effects (e.g. hybrid breeding); default: FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>var.additive.l</code></td>
<td>
<p>Variance of additive QTL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>var.dominant.l</code></td>
<td>
<p>Variance of dominante QTL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>var.qualitative.l</code></td>
<td>
<p>Variance of qualitative epistatic QTL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>var.quantitative.l</code></td>
<td>
<p>Variance of quantitative epistatic QTL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>effect.size.equal.add</code></td>
<td>
<p>Effect size of the QTLs in n.equal.additive</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>effect.size.equal.dom</code></td>
<td>
<p>Effect size of the QTLs in n.equal.dominant</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exclude.snps</code></td>
<td>
<p>Marker were no QTL are simulated on</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>randomSeed</code></td>
<td>
<p>Set random seed of the process</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shuffle.traits</code></td>
<td>
<p>Combine different traits into a joined trait</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shuffle.cor</code></td>
<td>
<p>Target Correlation between shuffeled traits</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replace.traits</code></td>
<td>
<p>If TRUE delete the simulated traits added before</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trait.name</code></td>
<td>
<p>Name of the trait generated</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove.invalid.qtl</code></td>
<td>
<p>Set to FALSE to deactive the automatic removal of QTLs on markers that do not exist</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bv.standard</code></td>
<td>
<p>Set TRUE to standardize trait mean and variance via bv.standardization()</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mean.target</code></td>
<td>
<p>Target mean</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>var.target</code></td>
<td>
<p>Target variance</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Set to FALSE to not display any prints</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is.maternal</code></td>
<td>
<p>Vector coding if a trait is caused by a maternal effect (Default: all FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is.paternal</code></td>
<td>
<p>Vector coding if a trait is caused by a paternal effect (Default: all FALSE)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Population-list with one or more additional new traits
</p>


<h3>Examples</h3>

<pre><code class="language-R">population &lt;- creating.diploid(nsnp=1000, nindi=100)
population &lt;- creating.trait(population, n.additive=100)
</code></pre>


</div>