<div class="container">

<table style="width: 100%;"><tr>
<td>calculate.bv</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate breeding values</h2>

<h3>Description</h3>

<p>Internal function to calculate the breeding value of a given individual
</p>


<h3>Usage</h3>

<pre><code class="language-R">calculate.bv(
  population,
  gen,
  sex,
  nr,
  activ_bv,
  import.position.calculation = NULL,
  decodeOriginsU = decodeOriginsR,
  store.effect.freq = FALSE,
  bit.storing = FALSE,
  nbits = 30,
  output_compressed = FALSE,
  bv.ignore.traits = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>population</code></td>
<td>
<p>Population list</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gen</code></td>
<td>
<p>Generation of the individual of interest</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sex</code></td>
<td>
<p>Sex of the individual of interest</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nr</code></td>
<td>
<p>Number of the individual of interest</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>activ_bv</code></td>
<td>
<p>traits to consider</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>import.position.calculation</code></td>
<td>
<p>Function to calculate recombination point into adjacent/following SNP</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>decodeOriginsU</code></td>
<td>
<p>Used function for the decoding of genetic origins [[5]]/[[6]]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>store.effect.freq</code></td>
<td>
<p>If TRUE store the allele frequency of effect markers per generation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bit.storing</code></td>
<td>
<p>Set to TRUE if the MoBPS (not-miraculix! bit-storing is used)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nbits</code></td>
<td>
<p>Bits available in MoBPS-bit-storing</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output_compressed</code></td>
<td>
<p>Set to TRUE to get a miraculix-compressed genotype/haplotype</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bv.ignore.traits</code></td>
<td>
<p>Vector of traits to ignore in the calculation of the genomic value (default: NULL; Only recommended for high number of traits and experienced users!)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>[[1]] true genomic value [[2]] allele frequency at QTL markers
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(ex_pop)
calculate.bv(ex_pop, gen=1, sex=1, nr=1, activ_bv = 1)
</code></pre>


</div>