<div class="container">

<table style="width: 100%;"><tr>
<td>breeding.intern</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Internal function to simulate one meiosis</h2>

<h3>Description</h3>

<p>Internal function to simulate one meiosis
</p>


<h3>Usage</h3>

<pre><code class="language-R">breeding.intern(
  info.parent,
  parent,
  population,
  mutation.rate = 10^-5,
  remutation.rate = 10^-5,
  recombination.rate = 1,
  recom.f.indicator = NULL,
  duplication.rate = 0,
  duplication.length = 0.01,
  duplication.recombination = 1,
  delete.same.origin = FALSE,
  gene.editing = FALSE,
  nr.edits = 0,
  gen.architecture = 0,
  decodeOriginsU = MoBPS::decodeOriginsR
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>info.parent</code></td>
<td>
<p>position of the parent in the dataset</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parent</code></td>
<td>
<p>list of information regarding the parent</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>population</code></td>
<td>
<p>Population list</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutation.rate</code></td>
<td>
<p>Mutation rate in each marker (default: 10^-5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remutation.rate</code></td>
<td>
<p>Remutation rate in each marker (default: 10^-5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>recombination.rate</code></td>
<td>
<p>Average number of recombination per 1 length unit (default: 1M)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>recom.f.indicator</code></td>
<td>
<p>Use step function for recombination map (transform snp.positions if possible instead)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>duplication.rate</code></td>
<td>
<p>Share of recombination points with a duplication (default: 0 - DEACTIVATED)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>duplication.length</code></td>
<td>
<p>Average length of a duplication (Exponentially distributed)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>duplication.recombination</code></td>
<td>
<p>Average number of recombinations per 1 length uit of duplication (default: 1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>delete.same.origin</code></td>
<td>
<p>If TRUE delete recombination points when genetic origin of adjacent segments is the same</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.editing</code></td>
<td>
<p>If TRUE perform gene editing on newly generated individual</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nr.edits</code></td>
<td>
<p>Number of edits to perform per individual</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gen.architecture</code></td>
<td>
<p>Used underlying genetic architecture (genome length in M)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>decodeOriginsU</code></td>
<td>
<p>Used function for the decoding of genetic origins [[5]]/[[6]]</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Inherited parent gamete
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(ex_pop)
child_gamete &lt;- breeding.intern(info.parent = c(1,1,1), parent = ex_pop$breeding[[1]][[1]][[1]],
                                population = ex_pop)
</code></pre>


</div>