<div class="container">

<table style="width: 100%;"><tr>
<td>compute.snps_single</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute genotype/haplotype in gene editing application</h2>

<h3>Description</h3>

<p>Internal function for the computation of genotypes &amp; haplotypes in gene editing application
</p>


<h3>Usage</h3>

<pre><code class="language-R">compute.snps_single(
  population,
  current.recombi,
  current.mut,
  current.ursprung,
  faster = TRUE,
  import.position.calculation = NULL,
  decodeOriginsU = decodeOriginsR
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>population</code></td>
<td>
<p>Population list</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>current.recombi</code></td>
<td>
<p>vector of currently activ recombination points</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>current.mut</code></td>
<td>
<p>vector of currently activ mutations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>current.ursprung</code></td>
<td>
<p>vector of currently activ origins</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>faster</code></td>
<td>
<p>If FALSE use slower version to compute markers between recombination points</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>import.position.calculation</code></td>
<td>
<p>Function to calculate recombination point into adjacent/following SNP</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>decodeOriginsU</code></td>
<td>
<p>Used function for the decoding of genetic origins [[5]]/[[6]]</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>haplotypes for the selected individual
</p>


</div>