<div class="container">

<table style="width: 100%;"><tr>
<td>compute.snps</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute genotype/haplotype</h2>

<h3>Description</h3>

<p>Internal function for the computation of genotypes &amp; haplotypes
</p>


<h3>Usage</h3>

<pre><code class="language-R">compute.snps(
  population,
  gen,
  sex,
  nr,
  faster = TRUE,
  import.position.calculation = NULL,
  from_p = 1,
  to_p = Inf,
  decodeOriginsU = decodeOriginsR,
  bit.storing = FALSE,
  nbits = 30,
  output_compressed = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>population</code></td>
<td>
<p>Population list</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gen</code></td>
<td>
<p>Generation of the individual to compute</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sex</code></td>
<td>
<p>Gender of the individual to compute</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nr</code></td>
<td>
<p>Number of the individual to compute</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>faster</code></td>
<td>
<p>If FALSE use slower version to compute markers between recombination points</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>import.position.calculation</code></td>
<td>
<p>Function to calculate recombination point into adjacent/following SNP</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>from_p</code></td>
<td>
<p>First SNP to consider</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>to_p</code></td>
<td>
<p>Last SNP to consider</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>decodeOriginsU</code></td>
<td>
<p>Used function for the decoding of genetic origins [[5]]/[[6]]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bit.storing</code></td>
<td>
<p>Set to TRUE if the MoBPS (not-miraculix! bit-storing is used)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nbits</code></td>
<td>
<p>Bits available in MoBPS-bit-storing</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output_compressed</code></td>
<td>
<p>Set to TRUE to get a miraculix-compressed genotype/haplotype</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>haplotypes for the selected individual
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(ex_pop)
compute.snps(ex_pop, gen=1, sex=1, nr=1)
</code></pre>


</div>