<div class="container">

<table style="width: 100%;"><tr>
<td>founder.simulation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Founder simulation</h2>

<h3>Description</h3>

<p>Function to generate founder genotypes
</p>


<h3>Usage</h3>

<pre><code class="language-R">founder.simulation(
  nindi = 100,
  sex.quota = 0.5,
  nsnp = 0,
  n.gen = 100,
  nfinal = NULL,
  sex.quota.final = NULL,
  big.output = FALSE,
  plot = TRUE,
  display.progress = TRUE,
  depth.pedigree = 7,
  dataset = NULL,
  vcf = NULL,
  chr.nr = NULL,
  bp = NULL,
  snp.name = NULL,
  hom0 = NULL,
  hom1 = NULL,
  bpcm.conversion = 0,
  freq = "beta",
  sex.s = "fixed",
  chromosome.length = NULL,
  length.before = 5,
  length.behind = 5,
  snps.equidistant = NULL,
  change.order = FALSE,
  snp.position = NULL,
  position.scaling = FALSE,
  bit.storing = FALSE,
  nbits = 30,
  randomSeed = NULL,
  miraculix = TRUE,
  miraculix.dataset = TRUE,
  template.chip = NULL,
  beta.shape1 = 1,
  beta.shape2 = 1,
  map = NULL,
  verbose = TRUE,
  vcf.maxsnp = Inf
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nindi</code></td>
<td>
<p>number of inidividuals to generate in a random dataset</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sex.quota</code></td>
<td>
<p>Share of newly added female individuals (deterministic if sex.s="fixed", alt: sex.s="random")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsnp</code></td>
<td>
<p>number of markers to generate in a random dataset</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.gen</code></td>
<td>
<p>Number of generations to simulate (default: 100)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nfinal</code></td>
<td>
<p>Number of final individuals to include (default: nindi)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sex.quota.final</code></td>
<td>
<p>Share of female individuals in the final generation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>big.output</code></td>
<td>
<p>Set to TRUE to export map, population list and pedigree relationship</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>Set to FALSE to not generate LD-decay plot and allele frequency spectrum</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>display.progress</code></td>
<td>
<p>Set FALSE to not display progress bars. Setting verbose to FALSE will automatically deactive progress bars</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>depth.pedigree</code></td>
<td>
<p>Depth of the pedigree in generations (default: 7)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dataset</code></td>
<td>
<p>SNP dataset, use "random", "allhetero" "all0" when generating a dataset via nsnp,nindi</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vcf</code></td>
<td>
<p>Path to a vcf-file used as input genotypes (correct haplotype phase is assumed!)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr.nr</code></td>
<td>
<p>Vector containing the assosiated chromosome for each marker (default: all on the same)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bp</code></td>
<td>
<p>Vector containing the physical position (bp) for each marker (default: 1,2,3...)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.name</code></td>
<td>
<p>Vector containing the name of each marker (default ChrXSNPY - XY chosen accordingly)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hom0</code></td>
<td>
<p>Vector containing the first allelic variant in each marker (default: 0)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hom1</code></td>
<td>
<p>Vector containing the second allelic variant in each marker (default: 1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bpcm.conversion</code></td>
<td>
<p>Convert physical position (bp) into a cM position (default: 0 - not done)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freq</code></td>
<td>
<p>frequency of allele 1 when randomly generating a dataset</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sex.s</code></td>
<td>
<p>Specify which newly added individuals are male (1) or female (2)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosome.length</code></td>
<td>
<p>Length of the newly added chromosome (default: 5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>length.before</code></td>
<td>
<p>Length before the first SNP of the dataset (default: 5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>length.behind</code></td>
<td>
<p>Length after the last SNP of the dataset (default: 5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snps.equidistant</code></td>
<td>
<p>Use equidistant markers (computationally faster! ; default: TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>change.order</code></td>
<td>
<p>If TRUE sort markers according to given marker positions</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.position</code></td>
<td>
<p>Location of each marker on the genetic map</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>position.scaling</code></td>
<td>
<p>Manual scaling of snp.position</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bit.storing</code></td>
<td>
<p>Set to TRUE if the MoBPS (not-miraculix! bit-storing is used)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nbits</code></td>
<td>
<p>Bits available in MoBPS-bit-storing</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>randomSeed</code></td>
<td>
<p>Set random seed of the process</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>miraculix</code></td>
<td>
<p>If TRUE use miraculix package for data storage, computations and dataset generation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>miraculix.dataset</code></td>
<td>
<p>Set FALSE to deactive miraculix package for dataset generation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>template.chip</code></td>
<td>
<p>Import genetic map and chip from a species ("cattle", "chicken", "pig")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta.shape1</code></td>
<td>
<p>First parameter of the beta distribution for simulating allele frequencies</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta.shape2</code></td>
<td>
<p>Second parameter of the beta distribution for simulating allele frequencies</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>map-file that contains up to 5 colums (Chromsome, SNP-id, M-position, Bp-position, allele freq - Everything not provides it set to NA). A map can be imported via MoBPSmaps::ensembl.map()</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Set to FALSE to not display any prints</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vcf.maxsnp</code></td>
<td>
<p>Maximum number of SNPs to include in the genotype file (default: Inf)</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">population &lt;- founder.simulation(nindi=100, nsnp=1000, n.gen=5)
</code></pre>


</div>