<div class="container">

<table style="width: 100%;"><tr>
<td>getAltTSS</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Annotate Peaks falling in Transcription Start Sites
(TSS) and identify alternatively regulated TSSs for each gene.</h2>

<h3>Description</h3>

<p><code>getAltTSS</code> Pulls out all peaks that fall in TSS,
annotates them with the name of gene, and identifies genes that have
evidence for alternatively regulated TSSs, including both type i (only some
of the open TSSs for a gene are significantly more (or less) accessible),
and type ii (multiple TSSs are significant different, with some being more
accessible and others less). Alternatively, this function  will return all
open TSSs with differential measurements if the returnAllTSS flag is set to
TRUE.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getAltTSS(
  completeDAPs,
  returnAllTSS = FALSE,
  nuancedTSS = TRUE,
  nuancedTSSGap = 150,
  threshold = 0.2,
  TxDb,
  OrgDb
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>completeDAPs</code></td>
<td>
<p>GRanges object that contains the differential
measurements across all peaks (unfiltered DAPs). Will also work with
data.frame or data.table version of a GRanges object. If you want
alternatively regulated TSSs, the object must include a column names 'FDR',
and 'Log2FC_C', which is standard for MOCHA differentials.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>returnAllTSS</code></td>
<td>
<p>Flag to return all TSSs with DAPs measurements,
without filtering for alternative TSS usage. If multiple TSSs fall within
the same tile, then that tile will be repeated for each TSS.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nuancedTSS</code></td>
<td>
<p>True/False flag to determine if alternative TSS genes
should be filtered out if all their differential TSS usage falls within too
small of a range. Default is TRUE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nuancedTSSGap</code></td>
<td>
<p>Minimum distance betweeen TSSs needed for them to
considered distinctly regulated TSSs. If two TSSs are too close, it is
unclear and highly unlikely that ATAC data can distinguish between them.
Default is 150 bp.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>FDR Threshold for determining significant vs non-significant
changes in accessibility. Following MOCHA's standards, default is 0.2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>TxDb</code></td>
<td>
<p>The TxDb-class transcript annotation
package for your organism (e.g. "TxDb.Hsapiens.UCSC.hg38.refGene"). This
must be installed. See
<a href="https://bioconductor.org/packages/release/data/annotation/">
Bioconductor AnnotationData Packages</a>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OrgDb</code></td>
<td>
<p>The OrgDb-class genome wide annotation
package for your organism (e.g. "org.Hs.eg.db"). This must be installed.
See <a href="https://bioconductor.org/packages/release/data/annotation/">
Bioconductor AnnotationData Packages</a></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>tpeaks A GRanges containing annotated peaks falling in TSS
</p>


</div>