<div class="container">

<table style="width: 100%;"><tr>
<td>coVar</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>coVar</h2>

<h3>Description</h3>

<p>Function to calculate ACDC covariances within a data pair for all samples
</p>


<h3>Usage</h3>

<pre><code class="language-R">coVar(dataPair, fullData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dataPair</code></td>
<td>
<p>column indices of two genes to calculate covariance between</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fullData</code></td>
<td>
<p>dataframe or matrix with samples as rows, all probes as columns; each entry should be numeric gene expression or other molecular data values</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Co-expression for a single sample, s, is defined as
</p>
<p style="text-align: center;"><code class="reqn">c_{s,j,k} \equiv \left(g_{s,j}-\bar{g_j}\right)\left(g_{s,k}-\bar{g_k}\right)</code>
</p>

<p>where <code class="reqn">g_{s,j}</code> denotes the expression of gene j in sample s and <code class="reqn">\bar{g_j}</code> denotes the mean expression of gene j in all samples.
</p>
<p>Denoting the sample size as N, coVar returns the co-expression profile across all samples:
</p>
<p style="text-align: center;"><code class="reqn">c_{j,k} = (c_{1,j,k}, c_{2,j,k}, ... , c_{N,j,k})</code>
</p>



<h3>Value</h3>

<p>Co-expression profile, or pairwise covariances for all samples, vector for given features
</p>


<h3>Author(s)</h3>

<p>Katelyn Queen, <a href="mailto:kjqueen@usc.edu">kjqueen@usc.edu</a>
</p>


<h3>References</h3>

<p>Martin P, et al. Novel aspects of PPARalpha-mediated regulation of lipid and xenobiotic metabolism revealed through a nutrigenomic study. <em>Hepatology</em>, in press, 2007.
</p>
<p>Queen K, Nguyen MN, Gilliland F, Chun S, Raby BA, Millstein J. ACDC: a general approach for detecting phenotype or exposure associated co-expression. <em>Frontiers in Medicine</em> (2023) 10. doi:10.3389/fmed.2023.1118824.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#load CCA package for example dataset
library(CCA)

# load dataset
data("nutrimouse")

# run function with first two samples
coVar(dataPair = c(1, 2), 
      fullData = nutrimouse$lipid)

</code></pre>


</div>