<div class="container">

<table style="width: 100%;"><tr>
<td>GCTA_parPlot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>GCTA_parPlot</h2>

<h3>Description</h3>

<p>GCTA_parPlot creates a graph of the output from the GCTA_par function, plotting average heritability of the first principal component of either co-expression or covariance of gene modules against information lost/percent reduction for both observed and permuted data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GCTA_parPlot(df, dataName = "", summaryType)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p>output from GCTA_par function with permutations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dataName</code></td>
<td>
<p>string of name of data for graph labels; default is blank</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>summaryType</code></td>
<td>
<p>one of "coexpression" or "covariance"; how modules were summarized for GCTA calculations</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Genome-wide Complex Trait Analysis (GCTA) is a suite of C++ functions. In order to use the GCTA functions, the user must specify the absolute path to the GCTA software, which can be downloaded from the Yang Lab website <a href="https://yanglab.westlake.edu.cn/software/gcta/#Download">here</a>.
</p>
<p>In GCTA_par, we use GCTA's Genomics REstricted Maximum Likelihood (GREML) method to estimate the average heritability of the first principal component of either co-expression or covariance of gene modules. The produced plot shows these heritability estimates at varying levels of dataset reduction, calculated for observed data in blue and permuted data in red. An information loss value of 0 represents the unreduced dataset, and an information loss level of 100 represents the data being reduced to the average expression of all features.
</p>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Author(s)</h3>

<p>Katelyn Queen, <a href="mailto:kjqueen@usc.edu">kjqueen@usc.edu</a>
</p>


<h3>References</h3>

<p>Millstein J, Battaglin F, Barrett M, Cao S, Zhang W, Stintzing S, et al. Partition: a surjective mapping approach for dimensionality reduction. <em>Bioinformatics</em> <strong>36</strong> (2019) 676â€“681. doi:10.1093/bioinformatics/ btz661.
</p>
<p>Yang J, Lee SH, Goddard ME, Visscher PM. GCTA: a tool for genome-wide complex trait analysis. Am J Hum Genet. 2011 Jan 7;88(1):76-82. doi: 10.1016/j.ajhg.2010.11.011. Epub 2010 Dec 17. PMID: 21167468; PMCID: PMC3014363.
</p>


<h3>See Also</h3>

<p>GCTA software - <a href="https://yanglab.westlake.edu.cn/software/gcta/#Overview">https://yanglab.westlake.edu.cn/software/gcta/#Overview</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# run OSCA_par and save output; input absolute path to OSCA software before running
## Not run: par &lt;- GCTA_par(df = geneExpressionData, 
          ILCincrement = 0.25, 
          fileLoc = "pathHere",
          gctaPath = "pathHere",
          summaryType = "coexpression",
          permute = TRUE,
          numNodes = 1)
## End(Not run)

# run function
## Not run: GCTA_parPlot(df=par, dataName = "Example Data", summaryType = "coexpression")

</code></pre>


</div>