<div class="container">

<table style="width: 100%;"><tr>
<td>tsne2clus</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>t-Stochastic Neighbor Embedding to Clusters</h2>

<h3>Description</h3>

<p>Finds clusters on a 2 dimensional map using
Density-based spatial clustering of applications with noise
(DBSCAN; Esther et al. 1996).
</p>


<h3>Usage</h3>

<pre><code class="language-R">tsne2clus(
  S.tsne,
  ann = NULL,
  labels,
  aest = NULL,
  eps_res = 100,
  eps_range = c(0, 4),
  min.clus.size = 10,
  group.names = "Groups",
  xlab = "x: tSNE(X)",
  ylab = "y: tSNE(X)",
  clus = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>S.tsne</code></td>
<td>
<p>Outcome of function "pca2tsne"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ann</code></td>
<td>
<p>Subjects' annotation data.
An incidence matrix assigning subjects to classes of biological
relevance.
Meant to tune cluster assignation via Biological Homogeneity Index (BHI).
If ann=NULL, the number of clusters is tuned with the
Silhouette index instead of BHI. Defaults to NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labels</code></td>
<td>
<p>Character vector with labels describing subjects.
Meant to assign aesthetics to the visual display of clusters.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aest</code></td>
<td>
<p>Data frame containing points shape and color.
Defaults to NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eps_res</code></td>
<td>
<p>How many eps values should be explored between the 
specified range?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eps_range</code></td>
<td>
<p>Vector containing the minimum and
maximum eps values to be explored. Defaults to c(0, 4).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.clus.size</code></td>
<td>
<p>Minimum size for a cluster to appear in the visual
display. Defaults to 10</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group.names</code></td>
<td>
<p>The title for the legend's key if 'aest' is specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>Name of the 'xlab'. Defaults to "x: tSNE(X)"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>Name of the 'ylab'. Defaults to "y: tSNE(X)"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clus</code></td>
<td>
<p>Should we do clustering? Defaults to TRUE. If false, only
point aesthetics are applied.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function takes the outcome of pca2tsne (or a list containing any
two-columns matrix) and finds clusters via DBSCAN.
It extends code from the MEREDITH (Taskesen et al. 2016) and
clValid (Datta &amp; Datta, 2018) R packages to tune DBSCAN parameters
with Silhouette or
Biological Homogeneity indexes.
</p>


<h3>Value</h3>

<p>A list with the results of the DBSCAN
clustering and (if argument 'plot'=TRUE) the corresponding
graphical displays.
</p>
<ul>
<li>
<p> dbscan.res: a list with the results of the (sparse) SVD,
containing:
</p>

<ul>
<li>
<p> cluster: Cluster partition.
</p>
</li>
<li>
<p> eps: Optimal eps according to the Silhouette or Biological
Homogeneity indexes criteria.
</p>
</li>
<li>
<p> SIL: Maximum peak in the trajectory of the Silhouette index.
</p>
</li>
<li>
<p> BHI: Maximum peak in the trajectory of the Biological
Homogeneity index.
</p>
</li>
</ul>
</li>
<li>
<p> clusters.plot: A ggplot object with the clusters' graphical display.
</p>
</li>
</ul>
<h3>References</h3>


<ul>
<li>
<p> Ester, Martin, Martin Ester, Hans-Peter Kriegel,
Jorg Sander, and Xiaowei Xu. 1996.
"A Density-Based Algorithm for Discovering Clusters in
Large Spatial Databases with Noise," 226_231.
</p>
</li>
<li>
<p> Hahsler, Michael, and Matthew Piekenbrock. 2017.
"Dbscan: Density Based Clustering of Applications with Noise
(DBSCAN) and Related Algorithms."
https://cran.r-project.org/package=dbscan.
</p>
</li>
<li>
<p> Datta, Susmita, and Somnath Datta. 2006. Methods for
Evaluating Clustering Algorithms for Gene Expression Data
Using a Reference Set of Functional Classes.
BMC Bioinformatics 7 (1). BioMed Central:397.
</p>
</li>
<li>
<p> Taskesen, Erdogan, Sjoerd M. H. Huisman, Ahmed Mahfouz,
Jesse H. Krijthe, Jeroen de Ridder, Anja van de Stolpe,
Erik van den Akker, Wim Verheagh, and Marcel J. T. Reinders. 2016.
Pan-Cancer Subtyping in a 2D-Map Shows Substructures
That Are Driven by Specific Combinations of Molecular
Characteristics. Scientific Reports 6 (1):24949.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">
library(MOSS)
library(viridis)
library(cluster)
library(annotate)

# Using the 'iris' data tow show cluster definition via BHI criterion.
set.seed(42)
data(iris)
# Scaling columns.
X &lt;- scale(iris[, -5])
# Calling pca2tsne to map the three variables onto a 2-D map.
Z &lt;- pca2tsne(X, perp = 30, n.samples = 1, n.iter = 1000)
# Using 'species' as previous knoledge to identify clusters.
ann &lt;- model.matrix(~ -1 + iris[, 5])
# Getting clusters.
tsne2clus(Z,
  ann = ann,
  labels = iris[, 5],
  aest = aest.f(iris[, 5]),
  group.names = "Species",
  eps_range = c(0, 3)
)

# Example of usage within moss.
set.seed(43)
sim_blocks &lt;- simulate_data()$sim_blocks
out &lt;- moss(sim_blocks[-4],
  tSNE = TRUE,
  cluster = list(eps_range = c(0, 4), eps_res = 100, min_clus_size = 1),
  plot = TRUE
)
out$clus_plot
out$clusters_vs_PCs

</code></pre>


</div>