<div class="container">

<table style="width: 100%;"><tr>
<td>SST</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Subset multiple hypothesis testing procedure to combine two steps of testing gene-environment interaction
in a two-step procedure.</h2>

<h3>Description</h3>

<p>Run <code>SST</code> to adjust for multiple testing
while combining two steps of the GxE interaction testing procedure. The procedure is applicable for
a multivariate phenotype, as well as a univariate phenotype.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SST(PVAL, Pg_thr_step1 = 0.005, FWER_step2 = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>PVAL</code></td>
<td>
<p>A data.frame with three columns.
The first column (PVAL$SNP) provides the name of all SNPs or genetic variants tested.
Second column (PVAL$G.P) contains the p-values of the variants obtained from testing
an overall marginal genetic
association between the multivariate phenotype and each genetic variant individually.
And the third column (PVAL$GE.P) contains the p-values obtained from testing overall GxE effect on the
multivariate phenotype in presence of possible marginal effect due to the genetic variant and
a marginal effect
due to the environmental variable. Number of rows in PVAL is the same as the number
of genetic variants, and it has the same structure as in the output of <code>mv_G_GE</code>.
No default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Pg_thr_step1</code></td>
<td>
<p>A positive real number between 0 and 1 providing the p-value threshold
to select the set of
promising SNPs in step 1. These selected SNPs will be tested for GxE effect in the second step.
Default is 0.005.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FWER_step2</code></td>
<td>
<p>A positive real number between 0 and 1 specifying the family-wise error rate
to be maintained in the second
step while identifying the genetic variants having a genome-wide significant overall GxE effect
on the multivariate phenotype. Default is 0.05.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The output is a vector of SNPs identified to have a genome-wide significant overall GxE effect.
</p>


<h3>References</h3>

<p>A Majumdar, KS Burch, S Sankararaman, B Pasaniuc, WJ Gauderman, JS Witte (2020)
A two-step approach to testing overall effect of gene-environment interaction for multiple phenotypes.
bioRxiv, doi: https://doi.org/10.1101/2020.07.06.190256
</p>


<h3>See Also</h3>

<p><code>WHT</code>, <code>mv_G_GE</code>
</p>


</div>