<div class="container">

<table style="width: 100%;"><tr>
<td>mv_G_GE</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Test for marginal overall genetic association with multivariate phenotype,
and test for overall GxE effect on the multivariate phenotype
in presence of marginal effect due to the genetic variant
and a marginal effect due to the environmental variable.</h2>

<h3>Description</h3>

<p>Run <code>mv_G_GE</code> to obtain two different sets of p-values, one from the
test for marginal overall genetic association with multiple phenotypes (using multivariate linear regression),
and the other
from the test of overall GxE effect on multivariate phenotype allowing for a possible
genetic effect
due to the genetic variant and an effect due to the environmental variable.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mv_G_GE(pheno, geno, env)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p>A numeric matrix or data.frame with the number of individuals (n) as the number of rows and the
number of phenotypes (k) as the number of columns. It contains the values of k phenotypes
(e.g. LDL and HDL) across the individuals. Each phenotype (e.g. LDL) must be individually
adjusted for relevant covariates (age, sex, principal components of
genetic ancestries, etc) beforehand. Therefore, each column of pheno matrix should be the adjusted
residuals
of the corresponding phenotype. Each final phenotype (column) should be continuous
and normally distributed. No default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>A numeric matrix/data.frame (for a batch of genetic variants), or a numeric vector (for a single genetic variant).
It contains the genotype values of the genetic variants/variant across the individuals.
For a batch of variants, columns correspond to variants, and rows correspond to n individuals.
For a SNP, three different ways of genotype coding are possible:
additive, dominant and recessive, where additive coding is more common. No default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>env</code></td>
<td>
<p>A vector of length n (number of individuals). It contains the values of the
environmental variable (e.g., frequency of alcohol consumption). It
can also contain factors, e.g., "yes" or "no" smoking status.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The output is a data.frame with three columns. First column is the name of
the SNPs or genetic variants. The main columns are as follows:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>G.P</code></td>
<td>
<p>P value of testing multivariate marginal genetic association
between the genetic variant
and the vector of phenotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GE.P</code></td>
<td>
<p>P value of testing multivariate overall GxE effect
in presence of possible marginal effect due
to the genetic variant and marginal effect
due to the environmental variable.</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>A Majumdar, KS Burch, S Sankararaman, B Pasaniuc, WJ Gauderman, JS Witte (2020)
A two-step approach to testing overall effect of gene-environment interaction for multiple phenotypes.
bioRxiv, doi: https://doi.org/10.1101/2020.07.06.190256
</p>


<h3>See Also</h3>

<p><code>WHT</code>, <code>SST</code>
</p>


</div>