<div class="container">

<table style="width: 100%;"><tr>
<td>create.mppData</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create a multi-parent population data object</h2>

<h3>Description</h3>

<p>This function combines all raw data sources in a single data object of class
<code>mppData</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">create.mppData(
  geno.off = NULL,
  geno.par = NULL,
  map = NULL,
  pheno = NULL,
  cross.ind = NULL,
  par.per.cross = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>geno.off</code></td>
<td>
<p><code>Character</code> marker score <code>matrix</code> of the offspring
with genotypes as row and markers as column.
<strong>Rows names must be the offspring genotypes identifiers similar to
the one used in <code>pheno</code>. The columns names must be the marker names
similar to the one used in <code>map</code>. Marker scores must be coded using one
letter per allele. For example, AA, CC, GG, etc. Missing values must be coded
<code>NA</code>.</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno.par</code></td>
<td>
<p><code>Character</code> marker score <code>matrix</code> of the parents
with genotypes as row and markers as column.
<strong>Rows names must be the parents genotypes identifiers similar to
the one used in <code>par.per.cross</code>. The columns names must be the marker
names similar to the one used in <code>map</code>. Marker scores must be coded
using one letter per allele. For example, AA, CC, GG, etc. Missing values
must be coded <code>NA</code>.</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>Three columns <code>data.frame</code> with: 1) <code>character</code> marker
identifiers; 2) <code>numeric</code> chromosome; 3) <code>numeric</code> positions in
centi-Morgan.<strong>The marker identifiers must be identical to the column
names of the maker matrices (<code>geno.off</code> and <code>geno.par</code>).
The chromosome identifiers must start by 1 and increase by 1 unit,
e.g. 1, 2, 3, ...</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p><code>Numeric matrix</code> with one column per trait and rownames
as genotypes identifiers. <strong>The genotypes identifiers must be identical
to the rownames of the offspring marker matrix (<code>geno.off</code>).</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cross.ind</code></td>
<td>
<p><code>Character</code> vector indicating to which cross each
genotype belongs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>par.per.cross</code></td>
<td>
<p>Three columns <code>Character matrix</code> specifying :
1) the cross indicators; 2) the parents 1 identifiers
of the crosses; 3) the parents 2 identifiers of the crosses.
<strong>The list of crosses must contain the same cross indicators as in
<code>cross.ind</code> and they must appear in the same order.
The list of parent identifiers must be the same to the rownames of
the argument <code>geno.par</code></strong>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a <code>list</code> of class <code>mppData</code> which contains the following elements
</p>
<table>
<tr style="vertical-align: top;">
<td><code>geno.off</code></td>
<td>
<p><code>Matrix</code> with offspring marker scores.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno.par</code></td>
<td>
<p><code>Matrix</code> with parents marker scores.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p><code>Matrix</code> with phenotypic trait values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p><code>Data.frame</code> with genetic marker information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cross.ind</code></td>
<td>
<p>Cross indicator.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>par.per.cross</code></td>
<td>
<p><code>Character matrix</code> information about cross and
the parents of the crosses.</p>
</td>
</tr>
</table>
<p>The <code>list</code> also contain other arguments that will be filled later in
the data processing.
</p>


<h3>Author(s)</h3>

<p>Vincent Garin
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(USNAM_geno)
data(USNAM_map)
data(USNAM_pheno)

geno.off &lt;- USNAM_geno[7:506, ]
geno.par &lt;- USNAM_geno[1:6, ]
map &lt;- USNAM_map
pheno &lt;-  USNAM_pheno
cross.ind &lt;- substr(rownames(pheno), 1, 4)
par.per.cross &lt;- cbind(unique(cross.ind), rep("B73", 5),
                       rownames(geno.par)[2:6])

mppData &lt;- create.mppData(geno.off = geno.off, geno.par = geno.par,
                          map = map, pheno = pheno, cross.ind = cross.ind,
                          par.per.cross = par.per.cross)
                          
</code></pre>


</div>