<div class="container">

<table style="width: 100%;"><tr>
<td>parent_cluster.mppData</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Parent clustering for <code>mppData</code> objects</h2>

<h3>Description</h3>

<p>Integrate the parent clustering information to the mppData
object. The parent clustering is necessary to compute the ancestral model.
If the parent clustering step is skipped, the ancestral model can not be
used but the other models (cross-specific, parental, and bi-allelic) can
still be computed.
</p>


<h3>Usage</h3>

<pre><code class="language-R">parent_cluster.mppData(mppData, par.clu = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mppData</code></td>
<td>
<p>An object of class <code>mppData</code>. the <code>mppData</code> must
have been processed using: <code>create.mppData</code>,
<code>QC.mppData</code>, <code>IBS.mppData</code>,
and <code>IBD.mppData</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>par.clu</code></td>
<td>
<p><code>Interger matrix</code> representing the results of a
parents genotypes clustering. The columns represent the parental lines and
the rows the markers. The columns names must be the same as the parents
list of the mppData object. The rownames must be the same as the map marker
list of the mppData object. At a particular position, parents with the same
value are assumed to inherit from the same ancestor. for more details,
see <code>par_clu</code>. Default = NULL.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>At a single marker position, two parents can be grouped into a similar
ancestral classes if we assume that they receive there allele from a common
ancestor. The parent clustering information (<code>par.clu</code>) describe parental
relatedness and which parent belong to which ancestral group. For example,
at marker i, we could have five parents (pA, pB, pC, pD, pE) and the following
clustering information (1, 2, 1, 2, 3). This means that pA and pC received
their allele from the same ancestor (A1). pB and pD also have a shared
ancestor (A2) who is different from (A1). And pE was not included in any
group and can be seen as an independent ancestral group (A3).
</p>
<p>The parent clustering information is provided via <code>par.clu</code>. It is an
<code>interger matrix</code> with markers in row and parents in columns.
At a particular marker position, parents with the same value are assumed to
inherit from the same ancestor. for more details, see <code>par_clu</code>.
</p>
<p>The marker positions that are considered as monomorphic given the parent
clustering information are set back to one allele per parent to still allow
the computation of the QTL allelic effect at those positions later.
</p>
<p>The parent clustering can be performed using the R package
'clusthaplo' that can be found there:
<a href="https://cran.r-project.org/src/contrib/Archive/clusthaplo/">https://cran.r-project.org/src/contrib/Archive/clusthaplo/</a>.
The 'clusthaplo' option is not integrated in this version of mppR. However,
a version of mppR with function calling clusthaplo can be found on github
<a href="https://github.com/vincentgarin/mppR">https://github.com/vincentgarin/mppR</a> (branch master).
</p>


<h3>Value</h3>

<p>An increased <code>mppData</code> object containing the the same elements
as the <code>mppData</code> object provided as argument and the
following new elements:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>par.clu</code></td>
<td>
<p><code>Integer matrix</code> with rows representing markers and
columns corresponding to the parents. At a single marker position, parents
with the same value were clustered in the same ancestral group.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.anc</code></td>
<td>
<p>Average number of ancestral clusters along the genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mono.anc</code></td>
<td>
<p>Positions for which the ancestral clustering was monomorphic.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Vincent Garin
</p>


<h3>See Also</h3>

<p><code>create.mppData</code>, <code>QC.mppData</code>,
<code>IBS.mppData</code>, <code>IBD.mppData</code>, <code>par_clu</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(mppData_init)
data(par_clu)

mppData &lt;- QC.mppData(mppData_init)
mppData &lt;- IBS.mppData(mppData = mppData)

mppData &lt;- IBD.mppData(mppData = mppData, type = 'RIL',
                       type.mating = 'selfing')
                       
mppData &lt;- parent_cluster.mppData(mppData = mppData, par.clu  = par_clu)                         
                       

</code></pre>


</div>