<div class="container">

<table style="width: 100%;"><tr>
<td>mppGE_SIM</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>MPP GxE Simple Interval Mapping</h2>

<h3>Description</h3>

<p>Computes single QTL models along the genome using an approximate mixed model
computation. An initial variance covariance (VCOV) structure is calculated
using function from the <code>nlme</code> package. Then, this information is used
to estimate the QTL global and within parental effect significance using a
Wald test.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mppGE_SIM(
  mppData,
  trait,
  VCOV = "UN",
  ref_par = NULL,
  n.cores = 1,
  maxIter = 100,
  msMaxIter = 100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mppData</code></td>
<td>
<p>An object of class <code>mppData</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trait</code></td>
<td>
<p><code>Character vector</code> specifying which traits (environments)
should be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>VCOV</code></td>
<td>
<p>VCOV <code>Character</code> expression defining the type of variance
covariance structure used. 'CS' for compound symmetry assuming a unique
genetic covariance between environments. 'CSE' for cross-specific within
environment error term. 'CS_CSE' for both compound symmetry plus
cross-specific within environment error term. 'UN' for unstructured
environmental variance covariance structure allowing a specific genotypic
covariance for each pair of environments. Default = 'UN'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref_par</code></td>
<td>
<p>Optional <code>Character</code> expression defining the parental
allele that will be used as reference for the parental model. Default = NULL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.cores</code></td>
<td>
<p><code>Numeric</code>. Specify here the number of cores you like to
use. Default = 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxIter</code></td>
<td>
<p>maximum number of iterations for the lme optimization algorithm.
Default = 100.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>msMaxIter</code></td>
<td>
<p>maximum number of iterations for the optimization step inside
the lme optimization. Default = 100.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The estimated model is the following:
</p>
<p><code class="reqn">\underline{y}_{icj} = E_{j} + C_{cj} + x_{i_{q}p} * \beta_{pj} + \underline{GE}_{icj} + \underline{e}_{icj}</code>
</p>
<p>For further details see the vignette.
</p>


<h3>Value</h3>

<p>Return:
</p>
<table><tr style="vertical-align: top;">
<td><code>SIM </code></td>
<td>
<p><code>Data.frame</code> of class <code>QTLprof</code>. with five columns :
1) QTL marker or in between position names; 2) chromosomes;
3) integer position indicators on the chromosome;
4) positions in centi-Morgan; and 5) -log10(p-val) of the global QTL effect
across environments 6) p-values of the within environment QTL effects
(one column per environment); and p-values of the within environment parental
QTL allelic effects (one column per parent environment combination).</p>
</td>
</tr></table>
<h3>Author(s)</h3>

<p>Vincent Garin
</p>


<h3>References</h3>

<p>Pinheiro J, Bates D, DebRoy S, Sarkar D, R Core Team (2021). nlme: Linear
and Nonlinear Mixed Effects Models_. R package version 3.1-152,
&lt;URL: https://CRAN.R-project.org/package=nlme&gt;.
</p>


<h3>See Also</h3>

<p><code>mppGE_CIM</code>,
<code>mppGE_proc</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(mppData_GE)

SIM &lt;- mppGE_SIM(mppData = mppData_GE, trait = c('DMY_CIAM', 'DMY_TUM'))

Qpos &lt;- QTL_select(Qprof = SIM, threshold = 3, window = 50)

plot(x = SIM, QTL = Qpos)

plot_allele_eff_GE(mppData = mppData_GE, nEnv = 2, EnvNames = c('CIAM', 'TUM'),
                   Qprof = SIM, Q.eff = 'par', QTL = Qpos, text.size = 14)

</code></pre>


</div>