<div class="container">

<table style="width: 100%;"><tr>
<td>mppData</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Complete <code>mppData</code> object
</h2>

<h3>Description</h3>

<p>Complete <code>mppData</code> object made from a sample data of the maize US nested
association mapping (NAM) population. This mppData object went through all
the steps of the data processing: <code>create.mppData</code>,
<code>QC.mppData</code>, <code>IBS.mppData</code>, <code>IBD.mppData</code>, <code>parent_cluster.mppData</code>. The mppData contain all the data
necessary for the QTL analysis procedures.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(mppData)</code></pre>


<h3>Format</h3>

<p><code>mppData</code>
</p>


<h3>Details</h3>

<p>The complete <code>mppData</code> object is a <code>list</code> containing the following
elements:
</p>

<ol>
<li>
<p>geno.IBS: IBS genotype marker matrix.
</p>
</li>
<li>
<p>geno.IBD: R/qtl cross.object containing the genotype within cross IBD probabilities.
</p>
</li>
<li>
<p>geno.id: List of genotypes.
</p>
</li>
<li>
<p>allele.ref: Matrix containing for each marker the most and least frequent marker scores and the two heterozygous scores.
</p>
</li>
<li>
<p>geno.par: Parents marker matrix.
</p>
</li>
<li>
<p>geno.par.clu: Parent marker data used to cluster the parents.
</p>
</li>
<li>
<p>par.clu: Parent clustering results.
</p>
</li>
<li>
<p>mono.anc: Positions for which the ancestral clustering was monomorphic.
</p>
</li>
<li>
<p>pheno: Phenotypic trait matrix.
</p>
</li>
<li>
<p>map: Genetic map corresponding to the geno (IBS, IBD, par) arguments.
</p>
</li>
<li>
<p>haplo.map: Genetic map corresponding to geno.par.clu.
</p>
</li>
<li>
<p>cross.ind: Vector indicating to which cross each genotype belongs.
</p>
</li>
<li>
<p>par.per.cross: Matrix with for each cross the parent 1 and 2.
</p>
</li>
<li>
<p>parents: Vector of parents.
</p>
</li>
<li>
<p>type: Type of population.
</p>
</li>
<li>
<p>n.cr: Number of crosses.
</p>
</li>
<li>
<p>n.par: Number of parents.
</p>
</li>
<li>
<p>n.anc: Average number of ancestral group along the genome.
</p>
</li>
<li>
<p>n.zigo: Number of possible allelic computations 2 (AA/BB) or 3 (AA/AB/BB).
</p>
</li>
<li>
<p>rem.mk: Removed markers in the data processing.
</p>
</li>
<li>
<p>rem.gen: Removed genotypes in the data processing.
</p>
</li>
<li>
<p>status: Indicates the level of progression in the data processing.
</p>
</li>
</ol>
<h3>See Also</h3>

<p><code>create.mppData</code>, <code>QC.mppData</code>,
<code>IBS.mppData</code>, <code>IBD.mppData</code>,
<code>parent_cluster.mppData</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">  data(mppData)
</code></pre>


</div>