<div class="container">

<table style="width: 100%;"><tr>
<td>mpp_CIM</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>MPP Composite Interval Mapping</h2>

<h3>Description</h3>

<p>Compute QTL models along the genome using cofactors representing other
genetic positions for control.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mpp_CIM(
  mppData,
  trait = 1,
  Q.eff = "cr",
  cofactors = NULL,
  window = 20,
  plot.gen.eff = FALSE,
  n.cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mppData</code></td>
<td>
<p>An object of class <code>mppData</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trait</code></td>
<td>
<p><code>Numerical</code> or <code>character</code> indicator to specify which
trait of the <code>mppData</code> object should be used. Default = 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Q.eff</code></td>
<td>
<p><code>Character</code> expression indicating the assumption concerning
the QTL effects: 1) "cr" for cross-specific; 2) "par" for parental; 3) "anc"
for ancestral; 4) "biall" for a bi-allelic. For more details see
<code>mpp_SIM</code>. Default = "cr".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cofactors</code></td>
<td>
<p>Object of class <code>QTLlist</code> representing a list of
selected position obtained with the function <code>QTL_select</code> or
vector of <code>character</code> marker positions names.
Default = NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>window</code></td>
<td>
<p><code>Numeric</code> distance (cM) on the left and the right of a
cofactor position where it is not included in the model. Default = 20.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.gen.eff</code></td>
<td>
<p><code>Logical</code> value. If <code>plot.gen.eff = TRUE</code>,
the function will save the decomposed genetic effects per cross/parent.
These results can be plotted with the function <code>plot.QTLprof</code>
to visualize a genome-wide decomposition of the genetic effects.
<strong>This functionality is only available for the cross-specific,
parental and ancestral models.</strong>
Default value = FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.cores</code></td>
<td>
<p><code>Numeric</code>. Specify here the number of cores you like to
use. Default = 1.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For more details about the different models, see documentation of the
function <code>mpp_SIM</code>. The function returns a -log10(p-value) QTL
profile.
</p>


<h3>Value</h3>

<p>Return:
</p>
<table><tr style="vertical-align: top;">
<td><code>CIM </code></td>
<td>
<p><code>Data.frame</code> of class <code>QTLprof</code>. with five columns :
1) QTL marker names; 2) chromosomes;
3) interger position indicators on the chromosome;
4) positions in centi-Morgan; and 5) -log10(p-val). And if
<code>plot.gen.eff = TRUE</code>, p-values of the cross or parental QTL effects.</p>
</td>
</tr></table>
<h3>Author(s)</h3>

<p>Vincent Garin
</p>


<h3>See Also</h3>

<p><code>mpp_SIM</code>, <code>QTL_select</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Cross-specific effect model
#############################

data(mppData)

SIM &lt;- mpp_SIM(mppData = mppData, Q.eff = "cr")

cofactors &lt;- QTL_select(Qprof = SIM, threshold = 3, window = 20)

CIM &lt;- mpp_CIM(mppData = mppData, Q.eff = "cr", cofactors = cofactors,
               window = 20, plot.gen.eff = TRUE)

plot(x = CIM)
plot(x = CIM, gen.eff = TRUE, mppData = mppData, Q.eff = "cr")

# Bi-allelic model
##################

cofactors &lt;- mppData$map[c(15, 63), 1]

CIM &lt;- mpp_CIM(mppData = mppData, Q.eff = "biall", cofactors = cofactors,
               window = 20)

plot(x = CIM, type = "h")
                               
</code></pre>


</div>