<div class="container">

<table style="width: 100%;"><tr>
<td>FASTmrMLM</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>To perform GWAS with FASTmrMLM method</h2>

<h3>Description</h3>

<p>FAST multi-locus random-SNP-effect Mixed Linear Model</p>


<h3>Usage</h3>

<pre><code class="language-R">FASTmrMLM(gen,phe,outATCG,genRaw,kk,psmatrix,svpal,svrad,svmlod,Genformat,CLO)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gen</code></td>
<td>
<p>genotype matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phe</code></td>
<td>
<p>phenotype matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outATCG</code></td>
<td>
<p>genotype for code 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genRaw</code></td>
<td>
<p>raw genotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kk</code></td>
<td>
<p>kinship matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>psmatrix</code></td>
<td>
<p>population structure matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>svpal</code></td>
<td>
<p>Critical P-value for selecting variable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>svrad</code></td>
<td>
<p>Search Radius in search of potentially associated QTN.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>svmlod</code></td>
<td>
<p>Critical LOD score for significant QTN.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Genformat</code></td>
<td>
<p>Format for genotypic codes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CLO</code></td>
<td>
<p>number of CPU.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Li Pei, and Zhang Yuan-Ming<br>
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R">G1=data(mrgen)
P1=data(mrphe)
G2=data(mrgenraw)
result=FASTmrMLM(mrgen,mrphe,outATCG=NULL,mrgenraw,kk=NULL,psmatrix=NULL,
0.01,20,3,1,CLO=1)
</code></pre>


</div>