<div class="container">

<table style="width: 100%;"><tr>
<td>PlotGraphWithModules</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Plot a graph with nodes in modules indicated by colors 
</h2>

<h3>Description</h3>

<p>Visualization of a graph with nodes in modules inferred from the clustering dendrogram by PlotDendrogram.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PlotGraphWithModules(Adj_directed, PlotDendrogramObj, 
                     GV, node.size = 8, arrow.size = 5,
                     label.size = 3,alpha = 1,...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Adj_directed</code></td>
<td>

<p>Adjacency matrix of a graph.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>PlotDendrogramObj</code></td>
<td>

<p>The graphical objects from PlotDendrogram.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GV</code></td>
<td>

<p>The number of genetic variants (SNPs/indels/CNVs/eQTL) in the input data matrix. For example, if the data has one SNPs/indels/CNV/eQTL in the first column, then GV = 1, if 2 SNPs/indels/CNVs/eQTL in the 1st and 2nd Column, then GV = 2, and so on. If no GV then GV = 0.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>node.size</code></td>
<td>

<p>The size of the nodes in the graph. Defaults to 8.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>arrow.size</code></td>
<td>

<p>The size of the arrows for directed network edges, in points. Defaults to 5.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label.size</code></td>
<td>

<p>The size of the node labels in points, as a numeric value, a vector of numeric values, or as a vertex attribute containing numeric values. Defaults to 3.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>

<p>The level of transparency of the edges and nodes. Defaults to 1 (no transparency).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Other arguments passed to ggnet2.
</p>
</td>
</tr>
</table>
<h3>Value</h3>


<ul><li> <p><code>PlotGraphWithModulesObj</code>: An object of class "graph" of the graph.
</p>
</li></ul>
<h3>Author(s)</h3>

<p>Md Bahadur Badsha (mbbadshar@gmail.com)
</p>


<h3>References</h3>

<p>1. Badsha MB, Martin EA and Fu AQ (2021). MRPC: An R package for inference of causal graphs. Frontiers in Genetics, 10:651812. 
</p>


<h3>See Also</h3>

<p>PlotDendrogram
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# Adjacency matrix from a graph in the example
Adj_directed &lt;- as(data_examples$complex$cont$withGV$graph,
                   "matrix")

# A clustering dendrogram with nodes grouped in colored modules
PlotDendrogramObj &lt;- PlotDendrogram(Adj_directed,
                                    minModuleSize = 5)
                  
# A graph object with nodes in modules
PlotGraphWithModulesObj &lt;- PlotGraphWithModules(Adj_directed,
                                                PlotDendrogramObj,
                                                GV = 14,
                                                node.size = 8,
                                                arrow.size = 5,
                                                label.size = 3,
                                                alpha = 1) 

# Plot the graph with nodes in different colors
plot(PlotGraphWithModulesObj) 

## End(Not run)
</code></pre>


</div>