<div class="container">

<table style="width: 100%;"><tr>
<td>RecallPrecision</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Calculate recall and precision for two graphs
</h2>

<h3>Description</h3>

<p>This function counts the number of true and false positives, and calculates recall (i.e., power) and precision (i.e., 1-FDR), which are defined as follows:
</p>
<p>Recall = (# edges correctly identified in inferred graph) / (# edges in true graph). 
</p>
<p>Precision = (# edges correctly identified in inferred graph) / (# edges in inferred graph).  
</p>


<h3>Usage</h3>

<pre><code class="language-R">RecallPrecision(g1, g2, GV, includeGV, edge.presence = 1.0, edge.direction = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>g1</code></td>
<td>

<p>First graph object, from the true graph
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>g2</code></td>
<td>

<p>Second graph object, from the inferred graph
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GV</code></td>
<td>

<p>The number of genetic variants (SNPs/indels/CNV/eQTLs) in the input data matrix. For example, if the data has one variant, which is in the first column, then GV = 1. If there are two variants, which are in the first and second Columns, then GV = 2. If there are no variants, then GV = 0.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>includeGV</code></td>
<td>

<p>If TRUE, include edges involving genetic variants (GV) when calculating recall and precision. If FALSE, exclude edges involving genetic variants (GV) when calculating recall and precision.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>edge.presence</code></td>
<td>

<p>The weight for an edge being present.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>edge.direction</code></td>
<td>

<p>The weight for the edge direction.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>We consider it more important to be able to identify the presence of an edge than to also get the direct correct. Therefore, we assign 1 as the default to an edge with the correct direction and 0.5 to an edge with the wrong direction or no direction (Badsha and Fu, 2019; Badsha et al., 2021).
</p>


<h3>Value</h3>

<p>A list of object that containing the following:
</p>

<ul>
<li> <p><code>Matrix</code>: Results store for TP and FP
</p>
</li>
<li> <p><code>TP</code>: Total found edges in the inferred graph and edge exists in the true graph.
</p>
</li>
<li> <p><code>FP</code>: Total found edges in the inferred graph but no edge exists in the true graph.
</p>
</li>
<li> <p><code>NTE</code>: Total number of edges in the true graph.
</p>
</li>
<li> <p><code>NIE</code>: Total number of edges in the inferred graph.
</p>
</li>
<li> <p><code>Recall</code>: Power, or sensitivity measures how many edges from the true graph a method can recover.
</p>
</li>
<li> <p><code>Precision</code>: Measures how many correct edges are recovered in the inferred graph.
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Md Bahadur Badsha (mbbadshar@gmail.com)
</p>


<h3>References</h3>

<p>1. Badsha MB and Fu AQ (2019). Learning causal biological networks with the principle of Mendelian randomization. Frontiers in Genetics, 10:460.
</p>
<p>2. Badsha MB, Martin EA and Fu AQ (2021). MRPC: An R package for inference of causal graphs. Frontiers in Genetics, 10:651812.
</p>


<h3>See Also</h3>

<p>aSHD: adjusted Structural Hamming Distance (aSHD)
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# True model
# True graph (V1 --&gt; T1 --&gt; T2 --&gt; T3) 
# Where V1 is a genetic variant (GV) and T1, T2, and T3 are phenotypes
tarmat_s1 &lt;- matrix(0,
                    nrow = 4,
                    ncol = 4)
                    
colnames(tarmat_s1) &lt;- c("V1", "T1", "T2", "T3")
rownames(tarmat_s1) &lt;- colnames(tarmat_s1)

# Create an adjacency matrix for the true graph
tarmat_s1[1, 2] &lt;- 1
tarmat_s1[2, 3] &lt;- 1
tarmat_s1[3, 4] &lt;- 1

# Graph object of the true graph
Truth &lt;- as(tarmat_s1,
            "graphNEL")

# Inferred graph (V1 --&gt; T1 &lt;-- T2 --&gt; T3)
# Where V1 is a genetic variant (GV) and T1, T2, and T3 are phenotypes
tarmat_s2 &lt;- matrix(0,
                    nrow = 4,
                    ncol = 4)
                    
colnames(tarmat_s2) &lt;- c("V1", "T1", "T2", "T3")
rownames(tarmat_s2) &lt;- colnames(tarmat_s2)

# Create an adjacency matrix for the inferred graph
tarmat_s2[1, 2] &lt;- 1
tarmat_s2[3, 2] &lt;- 1
tarmat_s2[3, 4] &lt;- 1

# Graph objects for the inferred graph
Inferred &lt;- as(tarmat_s2,
               "graphNEL")
         
# Recall and Precision
Recall_Precision &lt;- RecallPrecision(Truth,
                                   Inferred,
                                   GV = 1,
                                   includeGV = TRUE,
                                   edge.presence = 1.0,
                                   edge.direction = 0.5)
</code></pre>


</div>