<div class="container">

<table style="width: 100%;"><tr>
<td>SimulateData</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Simulate data under certain graphs
</h2>

<h3>Description</h3>

<p>This function simulates data using linear models for several graphs: the five basic topologies and three topologies that are common in biology, namely the multi-parent graph, the star graph and the layered graph. See details in Badsha and Fu (2019) and Badsha et al. (2021).
</p>


<h3>Usage</h3>

<pre><code class="language-R">SimulateData(N, p, model, b0.1, b1.1, b1.2, b1.3, sd.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>N</code></td>
<td>
<p>The number of observations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>Population frequency of the reference allele. Real number between 0 to 1, which is the number of a particular allele is present.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>The model for which data will be simulated. For example, if you want to generate data for model 0 you would type 'model0' into the function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b0.1</code></td>
<td>
<p>Intercept of b0.1 + b1.1*P1 + b1.2*P2 + b1.3*P3, where P1, P2, and P3 are the parents of the corresponding node.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b1.1</code></td>
<td>
<p>Slope of P1 for b0.1 + b1.1*P1 + b1.2*P2 + b1.3*P3, where P1, P2, and P3 are the parents of the corresponding node.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b1.2</code></td>
<td>
<p>Slope of P2 for b0.1 + b1.1*P1 + b1.2*P2 + b1.3*P3, where P1, P2, and P3 are the parents of the corresponding node.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b1.3</code></td>
<td>
<p>Slope of P3 for b0.1 + b1.1*P1 + b1.2*P2 + b1.3*P3, where P1, P2, and P3 are the parents of the corresponding node.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sd.1</code></td>
<td>
<p>Standard deviation for corresponding data generated nodes.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The first column of the input matrix is a genetic variant and the remaining columns are gene expression nodes.
</p>


<h3>Value</h3>

<p>Matrix
</p>


<h3>Author(s)</h3>

<p>Md Bahadur Badsha (mbbadshar@gmail.com)
</p>


<h3>References</h3>

<p>1. Badsha MB and Fu AQ (2019). Learning causal biological networks with the principle of Mendelian randomization. Frontiers in Genetics, 10:460.
</p>
<p>2. Badsha MB, Martin EA and Fu AQ (2021). MRPC: An R package for inference of causal graphs. Frontiers in Genetics, 10:651812. 
</p>


<h3>See Also</h3>

<p>MRPC; SimulateDataNP, which simulates data for a node with no parent; SimulateData1P for a node with one parent; SimulateData2P for a node with two parents.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># When there is one genetic variant, the 1st column of
# the simulated data matrix will be the variant and the remaining
# columns are the gene expression nodes.


## Model 0
simu_data_M0 &lt;- SimulateData(N = 10^3, 
                             p = 0.45,
                             'model0', 
                             b0.1 = 0,
                             b1.1 = 1, 
                             b1.2 = 1,
                             b1.3 = 1, 
                             sd.1 = 1)
                             
## Model 1
simu_data_M1 &lt;- SimulateData(N = 10^3, 
                             p = 0.45,
                             'model1', 
                             b0.1 = 0,
                             b1.1 = 1, 
                             b1.2 = 1,
                             b1.3 = 1, 
                             sd.1 = 1)
                             
## Model 2
simu_data_M2 &lt;- SimulateData(N = 10^3, 
                             p = 0.45,
                             'model2', 
                             b0.1 = 0,
                             b1.1 = 1, 
                             b1.2 = 1,
                             b1.3 = 1, 
                             sd.1 = 1)
                             
## Model 3
simu_data_M3 &lt;- SimulateData(N = 10^3, 
                             p = 0.45,
                             'model3', 
                             b0.1 = 0,
                             b1.1 = 1, 
                             b1.2 = 1,
                             b1.3 = 1, 
                             sd.1 = 1)
                             
## Model 4
simu_data_M4 &lt;- SimulateData(N = 10^3, 
                             p = 0.45,
                             'model4', 
                             b0.1 = 0,
                             b1.1 = 1, 
                             b1.2 = 1,
                             b1.3 = 1, 
                             sd.1 = 1)
                             
## Multiple Parent Model
simu_data_multiparent &lt;- SimulateData(N = 10^3, 
                                      p = 0.45,
                                      'multiparent', 
                                      b0.1 = 0,
                                      b1.1 = 1, 
                                      b1.2 = 1,
                                      b1.3 = 1, 
                                      sd.1 = 1)

## Star Model
simu_data_starshaped &lt;- SimulateData(N = 10^3, 
                                     p = 0.45,
                                     'starshaped', 
                                     b0.1 = 0,
                                     b1.1 = 1, 
                                     b1.2 = 1,
                                     b1.3 = 1, 
                                     sd.1 = 1)

## Layered Model
simu_data_layered &lt;- SimulateData(N = 10^3, 
                                  p = 0.45,
                                  'layered', 
                                  b0.1 = 0,
                                  b1.1 = 1, 
                                  b1.2 = 1,
                                  b1.3 = 1, 
                                  sd.1 = 1)
</code></pre>


</div>