<div class="container">

<table style="width: 100%;"><tr>
<td>data_examples</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Example data under simple and complex models
</h2>

<h3>Description</h3>

<p>Example data under the simple and complex graphs. Data may be continuous or discrete.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(data_examples)
</code></pre>


<h3>Details</h3>

<p>For each model, the graph and a simulated data matrix are available for both continuous and discrete data. 
</p>
<p><b>For continuous data with genetic information</b>: 1000 samples in row and 6 variables in column. First two columns are the genetic variants and remaning columns are gene expression.
</p>
<p><b>Continuous data without genetic information</b>: 1000 samples in row and 8 variables in column.
</p>
<p><b>Discrete data with genetic information</b>: 1000 samples in row and 6 variables in column. First column is the genetic variant and remaning columns are the gene expression.
</p>
<p><b>Discrete data without genetic information</b>: 1000 samples in row and 5 variables in column.
</p>
<p><b>Continuous data with genetic information for complex model</b>: 1000 samples in row and 22 variables in column. First 14 column is the genetic variants and remaning columns are the genes expression.   
</p>


<h3>Value</h3>

<p>A list that containing the numeric data matrix and components of a graph.
</p>

<ul>
<li> <p><code>simple</code>: Simple model.
</p>
</li>
<li> <p><code>complex</code>: Complex model.
</p>
</li>
<li> <p><code>cont</code>: Continuous.
</p>
</li>
<li> <p><code>disc</code>: Discrete.
</p>
</li>
<li> <p><code>withGV</code>: With genetic information.
</p>
</li>
<li> <p><code>withoutGV</code>: Without genetic information.
</p>
</li>
<li> <p><code>data</code>: Data matrix.
</p>
</li>
<li> <p><code>graph</code>: Components of a graph.
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Md Bahadur Badsha (mbbadshar@gmail.com)
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# Continuous data with genetic varitant (GV)
# load the data
data("data_examples")  
data &lt;- data_examples$simple$cont$withGV$data
# Extract the sample size
n &lt;- nrow(data)  

# Extract the node/column names
V &lt;- colnames(data)  

# Calculate Pearson correlation
suffStat_C &lt;- list(C = cor(data),
                   n = n)

# Infer the graph by MRPC
data.mrpc.cont.withGV &lt;- MRPC(data = data,
                              suffStat = suffStat_C, 
                              GV = 2,
                              FDR = 0.05, 
                              indepTest = 'gaussCItest',
                              labels = V,
                              FDRcontrol = 'LOND',
                              verbose = FALSE)
# Plot the results
par(mfrow = c(1, 2))
# plot the true graph
plot(data_examples$simple$cont$withGV$graph,
     main = "truth") 
# plot the inferred graph
plot(data.mrpc.cont.withGV,
     main = "inferred")    


# Continuous data without genetic information
# load the data
data("data_examples")    
data &lt;- data_examples$simple$cont$withoutGV$data
  
# Extract the sample size
n &lt;- nrow(data) 

# Extract the node/column names
V &lt;- colnames(data)

# Calculate Pearson correlation
suffStat_C &lt;- list(C = cor(data),
                   n = n)

# Infer the graph by MRPC
data.mrpc.cont.withoutGV &lt;- MRPC(data = data,
                                 suffStat = suffStat_C, 
                                 GV = 0,
                                 FDR = 0.05,
                                 indepTest = 'gaussCItest',
                                 labels = V, 
                                 FDRcontrol = 'LOND',
                                 verbose = FALSE)
# Plot the results
par(mfrow = c(1, 2))
# plot the true graph
plot(data_examples$simple$cont$withoutGV$graph,
     main = "truth") 
# plot the inferred graph
plot(data.mrpc.cont.withoutGV,
     main = "inferred") 

# Discrete data with genetic information
# load the data
data("data_examples")     
data &lt;- data_examples$simple$disc$withGV$data
  
# Extract the sample size
n &lt;- nrow(data)  

# Extract the node/column names
V &lt;- colnames(data)

suffStat_C &lt;- list (dm = data, adaptDF = FALSE, n.min = 1000)

# Infer the graph by MRPC
data.mrpc.disc.withGV &lt;- MRPC(data = data,
                              suffStat = suffStat_C, 
                              GV = 1,
                              FDR = 0.05, 
                              indepTest = 'disCItest',
                              labels = V, 
                              FDRcontrol = 'LOND',
                              verbose = FALSE)
# Plot the results
par (mfrow = c(1, 2))
# plot the true graph
plot(data_examples$simple$disc$withGV$graph,
     main = "truth") 
# Plot the inferred causal graph
plot(data.mrpc.disc.withGV,
     main = "inferred") 

# Discrete data without genetic information
# load the data
data("data_examples")    
data &lt;- data_examples$simple$disc$withoutGV$data
  
# Extract the sample size
n &lt;- nrow (data)  

# Extract the node/column names
V &lt;- colnames(data)   

suffStat_C &lt;- list (dm = data, adaptDF = FALSE, n.min = 1000)

# Infer the graph by MRPC
data.mrpc.disc.withoutGV &lt;- MRPC(data = data,
                                 suffStat = suffStat_C, 
                                 GV = 1,
                                 FDR = 0.05, 
                                 indepTest = 'disCItest',
                                 labels = V, 
                                 FDRcontrol = 'LOND',
                                 verbose = FALSE)
# Plot the results
par(mfrow = c(1, 2))
# plot the true graph
plot(data_examples$simple$disc$withoutGV$graph,
     main = "truth") 
# plot the inferred graph
plot(data.mrpc.disc.withoutGV,
     main = "inferred") 

# Continuous data with genetic information for complex model
# load the data
data("data_examples")   

# Graph without clustering
plot(data_examples$complex$cont$withGV$graph) 

# Adjacency matrix from directed example graph
Adj_directed &lt;- as(data_examples$complex$cont$withGV$graph,
                   "matrix")

# Plot of dendrogram with modules colors of nodes
PlotDendrogramObj &lt;- PlotDendrogram(Adj_directed,
                                    minModuleSize = 5)
                  
# Visualization of inferred graph with modules colors
PlotGraphWithModulesObj &lt;- PlotGraphWithModules(Adj_directed,
                                                PlotDendrogramObj,
                                                GV = 14,
                                                node.size = 8,
                                                arrow.size = 5,
                                                label.size = 3,
                                                alpha = 1) 
# plot 
plot(PlotGraphWithModulesObj)

# Run MRPC on the complex data set with ADDIS as the FDR control method.
data &lt;- data_examples$complex$cont$withGV$data
n &lt;- nrow (data)        # Number of rows
V &lt;- colnames(data)     # Column names

# Calculate Pearson correlation
suffStat_C &lt;- list(C = cor(data),
                   n = n)

# Infer the graph by MRPC
MRPC.addis &lt;- MRPC(data,
                   suffStat = suffStat_C,
                   GV = 14,
                   FDR = 0.05,
                   indepTest = 'gaussCItest',
                   labels = V,
                   FDRcontrol = 'ADDIS',
                   tau = 0.5,
                   lambda = 0.25,
                   verbose = FALSE)
                   
# Plot the true and inferred graphs.
par(mfrow = c(1, 2))
plot(data_examples$complex$cont$withGV$graph,
     main = 'True graph')
plot(MRPC.addis,
     main = 'Inferred graph')

## End(Not run)
  </code></pre>


</div>