<div class="container">

<table style="width: 100%;"><tr>
<td>mr_mvegger</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Multivariable MR-Egger method</h2>

<h3>Description</h3>

<p>The <code>mr_mvegger</code> function performs multivariable Mendelian randomization via the MR-Egger method. This is implemented by multivariable weighted linear regression.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mr_mvegger(
  object,
  orientate = 1,
  correl = FALSE,
  distribution = "normal",
  alpha = 0.05
)

## S4 method for signature 'MRMVInput'
mr_mvegger(
  object,
  orientate = 1,
  correl = FALSE,
  distribution = "normal",
  alpha = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>An <code>MRMVInput</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>orientate</code></td>
<td>
<p>The risk factor that genetic associations are orientated to. The univariable and multivariable versions of MR-Egger are both sensitive to the choice of parameterization of the genetic associations - which allele the associations are orientated with respect to (in other words, which allele is the effect allele). For univariable MR-Egger, this is resolved by setting the genetic associations with the exposure all to be positive. In multivariable MR-Egger, we have to choose which of the exposures to orientate the genetic associations to. The default option is <code>1</code>, meaning that genetic associations with the first exposure are set to be positive.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>correl</code></td>
<td>
<p>If the genetic variants are correlated, then this correlation can be accounted for. The matrix of correlations between must be provided in the <code>MRInput</code> object: the elements of this matrix are the correlations between the individual variants (diagonal elements are 1). If a correlation matrix is specified in the <code>MRInput</code> object, then <code>correl</code> is set to <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>distribution</code></td>
<td>
<p>The type of distribution used to calculate the confidence intervals. Options are <code>"normal"</code> (default) or <code>"t-dist"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>The significance level used to calculate the confidence interval. The default value is 0.05.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Multivariable MR-Egger is an extension of the MR-Egger method to deal with genetic variants that are associated with multiple risk factors.
</p>
<p>We implement the method using multivariable weighted linear regression. If the variants are correlated, the method is implemented using generalized weighted linear regression; this is hard coded using matrix algebra.
</p>
<p>The causal estimate is obtained by regression of the associations with the outcome on the associations with the risk factors, with the intercept estimated and weights being the inverse-variances of the associations with the outcome.
</p>


<h3>Value</h3>

<p>The output from the function is an <code>MVEgger</code> object containing:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>Model</code></td>
<td>
<p>A character string giving the type of model used (<code>"random"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Orientate</code></td>
<td>
<p>The number corresponding to the risk factor that the genetic associations are orientated to.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Exposure</code></td>
<td>
<p>A character vector with the names given to the exposure.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Outcome</code></td>
<td>
<p>A character string with the names given to the outcome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Correlation</code></td>
<td>
<p>The matrix of genetic correlations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Estimate</code></td>
<td>
<p>A vector of the causal estimates (slope coefficient).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>StdError.Est</code></td>
<td>
<p>Standard errors of the causal estimates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Pvalue.Est</code></td>
<td>
<p>The p-values associated with the estimates using a normal or t-distribution (as specified in <code>distribution</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CILower.Est</code></td>
<td>
<p>The lower bound of the causal estimates based on the estimated standard error and the significance level provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CIUpper.Est</code></td>
<td>
<p>The upper bound of the causal estimates based on the estimated standard error and the significance level provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Intercept</code></td>
<td>
<p>The value of the intercept estimate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>StdError.Int</code></td>
<td>
<p>Standard error of the intercept estimate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Pvalue.Int</code></td>
<td>
<p>The p-value associated with the intercept.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CILower.Int</code></td>
<td>
<p>The lower bound of the intercept based on the estimated standard error and the significance level provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CIUpper.Int</code></td>
<td>
<p>The upper bound of the intercept based on the estimated standard error and the significance level provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Pvalue</code></td>
<td>
<p>The p-values associated with the estimates (calculated as Estimate/StdError as per Wald test) using a normal or t-distribution (as specified in <code>distribution</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SNPs</code></td>
<td>
<p>The number of genetic variants (SNPs) included in the analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>RSE</code></td>
<td>
<p>The estimated residual standard error from the regression model.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Heter.Stat</code></td>
<td>
<p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value: the null hypothesis is that all genetic variants estimate the same causal parameter; rejection of the null is an indication that one or more variants may be pleiotropic.</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>Jessica Rees, Angela Wood, Stephen Burgess. Extending the MR-Egger method for multivariable Mendelian randomization to correct for both measured and unmeasured pleiotropy. Statistics in Medicine 2017; 36(29): 4705-4718. doi: 10.1002/sim.7492.
</p>


<h3>Examples</h3>

<pre><code class="language-R">mr_mvegger(mr_mvinput(bx = cbind(ldlc, hdlc, trig), bxse = cbind(ldlcse, hdlcse, trigse),
   by = chdlodds, byse = chdloddsse), orientate = 1)

</code></pre>


</div>