<div class="container">

<table style="width: 100%;"><tr>
<td>mr_input</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Inputting and formatting data for use in causal estimation</h2>

<h3>Description</h3>

<p>The <code>mr_input</code> function is required for inputting and formatting data for use in any of the estimation functions provided in this package. The <code>MRInput</code> class outputted by the function can also be viewed graphically using the <code>mr_plot</code> function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mr_input(
  bx = 0,
  bxse = 0,
  by = 0,
  byse = 0,
  correlation = matrix(),
  exposure = "exposure",
  outcome = "outcome",
  snps = "snp",
  effect_allele = NA,
  other_allele = NA,
  eaf = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>bx</code></td>
<td>
<p>A numeric vector of beta-coefficient values for genetic associations with the first variable (often referred to as the exposure, risk factor, or modifiable phenotype).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bxse</code></td>
<td>
<p>The standard errors associated with the beta-coefficients <code>bx</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>by</code></td>
<td>
<p>A numeric vector of beta-coefficient values for genetic associations with the second variable (often referred to as the outcome). For a disease outcome, the beta coefficients are log odds estimates from logistic regression analyses.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>byse</code></td>
<td>
<p>The standard errors associated with the beta-coefficients in <code>by</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>correlation</code></td>
<td>
<p>The matrix of correlations between genetic variants. If this variable is not provided, then we assume that genetic variants are uncorrelated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exposure</code></td>
<td>
<p>The name of the exposure variable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outcome</code></td>
<td>
<p>The name of the outcome variable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snps</code></td>
<td>
<p>The names of the genetic variants (SNPs) included in the analysis. The inputs <code>exposure</code>, <code>outcome</code>, and <code>snps</code> are not required, but may be useful for keeping track of various <code>MRInput</code> objects. They are also used by the <code>mr_plot</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>effect_allele</code></td>
<td>
<p>The name of the effect allele for each SNP. The beta-coefficients are the associations with the exposure and outcome per additional copy of the effect allele.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>other_allele</code></td>
<td>
<p>The name of the non-effect allele.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eaf</code></td>
<td>
<p>The expected allele frequencies (numeric). The slots <code>effect_allele</code>, <code>other_allele</code>, and <code>eaf</code> are neither required, nor currently used in the MendelianRandomization package. They are included for future compatibility with the MR-Base suite of functions.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The beta-coefficients are assumed to be estimated for uncorrelated (independent) genetic variants, although a correlation matrix can be specified if the variants are correlated in their distributions. We also assume that the beta-coefficients for associations with the exposure and with the outcome are uncorrelated (corresponding to a two-sample Mendelian randomization analysis), although correlation between associations with the exposure and with the outcome generally have little impact on causal estimates or standard errors.
</p>
<p>If the four variables are not all the same length, then an error message will be reported. The analyses will still try to run, but the output may be misleading. However, in some analyses (for example, the standard IVW and MR-Egger methods), the values of <code>bxse</code> are not used in the analysis, and can therefore safely be omitted (provided that the other variables are correctly labelled).
</p>


<h3>Value</h3>

<p>An MRInput object containing:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>betaX</code></td>
<td>
<p>The genetic associations with the exposure.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>betaXse</code></td>
<td>
<p>The corresponding standard errors.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>betaY</code></td>
<td>
<p>The genetic associations with the outcome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>betaYse</code></td>
<td>
<p>The corresponding standard errors.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>correlation</code></td>
<td>
<p>The matrix of genetic correlations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exposure</code></td>
<td>
<p>A character string giving the name given to the exposure.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outcome</code></td>
<td>
<p>A character string giving the name given to the outcome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snps</code></td>
<td>
<p>A vector of character strings with the names of the genetic variants.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>effect_allele</code></td>
<td>
<p>A vector of character strings with the names of the effect alleles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>other_allele</code></td>
<td>
<p>A vector of character strings with the names of the non-effect alleles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eaf</code></td>
<td>
<p>A numeric vector with the effect allele frequencies.</p>
</td>
</tr>
</table>
<h3>See Also</h3>

<p><code>extract.pheno.csv()</code> for a description of how an <code>MRInput</code> object can be extracted from PhenoScanner (<a href="http://www.phenoscanner.medschl.cam.ac.uk/">http://www.phenoscanner.medschl.cam.ac.uk/</a>).
</p>


</div>