<div class="container">

<table style="width: 100%;"><tr>
<td>MSCsimtester</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Validity tests of simulators of the multspecies coalescent model in phylogenomics.</h2>

<h3>Description</h3>

<p>The package performs comparisons of certain summary statistics for simulated gene tree samples to
theoretical predictions under the multispecies coalescent model.
The primary functions are <code>rootedTriple</code> for comparison of frequencies of topological rooted triples
on gene trees, and <code>pairwiseDist</code> and <code>ADtest</code> for comparison of the distributions of pairwise
distances between taxa on gene trees.
</p>


<h3>Details</h3>

<p>Required input is a collection of gene trees, stored as a <code>multiPhylo</code> object by the <code>ape</code> package,
and a rooted species tree, as a <code>Phylo</code> object,
with edge lengths in generations, together with constant population sizes for each edge.
</p>
<p><code>MSCsimtester</code> builds on the packages <code>ape</code> and <code>kSamples</code>.
</p>
<p>For further examples of use and citation purposes, see (Allman et al. 2019).
</p>


<h3>References</h3>

<p>Allman ES, Baños H, Rhodes JA (2019).
“Testing Multispecies Coalescent Simulators Using Summary Statistics.”
<em>arXiv:1908.01424</em>.
</p>


</div>