<div class="container">

<table style="width: 100%;"><tr>
<td>msea_sub</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>MSEA by Subramanian et al.</h2>

<h3>Description</h3>

<p>This function performs metabolite set enrichment analysis implemented in the same fashion as gene set enrichment analysis (Subramanian et al. 2005).
In this function, a permutation procedure is performed for a metabolite set rather than class label.
This procedure corresponds to a "gene set" of permutation type in GSEA-P software (Subramanian et al. 2007).
A leading-edge subset analysis is also undertaken following the standard GSEA procedure.
</p>


<h3>Usage</h3>

<pre><code class="language-R">msea_sub(M, D, y, maxiter = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>M</code></td>
<td>

<p>list of metbolite set name and metabolite IDs
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>D</code></td>
<td>

<p>data.frame(metabolite ID, data matix)
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>

<p>response variable (e.g. PC score)
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxiter</code></td>
<td>

<p>maximum number of iterations in random permutation (default=1000)
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>list of normalized enrichment score, p-value and q-value for metabolite set, and the results of leading edge subset
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Subramanian, A., Tamayo, P., Mootha, V. K., Mukherjee, S., Ebert, B. L., Gillette, M. A., Paulovich, A., Pomeroy, S. L., Golub, T. R., Lander, E. S. &amp; Mesirov, J. P. (2005)
Gene set enrichment analysis: A knowledge-based approach for interpreting genome-wide expression profiles. Proc. Natl. Acad. Sci. USA 102, 15545-15550.
</p>
<p>Subramanian, A., Kuehn, H., Gould, J., Tamayo, P., Mesirov, J.P. (2007) GSEA-P: A desktop application for Gene Set Enrichment Analysis. Bioinformatics, doi: 10.1093/bioinformatics/btm369.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(fasting)
data(pathway)

# pca and pca loading
pca &lt;- prcomp(fasting$X, scale=TRUE)
pca &lt;- pca_loading(pca)

# all detected metabolites
metabolites &lt;- colnames(fasting$X)

# statistically significant negatively correlated metabolites in PC1 loading
SIG &lt;- metabolites[pca$loading$R[,1] &lt; 0 &amp; pca$loading$p.value[,1] &lt; 0.05]
ALL &lt;- metabolites #all detected metabolites

# Set response variable
y &lt;- pca$x[,1]

# preparing dataframe
D &lt;- data.frame(ALL,t(fasting$X)) 		# preparing dataframe

# MSEA by Subramanian et al.
M &lt;- pathway$fasting
P &lt;- msea_sub(M,D,y, maxiter = 10) # iteration was set ato 10 for demonstration

</code></pre>


</div>