<div class="container">

<table style="width: 100%;"><tr>
<td>msea_ora</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>MSEA by over representation analysis</h2>

<h3>Description</h3>

<p>This function performs metabolite set enrichment analysis by over representation analysis (ORA).
Statistical hypothesis test of cross tabulation is performed by one-sided Fisher's exact test.
</p>


<h3>Usage</h3>

<pre><code class="language-R">msea_ora(SIG, ALL, M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>SIG</code></td>
<td>

<p>Metabolite names of significant metabolites
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ALL</code></td>
<td>

<p>Metabolite names of all detected metabolites
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>M</code></td>
<td>

<p>list of metabolite set name and metabolite name
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>list of p-value and q-value for metabolite set and selected (significant) metabolite IDs for each metabolite set
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Draghici S, Khatri P, Martins RP, Ostermeier GC, Krawetz SA.
Global functional profiling of gene expression.
Genomics. 2003 Feb;81(2):98-104.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Example1 : Metabolome data
data(fasting)
data(pathway)

# pca and pca loading
pca &lt;- prcomp(fasting$X, scale=TRUE)
pca &lt;- pca_loading(pca)

# all detected metabolites
metabolites &lt;- colnames(fasting$X)

# statistically significant negatively correlated metabolites in PC1 loading
SIG &lt;- metabolites[pca$loading$R[,1] &lt; 0 &amp; pca$loading$p.value[,1] &lt; 0.05]
ALL &lt;- metabolites #all detected metabolites

# metabolite set list
M &lt;- pathway$fasting

# MSEA by over representation analysis
B &lt;- msea_ora(SIG, ALL, M)
B$`Result of MSEA(ORA)`

## Example2 : Proteome data
data(covid19)
data(pathway)

X &lt;- covid19$X$proteomics
Y &lt;- covid19$Y
D &lt;- covid19$D
tau &lt;- covid19$tau

protein_name &lt;- colnames(X)

# pls-rog and pls-rog loading
plsrog &lt;- pls_rog(X,Y,D)
plsrog &lt;- plsrog_loading(plsrog)

# statistically significant proteins
index_prot &lt;- which(plsrog$loading$R[,1]&gt;0 &amp; plsrog$loading$p.value[,1]&lt;0.05)
sig_prot &lt;- protein_name[index_prot]

# detected proteins
protein_name &lt;- colnames(X)

# protein set list
M &lt;- pathway$covid19$proteomics

# MSEA by over representation analysis
B &lt;- msea_ora(sig_prot, protein_name, M)
B$`Result of MSEA(ORA)`

</code></pre>


</div>