<div class="container">

<table style="width: 100%;"><tr>
<td>MTAR</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Multiple-Traits Analysis of Rare-Variant Association Test</h2>

<h3>Description</h3>

<p>Test for association between a set of rare SNPs and multiple traits with input of summary statistics, possibly from overlap samples. The input number of SNPs in each trait can be different, MTAR function will analyze the union of SNPs that show up in at least one trait, and automatically handle the non-polymorphic SNPs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MTAR(
  U,
  V,
  MAF,
  MAF_UB = 0.05,
  zeta = NULL,
  genetic_cor.trait = NULL,
  rho.SNP = c(0, 0.5, 1),
  rho.trait = c(0, 0.5, 1),
  weight.SNP = c(1, 25)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>U</code></td>
<td>
<p>a numeric list, each sublist containing summary statistics U for each traits. The SNP IDs must be provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>V</code></td>
<td>
<p>a numeric list, each sublist containing the corresponding covariance matrix of summary statistics. If your original summary statistics are other format, please use Get_UV_from_data, Get_UV_from_varU or Get_UV_from_beta to generate the summary statistics <code class="reqn">U</code> and <code class="reqn">V</code> for MTAR.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF</code></td>
<td>
<p>a numeric vector containing minor allele frequency for the SNPs show up in at least one trait. The SNP IDs must be provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF_UB</code></td>
<td>
<p>a numeric value indicating the cutoff threshold of minor allele frequency for SNPs The default value is 0.05.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>zeta</code></td>
<td>
<p>a numeric matrix containing the sample correlation of Z-scores over a large number of independent null common SNPs across genome. The default value is NULL, where MTAR assumes there are no overlap samples. However, if there is overlapping in subjects, zeta must be provided. zeta can be estimated using MTAR::Get_zeta.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genetic_cor.trait</code></td>
<td>
<p>a numeric matrix containing the genetic correlation among traits. The default value of genetic_cor.trait is NULL, where an exchangeable correlation structure with the correlation coefficient denoted by rho.trait (<code class="reqn">\rho_2</code>) is assumed. In this case, there is no difference between cMTAR and iMTAR.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rho.SNP</code></td>
<td>
<p>a numeric vector containing all the possible values of <code class="reqn">\rho_1</code>. The default value is c(0, 0.5, 1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rho.trait</code></td>
<td>
<p>a numeric vector containing all the possible values of <code class="reqn">\rho_2</code>. The default value is c(0, 0.5, 1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weight.SNP</code></td>
<td>
<p>a numeric vector containing the parameters in Beta density function to calculate the weight among SNPs. The default value is c(1, 25).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>MTAR assumes that the genetic effect estimates <code class="reqn">\beta</code> has covariance matrix <code class="reqn">B</code>, which is a Kronecker product of two pre-specified matrices: among-variant effect covariance <code class="reqn">B_1</code> and among-trait effect covariance <code class="reqn">B_2</code>. An exchaneable correlation structure with the correlation coefficient denoted by rho.SNPs (<code class="reqn">\rho_1</code>) for <code class="reqn">B_1</code> is assumed. The default MTAR requires the input of genetic correlation matrix genetoc_cor.trait, if missing, then an exchaneable correlation structure for rho.trait (<code class="reqn">\rho_2</code>) is assumed. The default weight of <code class="reqn">B_1</code> is <code class="reqn">dBeta(MAF, 1, 25)</code>, which can be changed freely by users.
</p>


<h3>Value</h3>

<p>a list of p-values of MTAR-O, cMTAR, iMTAR and cctP as well as ancillary information. Here cctP is the Cauchy-combined p-value of SKAT and burden tests with default weight <code class="reqn">dBeta(MAF, 1, 25)</code>.
</p>


<h3>Author(s)</h3>

<p>Lan Luo
</p>
<p>Lan Luo
</p>


<h3>References</h3>

<p>Liu, Y., Chen, S., Li, Z., Morrison, A.C., Boerwinkle, E. and Lin, X., 2019. ACAT: A fast and powerful p value combination method for rare-variant analysis in sequencing studies. The American Journal of Human Genetics, 104(3), pp.410-421.
</p>
<p>Liu, Y. and Xie, J., 2019. Cauchy combination test: a powerful test with analytic p-value calculation under arbitrary dependency structures. Journal of the American Statistical Association, pp.1-18.
</p>
<p>Luo, L., Shen, J., Zhang, H., Chhibber, A., Mehrotra, D. V., Tang, Z., 2019. Multi-trait analysis of rare-variant association summary statistics using MTAR.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(MTAR.example)
attach(MTAR.example)
pval &lt;- MTAR(U = U, V = V, MAF = MAF, genetic_cor.trait = genetic_cor.trait,
             zeta = zeta)
pval
detach(MTAR.example)

</code></pre>


</div>