<div class="container">

<table style="width: 100%;"><tr>
<td>Get_zeta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate Covariances of Z-scores between Traits from Overlapping Samples</h2>

<h3>Description</h3>

<p>This function allows you to estimate the matrix <code class="reqn">\zeta</code> to adjust for the potential sample overlap in the data set. Here we applied LD pruning (<code class="reqn">r^2 &lt; 0.1</code> in 500kb region) on 1000 Genome genotype dataset (hg19) as a list of reference independent SNPs. The SNP ID is chr:pos.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Get_zeta(Zscore, pval_cutoff = 0.05, Indp_common_snp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Zscore</code></td>
<td>
<p>a numeric list, each sublist containing a vector of Z scores of SNPs with minor allele frequency (MAF) larger than 0.05. The chr:pos for each SNP is required.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pval_cutoff</code></td>
<td>
<p>a numeric value indicating the cutoff threshold of p-values. The default value is 0.05. Variants with p-value less than or equal to this threshold will be automatically removed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Indp_common_snp</code></td>
<td>
<p>a numeric list of independent common SNPs</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A <code class="reqn">K \times K</code> matrix <code class="reqn">\zeta</code>, where <code class="reqn">K</code> is the number of traits.
</p>


<h3>Author(s)</h3>

<p>Lan Luo
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(zeta.example)
attach(zeta.example)
# Downloading independent common SNPs from 1000Genome data set.
githubURL &lt;- "https://github.com/lan/MTAR/blob/master/indp_snps.1KG.rda?raw=true"
utils::download.file(githubURL,"1kgfile")
load("1kgfile")
zeta1 &lt;- Get_zeta(Zscore = Zscore, Indp_common_snp = indp_snps.1KG)
zeta1
detach(zeta.example)

</code></pre>


</div>