<div class="container">

<table style="width: 100%;"><tr>
<td>mPhen.sampleGeno</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Sample genotypes</h2>

<h3>Description</h3>

<p>...</p>


<h3>Usage</h3>

<pre><code class="language-R">mPhen.sampleGeno(n = 100, sampSize = 100, chr="0",pos = 1:n, 
snpids = paste(chr,pos,sep="_"),meanAlleleFreq=0.2, mu = 10,
samples =paste("id",1:sampSize,sep="_"),imputed = FALSE, dirichlet = 1)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>Number of genotypes to sample</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sampSize</code></td>
<td>
<p>Number of individuals to sample</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Name of chromosome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>Positions of genotypes on chromosome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpids</code></td>
<td>
<p>Ids of genotypes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>meanAlleleFreq</code></td>
<td>
<p>The mean allele frequency to simulate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu</code></td>
<td>
<p>A weight parameter which controls how close to the meanAlleleFreq the allele frequencies are sampled, via a beta distribution.  A higher number implies allele frequencies stay closer to mean</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>samples</code></td>
<td>
<p>The sample ids</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>imputed</code></td>
<td>
<p>Whether to simulate imputed data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dirichlet</code></td>
<td>
<p>The weight of a dirichlet distribution used to simulated imputed data</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns matrix of genotypes, with individuals by rows, and snps by column, or a 3 dimensional array if imputed is TRUE</p>


</div>