<div class="container">

<table style="width: 100%;"><tr>
<td>processVCFdata</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Process VCF data.</h2>

<h3>Description</h3>

<p>Check, annotate, and process variants imported from a list of VCF files, so that it can be used
to run a mutational signature analysis
</p>


<h3>Usage</h3>

<pre><code class="language-R">processVCFdata(
  vcfData,
  BSGenomeDb,
  chr_colName = "CHROM",
  pos_colName = "POS",
  ref_colName = "REF",
  alt_colName = "ALT",
  sample_colName = NULL,
  nucl_contextN = 3,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcfData</code></td>
<td>
<p>data.frame, includes mutation data from 2 or more samples</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BSGenomeDb</code></td>
<td>
<p>a BSGenomeDb-class object storing the genomic sequences and coordinates</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr_colName</code></td>
<td>
<p>string, name of the column including the chromosome (seq) name. Defaults to "CHROM"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos_colName</code></td>
<td>
<p>string, name of the column including the genomic coordinates/position. Defaults to "POS"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref_colName</code></td>
<td>
<p>string, name of the column including the reference nucleotide. Defaults to "REF"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alt_colName</code></td>
<td>
<p>string, name of the column including the variant nucleotide. Defaults to "ALT"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample_colName</code></td>
<td>
<p>string, name of the column including the sample ID. Can be NULL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nucl_contextN</code></td>
<td>
<p>integer, span (in nucelotides) of the context around the variants. Defaults to 3</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical, shall information about the ongoing analysis be printed to console</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>a data.frame including processed variants from VCF files
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li>
</ol>
</div>