<div class="container">

<table style="width: 100%;"><tr>
<td>attachContext</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Attach Nucleotide Context.</h2>

<h3>Description</h3>

<p>Retrieve the nucleotide context around each DNA variant based on the
genomic coordinates of the variant and a reference BSGenome database.
</p>


<h3>Usage</h3>

<pre><code class="language-R">attachContext(
  mutData,
  BSGenomeDb,
  chr_colName = "chr",
  start_colName = "start_position",
  end_colName = "end_position",
  nucl_contextN = 3,
  context_colName = "context"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mutData</code></td>
<td>
<p>data.frame storing mutation data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BSGenomeDb</code></td>
<td>
<p>a BSGenomeDb-class object, storing info about the genome of interest</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr_colName</code></td>
<td>
<p>string, name of the column storing seqNames. Defaults to "chr"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start_colName</code></td>
<td>
<p>string, name of the column storing start positions. Defaults to "start_position"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>end_colName</code></td>
<td>
<p>string, name of the column storing end positions. Defaults to "end_position"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nucl_contextN</code></td>
<td>
<p>integer, the span of nucleotides to be retrieved around the variant. Defaults to 3</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>context_colName</code></td>
<td>
<p>string, name of the column that will be storing the 
nucleotide context. Defaults to "context"</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>a modified data.frame including the nucleotide context in a new column
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li>
</ol>
</div>