<div class="container">

<table style="width: 100%;"><tr>
<td>cn.region.heatmap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A Function for Creating a Heatmap of DNA Copy Number Data</h2>

<h3>Description</h3>

<p>This function creates a heatmap of DNA copy number data for a given chromosomal region.
</p>


<h3>Usage</h3>

<pre><code class="language-R">cn.region.heatmap(cn.mat, gene.annot, plot.chr, plot.start, plot.stop,
  plot.list, sample.annot = NULL, sample.cluster = F, low.thresh = -2,
  high.thresh = 2, num.cols = 50, collist = c("blue", "white", "red"),
  annot.colors = c("black", "red", "green", "blue", "cyan"),
  plot.sample.annot = F, cytoband.colors = c("gray90", "gray60"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cn.mat</code></td>
<td>
<p>A matrix of gene-level DNA copy number data (rows = genes, columns = samples).
DNA methylation data can also be used.  Both row names (gene names) and column names (Sample IDs) must be given.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.annot</code></td>
<td>
<p>A three-column matrix containing gene position information.  Column 1 = chromosome number written in 
the form 'chr1' (note that chrX and chrY should be written chr23 and chr24), Column 2 = position (in base pairs), Column 3 = cytoband.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.chr</code></td>
<td>
<p>The chromosome used to define the region of interest.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.start</code></td>
<td>
<p>The genomic position (in base pairs) where the region starts.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.stop</code></td>
<td>
<p>The genomic position (in base pairs) where the region stops.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.list</code></td>
<td>
<p>A list produced by corr.list.compute().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.annot</code></td>
<td>
<p>An optional two-column matrix of sample annotation data.  Column 1 = sample IDs, Column 2 = sample annotation 
(e.g. tumor vs. normal).  If NULL, sample annot will be created using the common sample IDs and a single group ('1').  Default = NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.cluster</code></td>
<td>
<p>Logical values indicating whether the samples should be clustered.  Default = FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>low.thresh</code></td>
<td>
<p>Lower threshold for DNA copy number measurements.  All values less than low.thresh are set equal to low.thresh.  Default = -2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>high.thresh</code></td>
<td>
<p>Upper threshold for DNA copy number measurements.  All values greater than high.thresh are set equal to high.thresh.  Default = 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.cols</code></td>
<td>
<p>Number of distinct colors in the heatmap.  Default = 50.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>collist</code></td>
<td>
<p>Color scheme for displaying copy number values.  Default = ("blue", "white", "red").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annot.colors</code></td>
<td>
<p>Character vector used to define the color scheme for sample annotation.  Default = c("black", "red", "green", "blue", "cyan").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.sample.annot</code></td>
<td>
<p>Logical value used to specify whether the sample annotation information should be plotted.  Default = FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cytoband.colors</code></td>
<td>
<p>Character vector of length two used to define the color scheme for annotating the cytoband.  Default = c("gray90", "gray60").</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">exp.mat = tcga.exp.convert(exp.mat)

 cn.mat = tcga.cn.convert(cn.mat)

 prepped.data = data.prep(exp.mat, cn.mat, gene.annot, sample.annot, log.exp = FALSE)

 pd.exp = prepped.data[["exp"]]

 pd.cn = prepped.data[["cn"]]

 pd.ga = prepped.data[["gene.annot"]]

 pd.sa = prepped.data[["sample.annot"]]

 output.list = corr.list.compute(pd.exp, pd.cn, pd.ga, pd.sa)

 cn.region.heatmap(cn.mat = pd.cn, gene.annot = pd.ga, plot.chr = 11,

plot.start = 0e6, plot.stop = 135e6, sample.annot = pd.sa, plot.list = output.list)

</code></pre>


</div>