<div class="container">

<table style="width: 100%;"><tr>
<td>smooth.region.plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A Function for Plotting Smoothed Pearson Correlation Coefficients Genomewide</h2>

<h3>Description</h3>

<p>This function plots smoothed R or R^2 values produced by corr.list.compute() genomewide.
</p>


<h3>Usage</h3>

<pre><code class="language-R">smooth.region.plot(plot.list, plot.chr, plot.start, plot.stop,
  plot.column = "R^2", annot.colors = c("black", "red", "green", "blue",
  "cyan"), vert.pad = 0.05, ylim.low = NULL, ylim.high = NULL,
  plot.legend = TRUE, legend.loc = "topleft", lty.vec = NULL,
  lwd.vec = NULL, loess.span = 50, expand.size = 50,
  xaxis.label = "Position (Mb)", yaxis.label = NULL, main.label = NULL,
  axis.cex = 1, label.cex = 1, xaxis.line = 1.5, yaxis.line = 2.5,
  main.line = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>plot.list</code></td>
<td>
<p>A list produced by corr.list.compute().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.chr</code></td>
<td>
<p>The chromosome for which gene-level R or R^2 values will be plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.start</code></td>
<td>
<p>The genomic position (in base pairs) where the plot will start.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.stop</code></td>
<td>
<p>The genomic position (in base pairs) where the plot will stop.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.column</code></td>
<td>
<p>"R" or "R^2" depending on whether Pearson correlation coefficients or squared Pearson
correlation coefficients will be plotted.  Default = "R^2".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annot.colors</code></td>
<td>
<p>A vector of colors used for plotting values in different entries of plot.list.  Default = c("black", "red", "green", "blue", "cyan").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vert.pad</code></td>
<td>
<p>Amount of vertical white space in the plot.  Default = 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylim.low</code></td>
<td>
<p>Smallest value on the y-axis (used to control the range of values on the y-axis).  Default = NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylim.high</code></td>
<td>
<p>Largest value on the y-axis (used to control the range of values on the y-axis).  Default = NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.legend</code></td>
<td>
<p>Logical value specifying whether a legend should be included.  Default = FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend.loc</code></td>
<td>
<p>Character value specifing the location of the legend.  Default = "topright".  See See <code>legend</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lty.vec</code></td>
<td>
<p>Vector specifying line types for plotting values in different entries of plot.list.  Default = NULL.  See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lwd.vec</code></td>
<td>
<p>Vector specifying line widths for plotting values in different entries of  plot.list.  Default = NULL.  See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loess.span</code></td>
<td>
<p>A numerical value used to control the level of smoothing.  Smoothing is performed separately for each chromosome,
and loess.span effectively defines the number of genes in each smoothing window.  Default = 250.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>expand.size</code></td>
<td>
<p>A numerical value used to control smoothing at the ends of the region of interest.  Both ends of the region are artificially
extended by expand.size genes, smoothing is performed on the expanded region, and then the smoothed values are restricted to the size
of the original region.  Default = 50.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xaxis.label</code></td>
<td>
<p>Text used to label the x-axis of the plot.  Default = "Chromosome".  See <code>plot</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>yaxis.label</code></td>
<td>
<p>Text used to label the y-axis of the plot.  Default = NULL.  See <code>plot</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main.label</code></td>
<td>
<p>Text used to label the plot header.  Default = NULL.  See <code>plot</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>axis.cex</code></td>
<td>
<p>Numerical value used to specify the font size on the axes.  Default = 1.  See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label.cex</code></td>
<td>
<p>Numerical value used to specify the font size for the axis labels.  Default = 1.  See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xaxis.line</code></td>
<td>
<p>Numerical value used to specify location of xaxis.label.  Default = 0.  See <code>mtext</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>yaxis.line</code></td>
<td>
<p>Numerical value used to specify location of yaxis.label.  Default = 0.  See <code>mtext</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main.line</code></td>
<td>
<p>Numerical value used to specify location of main.label.  Default = 0.  See <code>mtext</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Creates a plot of gene-level R or R^2 values produced by corr.list.compute().
</p>


<h3>Examples</h3>

<pre><code class="language-R">exp.mat = tcga.exp.convert(exp.mat)

 cn.mat = tcga.cn.convert(cn.mat)

 prepped.data = data.prep(exp.mat, cn.mat, gene.annot, sample.annot, log.exp = FALSE)

 pd.exp = prepped.data[["exp"]]

 pd.cn = prepped.data[["cn"]]

 pd.ga = prepped.data[["gene.annot"]]

 pd.sa = prepped.data[["sample.annot"]]

 output.list = corr.list.compute(pd.exp, pd.cn, pd.ga, pd.sa)

 smooth.region.plot(plot.list = output.list, plot.chr = 11, plot.start = 0e6, plot.stop = 135e6)

</code></pre>


</div>