<div class="container">

<table style="width: 100%;"><tr>
<td>unsmooth.region.plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A Function for Plotting Pearson Correlation Coefficients in a Given Genomic Region</h2>

<h3>Description</h3>

<p>This function plots unsmoothed R or R^2 values produced by corr.list.compute() in a specified genomic region.
</p>


<h3>Usage</h3>

<pre><code class="language-R">unsmooth.region.plot(plot.list, plot.chr, plot.start, plot.stop,
  plot.column = "R", plot.points = TRUE, plot.lines = TRUE,
  gene.names = TRUE, annot.colors = c("black", "red", "green", "blue",
  "cyan"), vert.pad = 0.05, num.ticks = 5, ylim.low = NULL,
  ylim.high = NULL, pch.vec = NULL, lty.vec = NULL, lwd.vec = NULL,
  plot.legend = TRUE, legend.loc = "topright")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>plot.list</code></td>
<td>
<p>A list produced by corr.list.compute().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.chr</code></td>
<td>
<p>The chromosome for which gene-level R or R^2 values will be plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.start</code></td>
<td>
<p>The genomic position (in base pairs) where the plot will start.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.stop</code></td>
<td>
<p>The genomic position (in base pairs) where the plot will stop.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.column</code></td>
<td>
<p>"R" or "R^2" depending on whether Pearson correlation coefficients or squared Pearson
correlation coefficients will be plotted.  Default = "R^2".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.points</code></td>
<td>
<p>Logical value specifying whether points should be included in the plot.  Default = TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.lines</code></td>
<td>
<p>Logical values specifying whether points should be connected by lines.  Default = FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.names</code></td>
<td>
<p>Logical value specifying whether gene names should appear on the plot.  Default = FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annot.colors</code></td>
<td>
<p>A vector of colors used for plotting values in different entries of plot.list.  Default = c("black", "red", "green", "blue", "cyan").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vert.pad</code></td>
<td>
<p>Amount of vertical white space in the plot.  Default = 0.05.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.ticks</code></td>
<td>
<p>Number of ticks on the x-axis.  Default = 5.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylim.low</code></td>
<td>
<p>Smallest value on the y-axis (used to control the range of values on the y-axis).  Default = NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylim.high</code></td>
<td>
<p>Largest value on the y-axis (used to control the range of values on the y-axis).  Default = NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pch.vec</code></td>
<td>
<p>Vector specifying point characters for plotting values in different entries of plot.list.  Default = NULL.  See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lty.vec</code></td>
<td>
<p>Vector specifying line types for plotting values in different entries of plot.list.  Default = NULL.  See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lwd.vec</code></td>
<td>
<p>Vector specifying line widths for plotting values in different entries of  plot.list.  Default = NULL.  See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.legend</code></td>
<td>
<p>Logical value specifying whether a legend should be included.  Default = FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend.loc</code></td>
<td>
<p>Character value specifing the location of the legend.  Default = "topright".  See See <code>legend</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Creates a plot of gene-level R or R^2 values produced by corr.list.compute().
</p>


<h3>Examples</h3>

<pre><code class="language-R">exp.mat = tcga.exp.convert(exp.mat)

 cn.mat = tcga.cn.convert(cn.mat)

 prepped.data = data.prep(exp.mat, cn.mat, gene.annot, sample.annot, log.exp = FALSE)

 pd.exp = prepped.data[["exp"]]

 pd.cn = prepped.data[["cn"]]

 pd.ga = prepped.data[["gene.annot"]]

 pd.sa = prepped.data[["sample.annot"]]

 output.list = corr.list.compute(pd.exp, pd.cn, pd.ga, pd.sa)

 unsmooth.region.plot(plot.list = output.list, plot.chr = 11, plot.start = 69e6, plot.stop = 70.5e6)

</code></pre>


</div>