<div class="container">

<table style="width: 100%;"><tr>
<td>simulate_traits</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate phenotye data</h2>

<h3>Description</h3>

<p>This function simulates trait data from a genotype matrix.
</p>


<h3>Usage</h3>

<pre><code class="language-R">simulate_traits(
  genotype_matrix,
  n_causal = 1000,
  n_trait_specific = 10,
  n_pleiotropic = 10,
  H2 = 0.6,
  d = 2,
  rho = 0.8,
  marginal_correlation = 0.3,
  epistatic_correlation = 0.3,
  group_ratio_trait = 1,
  group_ratio_pleiotropic = 1,
  maf_threshold = 0.01,
  seed = 67132,
  logLevel = "INFO",
  logFile = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genotype_matrix</code></td>
<td>
<p>Genotype matrix with samples as rows, and SNPs as columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_causal</code></td>
<td>
<p>Number of SNPs that are causal.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_trait_specific</code></td>
<td>
<p>Number of causal SNPs with single trait epistatic effects.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_pleiotropic</code></td>
<td>
<p>Number of SNPs with epistatic effects on all traits.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>H2</code></td>
<td>
<p>Broad-sense heritability. Can be vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d</code></td>
<td>
<p>Number of traits.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rho</code></td>
<td>
<p>Proportion of heritability explained by additivity.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marginal_correlation</code></td>
<td>
<p>Correlation between the additive effects of the trait.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>epistatic_correlation</code></td>
<td>
<p>Correlation between the epistatic effects of the trait.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group_ratio_trait</code></td>
<td>
<p>Ratio of sizes of trait specific groups that interact, e.g. a ratio 1:3 would be value 3.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group_ratio_pleiotropic</code></td>
<td>
<p>Ratio of sizes of pleiotropic groups that interact, e.g. a ratio 1:3 would be value 3.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf_threshold</code></td>
<td>
<p>is a float parameter defining the threshold for the minor allele frequency not included in causal SNPs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>Random seed for simulation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logLevel</code></td>
<td>
<p>is a string parameter defining the log level for the logging package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logFile</code></td>
<td>
<p>is a string parameter defining the name of the log file for the logging output.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function takes a genotype matrix and simulates trait data under the following model:
beta_i ~ MN(0, V_i, I), i in { additive, epistatic, residual}
</p>
<p>The effect sizes follow a matrix normal distribution with no correlation between the samples but covariance between the effects for different traits
</p>


<h3>Value</h3>

<p>A list object containing the trait data, the genotype data, as well as the causal SNPs and summary statistics.
</p>


<h3>Examples</h3>

<pre><code class="language-R">p &lt;- 200
f &lt;- 10
g &lt;- 4
n &lt;- 100
d &lt;- 3
X &lt;- matrix(
    runif(p * n),
    ncol = p
)
data &lt;- simulate_traits(
    X, n_causal = f, n_trait_specific = g, n_pleiotropic = g, d = d, maf_threshold = 0,
    logLevel = "ERROR"
)
</code></pre>


</div>