<div class="container">

<table style="width: 100%;"><tr>
<td>MatchMap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Match a Phylostratigraphic Map or Divergence Map with a ExpressionMatrix</h2>

<h3>Description</h3>

<p>This function matches a <em>Phylostratigraphic Map</em> or <em>Divergence Map</em> only storing unique gene ids with a ExpressionMatrix
also storing only unique gene ids.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MatchMap(Map, ExpressionMatrix, remove.duplicates = FALSE, accumulate = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Map</code></td>
<td>
<p>a standard <em>Phylostratigraphic Map</em> or <em>Divergence Map</em> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ExpressionMatrix</code></td>
<td>
<p>a standard ExpressionMatrix object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove.duplicates</code></td>
<td>
<p>a logical value indicating whether duplicate gene ids should be removed from the data set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>accumulate</code></td>
<td>
<p>an accumulation function such as <code>mean()</code>, <code>median()</code>, or <code>min()</code>
to accumulate multiple expression levels that map to the same unique gene id present in the <code>ExpressionMatrix</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>In phylotranscriptomics analyses two major techniques are performed to
obtain standard <em>Phylostratigraphic map</em> or <em>Divergence map</em> objects.
</p>
<p>To obtain a <em>Phylostratigraphic Map</em>, <em>Phylostratigraphy</em> (Domazet-Loso et al., 2007) has to be performed. To obtain a <em>Divergence Map</em>, 
orthologous gene detection, Ka/Ks computations, and decilation (Quint et al., 2012; Drost et al., 2015) have to be performed.
</p>
<p>The resulting standard <em>Phylostratigraphic Map</em> or <em>Divergence Map</em> objects consist of 2 colums storing the phylostratum assignment 
or divergence stratum assignment of a given gene in column one, and the corresponding gene id of that gene on columns two.
</p>
<p>A standard ExpressionMatrix is a common gene expression matrix storing the gene ids or probe ids in the first column, and all experiments/stages/replicates in the following columns.
</p>
<p>The <em>MatchMap</em> function takes both standard datasets: <em>Map</em> and <em>ExpressionMatrix</em> to merge both data sets to obtain a standard PhyloExpressionSet or DivergenceExpressionSet object.
</p>
<p>This procedure is analogous to <code>merge</code>, but is customized to the <em>Phylostratigraphic Map</em>, <em>Divergence Map</em>, and <em>ExpressionMatrix</em> standards to allow a faster and more intuitive usage. 
</p>
<p>In case you work with an ExpressionMatrix that stores multiple expression levels for a unique gene id, you
can specify the <code>accumulation</code> argument to accumulate these multiple expression levels to obtain
one expression level for one unique gene.
</p>


<h3>Value</h3>

<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Domazet-Loso T, Brajkovic J, Tautz D (2007) A phylostratigraphy approach to uncover the genomic history of major adaptations in metazoan lineages. Trends Genet. 23: 533-9.
</p>
<p>Domazet-Loso T, Tautz D (2010) A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns. Nature 468: 815-8.
</p>
<p>Quint M., Drost H.G., Gabel A., Ullrich K.K., Boenn M., Grosse I. (2012) A transcriptomic hourglass in plant embryogenesis. Nature 490: 98-101.
</p>
<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012
</p>


<h3>Examples</h3>

<pre><code class="language-R">        
# load a standard PhyloExpressionSet
data(PhyloExpressionSetExample)
        
# in a standard PhyloExpressionSet, 
# column one and column two denote a standard 
# phylostratigraphic map
PhyloMap &lt;- PhyloExpressionSetExample[ , 1:2]
        
# look at the phylostratigraphic map standard
head(PhyloMap)
        
# in a standard PhyloExpressionSet, column two combined 
# with column 3 - N denote a standard ExpressionMatrix
ExpressionMatrixExample &lt;- PhyloExpressionSetExample[ , c(2,3:9)]
        
# these two data sets shall illustrate an example 
# phylostratigraphic map that is returned
# by a standard phylostratigraphy run, and a expression set 
# that is the result of expression data analysis 
# (background correction, normalization, ...)
        
# now we can use the MatchMap function to merge both data sets
# to obtain a standard PhyloExpressionSet
        
PES &lt;- MatchMap(PhyloMap, ExpressionMatrixExample)
        
# note that the function returns a head() 
# of the matched gene ids to enable
# the user to find potential mis-matches
        
# the entire procedure is analogous to merge() 
# with two data sets sharing the same gene ids 
# as column (primary key)
PES_merge &lt;- merge(PhyloMap, ExpressionMatrixExample)
        
        
        
</code></pre>


</div>