<div class="container">

<table style="width: 100%;"><tr>
<td>PlotRE</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot Relative Expression Levels</h2>

<h3>Description</h3>

<p>This function computes for each age category the corresponding relative expression profile.
</p>
<p>For each age category the corresponding relative expression profile is being computed as follows:
</p>
<p style="text-align: center;"><code class="reqn">f_js = ( e_js - e_j min ) / ( e_j max - e_j min )</code>
</p>

<p>where <code class="reqn">e_j min</code> and <code class="reqn">e_j max</code> denote the minimum/maximum <code>mean</code> expression level 
of phylostratum j over  developmental stages s. This linear transformation corresponds to 
a shift by <code class="reqn">e_j min</code> and a subsequent shrinkage by <code class="reqn">e_j max - e_j min</code>. 
As a result, the relative expression level <code class="reqn">f_js</code> of developmental stage s 
with minimum <code class="reqn">e_js</code> is 0, the relative expression level <code class="reqn">f_js</code> of the developmental 
stage s with maximum <code class="reqn">e_js</code> is 1, and the relative expression levels <code class="reqn">f_js</code> of 
all other stages s range between 0 and 1, accordingly.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PlotRE(
  ExpressionSet,
  Groups = NULL,
  modules = NULL,
  legendName = "age",
  xlab = "Ontogeny",
  ylab = "Relative Expression Level",
  main = "",
  y.ticks = 10,
  adjust.range = TRUE,
  alpha = 0.008,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Groups</code></td>
<td>
<p>a list containing the age categories for which mean expression levels shall be drawn.
For ex. evolutionary users can compare old phylostrata: PS1-3 (Class 1) and evolutionary young phylostrata: PS4-12 (Class 2). 
In this example, the list could be assigned as, <code>Groups = list(c(1:3), c(4:12))</code>. 
The group options is limited to 2 Groups.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>modules</code></td>
<td>
<p>a list storing three elements for specifying the modules: early, mid, and late. 
Each element expects a numeric vector specifying the developmental stages 
or experiments that correspond to each module. For example, 
<code>module</code> = <code>list(early = 1:2, mid = 3:5, late = 6:7)</code> devides a dataset storing seven developmental stages into 3 modules. Default is <code>modules = NULL</code>. 
But if specified, a shaded are will be drawn to illustrate stages corresponding to the mid module.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legendName</code></td>
<td>
<p>a character string specifying the legend title.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>label of x-axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>label of y-axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main</code></td>
<td>
<p>main text.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y.ticks</code></td>
<td>
<p>number of ticks that shall be drawn on the y-axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adjust.range</code></td>
<td>
<p>logical indicating whether or not the y-axis scale shall be adjusted to the same range in case two groups are specified. Default is <code>adjust.range = TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>transparency of the shaded area (between [0,1]). Default is <code>alpha = 0.1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>place holder for old version of PlotRE that was based on base graphics instead of ggplot2.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Studying the relative expression profiles of each phylostratum or divergence-stratum enables the detection
of common gene expression patterns shared by several phylostrata or divergence-strata.
</p>
<p>Finding similar relative expression profiles among phylostrata or divergence-strata suggests that 
phylostrata or divergence-strata sharing a similar relative expression profile are regulated by similar
gene regulatory elements. Hence, these common phylostrata or divergence-strata might govern similar processes in the given developmental time course.
</p>


<h3>Value</h3>

<p>a plot showing the relative expression profiles of phylostrata or divergence-strata belonging to the same group.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Domazet-Loso T and Tautz D. 2010. "A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns". Nature (468): 815-818.
</p>
<p>Quint M et al. 2012. "A transcriptomic hourglass in plant embryogenesis". Nature (490): 98-101.
</p>


<h3>See Also</h3>

<p><code>PlotBarRE</code>, <code>RE</code>, <code>REMatrix</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# read standard phylotranscriptomics data
data(PhyloExpressionSetExample)
data(DivergenceExpressionSetExample)

# example PhyloExpressionSet
PlotRE(PhyloExpressionSetExample,
       Groups = list(c(1:3), c(4:12)), 
       legendName = "PS")


# or you can choose any combination of groups
PlotRE(PhyloExpressionSetExample,
       Groups = list(c(1,7,9), c(2:6,8,10:12)),
       legendName = "PS")

    
# example DivergenceExpressionSet
PlotRE(DivergenceExpressionSetExample,
       Groups = list(c(1:5), c(6:10)), 
       legendName = "DS")


  
</code></pre>


</div>