<div class="container">

<table style="width: 100%;"><tr>
<td>tf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Transform Gene Expression Levels</h2>

<h3>Description</h3>

<p>This function transforms the gene expression set stored in an input 
PhloExpressionSet or DivergenceExpressionSet object and returns a 
PhloExpressionSet or DivergenceExpressionSet object with transformed expression levels. 
The resulting transformed PhloExpressionSet or DivergenceExpressionSet 
object can then be used for subsequent analyses based on transformed expression levels.
</p>


<h3>Usage</h3>

<pre><code class="language-R">tf(ExpressionSet, FUN)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ExpressionSet</code></td>
<td>
<p>a standard PhloExpressionSet or DivergenceExpressionSet object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FUN</code></td>
<td>
<p>any valid function that transformes gene expression levels.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Motivated by the dicussion raised by Piasecka et al., 2013, the influence of
gene expression transformation on the global phylotranscriptomics pattern does not seem negligible.
Hence, different transformations can result in qualitatively different <code>TAI</code> or <code>TDI</code>
patterns.
</p>
<p>Initially, the <code>TAI</code> and <code>TDI</code> formulas were defined for absolute expression levels.
So using the initial <code>TAI</code> and <code>TDI</code> formulas with transformed expression levels
might turn out in qualitatively different patterns when compared with non-transformed expression levels,
but might also belong to a different class of models, since different valid expression level transformation functions result in different patterns.
</p>
<p>The purpose of this function is to allow the user to study the qualitative impact of different transformation functions on 
the global <code>TAI</code> and <code>TDI</code> pattern, or on any subsequent phylotranscriptomics analysis.
</p>
<p>The examples using the <em>PhyloExpressionSetExample</em> data set show that using common gene expression 
transformation functions: <code>log2</code> (Quackenbush, 2001 and 2002), <code>sqrt</code> (Yeung et al., 2001), 
<code>boxcox</code>, or <em>inverse hyperbolic sine transformation</em>, each transformation results 
in qualitatively different patterns. 
Nevertheless, for each resulting pattern the statistical significance can be tested 
using either the <code>FlatLineTest</code> or <code>ReductiveHourglassTest</code> (Drost et al., 2014) 
to quantify the significance of interest.
</p>


<h3>Value</h3>

<p>a standard PhloExpressionSet or DivergenceExpressionSet object storing transformed gene expression levels.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Piasecka B, Lichocki P, Moretti S, et al. (2013) The hourglass and the early conservation modelsâ€“co-existing patterns of developmental constraints in vertebrates. PLoS Genet. 9(4): e1003476.
</p>
<p>Quint M., Drost H.G., Gabel A., Ullrich K.K., Boenn M., Grosse I. (2012) A transcriptomic hourglass in plant embryogenesis. Nature 490: 98-101.
</p>
<p>Domazet-Loso T., Tautz D. (2010) A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns. Nature 468: 815-8.
</p>
<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012
</p>
<p>K.Y. Yeung et al.: Model-based clustering and data transformations for gene expression data. Bioinformatics 2001, 17:977-987
</p>
<p>K.Y. Yeung et al.:  Supplement to Model-based clustering and data transformations for gene expression data - Data Transformations and the Gaussian mixture assumption. Bioinformatics 2001, 17:977-987
</p>
<p>P.A.C. Hoen et al.: Deep sequencing-based expression analysis shows major advances in robustness, resolution and inter-lab portability over five microarray platforms. Nucleic Acids Research 2008, Vol. 36, No. 21
</p>
<p>H.H. Thygesen et al.: Comparing transformation methods for DNA microarray data. BMC Bioinformatics 2004, 5:77
</p>
<p>John Quackenbush: Microarray data normalization and transformation. Nature Genetics 2002, 32:496-501
</p>
<p>John Quackenbush: Computational Analysis of Microarray Data. Nature Reviews 2001, 2:418-427
</p>
<p>R. Nadon and J. Shoemaker: Statistical issues with microarrays: processing and analysis. TRENDS in Genetics 2002, Vol. 18 No. 5:265-271
</p>
<p>B.P. Durbin et al.: A variance-stabilizing transformation for gene-expression microarray data. Bioinformatics 2002, 18:S105-S110
</p>
<p>J. M. Bland et al.: Transforming data. BMJ 1996, 312:770
</p>
<p>John B. Burbidge, Lonnie Magee and A. Leslie Robb (1988) Alternative Transformations to Handle Extreme Values of the Dependent Variable. Journal of the American Statistical Association, 83(401): 123-127.
</p>
<p>G. E. P. Box and D. R. Cox (1964) An Analysis of Transformations. Journal of the Royal Statistical Society. Series B (Methodological), 26(2): 211-252.
</p>


<h3>See Also</h3>

<p><code>TAI</code>, <code>TDI</code>, <code>FlatLineTest</code>, <code>ReductiveHourglassTest</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(PhyloExpressionSetExample)

# a simple example is to transform the gene expression levels
# of a given PhyloExpressionSet using a sqrt or log2 transformation

PES.sqrt &lt;- tf(PhyloExpressionSetExample, sqrt)

PES.log2 &lt;- tf(PhyloExpressionSetExample, log2)

# in case a given PhyloExpressionSet already stores gene expression levels
# that are log2 transformed and need to be re-transformed to absolute
# expression levels, to perform subsequent phylotranscriptomics analyses 
# (that are defined for absolute expression levels), one can re-transform
# a PhyloExpressionSet like this:

PES.absolute &lt;- tf(PES.log2 , function(x) 2^x)

# which should be the same as  PhyloExpressionSetExample :
head(PhyloExpressionSetExample)
head(PES.absolute)


# plotting the TAI using log2 transformed expression levels
# and performing the Flat Line Test to obtain the p-value
PlotPattern(ExpressionSet = tf(PhyloExpressionSetExample, log2), 
            type          = "l", 
            lwd           = 5, 
            TestStatistic = "FlatLineTest")

</code></pre>


</div>